<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Expression Quantitative Trait Locus (eQTL) Mapping: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css">
<script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/incubator/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicons/incubator/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicons/incubator/favicon-16x16.png">
<link rel="manifest" href="favicons/incubator/site.webmanifest">
<link rel="mask-icon" href="favicons/incubator/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/aio.html';">Instructor View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Expression Quantitative Trait Locus (eQTL) Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Expression Quantitative Trait Locus (eQTL) Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Expression Quantitative Trait Locus (eQTL) Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/aio.html">Instructor View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="review-eqtl-study.html">2. Genetic Drivers of Pancreatic Islet Function</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="load-explore-data.html">3. Load and explore the data</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="review-mapping-steps.html">4. Review Mapping Steps</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="map-one-eqtl.html">5. Mapping A Single Gene Expression Trait</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="map-many-eqtls.html">6. Mapping Many Gene Expression Traits</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="create-transcriptome-map.html">7. Creating A Transcriptome Map</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="full-transcriptome-map.html">8. Transcriptome Map of cis and trans eQTL</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="maximum-peaks-genes.html">9. Maximum eQTL Peaks and Nearby Genes</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="interpret-results.html">10. Interpreting qtl2 results</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="mediation-analysis.html">11. Mediation Analysis</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-introduction"><p>Content from <a href="introduction.html">Introduction</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/introduction.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are expression quantitative trait loci (eQTL)?</li>
<li>How are eQTL used in genetic studies?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe how an expression quantitative trait locus (eQTL) impacts
gene expression.</li>
<li>Describe how eQTL are used in genetic studies.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Differences in disease risk between individuals are often caused by
genetic variants. Identifying the effects of genetic variants is key to
understanding disease phenotypes and their underlying biology. The
effects of variants in many single-gene disorders, such as cystic
fibrosis, are generally well-characterized and their disease biology
well understood. For example, in cystic fibrosis, mutations in the
coding region of the CFTR gene alter the three-dimensional structure of
the chloride channel proteins in epithelial cells, affecting not only
chloride transport, but also sodium and potassium transport in the
lungs, pancreas and skin. The path from gene mutation to altered protein
to disease phenotype is relatively simple and well understood.</p>
<figure><img src="fig/single-gene-disease.png" alt="Single gene disorder" class="figure mx-auto d-block"><div class="figcaption">Single-gene diseases like cystic fibrosis are
relatively well understood. In cystic fibrosis, mutations in the coding
region of the CFTR gene result in a defective protein, leading to excess
mucus production that can damage the lungs and digestive system.</div>
</figure><p>The most common human disorders, however, involve many genes
interacting with each other and with the environment, a far more
complicated path to follow than the path from a single gene mutation to
its protein to a disease phenotype. Cardiovascular disease, Alzheimer’s
disease, arthritis, diabetes and cancer involve a complex interplay of
genes with environment, and their mechanisms are not well understood.
One method of understanding the relationship between genetic variants
and disease is a “Genome-wide association study (GWAS), which associates
genetic variants with disease traits. It is tempting to think that these
genetic variants would fall in coding regions. However, most GWAS
variants for common diseases like diabetes are located in
<strong>non-coding</strong> regions of the genome. These variants are
therefore likely to fall in regulatory sequences which are involved in
gene regulation.</p>
<figure><img src="fig/gwas-catalog.png" alt="Figures showing GWAS Catalog" class="figure mx-auto d-block"><div class="figcaption">GWAS variants such as SNPs are often in
non-coding regions of the genome, indicating that they regulate gene
expression.</div>
</figure><figure><img src="fig/cis-regulation-single-gene.png" alt="Figure showing regulation of gene by SNP" class="figure mx-auto d-block"><div class="figcaption">Here a non-coding SNP influences expression of a
gene, which in turn affects a disease phenotype or other outcome of
interest.</div>
</figure><!-- DMG: STOPPED HERE --><p>Gene regulation controls the quantity, timing and locale of gene
expression. Analyzing the association between gene expression and
genetic variants is known as expression quantitative trait locus (eQTL)
mapping. eQTL mapping searches for associations between the expression
of one or more genes and a genetic locus. Specifically, genetic variants
underlying eQTL peak explain some of the variation in gene expression
levels. eQTL studies can reveal the architecture of quantitative traits,
connect DNA sequence variation to phenotypic variation, and shed light
on transcriptional regulation and regulatory variation. Traditional
analytic techniques like linkage and association mapping can be applied
to thousands of gene expression traits (transcripts) in eQTL analysis,
such that gene expression can be mapped in much the same way as a
physiological phenotype like blood pressure or heart rate. Joining gene
expression and physiological phenotypes with genetic variation can
identify genes with variants affecting disease phenotypes.</p>
<p>To the simple diagram above we’ll add two more details. Non-coding
SNPs can regulate gene expression from nearby locatons on the same
chromosome (in cis):</p>
<figure><img src="fig/cis-regulation-gene1.png" alt="Figure showing SNP regulating gene which affects disease" class="figure mx-auto d-block"><div class="figcaption">Genetic variants like SNPs often affect gene
expression locally near the gene that they regulate (in cis).</div>
</figure><p>SNPs that affect gene expression from afar, often from a different
chromosome from the gene that they regulate are called distal (trans)
regulators.</p>
<figure><img src="fig/trans-regulation-gene2.png" alt="Figure showing trans regulation" class="figure mx-auto d-block"><div class="figcaption">Alternatively, SNPs often affect gene expression
distally from the gene that they regulate (in trans), often from a
different chromosome altogether.</div>
</figure><p>In this lesson we revisit genetic mapping of quantitative traits and
apply its methods to gene expression. The examples are from <a href="https://doi.org/10.1534/genetics.118.300864" class="external-link">Genetic Drivers of
Pancreatic Islet Function</a> by Keller, et al. This study offers
supporting evidence for type 2 diabetes-associated loci in human GWAS,
most of which affect pancreatic islet function. The study assessed
pancreatic islet gene expression in Diversity Outbred mice on either a
regular chow or high-fat, high-sugar diet. Islet mRNA abundance was
quantified and analyzed, and the study identified more than 18,000 eQTL
peaks.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>An expression quantitative trait locus (eQTL) explains part of the
variation in gene expression.</li>
<li>Traditional linkage and association mapping can be applied to gene
expression traits (transcripts).</li>
<li>Genetic variants, such as single nucleotide polymorphisms (SNPs),
that underlie eQTL illuminate transcriptional regulation and
variation.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-review-eqtl-study"><p>Content from <a href="review-eqtl-study.html">Genetic Drivers of Pancreatic Islet Function</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/review-eqtl-study.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is the hypothesis of an example eQTL study?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe an example eQTL study in Diversity Outbred mice.</li>
<li>State the hypothesis from an example eQTL study in Diversity Outbred
mice.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Genome-wide association studies (GWAS) often identify variants in
non-coding regions of the genome, indicating that regulation of gene
expression predominates in common diseases like type II diabetes. Most
of the more than 100 genetic loci associated with type II diabetes
affect the function of pancreatic islets, which produce insulin for
regulating blood glucose levels. Susceptibility to type II diabetes
(T2D) increases with obesity, such that T2D-associated genetic loci
operate mainly under conditions of obesity (See <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5937189/" class="external-link">Keller,
Mark P et al. “Genetic Drivers of Pancreatic Islet Function.” Genetics
vol. 209,1 (2018): 335-356</a>). Like most GWAS loci, the T2D-associated
genetic loci identified from genome-wide association studies (GWAS) have
very small effect sizes and odds ratios just slightly more than 1.</p>
<p>This study explored islet gene expression in diabetes. The authors
hypothesized that gene expression changes in response to dietary
challenge would reveal signaling pathways involved in stress responses.
The expression of many genes often map to the same locus, indicating
that expression of these genes is controlled in common. If their mRNAs
encode proteins with common physiological functions, the function of the
controlling gene(s) is revealed. Variation in expression of the
controlling gene(s), rather than a genetic variant, can be tested as an
immediate cause of a disease-related phenotype.</p>
<p>In this study, Diversity Outbred (DO) mice were fed a high-fat,
high-sugar diet as a stressor, sensitizing the mice to develop diabetic
traits. Body weight and plasma glucose, insulin, and triglyceride
measurements were taken biweekly. Food intake could be measured since
animals were individually housed. A glucose tolerance test at 18 weeks
of age provided measurement of dynamic glucose and insulin changes at 5,
15, 30, 60 and 120 minutes after glucose administration. Area under the
curve (AUC) was determined from these time points for both plasma
glucose and insulin levels. Homeostatic model assessment (HOMA) of
insulin resistance (IR) and pancreatic islet function (B) were
determined after the glucose tolerance test was given. Islet cells were
isolated from pancreas, and RNA extracted and libraries constructed from
isolated RNA for gene expression measurements.</p>
<p>Genome scans were performed with the leave-one-chromosome-out (LOCO)
method for kinship correction. Sex and experimental cohort (DO wave)
were used as covariates. The results of one scan for insulin area under
the curve (AUC) is shown below with a strong peak on chromosome 11. In
this lesson, we will look into genes located under this peak.</p>
<figure><img src="fig/insulin-auc-lod-plot.png" alt="LOD plot for insulin area under the curve" class="figure mx-auto d-block"><div class="figcaption">LOD plot for insulin area under the curve</div>
</figure><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div></section><section id="aio-load-explore-data"><p>Content from <a href="load-explore-data.html">Load and explore the data</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/load-explore-data.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What data are required for eqtl mapping?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To provide an example and exploration of data used for eqtl
mapping.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Load the libraries.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">ggbeeswarm</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">corrplot</span><span class="op">)</span></span>
<span><span class="co"># the following analysis is derived from supplementary </span></span>
<span><span class="co"># File S1 Attie_eQTL_paper_physiology.Rmd </span></span>
<span><span class="co"># by Daniel Gatti. See Data Dryad entry for more information.</span></span></code></pre>
</div>
<section><h2 class="section-heading" id="physiological-phenotypes">Physiological Phenotypes<a class="anchor" aria-label="anchor" href="#physiological-phenotypes"></a>
</h2>
<hr class="half-width">
<p>The complete data used in these analyses are available from <a href="https://doi.org/10.5061/dryad.pj105" class="external-link">Data Dryad</a>.</p>
<p>Load in the clinical phenotypes.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_clinical.phenotypes.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_clinical.phenotypes.RData', probable reason 'No such file
or directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<p>See the <a href="data/Attie-232_Attie_DO_Islets-dictionary.csv">data
dictionary</a> to see a description of each of these phenotypes. You can
also view a table of the data dictionary.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin_dict</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">description</span>, <span class="va">formula</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">kable</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin_dict' not found</code></pre>
</div>
<div class="section level3">
<h3 id="phenotype-distributions">Phenotype Distributions<a class="anchor" aria-label="anchor" href="#phenotype-distributions"></a>
</h3>
<p>Boxplots are a great way to view the distribution of the data and to
identify any outliers. We will be using the total area under the curve
of insulin from the glucose tolerance test (Ins_tAUC). We will also
log-transform the data using the [scale_y_log10()][scale_y_log10]
function.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot Insulin on a log 10 scale</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">pheno_clin</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">sex</span>, <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Insulin tAUC"</span>, y <span class="op">=</span> <span class="st">"Insulin tAUC"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<p>Another visualization that has become popular is the [Violin Plot][<a href="https://en.wikipedia.org/wiki/Violin_plot" class="external-link uri">https://en.wikipedia.org/wiki/Violin_plot</a>]. We can
create one using ggplot’s [geom_violin][<a href="https://ggplot2.tidyverse.org/reference/geom_violin.html" class="external-link uri">https://ggplot2.tidyverse.org/reference/geom_violin.html</a>].
Whereas the boxplot automatically adds the median, we must tell
<code>geom_violin()</code> which quantiles that we want to draw using
the argument <code>draw_quantiles = c(0.25, 0.5, 0.75)</code>. We have
also overlaid the data points using ggbeeswarm’s [geom_beeswarm][<a href="https://www.rdocumentation.org/packages/ggbeeswarm/versions/0.5.3/topics/geom_beeswarm" class="external-link uri">https://www.rdocumentation.org/packages/ggbeeswarm/versions/0.5.3/topics/geom_beeswarm</a>].
We have told <code>geom_beeswarm()</code> to plot the points using the
argument <code>alpha = 0.1</code>. The <code>alpha</code> argument
ranges between 0 (completely transparent) to 1 (completely opaque). A
value of 0.1 means mostly transparent.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot Insulin on a log 10 scale</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">pheno_clin</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">sex</span>, <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>draw_quantiles <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_beeswarm</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Insulin tAUC"</span>, y <span class="op">=</span> <span class="st">"Insulin tAUC"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1:</h3>
<div class="callout-content">
<p>How many orders of magnitude (powers of 10) does Insulin tAUC
span?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Insulin tAUC spans three orders of magnitude, from near 10 to over
1000.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2:</h3>
<div class="callout-content">
<p>Which sex has higher median Insulin tAUC values?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>Males have higher Insulin tAUC than females.</p>
</div>
</div>
</div>
</div>
<p>Both of the boxplot and the violin plot are useful visualizations
which you can use to get some sense of the distribution of your
data.</p>
</div>
<div class="section level3">
<h3 id="quality-control-of-data">Quality Control of Data<a class="anchor" aria-label="anchor" href="#quality-control-of-data"></a>
</h3>
<p>Many statistical tests rely upon the data having a “normal” (or
Gaussian) distribution. Many biological phenotypes do not follow this
distribution and must be transformed before analysis. This is why we
log-transformed the data in the plots above.</p>
<p>While we can “eyeball” the distributions in the violin plot, it would
be better to use a “quantile-quantile” plot.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<p>In these plots, the “quantiles” of the normal distribution are
plotted on the X-axis and the data are plotted on the Y-axis. The line
indicates the quantiles that would be followed by a normal distribution.
The untransformed data do <strong>not</strong> follow a normal
distribution because the points are far from the line.</p>
<p>Next, we will loag-transform the data and then create a
quantile-quantile plot.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Ins_tAUC <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3:</h3>
<div class="callout-content">
<p>Does the log transformation make the data more normally distributed?
Explain your answer.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>Yes. The log transformation makes the data more normally distributed
because the data points follow the normality line more closely.</p>
</div>
</div>
</div>
</div>
<div id="challenge-4" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4" class="callout-inner">
<h3 class="callout-title">Challenge 4:</h3>
<div class="callout-content">
<p>Do any data points look suspicious to you? Explain your answer.</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>The data points that deviate from the normality line would be worth
investigating. All data deviates somewhat from normality, but the three
lowest points in the male data plot would be worth investigating. They
may be real, but there may also have been mishap in the assay.</p>
</div>
</div>
</div>
</div>
<p>Another way to identify outliers is to standardize the data and look
for data points that are more than four standard deviations from the
mean.</p>
<p>To do this, we will log transform and standardize Insulin tAUC.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ins_tauc</span> <span class="op">=</span> <span class="va">pheno_clin</span> <span class="op">%&gt;%</span> </span>
<span>             <span class="fu">select</span><span class="op">(</span><span class="va">mouse</span>, <span class="va">sex</span>, <span class="va">Ins_tAUC</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">group_by</span><span class="op">(</span><span class="va">sex</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>             <span class="fu">mutate</span><span class="op">(</span>Ins_tAUC <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">Ins_tAUC</span><span class="op">)</span>,</span>
<span>                    Ins_tAUC <span class="op">=</span> <span class="fu">scale</span><span class="op">(</span><span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ins_tauc</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">sex</span>, y <span class="op">=</span> <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="op">-</span><span class="fl">4</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span>  <span class="fl">4</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Distribution of Standardized Ins_tAUC"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'ins_tauc' not found</code></pre>
</div>
<p>There are no data points outside of the four standard deviation
limits.</p>
</div>
</section><section><h2 class="section-heading" id="gene-expression-phenotypes">Gene Expression Phenotypes<a class="anchor" aria-label="anchor" href="#gene-expression-phenotypes"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the expression data along with annotations and metadata</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/dataset.islet.rnaseq.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/dataset.islet.rnaseq.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">dataset.islet.rnaseq</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># look at gene annotations</span></span>
<span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">annots</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>,<span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># look at raw counts</span></span>
<span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">raw</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># look at sample metadata</span></span>
<span><span class="co"># summarize mouse sex, birth dates and DO waves</span></span>
<span><span class="fu">table</span><span class="op">(</span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">samples</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"sex"</span>, <span class="st">"birthdate"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">table</span><span class="op">(</span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">samples</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"sex"</span>, <span class="st">"DOwave"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<p>In order to make reasonable gene comparisons between samples, the
count data need to be normalized. In the quantile-quantile (Q-Q) plot
below, count data for the first gene are plotted over a diagonal line
tracing a normal distribution for those counts. Notice that most of the
count data values lie off of this line, indicating that these gene
counts are not normally distributed.</p>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<p>Q-Q plots for the first six genes show that count data for these
genes are not normally distributed. They are also not on the same scale.
The y-axis values for each subplot range to 20,000 counts in the first
subplot, 250 in the second, 90 in the third, and so on.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">raw</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">ENSMUSG00000000001</span><span class="op">:</span><span class="va">ENSMUSG00000000058</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">'gene'</span>, values_to <span class="op">=</span> <span class="st">'value'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">gene</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'Count distribution for six genes'</span>,</span>
<span>         xlab <span class="op">=</span> <span class="st">'Normal percentiles'</span>, y <span class="op">=</span> <span class="st">'Count percentiles'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<p>Q-Q plots of the normalized expression data for the first six genes
show that the data values match the diagonal line well, meaning that
they are now normally distributed. They are also all on the same scale
now as well.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">expr</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">ENSMUSG00000000001</span><span class="op">:</span><span class="va">ENSMUSG00000000058</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">'gene'</span>, values_to <span class="op">=</span> <span class="st">'value'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">gene</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'Normalized count distribution for six genes'</span>,</span>
<span>         xlab <span class="op">=</span> <span class="st">'Normal percentiles'</span>, y <span class="op">=</span> <span class="st">'Count percentiles'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<p>Boxplots of raw counts for six example genes are shown at left below.
Notice that the median count values (horizontal black bar in each
boxplot) are not comparable between the genes because the counts are not
on the same scale. At right, boxplots for the same genes show normalized
count data on the same scale.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw</span> <span class="op">=</span> <span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">raw</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">select</span><span class="op">(</span><span class="va">ENSMUSG00000000001</span><span class="op">:</span><span class="va">ENSMUSG00000000058</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">'gene'</span>, values_to <span class="op">=</span> <span class="st">'value'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">mutate</span><span class="op">(</span>type <span class="op">=</span> <span class="st">'raw'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">norm</span> <span class="op">=</span> <span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">expr</span> <span class="op">%&gt;%</span> </span>
<span>         <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>         <span class="fu">select</span><span class="op">(</span><span class="va">ENSMUSG00000000001</span><span class="op">:</span><span class="va">ENSMUSG00000000058</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>         <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">'gene'</span>, values_to <span class="op">=</span> <span class="st">'value'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>         <span class="fu">mutate</span><span class="op">(</span>type <span class="op">=</span> <span class="st">'normalized'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">bind_rows</span><span class="op">(</span><span class="va">raw</span>, <span class="va">norm</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>type <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">type</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'raw'</span>, <span class="st">'normalized'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">gene</span>, <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">type</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'Count distributions for example genes'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in `bind_rows()` at dplyr/R/mutate.R:146:3:
! Argument 1 must be a data frame or a named atomic vector.</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">raw</span>, <span class="va">norm</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in rm(raw, norm): object 'raw' not found</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in rm(raw, norm): object 'norm' not found</code></pre>
</div>
<p>Have a look at the first several rows of normalized count data.</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># look at normalized counts</span></span>
<span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">expr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<p>The expression data loaded provides LOD peaks for the eQTL analyses
performed in this study. As a preview of what you will be doing next,
look at the first several rows of LOD peak values and extract the LOD
peaks for chromosome 11.</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># look at LOD peaks</span></span>
<span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">lod.peaks</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>,<span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># look at chromosome 11 LOD peaks</span></span>
<span><span class="va">chr11_peaks</span> <span class="op">&lt;-</span> <span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">annots</span> <span class="op">%&gt;%</span> </span>
<span>   <span class="fu">select</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">chr</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>   <span class="fu">filter</span><span class="op">(</span><span class="va">chr</span><span class="op">==</span><span class="st">"11"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>   <span class="fu">left_join</span><span class="op">(</span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">lod.peaks</span>, </span>
<span>             by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"chr"</span> <span class="op">=</span> <span class="st">"chrom"</span>, <span class="st">"gene_id"</span> <span class="op">=</span> <span class="st">"annot.id"</span><span class="op">)</span><span class="op">)</span> </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># look at the first several rows of chromosome 11 peaks</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">chr11_peaks</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'chr11_peaks' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># how many rows?</span></span>
<span><span class="fu">dim</span><span class="op">(</span><span class="va">chr11_peaks</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'chr11_peaks' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># how many rows have LOD scores?</span></span>
<span><span class="va">chr11_peaks</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">lod</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dim</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'chr11_peaks' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># sort chromosome 11 peaks by LOD score</span></span>
<span><span class="va">chr11_peaks</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">lod</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">head</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'chr11_peaks' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># range of LOD scores and positions</span></span>
<span><span class="fu">range</span><span class="op">(</span><span class="va">chr11_peaks</span><span class="op">$</span><span class="va">lod</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'chr11_peaks' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">range</span><span class="op">(</span><span class="va">chr11_peaks</span><span class="op">$</span><span class="va">pos</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'chr11_peaks' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># view LOD scores by position</span></span>
<span><span class="va">chr11_peaks</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">lod</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">lod</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'chr11_peaks' not found</code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-review-mapping-steps"><p>Content from <a href="review-mapping-steps.html">Review Mapping Steps</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/review-mapping-steps.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are the steps involved in running QTL mapping in Diversity
Outbred mice?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To understand the key steps running a QTL mapping analysis</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Before we begin to run QTL mapping on gene expression data to find
eQTLs, let’s review the main QTL mapping steps that we learnt in the <a href="https://smcclatchy.github.io/qtl-mapping/" class="external-link">QTL mapping course</a>.
As a reminder, we are using data from the Keller et al. <a href="https://academic.oup.com/genetics/article/209/1/335/5931013?login=false" class="external-link">paper</a>
that are freely available to <a href="doi:10.5061/dryad.pj105">download</a>.</p>
<p>Make sure that you are in your main directory. If you’re not sure
where you are working right now, you can check your working directory
with <code>getwd()</code>. If you are not in your main directory, run
<code>setwd("code")</code> in the Console or Session -&gt; Set Working
Directory -&gt; Choose Directory in the RStudio menu to set your working
directory to the code directory.</p>
<div class="section level3">
<h3 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h3>
<p>Below are the neccessary libraries that we require for this review.
They are already installed on your machines so go ahead an load them
using the following code:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">broom</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>The data for this tutorial has been saved as several R binary files
which contain several data objects, including phenotypes and mapping
data as well as the genoprobs.</p>
<p>Load the data in now by running the following command in your new
script.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##phenotypes</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_clinical.phenotypes.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_clinical.phenotypes.RData', probable reason 'No such file
or directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##mapping data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_mapping.data.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_mapping.data.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##genotype probabilities</span></span>
<span><span class="va">probs</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">"../data/attie_DO500_genoprobs_v5.rds"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in gzfile(file, "rb"): cannot open compressed file
'../data/attie_DO500_genoprobs_v5.rds', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in gzfile(file, "rb"): cannot open the connection</code></pre>
</div>
<p>We loaded in a few data objects today. Check the Environment tab to
see what was loaded. You should see a phenotypes object called
<code>pheno_clin</code> with 500 observations (in rows) of 32 variables
(in columns), an object called map (physical map), and an object called
probs (genotype probabilities).</p>
</div>
<div class="section level3">
<h3 id="phenotypes">Phenotypes<a class="anchor" aria-label="anchor" href="#phenotypes"></a>
</h3>
<p>In this data set, we have 20 phenotypes for 500 Diversity Outbred
mice. <code>pheno_clin</code> is a data frame containing the phenotype
data as well as covariate information. Click on the triangle to the left
of <code>pheno_clin</code> in the Environment pane to view its contents.
Run <code>names(pheno_clin)</code> to list the variables.</p>
<p><code>pheno_clin_dict</code> is the phenotype dictionary. This data
frame contains information on each variable in <code>pheno_clin</code>,
including <code>name</code>,<code>short name</code>,
<code>pheno_type</code>, <code>formula</code> (if used) and
<code>description</code>.</p>
<p>Since the paper is interested in type 2 diabetes and insulin
secretion, we will choose <code>insulin tAUC</code> (area under the
curve (AUC) which was calculated without any correction for baseline
differences) for this review</p>
<p>Many statistical models, including the QTL mapping model in qtl2,
expect that the incoming data will be normally distributed. You may use
transformations such as log or square root to make your data more
normally distributed. Here, we will log transform the data.</p>
<p>Here is a histogram of the untransformed data.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">Ins_tAUC</span>, main <span class="op">=</span> <span class="st">"Insulin Area Under the Curve"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<p>Now, let’s apply the <code>log()</code> function to this data to
correct the distribution.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">Ins_tAUC_log</span> <span class="op">&lt;-</span> <span class="fu">log</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">Ins_tAUC</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<p>Now, let’s make a histogram of the log-transformed data.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">Ins_tAUC_log</span>, main <span class="op">=</span> <span class="st">"insulin tAUC (log-transformed)"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<p>This looks much better!</p>
</div>
<div class="section level3">
<h3 id="the-marker-map">The Marker Map<a class="anchor" aria-label="anchor" href="#the-marker-map"></a>
</h3>
<p>The marker map for each chromosome is stored in the <code>map</code>
object. This is used to plot the LOD scores calculated at each marker
during QTL mapping. Each list element is a numeric vector with each
marker position in megabases (Mb). Here we are using the 69K grid marker
file. Often when there are numerous genotype arrays used in a study, we
interoplate all to a 69k grid file so we are able to combine all samples
across different array types.</p>
<p>Look at the structure of <code>map</code> in the Environment tab by
clicking the triangle to the left or by running <code>str(map)</code> in
the Console.</p>
</div>
<div class="section level3">
<h3 id="genotype-probabilities">Genotype probabilities<a class="anchor" aria-label="anchor" href="#genotype-probabilities"></a>
</h3>
<p>Each element of <code>probs</code> is a 3 dimensional array
containing the founder allele dosages for each sample at each marker on
one chromosome. These are the 8 state allelle probabilities (not 32)
using the 69k marker grid for same 500 DO mice that also have clinical
phenotypes. We have already calculated genotype probabilities for you,
so you can skip the step for <a href="https://smcclatchy.github.io/qtl-mapping/calc-genoprob/" class="external-link">calculating
genotype probabilities</a> and the optional step for calculating allele
probabilities.</p>
<p>Next, we look at the dimensions of <code>probs</code> for chromosome
1:</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span><span class="va">probs</span>, <span class="va">map</span>, ind <span class="op">=</span> <span class="fl">1</span>, chr <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<p>In the plot above, the founder contributions, which range between 0
and 1, are colored from white (= 0) to black (= 1.0). A value of ~0.5 is
grey. The markers are on the X-axis and the eight founders (denoted by
the letters A through H) on the Y-axis. Starting at the left, we see
that this sample has genotype GH because the rows for G &amp; H are
grey, indicating values of 0.5 for both alleles. Moving along the genome
to the right, the genotype becomes HH where where the row is black
indicating a value of 1.0. This is followed by CD, DD, DG, AD, AH, CE,
etc. The values at each marker sum to 1.0.</p>
</div>
<div class="section level3">
<h3 id="kinship-matrix">
<a href="https://smcclatchy.github.io/qtl-mapping/calc-kinship/" class="external-link">Kinship
Matrix</a><a class="anchor" aria-label="anchor" href="#kinship-matrix"></a>
</h3>
<p>The kinship matrix has already been calculated and loaded in
above</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="fu">heatmap</span><span class="op">(</span><span class="va">K</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'K' not found</code></pre>
</div>
<p>The figure above shows kinship between all pairs of samples. Light
yellow indicates low kinship and dark red indicates higher kinship.
Orange values indicate varying levels of kinship between 0 and 1. The
dark red diagonal of the matrix indicates that each sample is identical
to itself. The orange blocks along the diagonal may indicate close
relatives (i.e. siblings or cousins).</p>
</div>
<div class="section level3">
<h3 id="covariates">Covariates<a class="anchor" aria-label="anchor" href="#covariates"></a>
</h3>
<p>Next, we need to create additive covariates that will be used in the
mapping model. First, we need to see which covariates are significant.
In the data set, we have <code>sex</code>, <code>DOwave</code> (Wave
(i.e., batch) of DO mice) and <code>diet_days</code> (number of days on
diet) to test whether there are any gender, batch or diet effects.</p>
<p>First we are going to select out the covariates and phenotype we want
from <code>pheno_clin</code> data frame. Then reformat these selected
variables into a long format (using the <code>gather</code> command)
grouped by the phenotypes (in this case, we only have
<code>Ins_tAUC_log</code>)</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Tests for sex, wave and diet_days.</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="va">pheno_clin</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">mouse</span>, <span class="va">sex</span>, <span class="va">DOwave</span>, <span class="va">diet_days</span>, <span class="va">Ins_tAUC_log</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">gather</span><span class="op">(</span><span class="va">phenotype</span>, <span class="va">value</span>, <span class="op">-</span><span class="va">mouse</span>, <span class="op">-</span><span class="va">sex</span>, <span class="op">-</span><span class="va">DOwave</span>, <span class="op">-</span><span class="va">diet_days</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">group_by</span><span class="op">(</span><span class="va">phenotype</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">nest</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<p>Let’s create a linear model function that will regress the covariates
and the phenotype.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod_fxn</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">lm</span><span class="op">(</span><span class="va">value</span> <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span> <span class="op">+</span> <span class="va">diet_days</span>, data <span class="op">=</span> <span class="va">df</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<p>Now let’s apply that function to the data object <code>tmp</code>
that we created above.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">=</span> <span class="va">tmp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="va">mod_fxn</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>summ <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">model</span>, <span class="va">tidy</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">unnest</span><span class="op">(</span><span class="va">summ</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'tmp' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'tmp' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">term</span> <span class="op">!=</span> <span class="st">"(Intercept)"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>neg.log.p <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">term</span>, <span class="va">neg.log.p</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">phenotype</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Significance of Sex, Wave &amp; Diet Days on Phenotypes"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'tmp' not found</code></pre>
</div>
<p>We can see that sex and DOwave (especially the third batch) are
significant. Here DOwave is the group or batch number as not all mice
are in the experiment at the same time. Because of this, we now have to
correct for it.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># convert sex and DO wave (batch) to factors</span></span>
<span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">sex</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covar</span> <span class="op">=</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span>, data <span class="op">=</span> <span class="va">pheno_clin</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<p><strong>REMEMBER:</strong> the sample IDs must be in the rownames of
<code>pheno</code>, <code>addcovar</code>, <code>genoprobs</code> and
<code>K</code>. <code>qtl2</code> uses the sample IDs to align the
samples between objects.</p>
</div>
<div class="section level3">
<h3 id="performing-a-genome-scan">
<a href="https://smcclatchy.github.io/qtl-mapping/perform-genome-scan/" class="external-link">Performing
a genome scan</a><a class="anchor" aria-label="anchor" href="#performing-a-genome-scan"></a>
</h3>
<p>At each marker on the genotyping array, we will fit a model that
regresses the phenotype (insulin secretion AUC) on covariates and the
founder allele proportions. Note that this model will give us an
estimate of the effect of each founder allele at each marker. There are
eight founder strains that contributed to the DO, so we will get eight
founder allele effects.</p>
<div class="section level4">
<h4 id="permutations">[Permutations]<a class="anchor" aria-label="anchor" href="#permutations"></a>
</h4>
<p>First, we need to work out the signifcance level. Let’s find the
significance level for 0.1, 0.05 and 0.01.</p>
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">operm</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno <span class="op">=</span> <span class="va">pheno_clin</span><span class="op">[</span>,<span class="st">"Ins_tAUC_log"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                   addcovar<span class="op">=</span><span class="va">covar</span>,</span>
<span>                   n_perm<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span></code></pre>
</div>
<p><em>Note</em> DO NOT RUN THIS (it will take too long). Instead, I
have run it earlier and will load it in here. We will also perform a
summary to find the summary level for 0.1, 0.05 and 0.01</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/operm_Ins_tAUC_log_1000.Rdata"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>,alpha<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.05</span>, <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="genome-scan">Genome Scan<a class="anchor" aria-label="anchor" href="#genome-scan"></a>
</h4>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl</span> <span class="op">=</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>            pheno <span class="op">=</span> <span class="va">pheno_clin</span><span class="op">[</span>,<span class="st">"Ins_tAUC_log"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>            kinship <span class="op">=</span> <span class="va">K</span>, </span>
<span>            addcovar <span class="op">=</span> <span class="va">covar</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we plot the genome scan.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">qtl</span>, </span>
<span>           map <span class="op">=</span> <span class="va">map</span>, </span>
<span>           lodcolumn <span class="op">=</span> <span class="st">"Ins_tAUC_log"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'qtl' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span>           <span class="fu">add_threshold</span><span class="op">(</span><span class="va">map</span>,  <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>,alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'purple'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in .rangeNum(..., na.rm = na.rm, finite = finite, isNumeric = is.numeric): argument is missing, with no default</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span>           <span class="fu">add_threshold</span><span class="op">(</span><span class="va">map</span>,  <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in .rangeNum(..., na.rm = na.rm, finite = finite, isNumeric = is.numeric): argument is missing, with no default</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span>           <span class="fu">add_threshold</span><span class="op">(</span><span class="va">map</span>,  <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'blue'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in .rangeNum(..., na.rm = na.rm, finite = finite, isNumeric = is.numeric): argument is missing, with no default</code></pre>
</div>
<p>We can see a very strong peak on chromosome 11 with no other
distibuishable peaks.</p>
</div>
</div>
<div class="section level3">
<h3 id="finding-lod-peaks">
<a href="https://smcclatchy.github.io/qtl-mapping/find-lod-peaks/" class="external-link">Finding
LOD peaks</a><a class="anchor" aria-label="anchor" href="#finding-lod-peaks"></a>
</h3>
<p>We can find all of the peaks above the significance threshold using
the <a href="https://github.com/rqtl/qtl2/blob/master/R/find_peaks.R" class="external-link">find_peaks</a>
function.</p>
<p>The support interval is determined using the <a href="http://www.ncbi.nlm.nih.gov/pubmed/11560912" class="external-link">Bayesian Credible
Interval</a> and represents the region most likely to contain the
causative polymorphism(s). We can obtain this interval by adding a
<code>prob</code> argument to <a href="https://github.com/rqtl/qtl2/blob/master/R/find_peaks.R" class="external-link">find_peaks</a>.
We pass in a value of <code>0.95</code> to request a support interval
that contains the causal SNP 95% of the time.</p>
<p>In case there are multiple peaks are found on a chromosome, the
<code>peakdrop</code> argument allows us to the find the peak which has
a certain LOD score drop between other peaks.</p>
<p>Let’s find LOD peaks. Here we are choosing to find peaks with a LOD
score greater than 6.</p>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_threshold</span> <span class="op">=</span> <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'operm' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">qtl</span>, map <span class="op">=</span> <span class="va">map</span>, </span>
<span>                   threshold <span class="op">=</span> <span class="va">lod_threshold</span>, </span>
<span>                   peakdrop <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                   prob <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'qtl' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb53">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kable</span><span class="op">(</span><span class="va">peaks</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span> <span class="op">(</span><span class="op">-</span><span class="va">lodindex</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">arrange</span><span class="op">(</span><span class="va">chr</span>, <span class="va">pos</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Phenotype QTL Peaks with LOD &gt;= 6"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'peaks' not found</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="qtl-effects">QTL effects<a class="anchor" aria-label="anchor" href="#qtl-effects"></a>
</h3>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">maxmarg</span><span class="op">(</span><span class="va">probs</span>, </span>
<span>             map<span class="op">=</span><span class="va">map</span>, </span>
<span>             chr<span class="op">=</span><span class="va">peaks</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>             pos<span class="op">=</span><span class="va">peaks</span><span class="op">$</span><span class="va">pos</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, </span>
<span>             minprob <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>             return_char<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_pxg</span><span class="op">(</span><span class="va">g</span>, </span>
<span>         <span class="va">pheno_clin</span><span class="op">[</span>,<span class="va">peaks</span><span class="op">$</span><span class="va">lodcolumn</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, </span>
<span>         ylab<span class="op">=</span><span class="va">peaks</span><span class="op">$</span><span class="va">lodcolumn</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, </span>
<span>         sort<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'g' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs<span class="op">=</span><span class="va">probs</span><span class="op">[</span>,<span class="va">peaks</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                  pheno<span class="op">=</span><span class="va">pheno_clin</span><span class="op">[</span>,<span class="va">peaks</span><span class="op">$</span><span class="va">lodcolumn</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, drop<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb61">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coefCC</span><span class="op">(</span><span class="va">blup</span>, </span>
<span>       map<span class="op">=</span><span class="va">map</span>, </span>
<span>       columns<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,</span>
<span>       bgcolor<span class="op">=</span><span class="st">"gray95"</span>, </span>
<span>       legend<span class="op">=</span><span class="st">"bottomleft"</span>,</span>
<span>       scan1_output <span class="op">=</span> <span class="va">qtl</span></span>
<span>       <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'blup' not found</code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1:</h3>
<div class="callout-content">
<p>Now choose another phenotype in <code>pheno_clin</code> and perform
the same steps.<br>
1). Check the distribution. Does it need transforming? If so, how would
you do it? 2). Are there any sex, batch, diet effects?<br>
3). Run a genome scan with the genotype probabilities and kinship
provided.<br>
4). Plot the genome scan for this phenotype.<br>
5). Find the peaks above LOD score of 6.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" tabindex="-1"></a><span class="co">#1).</span></span>
<span id="cb63-2"><a href="#cb63-2" tabindex="-1"></a></span>
<span id="cb63-3"><a href="#cb63-3" tabindex="-1"></a><span class="fu">hist</span>(pheno_clin<span class="sc">$</span><span class="er">&lt;</span>pheno name<span class="sc">&gt;</span>)</span>
<span id="cb63-4"><a href="#cb63-4" tabindex="-1"></a></span>
<span id="cb63-5"><a href="#cb63-5" tabindex="-1"></a>pheno_clin<span class="sc">$</span><span class="er">&lt;</span>pheno name<span class="sc">&gt;</span>_log <span class="ot">&lt;-</span> <span class="fu">log</span>(pheno_clin<span class="sc">$</span><span class="er">&lt;</span>pheno name<span class="sc">&gt;</span>)</span>
<span id="cb63-6"><a href="#cb63-6" tabindex="-1"></a></span>
<span id="cb63-7"><a href="#cb63-7" tabindex="-1"></a><span class="fu">hist</span>(pheno_clin<span class="sc">$</span><span class="er">&lt;</span>pheno name<span class="sc">&gt;</span>_log)</span>
<span id="cb63-8"><a href="#cb63-8" tabindex="-1"></a></span>
<span id="cb63-9"><a href="#cb63-9" tabindex="-1"></a></span>
<span id="cb63-10"><a href="#cb63-10" tabindex="-1"></a><span class="co">#2).</span></span>
<span id="cb63-11"><a href="#cb63-11" tabindex="-1"></a></span>
<span id="cb63-12"><a href="#cb63-12" tabindex="-1"></a>tmp <span class="ot">=</span> pheno_clin <span class="sc">%&gt;%</span></span>
<span id="cb63-13"><a href="#cb63-13" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(mouse, sex, DOwave, diet_days, <span class="sc">&lt;</span>pheno name<span class="sc">&gt;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-14"><a href="#cb63-14" tabindex="-1"></a>        <span class="fu">gather</span>(phenotype, value, <span class="sc">-</span>mouse, <span class="sc">-</span>sex, <span class="sc">-</span>DOwave, <span class="sc">-</span>diet_days) <span class="sc">%&gt;%</span></span>
<span id="cb63-15"><a href="#cb63-15" tabindex="-1"></a>        <span class="fu">group_by</span>(phenotype) <span class="sc">%&gt;%</span></span>
<span id="cb63-16"><a href="#cb63-16" tabindex="-1"></a>        <span class="fu">nest</span>()</span>
<span id="cb63-17"><a href="#cb63-17" tabindex="-1"></a>mod_fxn <span class="ot">=</span> <span class="cf">function</span>(df) {</span>
<span id="cb63-18"><a href="#cb63-18" tabindex="-1"></a>  <span class="fu">lm</span>(value <span class="sc">~</span> sex <span class="sc">+</span> DOwave <span class="sc">+</span> diet_days, <span class="at">data =</span> df)</span>
<span id="cb63-19"><a href="#cb63-19" tabindex="-1"></a>}</span>
<span id="cb63-20"><a href="#cb63-20" tabindex="-1"></a>tmp <span class="ot">=</span> tmp <span class="sc">%&gt;%</span></span>
<span id="cb63-21"><a href="#cb63-21" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model =</span> <span class="fu">map</span>(data, mod_fxn)) <span class="sc">%&gt;%</span></span>
<span id="cb63-22"><a href="#cb63-22" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">summ =</span> <span class="fu">map</span>(model, tidy)) <span class="sc">%&gt;%</span></span>
<span id="cb63-23"><a href="#cb63-23" tabindex="-1"></a>  <span class="fu">unnest</span>(summ) </span>
<span id="cb63-24"><a href="#cb63-24" tabindex="-1"></a></span>
<span id="cb63-25"><a href="#cb63-25" tabindex="-1"></a>tmp</span>
<span id="cb63-26"><a href="#cb63-26" tabindex="-1"></a></span>
<span id="cb63-27"><a href="#cb63-27" tabindex="-1"></a>tmp <span class="sc">%&gt;%</span></span>
<span id="cb63-28"><a href="#cb63-28" tabindex="-1"></a>  <span class="fu">filter</span>(term <span class="sc">!=</span> <span class="st">"(Intercept)"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-29"><a href="#cb63-29" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neg.log.p =</span> <span class="sc">-</span><span class="fu">log10</span>(p.value)) <span class="sc">%&gt;%</span></span>
<span id="cb63-30"><a href="#cb63-30" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(term, neg.log.p)) <span class="sc">+</span></span>
<span id="cb63-31"><a href="#cb63-31" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb63-32"><a href="#cb63-32" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>phenotype) <span class="sc">+</span></span>
<span id="cb63-33"><a href="#cb63-33" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Significance of Sex, Wave &amp; Diet Days on Phenotypes"</span>) <span class="sc">+</span></span>
<span id="cb63-34"><a href="#cb63-34" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb63-35"><a href="#cb63-35" tabindex="-1"></a><span class="fu">rm</span>(tmp)</span>
<span id="cb63-36"><a href="#cb63-36" tabindex="-1"></a></span>
<span id="cb63-37"><a href="#cb63-37" tabindex="-1"></a><span class="co">#3).</span></span>
<span id="cb63-38"><a href="#cb63-38" tabindex="-1"></a></span>
<span id="cb63-39"><a href="#cb63-39" tabindex="-1"></a>qtl <span class="ot">=</span> <span class="fu">scan1</span>(<span class="at">genoprobs =</span> probs, </span>
<span id="cb63-40"><a href="#cb63-40" tabindex="-1"></a>           <span class="at">pheno =</span> pheno_clin[,<span class="st">"&lt;pheno name&gt;"</span>, <span class="at">drop =</span> <span class="cn">FALSE</span>], </span>
<span id="cb63-41"><a href="#cb63-41" tabindex="-1"></a>        <span class="at">kinship =</span> K, </span>
<span id="cb63-42"><a href="#cb63-42" tabindex="-1"></a>        <span class="at">addcovar =</span> covar)</span>
<span id="cb63-43"><a href="#cb63-43" tabindex="-1"></a></span>
<span id="cb63-44"><a href="#cb63-44" tabindex="-1"></a><span class="co">#4).</span></span>
<span id="cb63-45"><a href="#cb63-45" tabindex="-1"></a></span>
<span id="cb63-46"><a href="#cb63-46" tabindex="-1"></a><span class="fu">plot_scan1</span>(<span class="at">x =</span> qtl, <span class="at">map =</span> map, <span class="at">lodcolumn =</span> <span class="st">"&lt;pheno name&gt;"</span>)</span>
<span id="cb63-47"><a href="#cb63-47" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">6</span>, <span class="at">col =</span> <span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb63-48"><a href="#cb63-48" tabindex="-1"></a></span>
<span id="cb63-49"><a href="#cb63-49" tabindex="-1"></a><span class="co">#5). </span></span>
<span id="cb63-50"><a href="#cb63-50" tabindex="-1"></a></span>
<span id="cb63-51"><a href="#cb63-51" tabindex="-1"></a>peaks <span class="ot">=</span> <span class="fu">find_peaks</span>(<span class="at">scan1_output =</span> qtl, <span class="at">map =</span> map, </span>
<span id="cb63-52"><a href="#cb63-52" tabindex="-1"></a>               <span class="at">threshold =</span> lod_threshold, </span>
<span id="cb63-53"><a href="#cb63-53" tabindex="-1"></a>               <span class="at">peakdrop =</span> <span class="dv">4</span>, </span>
<span id="cb63-54"><a href="#cb63-54" tabindex="-1"></a>               <span class="at">prob =</span> <span class="fl">0.95</span>)</span></code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div></section><section id="aio-map-one-eqtl"><p>Content from <a href="map-one-eqtl.html">Mapping A Single Gene Expression Trait</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/map-one-eqtl.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I map one gene expression trait?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To run a QTL analysis for expression data</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>In this tutorial, we are going to use the gene expression data as our
phenotype to map QTLs. That is, we are looking to find any QTLs (cis or
trans) that are responsible for variation in gene expression across
samples.</p>
<p>We are using the same directory as our previous tutorial, which you
can check by typing <code>getwd()</code> in the Console. If you are not
in the correct directory, type in <code>setwd("code")</code> in the
console or Session -&gt; Set Working Directory -&gt; Choose Directory in
the RStudio menu to set your working directory to the main
directory.</p>
<p>If you have started a new R session, you will need to load the
libraries again. If you haven’t, the libraries listed below are already
loaded.</p>
<div class="section level3">
<h3 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h3>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">broom</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>In this lesson, we are loading in gene expression data for
<code>21,771</code> genes: <code>attie_DO500_expr.datasets.RData</code>
that includes normalised and raw gene expression data.
<code>dataset.islet.rnaseq</code> is the dataset you can download
directly from Dryad. Again, if you are using the same R session, you
will not need to the load the mapping, phenotypes and genotype
probabilities data, again.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># expression data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_expr.datasets.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_expr.datasets.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># data from paper</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/dataset.islet.rnaseq.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/dataset.islet.rnaseq.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># phenotypes</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_clinical.phenotypes.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_clinical.phenotypes.RData', probable reason 'No such file
or directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># mapping data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_mapping.data.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_mapping.data.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># genotype probabilities</span></span>
<span><span class="va">probs</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">"../data/attie_DO500_genoprobs_v5.rds"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in gzfile(file, "rb"): cannot open compressed file
'../data/attie_DO500_genoprobs_v5.rds', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in gzfile(file, "rb"): cannot open the connection</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="expression-data">Expression Data<a class="anchor" aria-label="anchor" href="#expression-data"></a>
</h3>
<p>Raw gene expression counts are in the <code>counts</code> data
object. These counts have been normalised and saved in the
<code>norm</code> data object. More information is about normalisation
is <a href="https://whoatemycookie.github.io/gene-expression-qtl/03-load-explore-data/index.html#gene-expression-phenotypes" class="external-link">here</a>.</p>
<p>To view the contents of either one of these data , click the table on
the right hand side either <code>norm</code> or <code>counts</code> in
the Environment tab. If you type in <code>names(counts)</code>, you will
see the names all start with <code>ENSMUSG</code>. These are Ensembl
IDs. If we want to see which gene these IDs correspond to, type in
<code>dataset.islet.rnaseq$annots</code>, which gives information about
each gene, including ensemble id, gene symbol as well as start &amp;
stop location of the gene and chromsome on which the gene lies.</p>
<p>Because we are working with the <code>insulin tAUC</code> phenotype,
let’s map the expression counts for <code>Hnf1b</code> which is known to
influence this phenotype is these data. First, we need to find the
Ensembl ID for this gene:</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">annots</span><span class="op">[</span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">annots</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Hnf1b"</span>,<span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<p>We can see that the ensembl ID of <code>Hnf1b</code> is
<code>ENSMUSG00000020679</code>. If we check the distribution for
<code>Hnf1b</code> expression data between the raw and normalised data,
we can see there distribution has been corrected. Here is the
distribution of the raw counts:</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">counts</span><span class="op">$</span><span class="va">ENSMUSG00000020679</span>, main <span class="op">=</span> <span class="st">"Hnf1b"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'counts' not found</code></pre>
</div>
<p>and here is the distribution of the normalised counts:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">norm</span><span class="op">$</span><span class="va">ENSMUSG00000020679</span>, main <span class="op">=</span> <span class="st">"Hnf1b"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in norm$ENSMUSG00000020679: object of type 'closure' is not subsettable</code></pre>
</div>
<p>The histogram indicates that distribution of these counts are
normalised.</p>
</div>
<div class="section level3">
<h3 id="the-marker-map">The Marker Map<a class="anchor" aria-label="anchor" href="#the-marker-map"></a>
</h3>
<p>We are using the same marker map as in the previous <a href="https://smcclatchy.github.io/eqtl-mapping/review-mapping-steps/index.html#the-marker-map" class="external-link">lesson</a></p>
</div>
<div class="section level3">
<h3 id="genotype-probabilities">Genotype probabilities<a class="anchor" aria-label="anchor" href="#genotype-probabilities"></a>
</h3>
<p>We have explored this earlier in th previous <a href="https://smcclatchy.github.io/eqtl-mapping/review-mapping-steps/index.html#genotype-probabilities" class="external-link">lesson</a>.
But, as a reminder, we have already calculated genotype probabilities
which we loaded above called <code>probs</code>. This contains the 8
state genotype probabilities using the 69k grid map of the same 500 DO
mice that also have clinical phenotypes.</p>
</div>
<div class="section level3">
<h3 id="kinship-matrix">
<a href="https://smcclatchy.github.io/qtl-mapping/calc-kinship/" class="external-link">Kinship
Matrix</a><a class="anchor" aria-label="anchor" href="#kinship-matrix"></a>
</h3>
<p>We have explored the kinship matrix in the previous <a href="https://smcclatchy.github.io/eqtl-mapping/review-mapping-steps/index.html#kinship-matrix" class="external-link">lesson</a>.
It has already been calculated and loaded in above.</p>
</div>
<div class="section level3">
<h3 id="covariates">Covariates<a class="anchor" aria-label="anchor" href="#covariates"></a>
</h3>
<p>Now let’s add the necessary covariates. For <code>Hnf1b</code>
expression data, let’s see which covariates are significant.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">###merging covariate data and expression data to test for sex, wave and diet_days.</span></span>
<span></span>
<span><span class="va">cov.counts</span> <span class="op">&lt;-</span> <span class="fu">merge</span><span class="op">(</span><span class="va">covar</span>, <span class="va">norm</span>, by<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="st">"row.names"</span><span class="op">)</span>, sort<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'covar' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#testing covairates on expression data</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="va">cov.counts</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">mouse</span>, <span class="va">sex</span>, <span class="va">DOwave</span>, <span class="va">diet_days</span>, <span class="va">ENSMUSG00000020679</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">gather</span><span class="op">(</span><span class="va">expression</span>, <span class="va">value</span>, <span class="op">-</span><span class="va">mouse</span>, <span class="op">-</span><span class="va">sex</span>, <span class="op">-</span><span class="va">DOwave</span>, <span class="op">-</span><span class="va">diet_days</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">group_by</span><span class="op">(</span><span class="va">expression</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">nest</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'cov.counts' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod_fxn</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">lm</span><span class="op">(</span><span class="va">value</span> <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span> <span class="op">+</span> <span class="va">diet_days</span>, data <span class="op">=</span> <span class="va">df</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="va">tmp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="va">mod_fxn</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>summ <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">model</span>, <span class="va">tidy</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">unnest</span><span class="op">(</span><span class="va">summ</span><span class="op">)</span> </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'tmp' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#  kable(tmp, caption = "Effects of Sex, Wave &amp; Diet Days on Expression")</span></span>
<span></span>
<span><span class="va">tmp</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'tmp' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">term</span> <span class="op">!=</span> <span class="st">"(Intercept)"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>neg.log.p <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">term</span>, <span class="va">neg.log.p</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">expression</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Significance of Sex, Wave &amp; Diet Days on Expression"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'tmp' not found</code></pre>
</div>
<p>We can see that sex and DOwave are significant. Here DOwave is the
group or batch number as not all mice were submitted for genotyping at
the same time. Because of this, we now have to correct for it.
Considering the paper included the covariate, <code>diet_days</code>, we
will include that as well.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># convert sex and DO wave (batch) to factors</span></span>
<span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">sex</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">diet_days</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covar</span> <span class="op">=</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span> <span class="op">+</span> <span class="va">diet_days</span>, data <span class="op">=</span> <span class="va">pheno_clin</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="performing-a-genome-scan">
<a href="https://smcclatchy.github.io/qtl-mapping/perform-genome-scan/" class="external-link">Performing
a genome scan</a><a class="anchor" aria-label="anchor" href="#performing-a-genome-scan"></a>
</h3>
<div class="section level4">
<h4 id="permutations">[Permutations]<a class="anchor" aria-label="anchor" href="#permutations"></a>
</h4>
<p>First, we need to work out the signifcance level. Let’s find the
signifance level for 0.1, 0.05 and 0.01.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">operm</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno <span class="op">=</span> <span class="va">norm</span><span class="op">[</span>,<span class="st">"ENSMUSG00000020679"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   addcovar<span class="op">=</span><span class="va">covar</span>,</span>
<span>                   n_perm<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span></code></pre>
</div>
<p><em>Note</em> DO NOT RUN THIS (it will take too long). Instead, I
have run it earlier and will load it in here. We will also perform a
summary to find the summary level for 0.1, 0.05 and 0.01</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/operm_ENSMUSG00000020679_1000.Rdata"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>,alpha<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.05</span>, <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="genome-scan">Genome Scan<a class="anchor" aria-label="anchor" href="#genome-scan"></a>
</h4>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl</span> <span class="op">=</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>            pheno <span class="op">=</span> <span class="va">norm</span><span class="op">[</span>,<span class="st">"ENSMUSG00000020679"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>            kinship <span class="op">=</span> <span class="va">K</span>, </span>
<span>            addcovar <span class="op">=</span> <span class="va">covar</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we plot the genome scan.</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">qtl</span>, </span>
<span>           map <span class="op">=</span> <span class="va">map</span>, </span>
<span>           lodcolumn <span class="op">=</span> <span class="st">"ENSMUSG00000020679"</span>,</span>
<span>           main <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">qtl</span><span class="op">)</span><span class="op">)</span></span>
<span>           <span class="fu">add_threshold</span><span class="op">(</span><span class="va">map</span>,  <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'purple'</span><span class="op">)</span></span>
<span>           <span class="fu">add_threshold</span><span class="op">(</span><span class="va">map</span>,  <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span>
<span>           <span class="fu">add_threshold</span><span class="op">(</span><span class="va">map</span>,  <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'blue'</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
<div class="section level3">
<h3 id="finding-lod-peaks">
<a href="https://smcclatchy.github.io/qtl-mapping/find-lod-peaks/" class="external-link">Finding
LOD peaks</a><a class="anchor" aria-label="anchor" href="#finding-lod-peaks"></a>
</h3>
<p>Let’s find LOD peaks</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_threshold</span> <span class="op">=</span> <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">peaks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">qtl</span>, </span>
<span>                   map <span class="op">=</span> <span class="va">map</span>, </span>
<span>                   threshold <span class="op">=</span> <span class="va">lod_threshold</span>, </span>
<span>                   peakdrop <span class="op">=</span> <span class="fl">4</span>, prob <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">peaks</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">lodindex</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">arrange</span><span class="op">(</span><span class="va">chr</span>, <span class="va">pos</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Phenotype QTL Peaks with LOD &gt;= 6"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="qtl-effects">QTL effects<a class="anchor" aria-label="anchor" href="#qtl-effects"></a>
</h3>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs<span class="op">=</span><span class="va">probs</span><span class="op">[</span>,<span class="va">peaks</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                  <span class="va">norm</span><span class="op">[</span>,<span class="va">peaks</span><span class="op">$</span><span class="va">lodcolumn</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, drop<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_coefCC</span><span class="op">(</span><span class="va">blup</span>, </span>
<span>       map<span class="op">=</span><span class="va">map</span>, </span>
<span>       columns<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,</span>
<span>       bgcolor<span class="op">=</span><span class="st">"gray95"</span>, </span>
<span>       legend<span class="op">=</span><span class="st">"bottomleft"</span>,</span>
<span>       scan1_output <span class="op">=</span> <span class="va">qtl</span> <span class="op">)</span></span></code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1:</h3>
<div class="callout-content">
<p>Now choose another gene expression trait in <code>norm</code> data
object and perform the same steps.<br>
1). Check the distribution of the raw counts and normalised counts 2).
Are there any sex, batch, diet effects?<br>
3). Run a genome scan with the genotype probabilities and kinship
provided.<br>
4). Plot the genome scan for this gene.<br>
5). Find the peaks above LOD score of 6.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>Replace <code>&lt;ensembl id&gt;</code> with your choice of gene
expression trait</p>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a><span class="co">#1).</span></span>
<span id="cb47-2"><a href="#cb47-2" tabindex="-1"></a></span>
<span id="cb47-3"><a href="#cb47-3" tabindex="-1"></a><span class="fu">hist</span>(pheno_clin<span class="sc">$</span><span class="er">&lt;</span>ensembl id<span class="sc">&gt;</span>)</span>
<span id="cb47-4"><a href="#cb47-4" tabindex="-1"></a></span>
<span id="cb47-5"><a href="#cb47-5" tabindex="-1"></a>pheno_clin<span class="sc">$</span><span class="er">&lt;</span>ensembl id<span class="sc">&gt;</span>_log <span class="ot">&lt;-</span> <span class="fu">log</span>(pheno_clin<span class="sc">$</span><span class="er">&lt;</span>ensembl id<span class="sc">&gt;</span>)</span>
<span id="cb47-6"><a href="#cb47-6" tabindex="-1"></a></span>
<span id="cb47-7"><a href="#cb47-7" tabindex="-1"></a><span class="fu">hist</span>(pheno_clin<span class="sc">$</span><span class="er">&lt;</span>ensembl id<span class="sc">&gt;</span>_log)</span>
<span id="cb47-8"><a href="#cb47-8" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" tabindex="-1"></a></span>
<span id="cb47-10"><a href="#cb47-10" tabindex="-1"></a><span class="co">#2).</span></span>
<span id="cb47-11"><a href="#cb47-11" tabindex="-1"></a></span>
<span id="cb47-12"><a href="#cb47-12" tabindex="-1"></a>tmp <span class="ot">=</span> pheno_clin <span class="sc">%&gt;%</span></span>
<span id="cb47-13"><a href="#cb47-13" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(mouse, sex, DOwave, diet_days, <span class="sc">&lt;</span>ensembl id<span class="sc">&gt;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-14"><a href="#cb47-14" tabindex="-1"></a>        <span class="fu">gather</span>(expression, value, <span class="sc">-</span>mouse, <span class="sc">-</span>sex, <span class="sc">-</span>DOwave, <span class="sc">-</span>diet_days) <span class="sc">%&gt;%</span></span>
<span id="cb47-15"><a href="#cb47-15" tabindex="-1"></a>        <span class="fu">group_by</span>(expression) <span class="sc">%&gt;%</span></span>
<span id="cb47-16"><a href="#cb47-16" tabindex="-1"></a>        <span class="fu">nest</span>()</span>
<span id="cb47-17"><a href="#cb47-17" tabindex="-1"></a>mod_fxn <span class="ot">=</span> <span class="cf">function</span>(df) {</span>
<span id="cb47-18"><a href="#cb47-18" tabindex="-1"></a>  <span class="fu">lm</span>(value <span class="sc">~</span> sex <span class="sc">+</span> DOwave <span class="sc">+</span> diet_days, <span class="at">data =</span> df)</span>
<span id="cb47-19"><a href="#cb47-19" tabindex="-1"></a>}</span>
<span id="cb47-20"><a href="#cb47-20" tabindex="-1"></a>tmp <span class="ot">=</span> tmp <span class="sc">%&gt;%</span></span>
<span id="cb47-21"><a href="#cb47-21" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model =</span> <span class="fu">map</span>(data, mod_fxn)) <span class="sc">%&gt;%</span></span>
<span id="cb47-22"><a href="#cb47-22" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">summ =</span> <span class="fu">map</span>(model, tidy)) <span class="sc">%&gt;%</span></span>
<span id="cb47-23"><a href="#cb47-23" tabindex="-1"></a>  <span class="fu">unnest</span>(summ) </span>
<span id="cb47-24"><a href="#cb47-24" tabindex="-1"></a></span>
<span id="cb47-25"><a href="#cb47-25" tabindex="-1"></a>tmp</span>
<span id="cb47-26"><a href="#cb47-26" tabindex="-1"></a></span>
<span id="cb47-27"><a href="#cb47-27" tabindex="-1"></a>tmp <span class="sc">%&gt;%</span></span>
<span id="cb47-28"><a href="#cb47-28" tabindex="-1"></a>  <span class="fu">filter</span>(term <span class="sc">!=</span> <span class="st">"(Intercept)"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-29"><a href="#cb47-29" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neg.log.p =</span> <span class="sc">-</span><span class="fu">log10</span>(p.value)) <span class="sc">%&gt;%</span></span>
<span id="cb47-30"><a href="#cb47-30" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(term, neg.log.p)) <span class="sc">+</span></span>
<span id="cb47-31"><a href="#cb47-31" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb47-32"><a href="#cb47-32" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>expression) <span class="sc">+</span></span>
<span id="cb47-33"><a href="#cb47-33" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Significance of Sex, Wave &amp; Diet Days on Phenotypes"</span>) <span class="sc">+</span></span>
<span id="cb47-34"><a href="#cb47-34" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb47-35"><a href="#cb47-35" tabindex="-1"></a><span class="fu">rm</span>(tmp)</span>
<span id="cb47-36"><a href="#cb47-36" tabindex="-1"></a></span>
<span id="cb47-37"><a href="#cb47-37" tabindex="-1"></a><span class="co">#3).</span></span>
<span id="cb47-38"><a href="#cb47-38" tabindex="-1"></a></span>
<span id="cb47-39"><a href="#cb47-39" tabindex="-1"></a>qtl <span class="ot">=</span> <span class="fu">scan1</span>(<span class="at">genoprobs =</span> probs, </span>
<span id="cb47-40"><a href="#cb47-40" tabindex="-1"></a>           <span class="at">pheno =</span> pheno_clin[,<span class="st">"&lt;ensembl id&gt;"</span>, <span class="at">drop =</span> <span class="cn">FALSE</span>], </span>
<span id="cb47-41"><a href="#cb47-41" tabindex="-1"></a>        <span class="at">kinship =</span> K, </span>
<span id="cb47-42"><a href="#cb47-42" tabindex="-1"></a>        <span class="at">addcovar =</span> covar)</span>
<span id="cb47-43"><a href="#cb47-43" tabindex="-1"></a></span>
<span id="cb47-44"><a href="#cb47-44" tabindex="-1"></a><span class="co">#4).</span></span>
<span id="cb47-45"><a href="#cb47-45" tabindex="-1"></a></span>
<span id="cb47-46"><a href="#cb47-46" tabindex="-1"></a><span class="fu">plot_scan1</span>(<span class="at">x =</span> qtl, <span class="at">map =</span> map, <span class="at">lodcolumn =</span> <span class="st">"&lt;ensembl id&gt;"</span>)</span>
<span id="cb47-47"><a href="#cb47-47" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">6</span>, <span class="at">col =</span> <span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb47-48"><a href="#cb47-48" tabindex="-1"></a></span>
<span id="cb47-49"><a href="#cb47-49" tabindex="-1"></a><span class="co">#5). </span></span>
<span id="cb47-50"><a href="#cb47-50" tabindex="-1"></a></span>
<span id="cb47-51"><a href="#cb47-51" tabindex="-1"></a>peaks <span class="ot">=</span> <span class="fu">find_peaks</span>(<span class="at">scan1_output =</span> qtl, <span class="at">map =</span> map, </span>
<span id="cb47-52"><a href="#cb47-52" tabindex="-1"></a>               <span class="at">threshold =</span> lod_threshold, </span>
<span id="cb47-53"><a href="#cb47-53" tabindex="-1"></a>               <span class="at">peakdrop =</span> <span class="dv">4</span>, </span>
<span id="cb47-54"><a href="#cb47-54" tabindex="-1"></a>               <span class="at">prob =</span> <span class="fl">0.95</span>)</span></code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div></section><section id="aio-map-many-eqtls"><p>Content from <a href="map-many-eqtls.html">Mapping Many Gene Expression Traits</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/map-many-eqtls.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I map many genes?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To map several genes at the same time</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h3>
<p>For this lesson, we need to install two more libraries and load them.
You can do this by typing the following code:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="fu">c</span><span class="op">(</span><span class="st">"AnnotationHub"</span>,<span class="st">"rtracklayer"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s install our libraries, and source two other R scripts.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">broom</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2ggplot</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#source("../code/gg_transcriptome_map.R")</span></span>
<span><span class="co">#source("../code/qtl_heatmap.R")</span></span></code></pre>
</div>
<p>Before we begin this lesson, we need to create another directory
called <code>results</code> in our main directory. You can do this by
clicking on the “Files” tab and navigate into the main directory. Then
select “New Folder” and name it “results”.</p>
</div>
<div class="section level3">
<h3 id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># expression data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_expr.datasets.RData"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># data from paper</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/dataset.islet.rnaseq.RData"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># phenotypes</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_clinical.phenotypes.RData"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># mapping data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_mapping.data.RData"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># genotype probabilities</span></span>
<span><span class="va">probs</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">"../data/attie_DO500_genoprobs_v5.rds"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="data-selection">Data Selection<a class="anchor" aria-label="anchor" href="#data-selection"></a>
</h3>
<p>For this lesson, lets choose a random set of 50 gene expression
phenotypes.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">norm</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sams</span> <span class="op">&lt;-</span> <span class="fu">sample</span><span class="op">(</span><span class="fu">length</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span>, <span class="fl">50</span>, replace <span class="op">=</span> <span class="cn">FALSE</span>, prob <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in sample.int(length(x), size, replace, prob): cannot take a sample larger than the population when 'replace = FALSE'</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">sams</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'sams' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene.info</span> <span class="op">&lt;-</span> <span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">annots</span><span class="op">[</span><span class="va">genes</span>,<span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rownames</span><span class="op">(</span><span class="va">gene.info</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NULL</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'gene.info' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kable</span><span class="op">(</span><span class="va">gene.info</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'gene.info' not found</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="expression-data">Expression Data<a class="anchor" aria-label="anchor" href="#expression-data"></a>
</h3>
<p>Lets check the distribution for the first 20 gene expression
phenotypes. If you would like to check the distribution of all 50 genes,
change <code>for(gene in genes[1:20])</code> in the code below to
<code>for(gene in genes)</code>.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">par</span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">gene</span> <span class="kw">in</span> <span class="va">genes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">hist</span><span class="op">(</span><span class="va">norm</span><span class="op">[</span>,<span class="va">gene</span><span class="op">]</span>, main <span class="op">=</span> <span class="va">gene</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span></code></pre>
</div>
<p>Check the distributions. Do they all have a normal distribution?</p>
<p>You will notice that the distribtion of some genes are skewed to the
left. This means that that only a small amount of samples have data and
therefore, will need to be removed. A suitable qc would be keeping
expression data that have at least 5% of the samples with more than 10
reads.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes_qc</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">as.numeric</span><span class="op">(</span><span class="fu">colSums</span><span class="op">(</span><span class="va">counts</span><span class="op">[</span> , <span class="va">genes</span><span class="op">]</span> <span class="op">&gt;</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">0.05</span> <span class="op">*</span> <span class="fu">nrow</span><span class="op">(</span><span class="va">counts</span><span class="op">[</span>,<span class="va">genes</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'counts' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes_qc</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'genes_qc' not found</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="the-marker-map">The Marker Map<a class="anchor" aria-label="anchor" href="#the-marker-map"></a>
</h3>
<p>We are using the same marker map as in the previous <a href="https://smcclatchy.github.io/eqtl-mapping/review-mapping-steps/index.html#the-marker-map" class="external-link">lesson</a></p>
</div>
<div class="section level3">
<h3 id="genotype-probabilities">Genotype probabilities<a class="anchor" aria-label="anchor" href="#genotype-probabilities"></a>
</h3>
<p>We have explored this earlier in th previous <a href="https://smcclatchy.github.io/eqtl-mapping/review-mapping-steps/index.html#genotype-probabilities" class="external-link">lesson</a>.
But, as a reminder, we have already calculated genotype probabilities
which we loaded above called <code>probs</code>. This contains the 8
state g enotype probabilities using the 69k grid map of the same 500 DO
mice that also have clinical phenotypes.</p>
</div>
<div class="section level3">
<h3 id="kinship-matrix">
<a href="https://smcclatchy.github.io/mapping/04-calc-kinship/" class="external-link">Kinship
Matrix</a><a class="anchor" aria-label="anchor" href="#kinship-matrix"></a>
</h3>
<p>We have explored the kinship matrix in the previous <a href="https://smcclatchy.github.io/eqtl-mapping/review-mapping-steps/index.html#kinship-matrix" class="external-link">lesson</a>.
It has already been calculated and loaded in above.</p>
</div>
<div class="section level3">
<h3 id="covariates">Covariates<a class="anchor" aria-label="anchor" href="#covariates"></a>
</h3>
<p>Now let’s add the necessary covariates. For these 50 gene expression
data, we will correct for <code>DOwave</code>,<code>sex</code> and
<code>diet_days</code>.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># convert sex and DO wave (batch) to factors</span></span>
<span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">sex</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">diet_days</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covar</span> <span class="op">=</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span> <span class="op">+</span> <span class="va">diet_days</span>, data <span class="op">=</span> <span class="va">pheno_clin</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="performing-a-genome-scan">
<a href="https://smcclatchy.github.io/mapping/06-perform-genome-scan/" class="external-link">Performing
a genome scan</a><a class="anchor" aria-label="anchor" href="#performing-a-genome-scan"></a>
</h3>
<p>Now lets perform the genome scan! We are also going to save our qtl
results in an <code>Rdata</code> file to be used in further lessons. We
will not perform permutations in this lesson as it will take too long.
Instead we will use 6, which is the LOD score used in the paper to
determine significance.</p>
</div>
<div class="section level3">
<h3 id="qtl-scans">QTL Scans<a class="anchor" aria-label="anchor" href="#qtl-scans"></a>
</h3>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl.file</span> <span class="op">=</span> <span class="st">"../results/gene.norm_qtl_cis.trans.Rdata"</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu">file.exists</span><span class="op">(</span><span class="va">qtl.file</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">load</span><span class="op">(</span><span class="va">qtl.file</span><span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="va">qtl</span> <span class="op">=</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                pheno <span class="op">=</span> <span class="va">norm</span><span class="op">[</span>,<span class="va">genes</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                kinship <span class="op">=</span> <span class="va">K</span>, </span>
<span>                addcovar <span class="op">=</span> <span class="va">covar</span>, </span>
<span>                cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span>    <span class="fu">save</span><span class="op">(</span><span class="va">qtl</span>, file <span class="op">=</span> <span class="va">qtl.file</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="qtl-plots">QTL plots<a class="anchor" aria-label="anchor" href="#qtl-plots"></a>
</h3>
<p>Let’s plot the first 20 gene expression phenotypes. If you would like
to plot all 50, change <code>for(i in 1:20)</code> in the code below to
<code>for(i in 1:ncol(qtl))</code>.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">par</span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">qtl</span>, </span>
<span>             map <span class="op">=</span> <span class="va">map</span>, </span>
<span>             lodcolumn <span class="op">=</span> <span class="va">i</span>, </span>
<span>             main <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">qtl</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="fu">abline</span><span class="op">(</span>h <span class="op">=</span> <span class="fl">6</span>, col <span class="op">=</span> <span class="fl">2</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'qtl' not found</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="qtl-peaks">QTL Peaks<a class="anchor" aria-label="anchor" href="#qtl-peaks"></a>
</h3>
<p>We are also going to save our peak results so we can use these again
else where.<br>
First, lets get out peaks with a LOD score greater than 6.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_threshold</span> <span class="op">=</span> <span class="fl">6</span></span>
<span><span class="va">peaks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">qtl</span>, </span>
<span>                   map <span class="op">=</span> <span class="va">map</span>, </span>
<span>                   threshold <span class="op">=</span> <span class="va">lod_threshold</span>, </span>
<span>                   peakdrop <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                   prob <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'qtl' not found</code></pre>
</div>
<p>We will save these peaks into a csv file.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kable</span><span class="op">(</span><span class="va">peaks</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">lodindex</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">arrange</span><span class="op">(</span><span class="va">chr</span>, <span class="va">pos</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Expression QTL (eQTL) Peaks with LOD &gt;= 6"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># write_csv(peaks, "../results/gene.norm_qtl_peaks_cis.trans.csv")</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'peaks' not found</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="qtl-peaks-figure">QTL Peaks Figure<a class="anchor" aria-label="anchor" href="#qtl-peaks-figure"></a>
</h3>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">qtl_heatmap</span><span class="op">(</span>qtl <span class="op">=</span> <span class="va">qtl</span>, map <span class="op">=</span> <span class="va">map</span>, low.thr <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in qtl_heatmap(qtl = qtl, map = map, low.thr = 3.5): could not find function "qtl_heatmap"</code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1:</h3>
<div class="callout-content">
<p>What do the qtl scans for all gene exression traits look like?
<em>Note:</em> Don’t worry, we’ve done the qtl scans for you!!! You can
read in this file, <code>../data/gene.norm_qtl_all.genes.Rdata</code>,
which are the <code>scan1</code> results for all gene expression
traits.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/gene.norm_qtl_all.genes.Rdata"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lod_threshold</span> <span class="op">=</span> <span class="fl">6</span></span>
<span><span class="va">peaks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">qtl.all</span>, </span>
<span>                map <span class="op">=</span> <span class="va">map</span>, </span>
<span>                threshold <span class="op">=</span> <span class="va">lod_threshold</span>, </span>
<span>                peakdrop <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                prob <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span><span class="fu">write_csv</span><span class="op">(</span><span class="va">peaks</span>, <span class="st">"../results/gene.norm_qtl_all.genes_peaks.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Heat Map</span></span>
<span><span class="fu">qtl_heatmap</span><span class="op">(</span>qtl <span class="op">=</span> <span class="va">qtl</span>, map <span class="op">=</span> <span class="va">map</span>, low.thr <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div></section><section id="aio-create-transcriptome-map"><p>Content from <a href="create-transcriptome-map.html">Creating A Transcriptome Map</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/create-transcriptome-map.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I create and interpret a transcriptome map?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe a transcriptome map.</li>
<li>Interpret a transcriptome map.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h3>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2convert</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2ggplot</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># source("../code/gg_transcriptome_map.R")</span></span></code></pre>
</div>
</div>
<section><h2 class="section-heading" id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<hr class="half-width">
<p>Load in the LOD peaks over 6 from previous lesson.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># REad in the LOD peaks from the previous lesson.</span></span>
<span><span class="va">lod_summary</span> <span class="op">&lt;-</span> <span class="fu">read.csv</span><span class="op">(</span><span class="st">"../results/gene.norm_qtl_peaks_cis.trans.csv"</span><span class="op">)</span></span></code></pre>
</div>
<p>In order to use the <code>ggtmap</code> function, we need to provide
specific column names. These are documented in the
<code>gg_transcriptome_map.R</code> file in the code directory of this
workshop. The required column names are:</p>
<ul>
<li>
<code>data</code>: data.frame (or tibble) with the following
columns:
<ul>
<li>
<code>ensembl</code>: (required) character string containing the
Ensembl gene ID.</li>
<li>
<code>qtl_chr</code>: (required) character string containing QTL
chromsome.</li>
<li>
<code>qtl_pos</code>: (required) floating point number containing
the QTL position in Mb.</li>
<li>
<code>qtl_lod</code>: (optional) floating point number containing
the LOD score.</li>
<li>
<code>gene_chr</code>: (optional) character string containing
transcript chromosome.</li>
<li>
<code>gene_start</code>: (optional) character string containing
transcript start position in Mb.</li>
<li>
<code>gene_end</code>: (optional) character string containing
transcript end position in Mb.</li>
</ul>
</li>
</ul>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get gene positions.</span></span>
<span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu">get_ensembl_genes</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in get_ensembl_genes(): could not find function "get_ensembl_genes"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>ensembl    <span class="op">=</span> <span class="va">ensembl</span><span class="op">$</span><span class="va">gene_id</span>, </span>
<span>                 gene_chr   <span class="op">=</span> <span class="fu">seqnames</span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span>, </span>
<span>                 gene_start <span class="op">=</span> <span class="fu">start</span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e-6</span>, </span>
<span>                 gene_end   <span class="op">=</span> <span class="fu">end</span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span>   <span class="op">*</span> <span class="fl">1e-6</span>,</span>
<span>                 stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'ensembl' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_summary</span> <span class="op">&lt;-</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span> </span>
<span>                 <span class="fu">rename</span><span class="op">(</span>lodcolumn <span class="op">=</span> <span class="st">"ensembl"</span>,</span>
<span>                        chr       <span class="op">=</span> <span class="st">"qtl_chr"</span>,</span>
<span>                        pos       <span class="op">=</span> <span class="st">"qtl_pos"</span>,</span>
<span>                        lod       <span class="op">=</span> <span class="st">"qtl_lod"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                 <span class="fu">left_join</span><span class="op">(</span><span class="va">df</span>, by <span class="op">=</span> <span class="st">"ensembl"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                 <span class="fu">mutate</span><span class="op">(</span>marker.id <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">qtl_chr</span>, <span class="va">qtl_pos</span> <span class="op">*</span> <span class="fl">1e6</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>,</span>
<span>                        gene_chr  <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">gene_chr</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                        qtl_chr   <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">qtl_chr</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'lod_summary' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<p>Some of the genes will have a QTL in the same location as the gene
and others will have a QTL on a chromosome where the gene is not
located.</p>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1:</h3>
<div class="callout-content">
<p>What do we call eQTL that are co-colocated with the gene? What do we
call eQTL that are located on a different chromosome than the gene?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>A cis-eQTL is an eQTL that is co-colocated with the gene. A
trans-eQTL is an eQTL that is located on a chromosome other than the
gene that was mapped.</p>
</div>
</div>
</div>
</div>
<p>We can tabulate the number of cis- and trans-eQTL that we have and
add this to our QTL summary table. A cis-eQTL occurs when the QTL peaks
is directly over the gene position. But what if it is 2 Mb away? Or 10
Mb? It’s possible that a gene may have a trans eQTL on the same
chromosome if the QTL is “far enough” from the gene. We have selected 4
Mb as a good rule of thumb.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_summary</span> <span class="op">&lt;-</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span> </span>
<span>                     <span class="fu">mutate</span><span class="op">(</span>cis <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">qtl_chr</span> <span class="op">==</span> <span class="va">gene_chr</span> <span class="op">&amp;</span> </span>
<span>                                  <span class="fu">abs</span><span class="op">(</span><span class="va">gene_start</span> <span class="op">-</span> <span class="va">qtl_pos</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">4</span>, </span>
<span>                                  <span class="st">"cis"</span>, <span class="st">"trans"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'lod_summary' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">count</span><span class="op">(</span><span class="va">lod_summary</span>, <span class="va">cis</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'lod_summary' not found</code></pre>
</div>
<div class="section level3">
<h3 id="plot-transcriptome-map">Plot Transcriptome Map<a class="anchor" aria-label="anchor" href="#plot-transcriptome-map"></a>
</h3>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggtmap</span><span class="op">(</span>data <span class="op">=</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span> </span>
<span>       <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span> <span class="op">&gt;=</span> <span class="fl">7.18</span><span class="op">)</span>, </span>
<span>              cis.points <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>              cis.radius <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in ggtmap(data = lod_summary %&gt;% filter(qtl_lod &gt;= 7.18), cis.points = TRUE, : could not find function "ggtmap"</code></pre>
</div>
<p>The plot above is called a “Transcriptome Map” because it shows the
positions of the genes (or transcripts) and their corresponding QTL. The
QTL position is shown on the X-axis and the gene position is shown on
the Y-axis. The chromosomes are listed along the top and right of the
plot. What type of QTL are the genes with QTL that are located along the
diagonal?</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Transcriptome maps aid in understanding gene expression
regulation.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section><section id="aio-full-transcriptome-map"><p>Content from <a href="full-transcriptome-map.html">Transcriptome Map of cis and trans eQTL</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/full-transcriptome-map.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I create a full transcriptome map?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Explain how to use markdown with the new lesson template</li>
<li>Demonstrate how to include pieces of code, figures, and nested
challenge blocks</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h3>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2ggplot</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># source("../code/gg_transcriptome_map.R")</span></span></code></pre>
</div>
</div>
<section><h2 class="section-heading" id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<hr class="half-width">
<p>Load in the RNA-seq eQTL mapping results.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># loading previous results</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/dataset.islet.rnaseq.RData"</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we need to format the column names of our eQTL results to that
the <code>ggtmap</code> function can use the results.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_summary</span> <span class="op">=</span> <span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">lod.peaks</span></span>
<span></span>
<span><span class="co"># Get gene positions.</span></span>
<span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu">get_ensembl_genes</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>ensembl    <span class="op">=</span> <span class="va">ensembl</span><span class="op">$</span><span class="va">gene_id</span>,</span>
<span>                 gene_chr   <span class="op">=</span> <span class="fu">seqnames</span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span>,</span>
<span>                 gene_start <span class="op">=</span> <span class="fu">start</span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e-6</span>,</span>
<span>                 gene_end   <span class="op">=</span> <span class="fu">end</span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span>   <span class="op">*</span> <span class="fl">1e-6</span>,</span>
<span>                 stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create eQTL table for transcriptome map function.</span></span>
<span><span class="va">lod_summary</span> <span class="op">&lt;-</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">rename</span><span class="op">(</span>annot.id  <span class="op">=</span> <span class="st">"ensembl"</span>,</span>
<span>                        chrom     <span class="op">=</span> <span class="st">"qtl_chr"</span>,</span>
<span>                        pos       <span class="op">=</span> <span class="st">"qtl_pos"</span>,</span>
<span>                        lod       <span class="op">=</span> <span class="st">"qtl_lod"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">left_join</span><span class="op">(</span><span class="va">df</span>, by <span class="op">=</span> <span class="st">"ensembl"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">mutate</span><span class="op">(</span>marker.id <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">qtl_chr</span>, <span class="va">qtl_pos</span> <span class="op">*</span> <span class="fl">1e6</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>,</span>
<span>                        gene_chr  <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">gene_chr</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                        qtl_chr   <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">qtl_chr</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">mutate</span><span class="op">(</span>cis <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">qtl_chr</span> <span class="op">==</span> <span class="va">gene_chr</span> <span class="op">&amp;</span> <span class="fu">abs</span><span class="op">(</span><span class="va">gene_start</span> <span class="op">-</span> <span class="va">qtl_pos</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">4</span>, <span class="st">"cis"</span>, <span class="st">"trans"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="plot-transcriptome-map">Plot Transcriptome Map<a class="anchor" aria-label="anchor" href="#plot-transcriptome-map"></a>
</h3>
<p>In the previous lesson, we mapped the QTL locations of 50 genes. In
this lesson, we will map the QTL positions of genes.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggtmap</span><span class="op">(</span>data <span class="op">=</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span> <span class="op">&gt;=</span> <span class="fl">7.18</span><span class="op">)</span>,</span>
<span>      cis.points <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      cis.radius <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<p>This transcriptome map is definitely a lot more crowded than the one
in the previous lesson. Again, the gene locations are shown on the
X-axis and the QTL locations are shown on the Y-axis.</p>
<div id="challenge-1-what-patterns-among-the-points-do-you-see-in-the-transcriptome-map" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-what-patterns-among-the-points-do-you-see-in-the-transcriptome-map" class="callout-inner">
<h3 class="callout-title">Challenge 1: What patterns among the points do you see in the transcriptome map?</h3>
<div class="callout-content">

</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>There are at least two patterns. One is the dense diagonal line of
cis-eQTL. The other is the increased density of QTL in vertical
lines.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2-what-would-a-vertical-band-in-the-transcriptome-map-mean" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-what-would-a-vertical-band-in-the-transcriptome-map-mean" class="callout-inner">
<h3 class="callout-title">Challenge 2: What would a vertical band in the transcriptome map mean?</h3>
<div class="callout-content">

</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>A vertical band indicates that one locus regulates the expression of
many genes.</p>
</div>
</div>
</div>
</div>
<p>Look at the transcriptome map. How many vertical bands do you see and
which chromosomes are they on?</p>
</div>
<div class="section level3">
<h3 id="qtl-density-plot">QTL Density Plot<a class="anchor" aria-label="anchor" href="#qtl-density-plot"></a>
</h3>
<p>In the transcriptome map above, we noticed vertical banding patterns
in the eQTL, which indicate that one locus may regulate the expression
of dozens of genes. How many genes are regulated by each locus and which
genes are they? In order to address this question, we need to make a
plot of the density of eQTL along the genome. This is like stacking up
the eQTL onto the X-axis.</p>
<p>We have provided a function to do this in the
<code>gg_transcriptome_map.R</code> file in the <code>code</code>
directory of this lesson. The function is called
<code>eqtl_density_plot</code> and takes the following arguments:</p>
<ul>
<li>
<code>data</code>: data.frame (or tibble) with the following
columns:
<ul>
<li>
<code>ensembl</code>: (required) character string containing the
Ensembl gene ID.</li>
<li>
<code>qtl_chr</code>: (required) character string containing QTL
chromsome.</li>
<li>
<code>qtl_pos</code>: (required) floating point number containing
the QTL position in Mb.</li>
<li>
<code>qtl_lod</code>: (optional) floating point number containing
the LOD score.</li>
<li>
<code>gene_chr</code>: (optional) character string containing
transcript chromosome.</li>
<li>
<code>gene_start</code>: (optional) character string containing
transcript start position in Mb.</li>
<li>
<code>gene_end</code>: (optional) character string containing
transcript end position in Mb.</li>
</ul>
</li>
<li>
<code>lod_thr</code>: numeric value that is the LOD above which QTL
will be retained. Default = 7.</li>
</ul>
<p>This function has been designed to use the same data structure as we
used to create the transcriptome map. First, we will look at the overall
QTL density for all peaks with LOD &gt; 7.18.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">lod_summary</span>, lod_thr <span class="op">=</span> <span class="fl">7.18</span><span class="op">)</span></span></code></pre>
</div>
<p>There are clearly some loci that have an enrichment of eQTL. We have
drawn a dashed line at <code>100"</code> as an arbitrary cutoff as a
potential cutoff to use when selecting peaks.</p>
<p>Compare this plot with the transcriptome map, in which we saw
vertical bands of eQTL. Do the peaks in the eQTL density plot match the
bands in the transcriptome map?</p>
<p>Next, we will look at the locations of the cis-eQTL. We must also
select a LOD threshold. We will use 7.18 since this is what was used in
the Keller et al. paper.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">filter</span><span class="op">(</span><span class="va">lod_summary</span>, <span class="va">cis</span> <span class="op">==</span> <span class="st">"cis"</span><span class="op">)</span>, lod_thr <span class="op">=</span> <span class="fl">7.18</span><span class="op">)</span></span></code></pre>
</div>
<p>In the plot above, there are many loci that have many genes
associated with their expression. Some loci have over 100 genes
associated with them. For example, there is a locus on chromosome 17
that may regulate over 100 genes. In this case, we are looking at
cis-eQTL, QTL which are co-located with the gene. What might this mean
biologically? Perhaps there is a mutation which causes a region of
chromatin to open, which leads to increased expression of a set of
genes. This increased expression may have biological consequences.</p>
<p>Next, we will create an eQTL density plot of the trans-eQTL. These
are QTL for which the gene and QTL are far from each other.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">filter</span><span class="op">(</span><span class="va">lod_summary</span>, <span class="va">cis</span> <span class="op">==</span> <span class="st">"trans"</span><span class="op">)</span>, lod_thr <span class="op">=</span> <span class="fl">7.18</span><span class="op">)</span></span></code></pre>
</div>
<p>In this case, we see much taller peaks than in the cis-eQTL density
plot and these peaks match the ones in the overall eQTL density
plot.</p>
<p>There are many potential explanations for a trans-eQTL band. There
may be a polymorphism in a transcription factor which alters the
expression of many other genes. Or there may be a polymorphism that
alters an amino acid and prevents it from binding properly to another
protein in a signalling cascade. Biologists are often interested in
these trans-eQTL bands, which are called “eQTL hotspots”. A set of over
100 genes with differential expression between genotypes may help us to
understand the biology behind variation in higher level phenotypes. It
is also possible that one of the genes with a cis-eQTL under the eQTL
hotspot regulates the expression of the remaining hotspot genes.</p>
</div>
</section><section><h2 class="section-heading" id="islet-rnaseq-eqtl-hotspots">Islet RNASeq eQTL Hotspots<a class="anchor" aria-label="anchor" href="#islet-rnaseq-eqtl-hotspots"></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="select-eqtl-hotspots">Select eQTL Hotspots<a class="anchor" aria-label="anchor" href="#select-eqtl-hotspots"></a>
</h3>
<p>There are several decisions to make when selecting eQTL hotspots.
What LOD threshold should you use to select the genes that comprise
hotspots? What number of genes should you use as a threshold to call a
peak an eQTL hotspot? In this case, the authors select a LOD of 7.18 and
decided that 100 trans-regulated genes was a useful threshold.</p>
<blockquote>
<p>TBD: Add references for eQTL module selection. Possibly augment
lesson?</p>
</blockquote>
<p>In order to identify eQTL hotspots, we will look at the density of
trans-eQTL along the genome using the <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/density" class="external-link">density</a>
function. We will filter to retain QTL with LOD &gt; 7.8 and will get
the position on each chromosome with the highest density of eQTL.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl_dens</span> <span class="op">=</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span> <span class="op">&gt;</span> <span class="fl">7.18</span> <span class="op">&amp;</span> <span class="va">cis</span> <span class="op">==</span> <span class="st">'trans'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">group_by</span><span class="op">(</span><span class="va">qtl_chr</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">summarize</span><span class="op">(</span>dens_x <span class="op">=</span> <span class="fu">density</span><span class="op">(</span><span class="va">qtl_pos</span>, adjust <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span><span class="op">$</span><span class="va">x</span>,</span>
<span>                       dens_y <span class="op">=</span> <span class="fu">density</span><span class="op">(</span><span class="va">qtl_pos</span>, adjust <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span><span class="op">$</span><span class="va">y</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">slice_max</span><span class="op">(</span><span class="va">dens_y</span>, n <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">qtl_dens</span></span></code></pre>
</div>
<p>In the table above, there is one row per chromosome. <code>x</code>
is the position on each chromosome with the highest density of eQTL.
<code>y</code> is the density, which we needed to obtain the
<code>x</code> position but is not used further.</p>
<p>Now that we have the location of maximum eQTL density on each
chromosome, we will count the number of eQTL within +/- 2 Mb of the
center.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hotspots</span> <span class="op">=</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">lod_summary</span>, <span class="fu">select</span><span class="op">(</span><span class="va">qtl_dens</span>, <span class="va">qtl_chr</span>, <span class="va">dens_x</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span> <span class="op">&gt;</span> <span class="fl">7.18</span> <span class="op">&amp;</span> <span class="va">cis</span> <span class="op">==</span> <span class="st">'trans'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">mutate</span><span class="op">(</span>pos_diff <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">dens_x</span> <span class="op">-</span> <span class="va">qtl_pos</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">pos_diff</span> <span class="op">&lt;=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pos_diff</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">rename</span><span class="op">(</span>dens_x <span class="op">=</span> <span class="st">'center'</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">hotspots</span><span class="op">)</span></span></code></pre>
</div>
<p>Now that we have a list of genes near the position of highest eQTL
density on each chromosome, we can count the number of genes in each
potential hotspot and retain the ones containing more than 100
genes.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hotspots</span> <span class="op">=</span> <span class="va">hotspots</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">count</span><span class="op">(</span><span class="va">qtl_chr</span>, <span class="va">center</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">n</span> <span class="op">&gt;=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">hotspots</span>, caption <span class="op">=</span> <span class="st">"Islet trans-eQTL hotspots"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="chr-11-eqtl-hotspot">Chr 11 eQTL Hotspot<a class="anchor" aria-label="anchor" href="#chr-11-eqtl-hotspot"></a>
</h3>
<p>From this analysis, we have identified five eQTL hotspots. Due to
time constraints, we will not examine all of them in depth. We will look
at the chromosome 11 eQTL hotspot in greater depth. It contains 163
trans-eQTL. First, we will get all of the genes with an eQTL in within
+/- 2 Mb of the chromosome 11 eQTL hotspot.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr11_mid</span> <span class="op">=</span>  <span class="va">hotspots</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_chr</span> <span class="op">==</span> <span class="st">'11'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">pull</span><span class="op">(</span><span class="va">center</span><span class="op">)</span></span>
<span><span class="va">chr11_eqtl</span> <span class="op">=</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span> <span class="op">&gt;</span>  <span class="fl">7.18</span> <span class="op">&amp;</span></span>
<span>                      <span class="va">qtl_chr</span> <span class="op">==</span> <span class="st">'11'</span> <span class="op">&amp;</span></span>
<span>                      <span class="fu">abs</span><span class="op">(</span><span class="va">chr11_mid</span> <span class="op">-</span> <span class="va">qtl_pos</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span> <span class="op">&amp;</span></span>
<span>                      <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">cis</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will filter to retain the cis-eQTL in the same interval. It
is possible that one or more genes near 71.5 Mb on chromosome 11 have a
cis-eQTL, which in turn alters the expression of the trans-eQTL
genes.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr11_cis</span> <span class="op">=</span> <span class="va">chr11_eqtl</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">filter</span><span class="op">(</span><span class="va">cis</span> <span class="op">==</span> <span class="st">'cis'</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">chr11_cis</span>, caption <span class="op">=</span> <span class="st">'Chr 11 cis-eQTL'</span><span class="op">)</span></span></code></pre>
</div>
<p>As you can see, there are cis-eQTL genes under the chromosome 11 eQTL
hotspot. This is a large number of candidate genes to screen. There may
be more genes with non-synonymous, splice, or stop mutations under the
eQTL hotspot as well. The Sanger Mouse Genomes website has been removed
and we are uncertain if it will be replaced. There are two websites
where you can find this information:</p>
<ol style="list-style-type: decimal">
<li>
<a href="http://nov2020.archive.ensembl.org/Mus_musculus/Info/Index" class="external-link">Ensembl</a>:
Once you search for a gene, you can select the “Variant Table” under
“Genetic Variation” in the left navigation panel.</li>
<li>
<a href="https://churchilllab.jax.org/foundersnps/search" class="external-link">Founder
Variant Search</a>: You can use this site to search for variants in
specific genes or genomic intervals.</li>
</ol>
<p>Next, we will get the expression of the genes in the chromosome 11
eQTL hotspot. The expression data is a numeric matrix, so we will use
the colnames to filter the genes.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr11_genes</span> <span class="op">=</span> <span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">expr</span><span class="op">[</span>,<span class="va">chr11_eqtl</span><span class="op">$</span><span class="va">ensembl</span><span class="op">]</span></span></code></pre>
</div>
<p>Next, we will join the eQTL and expression data and write it out to a
file. This way you will have all of the data for the hotspot in one
place.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr11_expr</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span>ensembl <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">chr11_genes</span><span class="op">)</span>,</span>
<span>              <span class="fu">data.frame</span><span class="op">(</span><span class="fu">t</span><span class="op">(</span><span class="va">chr11_genes</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">chr11_all</span> <span class="op">=</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">chr11_eqtl</span>, <span class="va">chr11_expr</span>, by <span class="op">=</span> <span class="st">'ensembl'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">write_csv</span><span class="op">(</span><span class="va">chr11_all</span>, file <span class="op">=</span> <span class="fu">file.path</span><span class="op">(</span><span class="st">'../results/'</span>, <span class="st">'chr11_eqtl_genes.csv'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section><section id="aio-maximum-peaks-genes"><p>Content from <a href="maximum-peaks-genes.html">Maximum eQTL Peaks and Nearby Genes</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/maximum-peaks-genes.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I display maximum eQTL peaks and nearby genes?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Explain how to use markdown with the new lesson template</li>
<li>Demonstrate how to include pieces of code, figures, and nested
challenge blocks</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<figure><img src="fig/LOD-plot-Hnf1b.png" alt="LOD plot for gene Hnf1b." class="figure mx-auto d-block"><div class="figcaption">LOD plot for gene Hnf1b.</div>
</figure><figure><img src="fig/SNP-association-Hnf1b.png" alt="SNP association plot for gene Hnf1b." class="figure mx-auto d-block"><div class="figcaption">SNP association plot for gene Hnf1b.</div>
</figure><div id="challenge-1-can-you-do-it" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-can-you-do-it" class="callout-inner">
<h3 class="callout-title">Challenge 1: Can you do it?</h3>
<div class="callout-content">
<p>What is the output of this command?</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">paste</span><span class="op">(</span><span class="st">"This"</span>, <span class="st">"new"</span>, <span class="st">"lesson"</span>, <span class="st">"looks"</span>, <span class="st">"good"</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Output </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "This new lesson looks good"</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="challenge-2-how-do-you-nest-solutions-within-challenge-blocks" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-how-do-you-nest-solutions-within-challenge-blocks" class="callout-inner">
<h3 class="callout-title">Challenge 2: how do you nest solutions within challenge blocks?</h3>
<div class="callout-content">

</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>You can add a line with at least three colons and a
<code>solution</code> tag.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div></section><section id="aio-interpret-results"><p>Content from <a href="interpret-results.html">Interpreting qtl2 results</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/interpret-results.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I interpret qtl2 results?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Interpret the relationship between sequence, expression and
phenotype variation from qtl2 mapping results.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p><img src="fig/SFig7_Keller_2018_chr11.png" alt="Attie et al, 2018 Supplementary Figure S7" class="figure">
::::::::::::::::::::::::::::::::::::: challenge</p>
<section><h2 class="section-heading" id="challenge-interpreting-qtl2-results">Challenge: Interpreting qtl2 results<a class="anchor" aria-label="anchor" href="#challenge-interpreting-qtl2-results"></a>
</h2>
<hr class="half-width">
<p>Refer to the figure above.</p>
<p>1). What does panel A show? What conclusions could you draw from
panel A?<br>
2). What does panel B show? What conclusions could you draw from panel
B?<br>
3). What does panel C show? What conclusions could you draw from panel
C?<br>
4). How are panels A through C related to one another? What story do
they tell together?</p>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<p>::::::::::::::::::::::::::::::::::::::::::::::::</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-mediation-analysis"><p>Content from <a href="mediation-analysis.html">Mediation Analysis</a></p>
<hr>
<p>Last updated on 2024-11-19 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/mediation-analysis.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is mediation analysis?</li>
<li>How do I explore causal relations with mediation analysis?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe mediation analysis as applied in genetics and
genomics.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>GWAS studies show that most disease-associated variants are found in
non-coding regions. This fact leads to the idea that regulation of gene
expression is an important mechanism enabling genetic variants to affect
complex traits. Mediation analysis can identify a causal chain from
genetic variants to molecular and clinical phenotypes. The graphic below
shows complete mediation, in which a predictor variable does not
directly impact the response variable, but does directly the mediator.
The mediator has a direct impact on the response variable. We would
observe the relationship between predictor and response, not knowing
that a mediator intervenes in this relationship.</p>
<p><img src="fig/complete-mediation.png" alt="In complete mediation an independent (predictor) variable influences the dependent (response) variable indirectly through a mediator variable." class="figure">
Mediation analysis is widely used in the social sciences including
psychology. In biomedicine, mediation analysis has been employed to
investigate how gene expression mediates the effects of genetic variants
on complex phenotypes and disease.</p>
<p>For example, a genetic variant (non-coding SNP) indirectly regulates
expression of gene 2 through a mediator, gene 1. The SNP regulates
expression of gene 1 in cis, and expression of gene 1 influences
expression of gene 2 in trans.</p>
<p><img src="fig/mediator-gene1.png" alt="A non-coding SNP affects expression of gene 1 in cis. Gene 1 mediates expression of gene 2." class="figure">
Instead of the expression of one gene impacting another, expression of
gene 1 in the graphic above could impact a physiological phenotype like
blood glucose. Expression of gene 1 would mediate the relationship
between the non-coding SNP and the glucose phenotype.</p>
<p>Gene Akr1e1 is located on chromosome 13 in mouse. How would you
interpret the LOD plot below? On which chromosome(s) would you expect to
find the driver gene(s)? The SNP(s)?</p>
<p><img src="fig/LOD-plot-Akr1e1.png" alt="Chromosome 13 gene Akr1e1 is affected by expression in both cis and trans by genes on chromosomes 13 and 4." class="figure">
Myo15b is located on chromosome 11. How would you interpret the
following LOD plot? On which chromosome(s) would you expect to find the
driver gene(s)? The SNP(s)?</p>
<p><img src="fig/LOD-plot-Myo15b.png" alt="Chromosome 11 gene Myo15b is affected by expression of a trans gene in the chromosome 2 hotspot." class="figure">
The <a href="https://churchilllab.jax.org/qtlviewer/attie/islets#" class="external-link">QTL
Viewer for the Attie islet data</a> integrates mediation into
exploration of the data. Below, mediation analysis identifies gene Hnf4a
as the chromosome 2 gene that impacts Myo15b expression.</p>
<figure><img src="fig/mediation-Hnf4a-Myo15b.png" alt="Mediating expression of Myo15b identifies Hnf4a as the gene that drops the LOD score from greater than 70 to less than 50." class="figure mx-auto d-block"><div class="figcaption">Mediating expression of Myo15b identifies Hnf4a
as the gene that drops the LOD score from greater than 70 to less than
50.</div>
</figure><div class="section level3">
<h3 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h3>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span><span class="co"># library(intermediate)</span></span></code></pre>
</div>
</div>
<section><h2 class="section-heading" id="load-functions">Load Functions<a class="anchor" aria-label="anchor" href="#load-functions"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rankZ</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">x</span> <span class="op">=</span> <span class="fu">rank</span><span class="op">(</span><span class="va">x</span>, na.last <span class="op">=</span> <span class="st">"keep"</span>, ties.method <span class="op">=</span> <span class="st">"average"</span><span class="op">)</span> <span class="op">/</span> <span class="op">(</span><span class="fu">sum</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span>
<span>  <span class="kw">return</span><span class="op">(</span><span class="fu">qnorm</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># expression data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_expr.datasets.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_expr.datasets.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># data from paper</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/dataset.islet.rnaseq.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/dataset.islet.rnaseq.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># phenotypes</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_clinical.phenotypes.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_clinical.phenotypes.RData', probable reason 'No such file
or directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># mapping data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_mapping.data.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_mapping.data.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># genotype probabilities</span></span>
<span><span class="va">probs</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">"../data/attie_DO500_genoprobs_v5.rds"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in gzfile(file, "rb"): cannot open compressed file
'../data/attie_DO500_genoprobs_v5.rds', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in gzfile(file, "rb"): cannot open the connection</code></pre>
</div>
<blockquote>
<p>Example from package. This will be removed, but I needed it for
now.</p>
</blockquote>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="co"># DOQTL liver protein expresion dataset</span></span>
<span>  <span class="fu">data</span><span class="op">(</span><span class="va">Tmem68</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Let us mediate Tmem68 to other proteins</span></span>
<span>  <span class="va">trgt</span> <span class="op">=</span> <span class="fu">matrix</span><span class="op">(</span><span class="va">Tmem68</span><span class="op">$</span><span class="va">target</span>, ncol <span class="op">=</span> <span class="fl">1</span>, dimnames <span class="op">=</span> <span class="fu">list</span><span class="op">(</span><span class="fu">names</span><span class="op">(</span><span class="va">Tmem68</span><span class="op">$</span><span class="va">target</span><span class="op">)</span>, <span class="st">'Tmem68'</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">annot</span> <span class="op">=</span> <span class="va">Tmem68</span><span class="op">$</span><span class="va">annotation</span></span>
<span>  <span class="fu">colnames</span><span class="op">(</span><span class="va">annot</span><span class="op">)</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span> <span class="op">=</span> <span class="st">'MIDDLE_POINT'</span></span>
<span>  <span class="va">med</span> <span class="op">&lt;-</span> <span class="fu">mediation.scan</span><span class="op">(</span>target     <span class="op">=</span> <span class="va">trgt</span>,</span>
<span>                        mediator   <span class="op">=</span> <span class="va">Tmem68</span><span class="op">$</span><span class="va">mediator</span>,</span>
<span>                        annotation <span class="op">=</span> <span class="va">annot</span>,</span>
<span>                        covar      <span class="op">=</span> <span class="va">Tmem68</span><span class="op">$</span><span class="va">covar</span>,</span>
<span>                        qtl.geno   <span class="op">=</span> <span class="va">Tmem68</span><span class="op">$</span><span class="va">qtl.geno</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Interactive plot</span></span>
<span>  <span class="fu">kplot</span><span class="op">(</span><span class="va">med</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="searching-for-candidate-genes">Searching for Candidate Genes<a class="anchor" aria-label="anchor" href="#searching-for-candidate-genes"></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="mapping-the-phenotype">Mapping the Phenotype<a class="anchor" aria-label="anchor" href="#mapping-the-phenotype"></a>
</h3>
<p>We will map the <code>Insulin tAUC</code> trait becuase it has a QTL
on chromosome 11. First, we will <code>rankZ</code> transform the
phenotypes because this helps to satisfy the model assumptions. We will
map using sex, generation, and the number of days on the diet as
additive covariates.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># NOTE: QTL is not nearly as large with untransformed phenotype.</span></span>
<span><span class="va">pheno_rz</span> <span class="op">=</span> <span class="va">pheno_clin</span> <span class="op">%&gt;%</span> </span>
<span>             <span class="fu">select</span><span class="op">(</span><span class="va">num_islets</span><span class="op">:</span><span class="va">weight_10wk</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>             <span class="fu">as.matrix</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_rz</span> <span class="op">=</span> <span class="fu">apply</span><span class="op">(</span><span class="va">pheno_rz</span>, <span class="fl">2</span>, <span class="va">rankZ</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_rz' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'covar' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">addcovar</span> <span class="op">=</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span> <span class="op">+</span> <span class="va">diet_days</span>, data <span class="op">=</span> <span class="va">covar</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'covar' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ins_qtl</span> <span class="op">=</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                pheno     <span class="op">=</span> <span class="va">pheno_rz</span><span class="op">[</span>,<span class="st">'Ins_tAUC'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>                addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">ins_qtl</span>, <span class="va">map</span>, main <span class="op">=</span> <span class="st">'Insulin tAUC'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'ins_qtl' not found</code></pre>
</div>
<p>There is a large peak on chromosome 11. Let’s look at the LOD score
and the location.</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span><span class="va">ins_qtl</span>, <span class="va">map</span>, threshold <span class="op">=</span> <span class="fl">10</span>, prob <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'ins_qtl' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'peaks' not found</code></pre>
</div>
<p>Let’s also estimate the founder allele effects for
<code>insulin tAUC</code>.</p>
<blockquote>
<p>TBD: This is slow. Precalculate and just show the plot?</p>
</blockquote>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span> <span class="op">=</span> <span class="st">'11'</span></span>
<span><span class="va">feff</span> <span class="op">=</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">pheno_rz</span><span class="op">[</span>,<span class="st">'Ins_tAUC'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                 kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coefCC</span><span class="op">(</span><span class="va">feff</span>, <span class="va">map</span>, scan1_output <span class="op">=</span> <span class="va">qtl</span>, legend <span class="op">=</span> <span class="st">'bottomleft'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'feff' not found</code></pre>
</div>
<p>Looking at the founder allele effects above, we can see that A/J,
BL6, 129, and NOD alleles contribute to higher insulin tAUC values.
Ideally, we are looking for a gene with a pattern of allele effects that
matches this pattern, or it’s inverse.</p>
<blockquote>
<p>TBD: Plot founder allele effects for two genes, one with a similar
pattern, one without.</p>
</blockquote>
</div>
</section><section><h2 class="section-heading" id="searching-for-cis-eqtl-under-a-physiological-qtl">Searching for cis-eQTL under a physiological QTL<a class="anchor" aria-label="anchor" href="#searching-for-cis-eqtl-under-a-physiological-qtl"></a>
</h2>
<hr class="half-width">
<p>When we have a physiological QTL, we often start by looking for genes
which contain missense, splice, or stop codon SNPs. This is a reasonable
first approach to selecting candidate genes and may pay off. However,
many QTL fall in intergenic regions, which suggests that the causal
variant(s) may be in regulatory regions. These regulatory regions are
not as well annotated as coding regions and it may be difficult to
identify them from sequence alone. However, if you have gene expression
data in a relevant tissue, you can look for genes in the QTL interval
which have a cis-eQTL. If the pattern of founder allele effects for a
candidate gene is similar to the allele effects of the physiological
QTL, then this gene is a good candidate.</p>
<blockquote>
<p>TBD: Need a figure with physiological QTL effect &amp; cis-eQTL
effects.</p>
</blockquote>
<p>We have a QTL for insulin tAUC that extends from Mb to Mb. This
support interval means that we expect the causal variant to be within
this interval 95% of the time. We would like to select a set of
candidate genes located near this QTL. If the causal variant is a coding
SNP, then the gene will be located within this interval. However, if the
causal variant is a regulatory SNP, then the gene that it regulates may
be up to 2 Mb away from the variant. This means that we should be
searching for genes within 2 Mb of the edges of the support
interval.</p>
<blockquote>
<p>TBD: Is there a REF for the +/- 2 Mb distance?</p>
</blockquote>
<p>First, we need to get the support interval and add 2 MB to each
side.</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ci</span> <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">ci_lo</span> <span class="op">-</span> <span class="fl">2</span>, <span class="va">peaks</span><span class="op">$</span><span class="va">ci_hi</span> <span class="op">+</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'peaks' not found</code></pre>
</div>
<p>Next, we filter the eQTL results to retain genes in this
interval.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cis_eqtl</span> <span class="op">=</span> <span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">lod.peaks</span> <span class="op">%&gt;%</span> </span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">chrom</span> <span class="op">==</span> <span class="st">'11'</span>  <span class="op">&amp;</span> </span>
<span>                      <span class="va">pos</span> <span class="op">&gt;=</span> <span class="va">ci</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&amp;</span> </span>
<span>                      <span class="va">pos</span> <span class="op">&lt;=</span> <span class="va">ci</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<p>There are r nrow(cis_eqtl) genes with cis-eQTL in this interval. That
is a lot of genes! We could estimate the founder allele effects for each
of the genes and stare at them pensively. Or we could add each gene to
the QTL mapping model and see if the LOD score changes.</p>
<div class="section level3">
<h3 id="mapping-the-phenotype-with-a-gene-as-as-covariate">Mapping the Phenotype with a Gene as as Covariate<a class="anchor" aria-label="anchor" href="#mapping-the-phenotype-with-a-gene-as-as-covariate"></a>
</h3>
<p>In QTL mediation analysis, we are hypothesizing that there is a
genomic variant which affects the expression of a gene, which in turn
affects the physiological phenotype.</p>
<blockquote>
<p>TBD: Add mediation figure here again?</p>
</blockquote>
<p>There is a gene called <em>Slfn3</em> under the Insulin tAUC peak.
Our hypothesis is that this gene may influence Insulin AUC becuase it
has a cis-eQTL in the same location. In order to test this hypothesis,
we will add the expression of this gene to the model as an additive
covariate. If <em>Slfn3</em> explains some of the variance in Insulin
tAUC, then the LOD should decrease. If it does not, then the LOD will
not change much.</p>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covar_gene</span> <span class="op">=</span> <span class="fu">cbind</span><span class="op">(</span><span class="va">norm</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">addcovar</span><span class="op">)</span>, <span class="st">'ENSMUSG00000018986'</span><span class="op">]</span>, <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'addcovar' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">slfn3_qtl</span> <span class="op">=</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                  pheno     <span class="op">=</span> <span class="va">pheno_rz</span><span class="op">[</span>,<span class="st">'Ins_tAUC'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                  kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>                  addcovar  <span class="op">=</span> <span class="va">covar_gene</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">ins_qtl</span>,   <span class="va">map</span>, chr <span class="op">=</span> <span class="va">chr</span>, main <span class="op">=</span> <span class="st">'Insulin tAUC with Slfn3'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'ins_qtl' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">slfn3_qtl</span>, <span class="va">map</span>, chr <span class="op">=</span> <span class="va">chr</span>, color <span class="op">=</span> <span class="st">'blue'</span>, add <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'slfn3_qtl' not found</code></pre>
</div>
<p>As you can see, the LOD decreased when we added the expression of
<em>Slfn3</em> into the model. Let’s see what the new LOD is on
chromosome 11.</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span><span class="va">qtl</span>, <span class="va">map</span>, threshold <span class="op">=</span> <span class="fl">6.5</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'qtl' not found</code></pre>
</div>
<p>The LOd decreased from 10.3 to 7.2.</p>
<blockquote>
<p>DMG: STOPPED HERE</p>
</blockquote>
</div>
</section><section><h2 class="section-heading" id="searching-for-causal-genes">Searching for Causal Genes<a class="anchor" aria-label="anchor" href="#searching-for-causal-genes"></a>
</h2>
<hr class="half-width">
<p>In mediation analysis, we don’t know the causal gene in advance. We
must fit a model like the one above for each gene that was measured. But
this would involve mapping the trait across the entire genome for each
gene. We don’t need to do this to identify causal genes under the
phenotype QTL. We only need to fit a model using each gene at the
position of the phenotype QTL.</p>
<p>Mediation analysis requires five pieces of data and metadata:</p>
<ol style="list-style-type: decimal">
<li>Target: the phenotype that has a QTL for which we are seeking causal
gene candidates.</li>
<li>Mediator: gene expression for genes that will be used in the
mediation analysis.</li>
<li>Annotation: gene annotation data for the mediator genes.</li>
<li>Covar: a set of covariates to use in the model.</li>
<li>QTL.Geno: the genotype probabilities at the Target maximum QTL.</li>
</ol>
<div id="challenge-1-can-you-do-it" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-can-you-do-it" class="callout-inner">
<h3 class="callout-title">Challenge 1: Can you do it?</h3>
<div class="callout-content">
<p>Visit the Attie islet data QTL viewer:<br><a href="https://churchilllab.jax.org/qtlviewer/attie/islets#" class="external-link uri">https://churchilllab.jax.org/qtlviewer/attie/islets#</a> 1.
Select Islet RNA for the current data set.<br>
2. Type in a gene symbol to search for.<br>
3. Click on the highest peak in the LOD plot.<br>
4. Look at the Effects in the bottom right panel.<br>
5. Select the Mediation tab and click the LOD peak again.<br>
6. Hover over the dots in the mediation plot to discover genes that
raise or lower the peak.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Mediation analysis investigates an intermediary between an
independent variable and its effect on a dependent variable.</li>
<li>Mediation analysis is used in high-throughput genomics studies to
identify molecular phenotypes, such as gene expression or methylation
traits, that mediate the effect of genetic variation on disease
phenotypes or other outcomes of interest.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.10" class="external-link">sandpaper (0.16.10)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.7" class="external-link">pegboard (0.7.7)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene expression, quantitative trait, genetic mapping, Diversity Outbred population",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/aio.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/aio.html",
  "dateCreated": "2024-11-07",
  "dateModified": "2024-11-19",
  "datePublished": "2024-11-19"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

