<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Expression Quantitative Trait Locus (eQTL) Mapping: Creating A Transcriptome Map</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicons/incubator/favicon-16x16.png"><link rel="manifest" href="favicons/incubator/site.webmanifest"><link rel="mask-icon" href="favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/create-transcriptome-map.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Expression Quantitative Trait Locus (eQTL) Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Expression Quantitative Trait Locus (eQTL) Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"></ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Expression Quantitative Trait Locus (eQTL) Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 57%" class="percentage">
    57%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 57%" aria-valuenow="57" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/create-transcriptome-map.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="review-eqtl-study.html">2. Genetic Drivers of Pancreatic Islet Function</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="load-explore-data.html">3. Load and Explore Data</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="map-one-eqtl.html">4. Mapping A Single Gene Expression Trait</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="map-many-eqtls.html">5. Mapping Many Gene Expression Traits</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="maximum-peaks-genes.html">6. Maximum eQTL Peaks and Nearby Genes</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        7. Creating A Transcriptome Map
        </span>
      
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="mediation-analysis.html">8. Mediation Analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>

                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="maximum-peaks-genes.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="mediation-analysis.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="maximum-peaks-genes.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Maximum eQTL Peaks
        </a>
        <a class="chapter-link float-end" href="mediation-analysis.html" rel="next">
          Next: Mediation Analysis...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Creating A Transcriptome Map</h1>
        <p>Last updated on 2025-01-14 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/create-transcriptome-map.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How do I create and interpret a transcriptome map?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Describe a transcriptome map.</li>
<li>Interpret a transcriptome map.</li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="local-and-distant-eqtl">Local and Distant eQTL<a class="anchor" aria-label="anchor" href="#local-and-distant-eqtl"></a></h3>
<p>In the previous lesson, we saw that the QTL peak for a gene can lie
directly over the gene that is being mapped. This is called a <em>local
eQTL</em> because the QTL peak is located near the gene. When the QTL
peak is located far from the gene being mapped, we call this a
<em>distal eQTL</em>. When the QTL peak is located on the same
chromosome as the gene, there are different heuristics regarding the
distance between the gene and its QTL peak that we use to call an eQTL
local or distal. This depends on the type of cross and the resolution of
the recombination block structure.</p>
<p>In this episode, we will create a table containing the QTL peaks for
all genes and the gene positions. We will then classify QTL into local
or distal and will make a plot of all off the eQTL in relation to the
gene positions. First let’s load the libraries and data again in case
these are no longer in your environment.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pheno</span>    <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'data/attie_do_pheno.rds'</span><span class="op">)</span></span>
<span><span class="va">addcovar</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'data/attie_do_addcovar.rds'</span><span class="op">)</span></span>
<span><span class="va">annot</span>    <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'data/attie_do_expr_annot.rds'</span><span class="op">)</span></span>
<span><span class="va">expr_rz</span>  <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'data/attie_do_expr_rz.rds'</span><span class="op">)</span></span>
<span><span class="va">probs</span>    <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'data/attie_DO500_genoprobs_v5.rds'</span><span class="op">)</span></span>
<span><span class="va">map</span>      <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'data/attie_do_map.rds'</span><span class="op">)</span></span>
<span><span class="va">K</span>        <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'data/attie_do_kinship.rds'</span><span class="op">)</span></span>
<span><span class="va">peaks</span>    <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'data/attie_do_eqtl_peaks.rds'</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="transcriptome-map">Transcriptome Map<a class="anchor" aria-label="anchor" href="#transcriptome-map"></a></h3>
<p>We have encapsulated the code to create a transcriptome map in a file
in this lesson. You can copy this file from the Github repository to use
in your eQTL mapping project. We will read this file in now.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">source</span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/smcclatchy/eqtl-mapping/refs/heads/main/episodes/code/gg_transcriptome_map.R"</span><span class="op">)</span></span></code></pre>
</div>
<p>This file loads in a function called <code>ggtmap</code>, which
requires the input data to be in a specific format.</p>
<p>In order to use the <code>ggtmap</code> function, we need to provide
specific column names. These are documented in the
<code>gg_transcriptome_map.R</code> file in the code directory of this
workshop. The required column names are:</p>
<ul><li>
<code>data</code>: data.frame (or tibble) with the following
columns:
<ul><li>
<code>gene_id</code>: (required) character string containing the
Ensembl gene ID.</li>
<li>
<code>qtl_chr</code>: (required) character string containing QTL
chromosome.</li>
<li>
<code>qtl_pos</code>: (required) floating point number containing
the QTL position in Mb.</li>
<li>
<code>qtl_lod</code>: (optional) floating point number containing
the LOD score.</li>
<li>
<code>gene_chr</code>: (optional) character string containing
transcript chromosome.</li>
<li>
<code>gene_start</code>: (optional) character string containing
transcript start position in Mb.</li>
<li>
<code>gene_end</code>: (optional) character string containing
transcript end position in Mb.</li>
</ul></li>
</ul><p>First, we will get the gene positions from the annotation and rename
the columns to match what <code>ggtmap</code> requires. We need to have
columns named <code>gene_id</code>, <code>gene_chr</code>, and
<code>gene_pos</code>. We must rename the columns because, when we are
finished, we will have <code>chr</code> and <code>pos</code> columns for
both the gene and its QTL. We will also add the <code>symbol</code>
column since it is nice to have gene symbols in the data.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_pos</span> <span class="op">&lt;-</span> <span class="va">annot</span> <span class="op">|&gt;</span></span>
<span>              <span class="fu">select</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">symbol</span>, </span>
<span>                     gene_chr   <span class="op">=</span> <span class="va">chr</span>, </span>
<span>                     gene_start <span class="op">=</span> <span class="va">start</span>,</span>
<span>                     gene_end   <span class="op">=</span> <span class="va">end</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will rename the columns in the filtered peaks and will join
them with the gene positions from above.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eqtl</span> <span class="op">&lt;-</span> <span class="va">peaks</span> <span class="op">|&gt;</span></span>
<span>          <span class="fu">select</span><span class="op">(</span>gene_id <span class="op">=</span> <span class="va">lodcolumn</span>,</span>
<span>                 qtl_chr <span class="op">=</span> <span class="va">chr</span>,</span>
<span>                 qtl_pos <span class="op">=</span> <span class="va">pos</span>,</span>
<span>                 qtl_lod <span class="op">=</span> <span class="va">lod</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>          <span class="fu">left_join</span><span class="op">(</span><span class="va">gene_pos</span>, by <span class="op">=</span> <span class="st">"gene_id"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>          <span class="fu">mutate</span><span class="op">(</span>marker.id <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">qtl_chr</span>,   <span class="va">qtl_pos</span> <span class="op">*</span> <span class="fl">1e6</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>,</span>
<span>                 gene_chr  <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">gene_chr</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                 qtl_chr   <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">qtl_chr</span>,  levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">gene_pos</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-1-how-many-genes-have-qtl-on-the-same-chromosome" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-how-many-genes-have-qtl-on-the-same-chromosome" class="callout-inner">
<h3 class="callout-title">Challenge 1: How many genes have QTL on the same chromosome?</h3>
<div class="callout-content">
<p>Write a command to count the number of genes which are located on the
same chromosome as their corresponding QTL peak.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="va">eqtl</span><span class="op">$</span><span class="va">qtl_chr</span> <span class="op">==</span> <span class="va">eqtl</span><span class="op">$</span><span class="va">gene_chr</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 15105</code></pre>
</div>
<p>15105 genes have QTL on the same chromosome.</p>
</div>
</div>
</div>
</div>
<p>We can tabulate the number of local and distal eQTL that we have and
add this to our QTL summary table. A local eQTL occurs when the QTL
peaks is directly over the gene position. But what if it is 2 Mb away?
Or 10 Mb? It’s possible that a gene may have a distal eQTL on the same
chromosome if the QTL is “far enough” from the gene. In <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller et
al</a>, the authors selected a 4 Mb distance from the corresponding gene
and we will use this threshold.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eqtl</span> <span class="op">&lt;-</span> <span class="va">eqtl</span> <span class="op">|&gt;</span> </span>
<span>          <span class="fu">mutate</span><span class="op">(</span>local <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">qtl_chr</span> <span class="op">==</span> <span class="va">gene_chr</span> <span class="op">&amp;</span> </span>
<span>                                 <span class="fu">abs</span><span class="op">(</span><span class="va">gene_start</span> <span class="op">-</span> <span class="va">qtl_pos</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">4</span>, </span>
<span>                                     <span class="cn">TRUE</span>, </span>
<span>                                     <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">count</span><span class="op">(</span><span class="va">eqtl</span>, <span class="va">local</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  local     n
1 FALSE 28875
2  TRUE 14066</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="plot-transcriptome-map">Plot Transcriptome Map<a class="anchor" aria-label="anchor" href="#plot-transcriptome-map"></a></h3>
<p>In <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller
et al</a>, the authors used a LOD threshold of 7.2 to select genes to
use in their transcriptome map. We will use this threshold to reproduce
their results as closely as possible.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eqtl_thr</span> <span class="op">=</span> <span class="fl">7.2</span></span>
<span></span>
<span><span class="fu">ggtmap</span><span class="op">(</span>data <span class="op">=</span> <span class="va">eqtl</span> <span class="op">|&gt;</span> </span>
<span>                <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span>      <span class="op">&gt;=</span> <span class="va">eqtl_thr</span><span class="op">)</span>, </span>
<span>                       local.points <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                       local.radius <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-tmap-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot above is called a <em>transcriptome map</em> because it
shows the positions of the genes (or transcripts) and their
corresponding QTL. The QTL position is shown on the X-axis and the gene
position is shown on the Y-axis. The chromosomes are listed along the
top and right of the plot.</p>
<div id="challenge-2-what-are-the-blue-points-on-the-diagonal" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-what-are-the-blue-points-on-the-diagonal" class="callout-inner">
<h3 class="callout-title">Challenge 2: What are the blue points on the diagonal?</h3>
<div class="callout-content">
<p>What type of QTL are the genes with QTL that are located along the
diagonal?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Points on the diagonal have QTL that are located in the same position
as the gene, and so are local eQTL.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3-are-there-any-genome-locations-with-many-qtl" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-are-there-any-genome-locations-with-many-qtl" class="callout-inner">
<h3 class="callout-title">Challenge 3: Are there any genome locations with many QTL?</h3>
<div class="callout-content">
<p>Look at the transcriptome map and see if you can find any vertical
stripes? What do these vertical stripes mean and what might cause
them?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>There are vertical stripes on chromosomes 2, 5, and 11. There may be
more, depending on how you look at the plot. Since the QTL position is
on the X-axis, these stripes represent genes which are located through
out the genome, but all have a QTL in the same location.</p>
<p>These stripes imply that there is some genomic feature at the QTL
position which regulates the expression levels of many genes. This might
be a transcription factor or some other molecule which can regulate
transcription, possibly through multiple steps.</p>
</div>
</div>
</div>
</div>
<p>If you look at the plot, there are vertical bands of points which
stack over a single QTL location. These are called <em>eQTL
hotspots</em>. Rather than look at the transcriptome map, it may be
easier to look at the density of the eQTL along the genome. We have
provided a function called <code>eqtl_density_plot</code> to do
this.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span><span class="va">eqtl</span>, lod_thr <span class="op">=</span> <span class="va">eqtl_thr</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-eqtl_density-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot above shows the mouse genome on the X-axis and the number of
transcripts in a 4 Mb window on the Y-axis. It is difficult to say how
many genes must be involved to call something and eQTL hotspot. There
are permutation-based methods which require a large amount of time and
memory. In this case, we have called hotspots involving more than 100
genes eQTL hotspots.</p>
<p>Note that there appears to be an eQTL hotspot on chromosome 7 in the
plot above, but this is not evident in the transcriptome map.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span><span class="fu">filter</span><span class="op">(</span><span class="va">eqtl</span>, <span class="va">local</span> <span class="op">==</span> <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>                  lod_thr <span class="op">=</span> <span class="va">eqtl_thr</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Distant eQTL Density"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-ldistant_eqtl_density-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span><span class="fu">filter</span><span class="op">(</span><span class="va">eqtl</span>, <span class="va">local</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span>, </span>
<span>                  lod_thr <span class="op">=</span> <span class="va">eqtl_thr</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Local eQTL Density"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-local_eqtl_density-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From these two plots, it appears that the eQTL hotspots on most
chromosomes contain distant eQTL.</p>
<div id="challenge-4-compare-results-to-keller-et-al." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-compare-results-to-keller-et-al." class="callout-inner">
<h3 class="callout-title">Challenge 4: Compare results to Keller et al.</h3>
<div class="callout-content">
<p>Look at figure 3 in <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller et
al</a>. How do the eQTL density plots compare to their results?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>The eQTL density plots are largely similar. The hotspot on chromosome
11 contains more genes in our analysis than in the paper.</p>
</div>
</div>
</div>
</div>
<p>How can we find the location of the eQTL hotspots and the genes which
lie within each eQTL? We have written a function called
<code>get_eqtl_hotspots</code> to help you and have provides it in
<code>gg_transcriptome_map.R</code>. It requires the eQTL data, a LOD
threshold to filter the peaks, the number of genes required to call a
locus a hotspot, and the radius in Mb around the hotspot to use when
selecting genes which are part of the hotspot. We will use a 2 MB radius
around the hotspot, but may refine that later.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hotspots</span> <span class="op">&lt;-</span> <span class="fu">get_eqtl_hotspots</span><span class="op">(</span>data           <span class="op">=</span> <span class="va">eqtl</span>, </span>
<span>                              lod_thr        <span class="op">=</span> <span class="fl">7</span>, </span>
<span>                              hotspot_thr    <span class="op">=</span> <span class="fl">200</span>,</span>
<span>                              hotspot_radius <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s see how many hotspots we have and how many genes are in each
hotspot.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sapply</span><span class="op">(</span><span class="va">hotspots</span>, <span class="va">nrow</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  2   5   7  11
271 257 273 268 </code></pre>
</div>
<p>In the table above, the first row of values are the chromosomes on
which hotspots occur. The second row contains the number of genes in
each hotspot.</p>
<p>Where does each hotspot occur? We can get this information by taking
the mean of the eQTL positions in each hotspot.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sapply</span><span class="op">(</span><span class="va">hotspots</span>, <span class="kw">function</span><span class="op">(</span><span class="va">z</span><span class="op">)</span> <span class="op">{</span> <span class="fu">mean</span><span class="op">(</span><span class="va">z</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span> <span class="op">}</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        2         5         7        11
164.00699 146.30691  45.49353  70.47718 </code></pre>
</div>
<p>Let’s look at some of the genes in the chromosome 2 hotspot.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             gene_id qtl_chr  qtl_pos   qtl_lod        symbol gene_chr
1 ENSMUSG00000082514       2 162.0824 10.816809       Gm11452        2
2 ENSMUSG00000021902       2 162.3910  7.192061          Phf7       14
3 ENSMUSG00000021285       2 162.3945  7.576862      Ppp1r13b       12
4 ENSMUSG00000094856       2 162.5281 51.759581       Gm21962        3
5 ENSMUSG00000053141       2 162.6222 79.169112         Ptprt        2
6 ENSMUSG00000087267       2 162.6222 74.916874 4933427J07Rik        2
  gene_start  gene_end   marker.id local
1   162.9079 162.90861 2_162082435  TRUE
2    31.2377  31.25122 2_162391032 FALSE
3   111.8285 111.90805 2_162394483 FALSE
4   137.6715 137.67254 2_162528109 FALSE
5   161.5220 162.66115 2_162622197  TRUE
6   128.9557 128.95786 2_162622197 FALSE</code></pre>
</div>
<p>Let’s also look at the positions of the eQTL for these genes by
making a histogram of the positions.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">par</span><span class="op">(</span>plt <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.99</span>, <span class="fl">0.22</span>, <span class="fl">0.9</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="fu">table</span><span class="op">(</span><span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span>, </span>
<span>     las    <span class="op">=</span> <span class="fl">2</span>,</span>
<span>     main   <span class="op">=</span> <span class="st">"Positions of Chr 2 eQTL"</span>,</span>
<span>     xlab   <span class="op">=</span> <span class="st">""</span>,</span>
<span>     ylab   <span class="op">=</span> <span class="st">"Number of Genes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-hotspot_chr2_pos_hist-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From the plot above, we can see that many genes have eQTL which stack
up around 164 Mb. Are the other genes part of the same regulatory
network?</p>
<p>Let’s look at the correlation of all of these genes with each other.
First, we will get the genes from the expression data.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_2</span> <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">gene_id</span><span class="op">]</span></span></code></pre>
</div>
<p>Next, we will get the correlation of the genes with each other an
make a heatmap.</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor_2</span> <span class="op">&lt;-</span> <span class="fu">cor</span><span class="op">(</span><span class="va">expr_2</span><span class="op">)</span></span>
<span><span class="fu">heatmap</span><span class="op">(</span><span class="va">cor_2</span>, symm <span class="op">=</span> <span class="cn">TRUE</span>, main <span class="op">=</span> <span class="st">"Correlation of Chr2 Hotspot Genes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-chr2_hotspot_cor_heatmap-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot above shows a clustered heatmap in which red indicates
positive correlation and yellow indicates negative correlation.There are
“dendrograms” on the top and left, indicating the sort order.</p>
<p>We can split the genes into clusters by telling R how many clusters
we want. It will use the dendrogram to select a split which gives the
requested number of clusters. We are arbitrarily selecting three
clusters.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cl</span>     <span class="op">&lt;-</span> <span class="fu">hclust</span><span class="op">(</span><span class="fu">dist</span><span class="op">(</span><span class="va">cor_2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">cl</span>, labels <span class="op">=</span> <span class="cn">FALSE</span>, hang <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, main <span class="op">=</span> <span class="st">"Clustering of Chr2 Hotspot Genes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-clust_chr2_hotspot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cl_cut</span> <span class="op">&lt;-</span> <span class="fu">cutree</span><span class="op">(</span>tree <span class="op">=</span> <span class="va">cl</span>, k <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s look at the number of genes and the median eQTL position of
these clusters.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cl_spl</span> <span class="op">&lt;-</span> <span class="fu">split</span><span class="op">(</span><span class="va">cl_cut</span>, <span class="va">cl_cut</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">subcl</span> <span class="kw">in</span> <span class="va">cl_spl</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">cl_annot</span> <span class="op">&lt;-</span> <span class="fu">filter</span><span class="op">(</span><span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span>, <span class="va">gene_id</span> <span class="op">%in%</span> <span class="fu">names</span><span class="op">(</span><span class="va">subcl</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">cl_expr</span>  <span class="op">&lt;-</span> <span class="va">expr_2</span><span class="op">[</span>,<span class="fu">names</span><span class="op">(</span><span class="va">subcl</span><span class="op">)</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="fu">print</span><span class="op">(</span><span class="fu">paste</span><span class="op">(</span><span class="fu">nrow</span><span class="op">(</span><span class="va">cl_annot</span><span class="op">)</span>, <span class="fu">median</span><span class="op">(</span><span class="va">cl_annot</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "72 164.0704045"
[1] "94 164.022416"
[1] "105 164.022416"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">cl_spl</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, let’s make the heatmap again and place the cluster colors on
the axes.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">'black'</span>, <span class="st">'red'</span>, <span class="st">'blue'</span>, <span class="st">'orange'</span><span class="op">)</span><span class="op">[</span><span class="va">cl_cut</span><span class="op">]</span></span>
<span><span class="fu">heatmap</span><span class="op">(</span><span class="va">cor_2</span>, symm <span class="op">=</span> <span class="cn">TRUE</span>, RowSideColors <span class="op">=</span> <span class="va">cols</span>, ColSideColors <span class="op">=</span> <span class="va">cols</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-unnamed-chunk-2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In this case, we will filter the chromosome 2 hotspot to only include
the genes with an eQTL between 163.5 and 164.5 Mb.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hot_2</span> <span class="op">&lt;-</span> <span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">filter</span><span class="op">(</span><span class="fu">abs</span><span class="op">(</span><span class="va">qtl_pos</span> <span class="op">-</span> <span class="fl">164</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre>
</div>
<p>How many genes are there in the chromosome 2 hotspot now?</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">hot_2</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 149</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="principal-component-of-eqtl-hotspot-genes">Principal Component of eQTL Hotspot Genes<a class="anchor" aria-label="anchor" href="#principal-component-of-eqtl-hotspot-genes"></a></h3>
<p>One way to summarize the expression of genes in an eQTL hotspot is to
take the first principal component (PC1) of all of the genes in the
hotspot. PC1 should capture the largest amount of variance in the eQTL
hotspot and we can use this as a phenotype to map the hotspot.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_2</span> <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">hot_2</span><span class="op">$</span><span class="va">gene_id</span><span class="op">]</span></span>
<span><span class="va">pca_2</span>  <span class="op">&lt;-</span> <span class="fu">princomp</span><span class="op">(</span><span class="va">expr_2</span><span class="op">)</span></span>
<span><span class="va">pc1_2</span>  <span class="op">&lt;-</span> <span class="va">pca_2</span><span class="op">$</span><span class="va">scores</span><span class="op">[</span>,<span class="fl">1</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>We can then map PC1 as a phenotype and we should have a large QTL
peak in the same location as the hotspot.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pc1_lod</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                 pheno     <span class="op">=</span> <span class="va">pc1_2</span>,</span>
<span>                 kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">pc1_lod</span>, <span class="va">map</span>, main <span class="op">=</span> <span class="st">"PC1 of Chr 2 Hotspot"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-map_pc1_2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We can also look at the founder allele effects at the peak. We will
use a function in <code>gg_transcriptome_map.R</code> called
<code>plot_fit1()</code>.</p>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">find_peaks</span><span class="op">(</span><span class="va">pc1_lod</span>, <span class="va">map</span>, threshold <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="va">pr</span>    <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                          map       <span class="op">=</span> <span class="va">map</span>, </span>
<span>                          chr       <span class="op">=</span> <span class="va">peaks</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, </span>
<span>                          pos       <span class="op">=</span> <span class="va">peaks</span><span class="op">$</span><span class="va">pos</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">mod</span>   <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>,</span>
<span>              pheno     <span class="op">=</span> <span class="va">pc1_2</span>, </span>
<span>              kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">peaks</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>, </span>
<span>              addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_fit1</span><span class="op">(</span><span class="va">mod</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Chr 2 Hotspot Allele Effects"</span>,</span>
<span>       x     <span class="op">=</span> <span class="st">"Founder"</span>,</span>
<span>       y     <span class="op">=</span> <span class="st">"Founder Allele Effects"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/create-transcriptome-map-rendered-pc1_2_eff-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From the plot above, we can see that NZO and WSB have allele effects
which are different from the other strains. The direction of the allele
effects is arbitrary in the principal components, so we can’t determine
the direction of the allele effects from this plot.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>Transcriptome maps aid in understanding gene expression
regulation.</li>
<li>Local eQTL occur more frequently than distant eQTL.</li>
<li>Local eQTL appear along the diagonal in a transcriptome map and
distant eQTL appear on the off-diagonal.</li>
<li>Stacks of eQTL which appear over a single locus are called <em>eQTL
hotspots</em> and represent sets of genes which are transcriptionally
regulated by a single locus.</li>
<li>The first principal component of genes in and eQTL hotspot can be
used to summarize the genes in the hotspot.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>



      </div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="maximum-peaks-genes.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="mediation-analysis.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="maximum-peaks-genes.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Maximum eQTL Peaks
        </a>
        <a class="chapter-link float-end" href="mediation-analysis.html" rel="next">
          Next: Mediation Analysis...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/create-transcriptome-map.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.10" class="external-link">sandpaper (0.16.10)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.7" class="external-link">pegboard (0.7.7)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/create-transcriptome-map.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene expression, quantitative trait, genetic mapping, Diversity Outbred population",
  "name": "Creating A Transcriptome Map",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/create-transcriptome-map.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/create-transcriptome-map.html",
  "dateCreated": "2024-11-07",
  "dateModified": "2025-01-14",
  "datePublished": "2025-01-14"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

