<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Expression Quantitative Trait Locus (eQTL) Mapping: Transcriptome Map of cis and trans eQTL</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicons/incubator/favicon-16x16.png"><link rel="manifest" href="favicons/incubator/site.webmanifest"><link rel="mask-icon" href="favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/full-transcriptome-map.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Expression Quantitative Trait Locus (eQTL) Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Expression Quantitative Trait Locus (eQTL) Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"></ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Expression Quantitative Trait Locus (eQTL) Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 69%" class="percentage">
    69%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 69%" aria-valuenow="69" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/full-transcriptome-map.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="review-eqtl-study.html">2. Genetic Drivers of Pancreatic Islet Function</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="load-explore-data.html">3. Load and Explore Data</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="map-one-eqtl.html">4. Mapping A Single Gene Expression Trait</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="map-many-eqtls.html">5. Mapping Many Gene Expression Traits</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="create-transcriptome-map.html">6. Creating A Transcriptome Map</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        7. Transcriptome Map of cis and trans eQTL
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#load-data">Load Data</a></li>
<li><a href="#islet-rnaseq-eqtl-hotspots">Islet RNASeq eQTL Hotspots</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="maximum-peaks-genes.html">8. Maximum eQTL Peaks and Nearby Genes</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="interpret-results.html">9. Interpreting qtl2 results</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="mediation-analysis.html">10. Mediation Analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>

                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="create-transcriptome-map.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="maximum-peaks-genes.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="create-transcriptome-map.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Creating A
        </a>
        <a class="chapter-link float-end" href="maximum-peaks-genes.html" rel="next">
          Next: Maximum eQTL Peaks...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Transcriptome Map of cis and trans eQTL</h1>
        <p>Last updated on 2024-11-29 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/full-transcriptome-map.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How do I create a full transcriptome map?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Explain how to use markdown with the new lesson template</li>
<li>Demonstrate how to include pieces of code, figures, and nested
challenge blocks</li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a></h3>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2ggplot</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># source("../code/gg_transcriptome_map.R")</span></span></code></pre>
</div>
</div>
<section><h2 class="section-heading" id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a></h2>
<hr class="half-width"><p>Load in the RNA-seq eQTL mapping results.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># loading previous results</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/dataset.islet.rnaseq.RData"</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we need to format the column names of our eQTL results to that
the <code>ggtmap</code> function can use the results.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_summary</span> <span class="op">=</span> <span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">lod.peaks</span></span>
<span></span>
<span><span class="co"># Get gene positions.</span></span>
<span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu">get_ensembl_genes</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>ensembl    <span class="op">=</span> <span class="va">ensembl</span><span class="op">$</span><span class="va">gene_id</span>,</span>
<span>                 gene_chr   <span class="op">=</span> <span class="fu">seqnames</span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span>,</span>
<span>                 gene_start <span class="op">=</span> <span class="fu">start</span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e-6</span>,</span>
<span>                 gene_end   <span class="op">=</span> <span class="fu">end</span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span>   <span class="op">*</span> <span class="fl">1e-6</span>,</span>
<span>                 stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create eQTL table for transcriptome map function.</span></span>
<span><span class="va">lod_summary</span> <span class="op">&lt;-</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">rename</span><span class="op">(</span>annot.id  <span class="op">=</span> <span class="st">"ensembl"</span>,</span>
<span>                        chrom     <span class="op">=</span> <span class="st">"qtl_chr"</span>,</span>
<span>                        pos       <span class="op">=</span> <span class="st">"qtl_pos"</span>,</span>
<span>                        lod       <span class="op">=</span> <span class="st">"qtl_lod"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">left_join</span><span class="op">(</span><span class="va">df</span>, by <span class="op">=</span> <span class="st">"ensembl"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">mutate</span><span class="op">(</span>marker.id <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">qtl_chr</span>, <span class="va">qtl_pos</span> <span class="op">*</span> <span class="fl">1e6</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>,</span>
<span>                        gene_chr  <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">gene_chr</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                        qtl_chr   <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">qtl_chr</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">mutate</span><span class="op">(</span>cis <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">qtl_chr</span> <span class="op">==</span> <span class="va">gene_chr</span> <span class="op">&amp;</span> <span class="fu">abs</span><span class="op">(</span><span class="va">gene_start</span> <span class="op">-</span> <span class="va">qtl_pos</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">4</span>, <span class="st">"cis"</span>, <span class="st">"trans"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="plot-transcriptome-map">Plot Transcriptome Map<a class="anchor" aria-label="anchor" href="#plot-transcriptome-map"></a></h3>
<p>In the previous lesson, we mapped the QTL locations of 50 genes. In
this lesson, we will map the QTL positions of genes.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggtmap</span><span class="op">(</span>data <span class="op">=</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span> <span class="op">&gt;=</span> <span class="fl">7.18</span><span class="op">)</span>,</span>
<span>      cis.points <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      cis.radius <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<p>This transcriptome map is definitely a lot more crowded than the one
in the previous lesson. Again, the gene locations are shown on the
X-axis and the QTL locations are shown on the Y-axis.</p>
<div id="challenge-1-what-patterns-among-the-points-do-you-see-in-the-transcriptome-map" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-what-patterns-among-the-points-do-you-see-in-the-transcriptome-map" class="callout-inner">
<h3 class="callout-title">Challenge 1: What patterns among the points do you see in the transcriptome map?</h3>
<div class="callout-content">

</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>There are at least two patterns. One is the dense diagonal line of
cis-eQTL. The other is the increased density of QTL in vertical
lines.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2-what-would-a-vertical-band-in-the-transcriptome-map-mean" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-what-would-a-vertical-band-in-the-transcriptome-map-mean" class="callout-inner">
<h3 class="callout-title">Challenge 2: What would a vertical band in the transcriptome map mean?</h3>
<div class="callout-content">

</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>A vertical band indicates that one locus regulates the expression of
many genes.</p>
</div>
</div>
</div>
</div>
<p>Look at the transcriptome map. How many vertical bands do you see and
which chromosomes are they on?</p>
</div>
<div class="section level3">
<h3 id="qtl-density-plot">QTL Density Plot<a class="anchor" aria-label="anchor" href="#qtl-density-plot"></a></h3>
<p>In the transcriptome map above, we noticed vertical banding patterns
in the eQTL, which indicate that one locus may regulate the expression
of dozens of genes. How many genes are regulated by each locus and which
genes are they? In order to address this question, we need to make a
plot of the density of eQTL along the genome. This is like stacking up
the eQTL onto the X-axis.</p>
<p>We have provided a function to do this in the
<code>gg_transcriptome_map.R</code> file in the <code>code</code>
directory of this lesson. The function is called
<code>eqtl_density_plot</code> and takes the following arguments:</p>
<ul><li>
<code>data</code>: data.frame (or tibble) with the following
columns:
<ul><li>
<code>ensembl</code>: (required) character string containing the
Ensembl gene ID.</li>
<li>
<code>qtl_chr</code>: (required) character string containing QTL
chromsome.</li>
<li>
<code>qtl_pos</code>: (required) floating point number containing
the QTL position in Mb.</li>
<li>
<code>qtl_lod</code>: (optional) floating point number containing
the LOD score.</li>
<li>
<code>gene_chr</code>: (optional) character string containing
transcript chromosome.</li>
<li>
<code>gene_start</code>: (optional) character string containing
transcript start position in Mb.</li>
<li>
<code>gene_end</code>: (optional) character string containing
transcript end position in Mb.</li>
</ul></li>
<li>
<code>lod_thr</code>: numeric value that is the LOD above which QTL
will be retained. Default = 7.</li>
</ul><p>This function has been designed to use the same data structure as we
used to create the transcriptome map. First, we will look at the overall
QTL density for all peaks with LOD &gt; 7.18.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">lod_summary</span>, lod_thr <span class="op">=</span> <span class="fl">7.18</span><span class="op">)</span></span></code></pre>
</div>
<p>There are clearly some loci that have an enrichment of eQTL. We have
drawn a dashed line at <code>100"</code> as an arbitrary cutoff as a
potential cutoff to use when selecting peaks.</p>
<p>Compare this plot with the transcriptome map, in which we saw
vertical bands of eQTL. Do the peaks in the eQTL density plot match the
bands in the transcriptome map?</p>
<p>Next, we will look at the locations of the cis-eQTL. We must also
select a LOD threshold. We will use 7.18 since this is what was used in
the Keller et al. paper.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">filter</span><span class="op">(</span><span class="va">lod_summary</span>, <span class="va">cis</span> <span class="op">==</span> <span class="st">"cis"</span><span class="op">)</span>, lod_thr <span class="op">=</span> <span class="fl">7.18</span><span class="op">)</span></span></code></pre>
</div>
<p>In the plot above, there are many loci that have many genes
associated with their expression. Some loci have over 100 genes
associated with them. For example, there is a locus on chromosome 17
that may regulate over 100 genes. In this case, we are looking at
cis-eQTL, QTL which are co-located with the gene. What might this mean
biologically? Perhaps there is a mutation which causes a region of
chromatin to open, which leads to increased expression of a set of
genes. This increased expression may have biological consequences.</p>
<p>Next, we will create an eQTL density plot of the trans-eQTL. These
are QTL for which the gene and QTL are far from each other.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">filter</span><span class="op">(</span><span class="va">lod_summary</span>, <span class="va">cis</span> <span class="op">==</span> <span class="st">"trans"</span><span class="op">)</span>, lod_thr <span class="op">=</span> <span class="fl">7.18</span><span class="op">)</span></span></code></pre>
</div>
<p>In this case, we see much taller peaks than in the cis-eQTL density
plot and these peaks match the ones in the overall eQTL density
plot.</p>
<p>There are many potential explanations for a trans-eQTL band. There
may be a polymorphism in a transcription factor which alters the
expression of many other genes. Or there may be a polymorphism that
alters an amino acid and prevents it from binding properly to another
protein in a signalling cascade. Biologists are often interested in
these trans-eQTL bands, which are called “eQTL hotspots”. A set of over
100 genes with differential expression between genotypes may help us to
understand the biology behind variation in higher level phenotypes. It
is also possible that one of the genes with a cis-eQTL under the eQTL
hotspot regulates the expression of the remaining hotspot genes.</p>
</div>
</section><section><h2 class="section-heading" id="islet-rnaseq-eqtl-hotspots">Islet RNASeq eQTL Hotspots<a class="anchor" aria-label="anchor" href="#islet-rnaseq-eqtl-hotspots"></a></h2>
<hr class="half-width"><div class="section level3">
<h3 id="select-eqtl-hotspots">Select eQTL Hotspots<a class="anchor" aria-label="anchor" href="#select-eqtl-hotspots"></a></h3>
<p>There are several decisions to make when selecting eQTL hotspots.
What LOD threshold should you use to select the genes that comprise
hotspots? What number of genes should you use as a threshold to call a
peak an eQTL hotspot? In this case, the authors select a LOD of 7.18 and
decided that 100 trans-regulated genes was a useful threshold.</p>
<blockquote>
<p>TBD: Add references for eQTL module selection. Possibly augment
lesson?</p>
</blockquote>
<p>In order to identify eQTL hotspots, we will look at the density of
trans-eQTL along the genome using the <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/density" class="external-link">density</a>
function. We will filter to retain QTL with LOD &gt; 7.8 and will get
the position on each chromosome with the highest density of eQTL.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl_dens</span> <span class="op">=</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span> <span class="op">&gt;</span> <span class="fl">7.18</span> <span class="op">&amp;</span> <span class="va">cis</span> <span class="op">==</span> <span class="st">'trans'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">group_by</span><span class="op">(</span><span class="va">qtl_chr</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">summarize</span><span class="op">(</span>dens_x <span class="op">=</span> <span class="fu">density</span><span class="op">(</span><span class="va">qtl_pos</span>, adjust <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span><span class="op">$</span><span class="va">x</span>,</span>
<span>                       dens_y <span class="op">=</span> <span class="fu">density</span><span class="op">(</span><span class="va">qtl_pos</span>, adjust <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span><span class="op">$</span><span class="va">y</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">slice_max</span><span class="op">(</span><span class="va">dens_y</span>, n <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">qtl_dens</span></span></code></pre>
</div>
<p>In the table above, there is one row per chromosome. <code>x</code>
is the position on each chromosome with the highest density of eQTL.
<code>y</code> is the density, which we needed to obtain the
<code>x</code> position but is not used further.</p>
<p>Now that we have the location of maximum eQTL density on each
chromosome, we will count the number of eQTL within +/- 2 Mb of the
center.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hotspots</span> <span class="op">=</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">lod_summary</span>, <span class="fu">select</span><span class="op">(</span><span class="va">qtl_dens</span>, <span class="va">qtl_chr</span>, <span class="va">dens_x</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span> <span class="op">&gt;</span> <span class="fl">7.18</span> <span class="op">&amp;</span> <span class="va">cis</span> <span class="op">==</span> <span class="st">'trans'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">mutate</span><span class="op">(</span>pos_diff <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">dens_x</span> <span class="op">-</span> <span class="va">qtl_pos</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">pos_diff</span> <span class="op">&lt;=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pos_diff</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">rename</span><span class="op">(</span>dens_x <span class="op">=</span> <span class="st">'center'</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">hotspots</span><span class="op">)</span></span></code></pre>
</div>
<p>Now that we have a list of genes near the position of highest eQTL
density on each chromosome, we can count the number of genes in each
potential hotspot and retain the ones containing more than 100
genes.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hotspots</span> <span class="op">=</span> <span class="va">hotspots</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">count</span><span class="op">(</span><span class="va">qtl_chr</span>, <span class="va">center</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">n</span> <span class="op">&gt;=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">hotspots</span>, caption <span class="op">=</span> <span class="st">"Islet trans-eQTL hotspots"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="chr-11-eqtl-hotspot">Chr 11 eQTL Hotspot<a class="anchor" aria-label="anchor" href="#chr-11-eqtl-hotspot"></a></h3>
<p>From this analysis, we have identified five eQTL hotspots. Due to
time constraints, we will not examine all of them in depth. We will look
at the chromosome 11 eQTL hotspot in greater depth. It contains 163
trans-eQTL. First, we will get all of the genes with an eQTL in within
+/- 2 Mb of the chromosome 11 eQTL hotspot.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr11_mid</span> <span class="op">=</span>  <span class="va">hotspots</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_chr</span> <span class="op">==</span> <span class="st">'11'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">pull</span><span class="op">(</span><span class="va">center</span><span class="op">)</span></span>
<span><span class="va">chr11_eqtl</span> <span class="op">=</span> <span class="va">lod_summary</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span> <span class="op">&gt;</span>  <span class="fl">7.18</span> <span class="op">&amp;</span></span>
<span>                      <span class="va">qtl_chr</span> <span class="op">==</span> <span class="st">'11'</span> <span class="op">&amp;</span></span>
<span>                      <span class="fu">abs</span><span class="op">(</span><span class="va">chr11_mid</span> <span class="op">-</span> <span class="va">qtl_pos</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span> <span class="op">&amp;</span></span>
<span>                      <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">cis</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will filter to retain the cis-eQTL in the same interval. It
is possible that one or more genes near 71.5 Mb on chromosome 11 have a
cis-eQTL, which in turn alters the expression of the trans-eQTL
genes.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr11_cis</span> <span class="op">=</span> <span class="va">chr11_eqtl</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">filter</span><span class="op">(</span><span class="va">cis</span> <span class="op">==</span> <span class="st">'cis'</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">chr11_cis</span>, caption <span class="op">=</span> <span class="st">'Chr 11 cis-eQTL'</span><span class="op">)</span></span></code></pre>
</div>
<p>As you can see, there are cis-eQTL genes under the chromosome 11 eQTL
hotspot. This is a large number of candidate genes to screen. There may
be more genes with non-synonymous, splice, or stop mutations under the
eQTL hotspot as well. The Sanger Mouse Genomes website has been removed
and we are uncertain if it will be replaced. There are two websites
where you can find this information:</p>
<ol style="list-style-type: decimal"><li>
<a href="http://nov2020.archive.ensembl.org/Mus_musculus/Info/Index" class="external-link">Ensembl</a>:
Once you search for a gene, you can select the “Variant Table” under
“Genetic Variation” in the left navigation panel.</li>
<li>
<a href="https://churchilllab.jax.org/foundersnps/search" class="external-link">Founder
Variant Search</a>: You can use this site to search for variants in
specific genes or genomic intervals.</li>
</ol><p>Next, we will get the expression of the genes in the chromosome 11
eQTL hotspot. The expression data is a numeric matrix, so we will use
the colnames to filter the genes.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr11_genes</span> <span class="op">=</span> <span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">expr</span><span class="op">[</span>,<span class="va">chr11_eqtl</span><span class="op">$</span><span class="va">ensembl</span><span class="op">]</span></span></code></pre>
</div>
<p>Next, we will join the eQTL and expression data and write it out to a
file. This way you will have all of the data for the hotspot in one
place.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr11_expr</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span>ensembl <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">chr11_genes</span><span class="op">)</span>,</span>
<span>              <span class="fu">data.frame</span><span class="op">(</span><span class="fu">t</span><span class="op">(</span><span class="va">chr11_genes</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">chr11_all</span> <span class="op">=</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">chr11_eqtl</span>, <span class="va">chr11_expr</span>, by <span class="op">=</span> <span class="st">'ensembl'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">write_csv</span><span class="op">(</span><span class="va">chr11_all</span>, file <span class="op">=</span> <span class="fu">file.path</span><span class="op">(</span><span class="st">'../results/'</span>, <span class="st">'chr11_eqtl_genes.csv'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="create-transcriptome-map.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="maximum-peaks-genes.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="create-transcriptome-map.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Creating A
        </a>
        <a class="chapter-link float-end" href="maximum-peaks-genes.html" rel="next">
          Next: Maximum eQTL Peaks...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/full-transcriptome-map.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.10" class="external-link">sandpaper (0.16.10)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.7" class="external-link">pegboard (0.7.7)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/full-transcriptome-map.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene expression, quantitative trait, genetic mapping, Diversity Outbred population",
  "name": "Transcriptome Map of cis and trans eQTL",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/full-transcriptome-map.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/full-transcriptome-map.html",
  "dateCreated": "2024-11-07",
  "dateModified": "2024-11-29",
  "datePublished": "2024-11-29"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

