<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Expression Quantitative Trait Locus (eQTL) Mapping: Mediation Analysis</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicons/incubator/favicon-16x16.png"><link rel="manifest" href="favicons/incubator/site.webmanifest"><link rel="mask-icon" href="favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/mediation-analysis.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Expression Quantitative Trait Locus (eQTL) Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Expression Quantitative Trait Locus (eQTL) Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"></ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Expression Quantitative Trait Locus (eQTL) Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 91%" class="percentage">
    91%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 91%" aria-valuenow="91" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/mediation-analysis.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="review-eqtl-study.html">2. Genetic Drivers of Pancreatic Islet Function</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="load-explore-data.html">3. Load and Explore Data</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="map-one-eqtl.html">4. Mapping A Single Gene Expression Trait</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="map-many-eqtls.html">5. Mapping Many Gene Expression Traits</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="maximum-peaks-genes.html">6. Maximum eQTL Peaks and Nearby Genes</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="create-transcriptome-map.html">7. Creating A Transcriptome Map</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        8. Mediation Analysis
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#introduction">Introduction</a></li>
<li><a href="#using-gene-expression-as-a-covariate">Using Gene Expression as a Covariate</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>

                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="create-transcriptome-map.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="create-transcriptome-map.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Creating A
        </a>
        <a class="chapter-link float-end" href="index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Mediation Analysis</h1>
        <p>Last updated on 2024-12-11 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/mediation-analysis.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What is mediation analysis?</li>
<li>How do I explore causal relations with mediation analysis?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Describe mediation analysis as applied in genetics and
genomics.</li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pheno</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_pheno.rds'</span><span class="op">)</span></span>
<span><span class="va">pheno_dict</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_pheno_dict.rds'</span><span class="op">)</span></span>
<span><span class="va">covar</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_covar.rds'</span><span class="op">)</span></span>
<span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span>
<span><span class="va">addcovar</span>   <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span>, data <span class="op">=</span> <span class="va">covar</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">annot</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_expr_annot.rds'</span><span class="op">)</span></span>
<span><span class="va">expr_rz</span>    <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_expr_rz.rds'</span><span class="op">)</span></span>
<span><span class="va">map</span>        <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_map.rds'</span><span class="op">)</span></span>
<span><span class="va">probs</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="fu">url</span><span class="op">(</span><span class="st">'https://thejacksonlaboratory.box.com/shared/static/4hy4hbjyrxjbrzh570i4g02r62bx3lgk.rds'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readRDS(file = url("https://thejacksonlaboratory.box.com/shared/static/4hy4hbjyrxjbrzh570i4g02r62bx3lgk.rds")): cannot open the connection to 'https://thejacksonlaboratory.box.com/shared/static/4hy4hbjyrxjbrzh570i4g02r62bx3lgk.rds'</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">K</span>          <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_kinship.rds'</span><span class="op">)</span></span>
<span><span class="va">ins_tauc</span>   <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">[</span>, <span class="st">'Ins_tAUC'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC_log</span> <span class="op">&lt;-</span> <span class="fu">log</span><span class="op">(</span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC</span><span class="op">)</span></span>
<span><span class="va">ins_lod</span>    <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/ins_tauc_lod.rds'</span><span class="op">)</span></span>
<span><span class="va">eperm</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/ENSMUSG00000020679_perm_1000.rds'</span><span class="op">)</span></span>
<span><span class="va">hotspots</span>   <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/eqtl_hotspots.rds'</span><span class="op">)</span></span>
<span><span class="va">pc1_2</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/eqtl_hotspot_chr2_pc1.rds'</span><span class="op">)</span></span>
<span><span class="va">pc1_lod</span>    <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/eqtl_hotspot_chr2_pc1_lod.rds'</span><span class="op">)</span></span></code></pre>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a></h2>
<hr class="half-width"><p>GWAS studies show that most disease-associated variants are found in
non-coding regions. This fact leads to the idea that regulation of gene
expression is an important mechanism enabling genetic variants to affect
complex traits. Mediation analysis can identify a causal chain from
genetic variants to molecular and clinical phenotypes. The graphic below
shows complete mediation, in which a predictor variable does not
directly impact the response variable. Rather, it directly influences
the mediator (path a). The mediator has a direct impact on the response
variable (path b). We would see (path c) the relationship between
predictor and response, not knowing that a mediator intervenes in this
relationship.</p>
<figure><img src="fig/mediation-analysis.png" alt="Mediation Analysis" class="figure mx-auto d-block"><div class="figcaption">In complete mediation an predictor variable
influences a response variable indirectly through a mediator
variable.</div>
</figure><p>Mediation analysis is widely used in the social sciences including
psychology. In biomedicine, mediation analysis has been employed to
investigate how gene expression mediates the effects of genetic variants
on complex phenotypes and disease.</p>
<p>For example, a genetic variant (non-coding SNP) indirectly regulates
expression of gene 2 through a mediator, gene 1. The SNP regulates
expression of gene 1 locally, and expression of gene 1 influences
expression of gene 2 distally.</p>
<figure><img src="fig/mediation-trans-noncoding-SNP.png" alt="mediation figure" class="figure mx-auto d-block"><div class="figcaption">A non-coding SNP affects expression of gene 1
locally. Gene 1 mediates expression of gene 2 distally.</div>
</figure><p>Instead of the expression of one gene impacting another, expression
of gene 1 in the graphic above could impact a physiological phenotype
like blood glucose. Expression of gene 1 would mediate the relationship
between the non-coding SNP and the glucose phenotype.</p>
<p>Gene Akr1e1 is located on chromosome 13 in mouse.</p>
<figure><img src="fig/LOD-plot-Akr1e1.png" alt="Akr1e1 QTL plot" class="figure mx-auto d-block"><div class="figcaption">Chromosome 13 gene Akr1e1 is affected by
expression in both local and distant by genes on chromosomes 13 and
4.</div>
</figure><div id="challenge-1-interpreting-akr1e1-eqtl-plot." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-interpreting-akr1e1-eqtl-plot." class="callout-inner">
<h3 class="callout-title">Challenge 1: Interpreting <em>Akr1e1</em> eQTL
plot.</h3>
<div class="callout-content">
<p>How would you interpret the <em>Akr1e1</em> LOD plot above? On which
chromosome(s) would you expect to find the driver gene(s)? The
SNP(s)?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p><em>Ark1e1</em> expression is regulated by some variants on
chromosome 13 and other variants on chromosome 4. The largest driver
gene occurs on chromosome 4 because the LOD peak is higher there,
indicating that the genotype on chromosome 4 explains more of the
variance than the one on chromosome 13.</p>
</div>
</div>
</div>
</div>
<p>Myo15b is located on chromosome 11.</p>
<figure><img src="fig/LOD-plot-Myo15b.png" alt="Myo15b QTL plot" class="figure mx-auto d-block"><div class="figcaption">Chromosome 11 gene Myo15b is affected by
expression of a gene in the chromosome 2 hotspot.</div>
</figure><div id="challenge-2-interpreting-myo15b-eqtl-plot." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-interpreting-myo15b-eqtl-plot." class="callout-inner">
<h3 class="callout-title">Challenge 2: Interpreting <em>Myo15b</em> eQTL
plot.</h3>
<div class="callout-content">
<p>How would you interpret the <em>Myo15b</em> LOD plot above? On which
chromosome(s) would you expect to find the driver gene(s)? The
SNP(s)?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p><em>Myo15b</em> is regulated by variants on chromosome 2 because the
largest LOD peak is located there.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="using-gene-expression-as-a-covariate">Using Gene Expression as a Covariate<a class="anchor" aria-label="anchor" href="#using-gene-expression-as-a-covariate"></a></h2>
<hr class="half-width"><p>When we are searching for a gene whose expression influences a
physiological phenotype, we may not be looking for a missense SNP in the
the gene. Instead, we may be looking for a SNP in a regulatory region
which influence the expression of the gene, which in turn influences the
physiological phenotype. However, the exact sequence of regulatory
regions and the genes that they influence is not as well annotated as
protein-coding regions of genes. So, rather than search for the
regulatory SNP, we will search for genes which influence the trait and
we will not seek to identify the causal SNP.</p>
<p>Next, we are going to create our own mediation function. We will walk
through it step-by-step so that you understand why we are performing
each step.</p>
<p>Mediation analysis bring together a lot of information, to the
function will have a LOT of variables.</p>
<p>First, we need to pass in the chromosome, position, and LOD score of
the QTL at which we are mapping. Remember, in a mediation scan, we map a
phenotype at <strong>ONE</strong> locus and use different genes as
covariates.</p>
<p>Next, since we are mapping, we need to pass in the genoprobs, the
original phenotype which produced the QTL peak, the kinship matrices,
the covariates matrix, and the markers map. In this case, our
“phenotype” will be PC1 from the genes in the chromosome 2 eQTL
hotspot.</p>
<p>The last two arguments will be the expression matrix and the gene
annotation.</p>
<p>With all of those pieces in place, we will build the function
below.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform mediation analysis on one locus using all of the available genes on</span></span>
<span><span class="co"># the corresponding chromosome.</span></span>
<span><span class="co"># qtl_chr: character string containing one of the chromosome names in genoprobs.</span></span>
<span><span class="co">#      Must be 1:19 or X.</span></span>
<span><span class="co"># qtl_pos: floating point number that is the Mb position of the QTL to map at.</span></span>
<span><span class="co"># qtl_lod: floating point number that is the LOD score at the QTL.</span></span>
<span><span class="co"># genoprobs: qtl2-style list containing genoprobs for all chromosomes.</span></span>
<span><span class="co"># pheno: numeric matrix containing one column with the phenotype values. Sample</span></span>
<span><span class="co">#        IDs must be in rownames.</span></span>
<span><span class="co"># K: qtl2-style list of LOCO kinship matrices.</span></span>
<span><span class="co"># addcovar: numeric matrix of additive covariate created with something like</span></span>
<span><span class="co">#           model.matrix(). Must be ready to pass into scan1() directly.</span></span>
<span><span class="co"># map: qtl2-style list containing marker map. Markers must match the markers in</span></span>
<span><span class="co">#      genoprobs.</span></span>
<span><span class="co"># expr: numeric matrix of normalized expression values. Sample in rows and genes</span></span>
<span><span class="co">#       in columns.</span></span>
<span><span class="co"># annot: data.frame containing the gene annotation. Genes in rows must be in the</span></span>
<span><span class="co">#        same order as the genes in columns in "expr".</span></span>
<span><span class="va">mediation</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">qtl_chr</span>, <span class="va">qtl_pos</span>, <span class="va">qtl_lod</span>, <span class="va">genoprobs</span>, <span class="va">pheno</span>, <span class="va">K</span>, <span class="va">addcovar</span>, </span>
<span>                      <span class="va">map</span>, <span class="va">expr</span>, <span class="va">annot</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>  <span class="co"># Get the genoprobs at the QTL peak.</span></span>
<span>  <span class="va">pr</span> <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                         map       <span class="op">=</span> <span class="va">map</span>, </span>
<span>                         chr       <span class="op">=</span> <span class="va">qtl_chr</span>, </span>
<span>                         pos       <span class="op">=</span> <span class="va">qtl_pos</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Get the genes on chromosome 11.</span></span>
<span>  <span class="va">genes_chr</span> <span class="op">&lt;-</span> <span class="va">annot</span> <span class="op">|&gt;</span></span>
<span>                 <span class="fu">filter</span><span class="op">(</span><span class="va">chr</span> <span class="op">==</span> <span class="va">qtl_chr</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                 <span class="fu">pull</span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span></span>
<span>  <span class="va">expr_chr</span>  <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>, <span class="va">genes_chr</span><span class="op">]</span></span>
<span></span>
<span>  <span class="co"># Subset the covariates to include only the samples in the expression data.</span></span>
<span>  <span class="va">addcovar_expr</span> <span class="op">&lt;-</span> <span class="va">addcovar</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">expr_chr</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span>  <span class="co"># Create the results.</span></span>
<span>  <span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>gene_id  <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">expr_chr</span><span class="op">)</span>,</span>
<span>                        base_lod <span class="op">=</span> <span class="va">qtl_lod</span>,</span>
<span>                        med_lod  <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">left_join</span><span class="op">(</span><span class="fu">select</span><span class="op">(</span><span class="va">annot</span>, <span class="va">gene_id</span>, <span class="va">symbol</span>, <span class="va">middle</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Loop through each gene, add it to the covariates, and map it at the QTL </span></span>
<span>  <span class="co"># marker.</span></span>
<span>  <span class="va">t_init</span> <span class="op">&lt;-</span> <span class="fu">proc.time</span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu">ncol</span><span class="op">(</span><span class="va">expr_chr</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    </span>
<span>    <span class="co"># Show progress.</span></span>
<span>    <span class="kw">if</span><span class="op">(</span><span class="va">i</span> <span class="op">%%</span> <span class="fl">50</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>      </span>
<span>      <span class="va">t2</span> <span class="op">&lt;-</span> <span class="fu">proc.time</span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span></span>
<span>      <span class="fu">print</span><span class="op">(</span><span class="fu">paste</span><span class="op">(</span><span class="va">i</span>, <span class="st">'of'</span>, <span class="fu">ncol</span><span class="op">(</span><span class="va">expr_chr</span><span class="op">)</span>, <span class="st">":"</span>, <span class="op">(</span><span class="va">t2</span> <span class="op">-</span> <span class="va">t_init</span><span class="op">)</span>, <span class="st">"seconds elapsed."</span><span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>    <span class="op">}</span> <span class="co"># for(i)</span></span>
<span>  </span>
<span>    <span class="co"># Create covariates with the current gene's expression.</span></span>
<span>    <span class="va">addcovar_tmp</span> <span class="op">=</span> <span class="fu">cbind</span><span class="op">(</span><span class="va">addcovar_expr</span>, <span class="va">expr_chr</span><span class="op">[</span>,<span class="va">i</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Fit the QTL model at the QTL marker with the new covariates.</span></span>
<span>    <span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>,</span>
<span>                pheno     <span class="op">=</span> <span class="va">pheno</span>,</span>
<span>                kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">qtl_chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                addcovar  <span class="op">=</span> <span class="va">addcovar_tmp</span><span class="op">)</span></span>
<span>  </span>
<span>    <span class="co"># Add the LOD to the results.</span></span>
<span>    <span class="va">results</span><span class="op">$</span><span class="va">med_lod</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">lod</span></span>
<span>    </span>
<span>  <span class="op">}</span> <span class="co"># for(i)</span></span>
<span></span>
<span>  <span class="co"># Subtract the meditation LODs from the baseline LOD.</span></span>
<span>  <span class="va">results</span> <span class="op">&lt;-</span> <span class="va">results</span> <span class="op">|&gt;</span></span>
<span>               <span class="fu">mutate</span><span class="op">(</span>lod_drop <span class="op">=</span> <span class="va">med_lod</span> <span class="op">-</span> <span class="va">base_lod</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Return the results.</span></span>
<span>  <span class="kw">return</span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span></span>
<span><span class="op">}</span> <span class="co"># mediation()</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl_pos</span> <span class="op">&lt;-</span> <span class="fu">mean</span><span class="op">(</span><span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">med_2</span> <span class="op">&lt;-</span> <span class="fu">mediation</span><span class="op">(</span>qtl_chr  <span class="op">=</span> <span class="st">"2"</span>, </span>
<span>                   qtl_pos  <span class="op">=</span> <span class="va">qtl_pos</span>,</span>
<span>                   qtl_lod  <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">pc1_lod</span><span class="op">)</span>, </span>
<span>                   genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno    <span class="op">=</span> <span class="va">pc1_2</span>, </span>
<span>                   K        <span class="op">=</span> <span class="va">K</span>, </span>
<span>                   addcovar <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                   map      <span class="op">=</span> <span class="va">map</span>, </span>
<span>                   expr     <span class="op">=</span> <span class="va">expr_rz</span>, </span>
<span>                   annot    <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in mediation(qtl_chr = "2", qtl_pos = qtl_pos, qtl_lod = max(pc1_lod), : object 'probs' not found</code></pre>
</div>
<p>Now that we have the change in LOD scores associated with using each
gene as a covariate, we can plot the LOD change along chromosome 11.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">end</span>  <span class="op">&lt;-</span> <span class="fu">ceiling</span><span class="op">(</span><span class="fu">max</span><span class="op">(</span><span class="va">med_2</span><span class="op">$</span><span class="va">middle</span> <span class="op">/</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'med_2' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">med_2</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">middle</span>, <span class="va">lod_drop</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="va">end</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mediation Analysis for Chr 2 PC1"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"Position (Mb)"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"LOD Drop"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>size <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'med_2' not found</code></pre>
</div>
<p>In the plot above, the position of each gene along chromosome 2 is
shown on the X-axis and the LOD drop is shown on the Y-axes. Most genes
don’t change the LOD score by very much. It seems that, by chance, LOD
scores can vary by plus or minus 3. In this case, we are searching for
genes which decrease the LOD score by the largest amount. In this case,
there is one gene which reduces the LOD .</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">med_2</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">lod_drop</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">20</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'med_2' not found</code></pre>
</div>
<div id="challenge-3-keller-et-al-chromosome-2-hotspot-candidate-gene" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-keller-et-al-chromosome-2-hotspot-candidate-gene" class="callout-inner">
<h3 class="callout-title">Challenge 3: Keller et al chromosome 2 hotspot candidate gene?</h3>
<div class="callout-content">
<p>Look at Figure 4C in <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller et
al</a> and see which gene the selected in their mediaiton analysis.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>The authors found that <em>Hnf4a</em> reduces the LOD more than any
other gene.</p>
</div>
</div>
</div>
</div>
<div id="challenge-4-look-up-dpp4-in-pubmed" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-look-up-dpp4-in-pubmed" class="callout-inner">
<h3 class="callout-title">Challenge 4: Look up <em>Dpp4</em> in <a href="https://pubmed.ncbi.nlm.nih.gov/" class="external-link">Pubmed</a>
</h3>
<div class="callout-content">
<p>or <a href="https://www.alliancegenome.org/" class="external-link">Alliance Genome</a> and
see if it has any known association with type 2 diabetes or insulin
metabolism.</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p><a href="https://www.alliancegenome.org/gene/MGI:94919" class="external-link">Alliance
Genome</a> says that “Human ortholog(s) of this gene implicated in …
type 2 diabetes mellitus.”</p>
</div>
</div>
</div>
</div>
<p>Let’s look more closely at <em>Hnf4a</em>. First, let’s map the
expresson of <em>Hnf4a</em>.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hnf4a</span>     <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Hnf4a"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="va">hnf4a_lod</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                  pheno     <span class="op">=</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">hnf4a</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                  kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>                  addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="va">hnf4a_lod</span>, <span class="va">map</span>, main <span class="op">=</span> <span class="st">"Hnf4a Genome Scan"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'hnf4a_lod' not found</code></pre>
</div>
<p>In the plot above, we can see that <em>Hnf4a</em> has a QTL peak on
chromosome 2.</p>
<div id="challenge-5-find-the-hnf4a-qtl-peak-location." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5-find-the-hnf4a-qtl-peak-location." class="callout-inner">
<h3 class="callout-title">Challenge 5: Find the <em>Hnf4a</em> QTL peak
location.</h3>
<div class="callout-content">
<p>Use find_peaks() to find the position of the <em>Hnf4a</em> QTL peak.
What kind of eQTL would you call this peak? Local or distant? How does
this peak location compare with the location of the chromosome 2 eQTL
hotspot?</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span><span class="va">hnf4a_lod</span>, map <span class="op">=</span> <span class="va">map</span>, threshold <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'hnf4a_lod' not found</code></pre>
</div>
<p><em>Hnf4a</em> has a QTL peak on chromosome 2 at 164 Mb.</p>
<p>This is the same position as the chromosome 2 eQTL hotspot.</p>
<p>Since <em>Hnf4a</em> is located at 163.4 Mb and its largest eQTL is
located at 164 Mb, we could call this a local eQTL.</p>
</div>
</div>
</div>
</div>
<p>These are the primary steps in mediation analysis. As you can see,
sometimes, meditation analysis can point to a candidate gene that is
plausible. It is always important to remember that there may be more
than one causal gene and that causal genes may contain missense or
splice SNPs which affect gene expression levels.</p>
<div class="section level3">
<h3 id="interactive-qtl-viewer">Interactive QTL Viewer<a class="anchor" aria-label="anchor" href="#interactive-qtl-viewer"></a></h3>
<p>The <a href="https://churchilllab.jax.org/qtlviewer/attie/islets#" class="external-link">QTL
Viewer</a> for the Attie islet data integrates mediation into
exploration of the data. Below, mediation analysis identifies gene Hnf4a
as the chromosome 2 gene that impacts Myo15b expression.</p>
<figure><img src="fig/mediation-Hnf4a-Myo15b.png" alt="Hnf4a mediation" class="figure mx-auto d-block"><div class="figcaption">Mediating expression of Myo15b identifies Hnf4a
as the gene that drops the LOD score from greater than 70 to less than
50.</div>
</figure><div id="challenge-6-perform-mediation-at-another-eqtl-hotspot." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-perform-mediation-at-another-eqtl-hotspot." class="callout-inner">
<h3 class="callout-title">Challenge 6: Perform mediation at another eQTL hotspot.</h3>
<div class="callout-content">
<p>Select another eQTL hotspot and perform mediation analysis.</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<p>Set “hot_chr” to one of the other eQTL hotspot locations: 5, 7, or
11</p>
<p>First, we look at the positions of the eQTL in this region.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hot_chr</span> <span class="op">&lt;-</span> <span class="st">"11"</span></span>
<span><span class="va">hot</span>     <span class="op">&lt;-</span> <span class="va">hotspots</span><span class="op">[[</span><span class="va">hot_chr</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="fu">table</span><span class="op">(</span><span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span>, las <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/mediation-analysis-rendered-challenge6a-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Next, we subset the genes to select ones which have eQTL at the same
location. Some of the eQTL hotspots have large numbers of of genes at
more than one marker. Feel free to explore different markers. Below, we
sort the markers by the number of genes with eQTL at each marker so that
we can select ones to pursue.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sort</span><span class="op">(</span><span class="fu">table</span><span class="op">(</span><span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
68.122257  68.42035 68.640199 68.660126 68.825289 68.888033 68.889671  68.89049
        1         1         1         1         1         1         1         1
 69.21825 69.224642 69.231034 69.244859 69.505599  69.66328 69.768364 69.869428
        1         1         1         1         1         1         1         1
70.196497  70.60917 70.616235 70.782062 71.198629 71.422464 71.425934 71.432875
        1         1         1         1         1         1         1         1
71.510771 71.793079 71.952385 68.656091 69.198036 69.733033 69.844926 69.876397
        1         1         1         2         2         2         2         2
69.924127 70.772323 71.449172 71.675712  71.83438 69.131341  69.23423 69.592546
        2         2         2         2         2         3         3         3
69.876917 69.978753 71.276599 71.418821 71.429405 71.937774 68.892129 69.416362
        3         3         3         3         3         3         4         4
71.658293 71.803404 71.859513 68.987403 69.542556 69.796313 69.064645 69.227838
        4         4         4         5         5         5         6         6
70.661194 70.750402 70.785072  71.12066 69.484146 69.837051 71.782692 69.810287
        6         6         6         6         8         9        10        14
71.813729 69.829176 71.693132 71.354568
       14        22        26        31 </code></pre>
</div>
<p>Set the thresholds below to match the markers which you would like to
include.</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hot</span> <span class="op">&lt;-</span> <span class="va">hot</span><span class="op">[</span><span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span> <span class="op">&gt;=</span> <span class="fl">71.3</span> <span class="op">&amp;</span> <span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span> <span class="op">&lt;=</span> <span class="fl">71.7</span>,<span class="op">]</span></span>
<span><span class="fu">nrow</span><span class="op">(</span><span class="va">hot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 75</code></pre>
</div>
<p>Next, we get the expression of the genes at the marker(s) we are
pursuing and calculate the first pricipal component.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_hot</span> <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">hot</span><span class="op">$</span><span class="va">gene_id</span><span class="op">]</span></span>
<span><span class="va">hot_pca</span>  <span class="op">&lt;-</span> <span class="fu">princomp</span><span class="op">(</span><span class="va">expr_hot</span><span class="op">)</span></span>
<span><span class="va">pc1_hot</span>  <span class="op">&lt;-</span> <span class="va">hot_pca</span><span class="op">$</span><span class="va">scores</span><span class="op">[</span>,<span class="fl">1</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>Plot the genome scan of PC1.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pc1_lod</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                 pheno     <span class="op">=</span> <span class="va">pc1_hot</span>,</span>
<span>                 kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">pc1_lod</span>, <span class="va">map</span>, main <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chr"</span>, <span class="va">hot_chr</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/mediation-analysis-rendered-challenge6d-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>At this point, we have everything that we need to perform mediation
analysis.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl_pos</span> <span class="op">&lt;-</span> <span class="fu">median</span><span class="op">(</span><span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">med_hot</span> <span class="op">&lt;-</span> <span class="fu">mediation</span><span class="op">(</span>qtl_chr  <span class="op">=</span> <span class="va">hot_chr</span>, </span>
<span>                     qtl_pos  <span class="op">=</span> <span class="va">qtl_pos</span>,</span>
<span>                     qtl_lod  <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">pc1_lod</span><span class="op">)</span>, </span>
<span>                     genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                     pheno    <span class="op">=</span> <span class="va">pc1_hot</span>, </span>
<span>                     K        <span class="op">=</span> <span class="va">K</span>, </span>
<span>                     addcovar <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                     map      <span class="op">=</span> <span class="va">map</span>, </span>
<span>                     expr     <span class="op">=</span> <span class="va">expr_rz</span>, </span>
<span>                     annot    <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in mediation(qtl_chr = hot_chr, qtl_pos = qtl_pos, qtl_lod = max(pc1_lod), : object 'probs' not found</code></pre>
</div>
<p>Now that we have completed the mediation analysis, we plot the
results.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">end</span>  <span class="op">&lt;-</span> <span class="fu">ceiling</span><span class="op">(</span><span class="fu">max</span><span class="op">(</span><span class="va">med_hot</span><span class="op">$</span><span class="va">middle</span> <span class="op">/</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'med_hot' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">med_hot</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">middle</span>, <span class="va">lod_drop</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="va">end</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">"Mediation Analysis for Chr "</span>, <span class="va">hot_chr</span>, <span class="st">" PC1"</span><span class="op">)</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"Position (Mb)"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"LOD Drop"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>size <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'med_hot' not found</code></pre>
</div>
<p>Let’s list the genes with LOD drops less than -10.</p>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">filter</span><span class="op">(</span><span class="va">med_hot</span>, <span class="va">lod_drop</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'med_hot' not found</code></pre>
</div>
<p>here are three genes with large LOD drops. In figure S7 of <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller et
al</a>, they select <em>Sat2</em> as a candidate gene.</p>
<p>But notice that there are also gene with large LOD increases. Let’s
list out the genes with LOD increases over 12.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">filter</span><span class="op">(</span><span class="va">med_hot</span>, <span class="va">lod_drop</span> <span class="op">&gt;</span> <span class="fl">12</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'med_hot' not found</code></pre>
</div>
<p>There are three genes with large LOD increase. If a LOD drop means
that the gene is absorbing variance and may be a causal mediator, what
does a LOD increase mean?</p>
<p>Let’s start by looking at the allele effects of the hotspot PC1 at
the QTL.</p>
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pr</span>      <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, map <span class="op">=</span> <span class="va">map</span>, chr <span class="op">=</span> <span class="va">hot_chr</span>, pos <span class="op">=</span> <span class="va">qtl_pos</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pc1_eff</span> <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>,</span>
<span>                pheno     <span class="op">=</span> <span class="va">pc1_hot</span>,</span>
<span>                kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">hot_chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pr' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, <span class="va">pc1_eff</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>, xaxt <span class="op">=</span> <span class="st">"n"</span>, main <span class="op">=</span> <span class="st">"PC1"</span>, xlab <span class="op">=</span> <span class="st">"Founder"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"Allele Effects"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pc1_eff' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">axis</span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, at <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, labels <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">pc1_eff</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pc1_eff' not found</code></pre>
</div>
<p>From the plot above, we can see that CAST (F) and PWK (G) have lower
effects than the other founders. Remember that the sign of the PC is
arbitrary, so the effects may be positive.</p>
<p>Next, let’s look at the allele effects of the gene with the highest
LOD increase, <em>Rnf167</em>.</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_id</span>  <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Rnf167"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span></span>
<span></span>
<span><span class="va">gene_lod</span> <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">pr</span>,</span>
<span>                  pheno     <span class="op">=</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">gene_id</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                  kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">hot_chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                  addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pr' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, <span class="va">gene_lod</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,  xaxt <span class="op">=</span> <span class="st">"n"</span>, main <span class="op">=</span> <span class="st">"Rnf167"</span>, xlab <span class="op">=</span> <span class="st">"Founder"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"Allele Effects"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'gene_lod' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">axis</span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, at <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, labels <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">pc1_eff</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pc1_eff' not found</code></pre>
</div>
<p><em>Rnf167</em> has allele effects which seem quite correlated with
PC1. Both CAST (F) and PWK (G) have different allele effects from the
other strains.</p>
<p>Next, let’s look at <em>Sat2</em>, which is the gene with the largest
LOD <strong>drop</strong>.</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_id</span>  <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Sat2"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="va">pr</span>       <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, map <span class="op">=</span> <span class="va">map</span>, chr <span class="op">=</span> <span class="va">hot_chr</span>, pos <span class="op">=</span> <span class="va">qtl_pos</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'probs' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_lod</span> <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">pr</span>,</span>
<span>                  pheno     <span class="op">=</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">gene_id</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                  kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">hot_chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                  addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pr' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="va">gene_lod</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,xaxt <span class="op">=</span> <span class="st">"n"</span>, main <span class="op">=</span> <span class="st">"Sat2"</span>, xlab <span class="op">=</span> <span class="st">"Founder"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"Allele Effects"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'gene_lod' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">axis</span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, at <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, labels <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">pc1_eff</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pc1_eff' not found</code></pre>
</div>
<p>The allele effects for <em>Sat2</em> appear to be less correlated,
with PWK (G) having lower effects than the other strains.</p>
<p>Let’s look at the correlation of <em>Sat2</em>, <em>Rnf167</em>, and
the hotspot PC1.</p>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_tmp</span> <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="fu">c</span><span class="op">(</span><span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Sat2"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span>, <span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Rnf167"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">expr_tmp</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sat2"</span>, <span class="st">"Rnf167"</span><span class="op">)</span></span>
<span><span class="va">expr_tmp</span> <span class="op">&lt;-</span> <span class="fu">cbind</span><span class="op">(</span><span class="va">expr_tmp</span>, <span class="va">pc1_hot</span><span class="op">)</span></span>
<span><span class="fu">cor</span><span class="op">(</span><span class="va">expr_tmp</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             Sat2     Rnf167    Comp.1
Sat2    1.0000000 -0.1007433 0.4116323
Rnf167 -0.1007433  1.0000000 0.4086829
Comp.1  0.4116323  0.4086829 1.0000000</code></pre>
</div>
<p>Both genes have almost identical correlation (0.41) with PC1, but are
not strongly correlated with each other (-0.1).</p>
<p>Let’s look at the relationship between the correlation between PC1
and all of the genes on chromosome 11.</p>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_11</span> <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">chr</span> <span class="op">==</span> <span class="st">'11'</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span><span class="op">]</span></span>
<span><span class="va">cor_11</span>  <span class="op">&lt;-</span> <span class="fu">cor</span><span class="op">(</span><span class="va">pc1_hot</span>, <span class="va">expr_11</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in cor(pc1_hot, expr_11): the standard deviation is zero</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="va">med_hot</span><span class="op">$</span><span class="va">lod_drop</span>, <span class="va">cor_11</span>, xlab <span class="op">=</span> <span class="st">"LOD Drop"</span>, ylab <span class="op">=</span> <span class="st">"Correlation"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'med_hot' not found</code></pre>
</div>
<p>There seems to be a relationship between the absolute LOD drop and
the correlation between the gene and PC1.</p>
</div>
</div>
</div>
</div>
<p>Mediation analysis is one way of identifying candidate genes under
QTL peaks. In this episode, we have focused on eQTL hotspots, but you
could do this with physiological phenotypes as well.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>Mediation analysis investigates an intermediary between an
independent variable and its effect on a dependent variable.</li>
<li>Mediation analysis is used in high-throughput genomics studies to
identify molecular phenotypes, such as gene expression or methylation
traits, that mediate the effect of genetic variation on disease
phenotypes or other outcomes of interest.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="create-transcriptome-map.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="create-transcriptome-map.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Creating A
        </a>
        <a class="chapter-link float-end" href="index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/mediation-analysis.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.10" class="external-link">sandpaper (0.16.10)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.7" class="external-link">pegboard (0.7.7)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/mediation-analysis.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene expression, quantitative trait, genetic mapping, Diversity Outbred population",
  "name": "Mediation Analysis",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/mediation-analysis.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/mediation-analysis.html",
  "dateCreated": "2024-11-07",
  "dateModified": "2024-12-11",
  "datePublished": "2024-12-11"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

