<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Expression Quantitative Trait Locus (eQTL) Mapping: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png">
<link rel="manifest" href="../favicons/incubator/site.webmanifest">
<link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Expression Quantitative Trait Locus (eQTL) Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Expression Quantitative Trait Locus (eQTL) Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr></ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Expression Quantitative Trait Locus (eQTL) Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="review-eqtl-study.html">2. Genetic Drivers of Pancreatic Islet Function</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="load-explore-data.html">3. Load and Explore Data</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="map-one-eqtl.html">4. Mapping A Single Gene Expression Trait</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="map-many-eqtls.html">5. Mapping Many Gene Expression Traits</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="maximum-peaks-genes.html">6. Maximum eQTL Peaks and Nearby Genes</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="create-transcriptome-map.html">7. Creating A Transcriptome Map</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="interpret-results.html">8. Interpreting qtl2 results</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="mediation-analysis.html">9. Mediation Analysis</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
</ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-introduction"><p>Content from <a href="introduction.html">Introduction</a></p>
<hr>
<p>Last updated on 2024-12-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/introduction.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 15 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are expression quantitative trait loci (eQTL)?</li>
<li>How are eQTL used in genetic studies?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe how an expression quantitative trait locus (eQTL) impacts
gene expression.</li>
<li>Describe how eQTL are used in genetic studies.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Differences in disease risk between individuals are often caused by
genetic variants. Identifying the effects of genetic variants is key to
understanding disease phenotypes and their underlying biology. The
effects of variants in many single-gene disorders, such as cystic
fibrosis, are generally well-characterized and their disease biology
well understood. For example, in cystic fibrosis, mutations in the
coding region of the CFTR gene alter the three-dimensional structure of
the chloride channel proteins in epithelial cells, affecting not only
chloride transport, but also sodium and potassium transport in the
lungs, pancreas and skin. The path from gene mutation to altered protein
to disease phenotype is relatively simple and well understood.</p>
<figure><img src="../fig/cystic-fibrosis.png" alt="Cystic fibrosis is caused by a mutation in the CFTR gene that prevents passage of chloride ions from the airway lumen of the lungs. This results in a loss of water from the airway lumens and a thick sticky mucus in the airway." class="figure mx-auto d-block"><div class="figcaption">Single-gene diseases like cystic fibrosis are
relatively well understood. In cystic fibrosis mutations in the coding
region of the CFTR gene result in a defective protein. This leads to
excess mucus production that can damage the lungs and digestive
system.</div>
</figure><p>Created in <a href="https://BioRender.com" class="external-link">BioRender</a></p>
<p>The most common human disorders, however, involve many genes
interacting with each other and with the environment, a far more
complicated path to follow than the path from a single gene mutation to
its protein to a disease phenotype. Cardiovascular disease, Alzheimer’s
disease, arthritis, diabetes and cancer involve a complex interplay of
genes with environment, and their mechanisms are not well understood.
One method of understanding the relationship between genetic variants
and disease is a genome-wide association study (GWAS), which associates
genetic variants with disease traits. It is tempting to think that these
genetic variants would fall in coding regions. However, most GWAS
variants for common diseases like diabetes are located in
<strong>non-coding</strong> regions of the genome. These variants are
therefore likely to fall in regulatory sequences which are involved in
gene regulation.</p>
<p><img src="../fig/gwas-catalog.png" alt="Figures showing the GWAS Catalog" class="figure"> Excerpted from the <a href="https://www.ebi.ac.uk/gwas/" class="external-link">GWAS Catalog</a></p>
<p>Gene regulation controls the quantity, timing and locale of gene
expression. Analyzing the association between gene expression and
genetic variants is known as expression quantitative trait locus (eQTL)
mapping.</p>
<p><img src="../fig/noncoding-SNP.png" alt="Figure showing regulation of gene expression by a SNP" class="figure"> Created
in <a href="https://BioRender.com" class="external-link">BioRender</a></p>
<p>eQTL mapping searches for associations between the expression of one
or more genes and a genetic locus. Specifically, genetic variants
underlying eQTL peaks explain some of the variation in gene expression
levels. eQTL studies can reveal the architecture of quantitative traits,
connect DNA sequence variation to phenotypic variation, and shed light
on transcriptional regulation and regulatory variation. Traditional
analytic techniques like linkage and association mapping can be applied
to thousands of gene expression traits (transcripts) in eQTL analysis,
such that gene expression can be mapped in much the same way as a
physiological phenotype like blood pressure or heart rate. Joining gene
expression and physiological phenotypes with genetic variation can
identify genes with variants affecting disease phenotypes.</p>
<p>To the simple diagram above we’ll add two more details. Non-coding
SNPs can regulate gene expression from nearby locations on the same
chromosome:</p>
<p><img src="../fig/cis-noncoding-SNP.png" alt="Figure showing SNP regulating gene which affects disease" class="figure">
Created in <a href="https://BioRender.com" class="external-link">BioRender</a></p>
<p>SNPs that affect gene expression from afar, often from a different
chromosome from the gene that they regulate are called distal
regulators.</p>
<figure><img src="../fig/trans-noncoding-SNP.png" alt="Figure showing distal regulation" class="figure mx-auto d-block"><div class="figcaption">SNPs can affect gene expression distally from
the gene that they regulate, often from a different chromosome
altogether.</div>
</figure><p>Created in <a href="https://BioRender.com" class="external-link">BioRender</a></p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>An expression quantitative trait locus (eQTL) explains part of the
variation in gene expression.</li>
<li>Traditional linkage and association mapping can be applied to gene
expression traits (transcripts).</li>
<li>Genetic variants, such as single nucleotide polymorphisms (SNPs),
that underlie eQTL illuminate transcriptional regulation and
variation.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-review-eqtl-study"><p>Content from <a href="review-eqtl-study.html">Genetic Drivers of Pancreatic Islet Function</a></p>
<hr>
<p>Last updated on 2024-12-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/review-eqtl-study.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 15 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is the hypothesis of an example eQTL study?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe an example eQTL study in Diversity Outbred mice.</li>
<li>State the hypothesis from an example eQTL study in Diversity Outbred
mice.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Genome-wide association studies (GWAS) often identify variants in
non-coding regions of the genome, indicating that regulation of gene
expression predominates in common diseases like type II diabetes. In
type II diabetes (T2D) the pancreas produces less insulin and the body
becomes less responsive to insulin.</p>
<p><img src="../fig/healthy-vs-T2D.png" alt="Figure showing Type 2 diabetes and insulin." class="figure"> Created in <a href="https://BioRender.com" class="external-link">BioRender</a></p>
<p>Most of the more than 100 genetic loci associated with T2D affect the
function of pancreatic islets. This study offers supporting evidence for
T2D-associated loci in human GWAS, most of which affect pancreatic islet
function. Pancreatic islet cells produce endocrine hormones including
insulin. A feedback loop of glucose and insulin activates beta cells
that produce insulin and inhibits alpha cells in the pancreas.</p>
<p><img src="../fig/pancreatic-islet.png" alt="Figure showing the Islet of Langerhans in the pancreas, which is composed of alpha, beta, delta and gamma cells." class="figure">
Created in <a href="https://BioRender.com" class="external-link">BioRender</a></p>
<p>Susceptibility to type II diabetes (T2D) increases with obesity, such
that T2D-associated genetic loci operate mainly under conditions of
obesity (See <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5937189/" class="external-link">Keller,
Mark P et al. “Genetic Drivers of Pancreatic Islet Function.” Genetics
vol. 209,1 (2018): 335-356</a>). Like most GWAS loci, the T2D-associated
genetic loci identified from GWAS have very small effect sizes and odds
ratios just slightly more than 1.</p>
<p>This study explored islet gene expression in diabetes. The authors
hypothesized that gene expression changes in response to dietary
challenge would reveal signaling pathways involved in stress responses.
The expression of many genes often map to the same locus, indicating
that expression of these genes is controlled in common. If their mRNAs
encode proteins with common physiological functions, the function of the
controlling gene(s) is revealed. Variation in expression of the
controlling gene(s), rather than a genetic variant, can be tested as an
immediate cause of a disease-related phenotype.</p>
<p>In this study, Diversity Outbred (DO) mice were fed a high-fat,
high-sugar diet as a stressor, sensitizing the mice to develop diabetic
traits. Body weight and plasma glucose, insulin, and triglyceride
measurements were taken biweekly. Food intake could be measured since
animals were individually housed. A glucose tolerance test at 18 weeks
of age provided measurement of dynamic glucose and insulin changes at 5,
15, 30, 60 and 120 minutes after glucose administration. Area under the
curve (AUC) was determined from these time points for both plasma
glucose and insulin levels.</p>
<figure><img src="../fig/gtt_auc_example.png" alt="Figure showing glucose and insulin changes over time." class="figure mx-auto d-block"><div class="figcaption">Glucose and Insulin glucose tolerance test
plots</div>
</figure><p>In the figure above, time is plotted on the X-axis and glucose or
insulin levels are plotted on the Y-axis. Blood is taken at the
beginning of the test and at several time points after the glucose
bolus. The glucose or insulin AUC is the area under the measured points
down to the baseline level. A diversity of responses to the glucose
tolerance test is illustrated below.</p>
<figure><img src="../fig/insulin-area-under-curve.png" alt="A plot showing four different curves reflecting insulin levels after administration of glucose starting at time zero" class="figure mx-auto d-block"><div class="figcaption">Insulin curves following glucose tolerance
test</div>
</figure><p><a href="https://en.wikipedia.org/wiki/Homeostatic_model_assessment" class="external-link">Homeostatic
model assessment</a> (HOMA) quantifies insulin resistance (IR) and beta
cell (<span class="math inline">\(\beta\)</span>) function. For IR, the
equation quantifying insulin resistance is the product of glucose and
insulin in mg/dL divided by 405.</p>
<p><span class="math inline">\(HOMA-IR = (glucose \times insulin) /
405\)</span></p>
<p>For beta cell function, the equation is</p>
<p><span class="math inline">\(HOMA-\beta = (360 \times insulin) /
(glucose - 63)\)</span></p>
<p>expressed as a percentage. Insulin resistance and beta cell function
were determined after the glucose tolerance test was given. Islet cells
were isolated from pancreas, and RNA extracted and libraries constructed
from isolated RNA for gene expression measurements.</p>
<p>Genome scans were performed with the leave-one-chromosome-out (LOCO)
method for kinship correction. Sex and experimental cohort (DO wave)
were used as covariates. The results of one scan for insulin area under
the curve (AUC) is shown below with a strong peak on chromosome 11. In
this lesson, we will look into genes located under this peak.</p>
<figure><img src="../fig/insulin-auc-lod-plot.png" alt="Insulin AUC QTL plot" class="figure mx-auto d-block"><div class="figcaption">LOD plot for insulin area under the curve</div>
</figure><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Many GWAS studies identify SNPs which to no lie within coding
regions of the genome.</li>
<li>This implies that the SNPs lie within regulatory sequences which
affect gene expression levels.</li>
<li>Merging gene expression with physiological trait QTL mapping can
help to identify genes which affect physiological trait variation.</li>
</ul>
</div>
</div>
</div></section><section id="aio-load-explore-data"><p>Content from <a href="load-explore-data.html">Load and Explore Data</a></p>
<hr>
<p>Last updated on 2024-12-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/load-explore-data.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 45 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What data are required for eqtl mapping?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To provide an example and exploration of data used for eqtl
mapping.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">ggbeeswarm</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">DESeq2</span><span class="op">)</span></span></code></pre>
</div>
<section><h2 class="section-heading" id="physiological-phenotypes">Physiological Phenotypes<a class="anchor" aria-label="anchor" href="#physiological-phenotypes"></a>
</h2>
<hr class="half-width">
<p>You should have downloaded data files already when following the <a href="https://smcclatchy.github.io/eqtl-mapping/index.html" class="external-link">setup
instructions</a>. The complete data used in these analyses are available
from <a href="https://doi.org/10.5061/dryad.pj105" class="external-link">Data Dryad</a>. The
files we will use are in a simpler format than those on Data Dryad.</p>
<p>Load in the physiological phenotypes.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the data</span></span>
<span><span class="va">pheno</span>       <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_pheno.rds'</span><span class="op">)</span></span>
<span><span class="va">pheno_dict</span>  <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_pheno_dict.rds'</span><span class="op">)</span></span>
<span><span class="va">covar</span>       <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_covar.rds'</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="physiological-phenotypes-1">Physiological Phenotypes<a class="anchor" aria-label="anchor" href="#physiological-phenotypes-1"></a>
</h2>
<hr class="half-width">
<p>In this data set, we have 20 phenotypes for 500 Diversity Outbred
mice. <code>pheno</code> is a data frame containing the phenotype data
as well as covariates. Click on the triangle to the left of
<code>pheno</code> in the Environment pane to view its contents. Run
<code>names(pheno)</code> to list the variables.</p>
<p><code>pheno_dict</code> is the phenotype dictionary. This data frame
contains information on each variable in <code>pheno</code>, including
<code>name</code>, <code>short name</code>,
<code>pheno_type</code>,<br><code>formula</code> (if used) and <code>description</code>. You can
view a table of the data dictionary.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_dict</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">description</span>, <span class="va">formula</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">kable</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<table class="table">
<colgroup>
<col width="6%">
<col width="82%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">description</th>
<th align="left">formula</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">mouse</td>
<td align="left">Animal identifier.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">sex</td>
<td align="left">Male (M) or female (F).</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">sac_date</td>
<td align="left">Date when mouse was sacrificed; used to compute days on
diet, using birth dates.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">partial_inflation</td>
<td align="left">Some mice showed a partial pancreatic inflation which
would negatively effect the total number of islets collected from these
mice.</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">coat_color</td>
<td align="left">Visual inspection by Kathy Schuler on coat color.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">oGTT_date</td>
<td align="left">Date the oGTT was performed.</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">FAD_NAD_paired</td>
<td align="left">A change in the method that was used to make this
measurement by Matt Merrins’ lab. Paired was the same islet for the
value at 3.3mM vs. 8.3mM glucose; unpaired was where averages were used
for each glucose concentration and used to compute ratio.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">FAD_NAD_filter_set</td>
<td align="left">A different filter set was used on the microscope to
make the fluorescent measurement; may have influenced the values.</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">crumblers</td>
<td align="left">Some mice store food in their bedding (hoarders) which
would be incorrectly interpreted as consumed.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">birthdate</td>
<td align="left">Birth date</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">diet_days</td>
<td align="left">Number of days.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">num_islets</td>
<td align="left">Total number of islets harvested per mouse; negatively
impacted by those with partial inflation.</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">Ins_per_islet</td>
<td align="left">Amount of insulin per islet in units of
ng/ml/islet.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">WPIC</td>
<td align="left">Derived number; equal to total number of islets times
insulin content per islet.</td>
<td align="left">Ins_per_islet * num_islets</td>
</tr>
<tr class="odd">
<td align="left">HOMA_IR_0min</td>
<td align="left">glucose*insulin/405 at time t=0 for the oGTT</td>
<td align="left">Glu_0min * Ins_0min / 405</td>
</tr>
<tr class="even">
<td align="left">HOMA_B_0min</td>
<td align="left">360 * Insulin / (Glucose - 63) at time t=0 for the
oGTT</td>
<td align="left">360 * Ins_0min / (Glu_0min - 63)</td>
</tr>
<tr class="odd">
<td align="left">Glu_tAUC</td>
<td align="left">Area under the curve (AUC) calculation without any
correction for baseline differences.</td>
<td align="left">complicated</td>
</tr>
<tr class="even">
<td align="left">Ins_tAUC</td>
<td align="left">Area under the curve (AUC) calculation without any
correction for baseline differences.</td>
<td align="left">complicated</td>
</tr>
<tr class="odd">
<td align="left">Glu_6wk</td>
<td align="left">Plasma glucose with units of mg/dl; fasting.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Ins_6wk</td>
<td align="left">Plasma insulin with units of ng/ml; fasting.</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">TG_6wk</td>
<td align="left">Plasma triglyceride (TG) with units of mg/dl;
fasting.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Glu_10wk</td>
<td align="left">Plasma glucose with units of mg/dl; fasting.</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">Ins_10wk</td>
<td align="left">Plasma insulin with units of ng/ml; fasting.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">TG_10wk</td>
<td align="left">Plasma triglyceride (TG) with units of mg/dl;
fasting.</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">Glu_14wk</td>
<td align="left">Plasma glucose with units of mg/dl; fasting.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Ins_14wk</td>
<td align="left">Plasma insulin with units of ng/ml; fasting.</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">TG_14wk</td>
<td align="left">Plasma triglyceride (TG) with units of mg/dl;
fasting.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">food_ave</td>
<td align="left">Average food consumption over the measurements made for
each mouse.</td>
<td align="left">complicated</td>
</tr>
<tr class="odd">
<td align="left">weight_2wk</td>
<td align="left">Body weight at indicated date; units are gm.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">weight_6wk</td>
<td align="left">Body weight at indicated date; units are gm.</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">weight_10wk</td>
<td align="left">Body weight at indicated date; units are gm.</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">DOwave</td>
<td align="left">Wave (i.e., batch) of DO mice</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<p>Since the paper is interested in type 2 diabetes and insulin
secretion, we will choose insulin AUC (area under the curve which was
calculated without any correction for baseline differences) for this
review.</p>
<div class="section level3">
<h3 id="phenotype-distributions">Phenotype Distributions<a class="anchor" aria-label="anchor" href="#phenotype-distributions"></a>
</h3>
<p>Many statistical models, including the QTL mapping model in
<code>qtl2</code>, expect that the incoming data will be normally
distributed. You may use transformations such as log or square root to
make your data more normally distributed. Here, we will log transform
the data.</p>
<p>Let’s make a variable for insulin AUC so that we don’t have to type
as much.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ins_tauc</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">[</span>, <span class="st">'Ins_tAUC'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>Next, let’s look at the distribution of insulin AUC using a
histogram.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">ins_tauc</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, </span>
<span>     breaks <span class="op">=</span> <span class="fl">20</span>,</span>
<span>     main   <span class="op">=</span> <span class="st">"Insulin Area Under the Curve"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-hist_untransformed-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>This is clearly <strong>not</strong> normally distributed. In fact,
this type of distribution is often log normal.</p>
<p>Now, let’s apply the <code>log()</code> function to this data in an
effort to make the distribution more normal. By default the
<code>log()</code> function calculates the natural log (base
<em>e</em>).</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC_log</span> <span class="op">&lt;-</span> <span class="fu">log</span><span class="op">(</span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s make a histogram of the log-transformed data.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC_log</span>, </span>
<span>     breaks <span class="op">=</span> <span class="fl">20</span>,</span>
<span>     main   <span class="op">=</span> <span class="st">"insulin AUC (log-transformed)"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-hist_log_transform-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>This looks much better! The data has a somewhat Gaussian shape.
Technically, the assumptions of a linear model require that the
<strong>residuals</strong> be normally distributed. In practice,
transforming the input data to be normally distributed helps to make the
residuals normally distributed. As a reminder, a residual is the
vertical distance from a data point to the line described by a linear
model.</p>
<figure><img src=".././fig/residual.png" alt="a residual for one data point located above the line described by a linear model" class="figure mx-auto d-block"><div class="figcaption">A linear model showing one residual</div>
</figure><p>Boxplots are another great way to view the distribution of the data
and to identify any outliers. We will log-transform insulin AUC using
the <a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link"><code>scale_y_log10()</code></a>
function. This transforms the data using base 10 and creates a base 10
y-axis for plotting. We have overlaid the data points using
<code>ggbeeswarm</code>’s <a href="https://www.rdocumentation.org/packages/ggbeeswarm/versions/0.7.2/topics/geom_beeswarm" class="external-link"><code>geom_beeswarm</code></a>.
We have told <code>geom_beeswarm()</code> to plot the points with some
transparency using the argument <code>alpha = 0.2</code>. The
<code>alpha</code> argument ranges between 0 (completely transparent) to
1 (completely opaque). A value of 0.1 means mostly transparent.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot Insulin on a log 10 scale</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">pheno</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">sex</span>, <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_beeswarm</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Insulin area under the curve"</span>, y <span class="op">=</span> <span class="st">"insulin AUC"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-Ins_tAUC_boxplot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>How many orders of magnitude (powers of 10) does insulin AUC
span?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>insulin AUC spans three orders of magnitude, from near 10 to over
1000.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p>Which sex has higher median insulin AUC values?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>Males have higher insulin AUC than females.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3</h3>
<div class="callout-content">
<p>What does the heavy line bisecting the boxes indicate?<br>
What do the lines at the top and bottom of the boxes indicate?<br>
What does the <em>whisker</em> extending from the top and bottom of the
boxes indicate?<br>
What do the black dots extending from the whiskers indicate?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>The heavy line bisecting the boxes shows the median value (not the
mean!). Half of the data points are above and half are below this
line.<br>
The lines at the top and bottom of the boxes indicate the first and 3rd
quartiles of the data (the <em>hinges</em>). One-fourth of the data
points are beneath the box and another one-fourth are above the box. The
box itself contains 50% of the data points.<br>
The whiskers represent some multiple of the interquartile range (IQR),
which is the height of the box between the first and third quartiles.
<code>geom_boxplot()</code> produces <em>Tukey-style boxplots</em> in
which the whiskers are 1.5 <span class="math inline">\(\times\)</span>
the IQR. Any data points that lie beyond the whiskers are considered
outliers and are shown as heavy black dots.</p>
</div>
</div>
</div>
</div>
<p>The boxplot is a useful plot to visualize the distribution of your
data.</p>
</div>
<div class="section level3">
<h3 id="quality-control-of-data">Quality Control of Data<a class="anchor" aria-label="anchor" href="#quality-control-of-data"></a>
</h3>
<p>Many statistical tests rely upon the data having a normal (or
Gaussian) distribution. Many biological phenotypes do not follow this
distribution and must be transformed before analysis. This is why we
log-transformed the data in the plots above.</p>
<p>While we can “eyeball” the distributions in the boxplot, it would be
better to use a <em>quantile-quantile</em> plot.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Quantile-Quantile Plot of Ins_tAUC"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"Normal Quantiles"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"Ins_tAUC"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-qqplot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In these plots, the <em>quantiles</em> of the normal distribution are
plotted on the X-axis and the data are plotted on the Y-axis. A quantile
evenly divides data observations into a specific number of groups. The
boxplot above evenly divides the observations into quartiles, a quantile
containing four groups that each contain one-fourth of the data. A
percentile similarly divides data into 100 quantiles, with each
percentile containing 1% of the observations.</p>
<p>In a quantile-quantile (Q-Q) plot the straight line indicates the
quantiles a normal distribution would follow. The untransformed insulin
AUC data values do <strong>not</strong> follow a normal distribution
because the points are far from the line. If they were reasonably
normally distributed, most of the data points would fall directly on the
straight line.</p>
<p>Next, we will log-transform the data and then create a
quantile-quantile plot.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Ins_tAUC <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Quantile-Quantile Plot of log(Ins_tAUC)"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"Normal Quantiles"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"log(Ins_tAUC)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-qqplot_log-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-4" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4" class="callout-inner">
<h3 class="callout-title">Challenge 4</h3>
<div class="callout-content">
<p>Does the log transformation make the data more normally distributed?
Explain your answer.</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>Yes. The log transformation makes the data more normally distributed
because the data points follow the normality line more closely.</p>
</div>
</div>
</div>
</div>
<div id="challenge-5" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5" class="callout-inner">
<h3 class="callout-title">Challenge 5</h3>
<div class="callout-content">
<p>Do any data points look suspicious to you? Explain your answer.</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<p>The data points that deviate from the normality line would be worth
investigating. All data deviates somewhat from normality, but the three
lowest points in the male data plot would be worth investigating. They
may be real, but there may also have been mishap in the assay.</p>
</div>
</div>
</div>
</div>
<p>Another way to identify outliers is to standardize the data and look
for data points that are more than four standard deviations from the
mean.</p>
<p>To do this, we will log transform and standardize insulin AUC.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ins_tauc</span> <span class="op">=</span> <span class="va">pheno</span> <span class="op">|&gt;</span> </span>
<span>             <span class="fu">select</span><span class="op">(</span><span class="va">mouse</span>, <span class="va">sex</span>, <span class="va">Ins_tAUC</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">group_by</span><span class="op">(</span><span class="va">sex</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>             <span class="fu">mutate</span><span class="op">(</span>Ins_tAUC <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">Ins_tAUC</span><span class="op">)</span>,</span>
<span>                    Ins_tAUC <span class="op">=</span> <span class="fu">scale</span><span class="op">(</span><span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ins_tauc</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">sex</span>, y <span class="op">=</span> <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_beeswarm</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="op">-</span><span class="fl">4</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span>  <span class="fl">4</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Distribution of Standardized Ins_tAUC"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-pheno_std-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>There are no data points outside of the four standard deviation
limits.</p>
</div>
</section><section><h2 class="section-heading" id="gene-expression-data">Gene Expression Data<a class="anchor" aria-label="anchor" href="#gene-expression-data"></a>
</h2>
<hr class="half-width">
<p>Let’s read in the gene expression data.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_expr_annot.rds'</span><span class="op">)</span></span>
<span><span class="va">raw</span>   <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_expr_raw.rds'</span><span class="op">)</span></span></code></pre>
</div>
<p>We have loaded in two data objects:</p>
<ol style="list-style-type: decimal">
<li>
<code>annot</code>: a data frame containing gene annotation,
and</li>
<li>
<code>raw</code>: a numeric matrix containing the un-normalized
expression counts.</li>
</ol>
<div id="challenge-6-how-many-samples-and-genes-are-there" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-how-many-samples-and-genes-are-there" class="callout-inner">
<h3 class="callout-title">Challenge 6: How many samples and genes are there?</h3>
<div class="callout-content">

</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" data-bs-parent="#accordionSolution6" aria-labelledby="headingSolution6">
<div class="accordion-body">
<p>Use the <code>dim</code> command or the Environment tab to determine
the number of samples and genes in <code>raw</code>.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">raw</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1]   378 21771</code></pre>
</div>
<p>There are 378 samples and 21,771 genes.</p>
</div>
</div>
</div>
</div>
<p>The expression objects that we have loaded in are organized such that
the transcripts and samples are aligned between the objects. The figure
below may help you to visualize the relationship between the expression,
annotation, and covariates.</p>
<figure><img src="../fig/expr_sample_annot.png" alt="Figure showing relationship between samples, expression, and transcripts." class="figure mx-auto d-block"><div class="figcaption">Data Diagram</div>
</figure><p>Let’s look at the rows in the gene annotation object.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">annot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                              gene_id symbol chr     start       end strand
ENSMUSG00000000001 ENSMUSG00000000001  Gnai3   3 108.10728 108.14615     -1
ENSMUSG00000000028 ENSMUSG00000000028  Cdc45  16  18.78045  18.81199     -1
ENSMUSG00000000037 ENSMUSG00000000037  Scml2   X 161.11719 161.25821      1
ENSMUSG00000000049 ENSMUSG00000000049   Apoh  11 108.34335 108.41440      1
ENSMUSG00000000056 ENSMUSG00000000056   Narf  11 121.23725 121.25586      1
ENSMUSG00000000058 ENSMUSG00000000058   Cav2   6  17.28119  17.28911      1
                      middle nearest.marker.id        biotype      module
ENSMUSG00000000001 108.12671       3_108090236 protein_coding   darkgreen
ENSMUSG00000000028  18.79622       16_18817262 protein_coding        grey
ENSMUSG00000000037 161.18770       X_161182677 protein_coding        grey
ENSMUSG00000000049 108.37887      11_108369225 protein_coding greenyellow
ENSMUSG00000000056 121.24655      11_121200487 protein_coding       brown
ENSMUSG00000000058  17.28515        6_17288298 protein_coding       brown
                   hotspot
ENSMUSG00000000001    &lt;NA&gt;
ENSMUSG00000000028    &lt;NA&gt;
ENSMUSG00000000037    &lt;NA&gt;
ENSMUSG00000000049    &lt;NA&gt;
ENSMUSG00000000056    &lt;NA&gt;
ENSMUSG00000000058    &lt;NA&gt;</code></pre>
</div>
<p>There are many columns in the gene annotation file, including the
Ensembl ID, gene symbol, chromosome, start and end of the gene.</p>
<p>Next, let’s look at the sample covariates.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">covar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>      mouse sex DOwave diet_days
DO021 DO021   F      1       112
DO022 DO022   F      1       112
DO023 DO023   F      1       112
DO024 DO024   F      1       112
DO025 DO025   F      1       114
DO026 DO026   F      1       114</code></pre>
</div>
<p>The sample covariates have information about the sex and DO
generation, indicated as <code>DOwave</code>, of each mouse. These are
<em>metadata</em>, or data about the data.</p>
<p>In order to make reasonable gene comparisons between samples, the
count data needs to be normalized. In the quantile-quantile (Q-Q) plot
below, count data for the first gene are plotted over a diagonal line
tracing a normal distribution for those counts. Notice that most of the
count data values lie off of this line, indicating that these gene
counts are not normally distributed.</p>
<figure><img src="../fig/load-explore-data-rendered-view_manual_qqplot_raw-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Q-Q plots for the first six genes show that count data for these
genes are not normally distributed. They are also not on the same scale.
The y-axis values for each subplot range to 20,000 counts in the first
subplot, 250 in the second, 90 in the third, and so on.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">ENSMUSG00000000001</span><span class="op">:</span><span class="va">ENSMUSG00000000058</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">'gene'</span>, values_to <span class="op">=</span> <span class="st">'value'</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">gene</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'Count distribution for six genes'</span>,</span>
<span>         xlab <span class="op">=</span> <span class="st">'Normal percentiles'</span>, y <span class="op">=</span> <span class="st">'Count percentiles'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-view_qqplots_raw-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Since each gene has a different distribution, we will need to
normalize the gene expression data. We will do this in a future
lesson.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>It is important to inspect the phenotype distributions and to
transform them to be nearly normal.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-map-one-eqtl"><p>Content from <a href="map-one-eqtl.html">Mapping A Single Gene Expression Trait</a></p>
<hr>
<p>Last updated on 2024-12-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/map-one-eqtl.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are the steps in QTL mapping?</li>
<li>How do I map one gene expression trait?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Review the steps in QTL mapping.</li>
<li>Run a QTL analysis for expression data.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="expression-data">Expression Data<a class="anchor" aria-label="anchor" href="#expression-data"></a>
</h2>
<hr class="half-width">
<p>In this lesson we review mapping steps and apply those steps to a
gene expression trait. In a previous lesson, we loaded in the raw
transcript expression data and noticed that the distribution of each
gene was non-Gaussian and different.</p>
<p>There is another issue that we must also address. Each sample has a
different total number of counts. This affects our ability to compare
values between samples. For example, say that we look at the expression
of Gene1 in two samples and find that both samples have 500 counts for
Gene1. It appears that Gene1 is equally expressed in both samples.
However, suppose that the total counts (<em>i.e.</em> the sum of counts
for all genes in each sample) is 10 million for sample 1 and 20 million
for sample 2. The sum of all counts across all genes in a sample is also
called the <em>library size</em>. Then we need to scale the counts for
Gene1 by the total counts. This is shown in the table below.</p>
<table class="table">
<thead><tr class="header">
<th>Sample</th>
<th>Gene1 Counts</th>
<th>Total Counts</th>
<th>Proportion</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>500</td>
<td>10e6</td>
<td>5e-05</td>
</tr>
<tr class="even">
<td>2</td>
<td>500</td>
<td>20e6</td>
<td>2.5e-05</td>
</tr>
</tbody>
</table>
<p>In this case, we can see that Gene1 has lower expression in sample 2
compared to sample 1. Although the actual adjustment for library size
(i.e. total counts) is more complicated, this is the rationale for
adjusting each sample.</p>
<p>Let’s plot a histogram of the total counts in each sample.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="fu">rowSums</span><span class="op">(</span><span class="va">raw</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e-6</span>,</span>
<span>     breaks <span class="op">=</span> <span class="fl">50</span>, </span>
<span>     main   <span class="op">=</span> <span class="st">"Histogram of Total Counts per Sample (raw counts)"</span>,</span>
<span>     xlab   <span class="op">=</span> <span class="st">"Total Counts (Millions)"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-hist_total_counts-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>As you can see, total counts range from 15 to 50 million reads. The
distribution of counts seems to be bimodal as well, which is
troubling.</p>
<p>Perhaps we should plot total counts versus the batch information that
we have in the covariates. Recall that there are 500 mice in the
covariate data. The mouse IDs are in the rownames of the raw expression
data, but not all 500 mice have expression data.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="va">covar</span><span class="op">$</span><span class="va">mouse</span> <span class="op">%in%</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">raw</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 378</code></pre>
</div>
<p>Let’s subset the covariates to include only those with expression
data.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_covar</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">covar</span>, <span class="va">mouse</span> <span class="op">%in%</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">raw</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">expr_covar</span> <span class="op">&lt;-</span> <span class="va">expr_covar</span><span class="op">[</span><span class="fu">match</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">raw</span><span class="op">)</span>, <span class="va">expr_covar</span><span class="op">$</span><span class="va">mouse</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">expr_covar</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">expr_covar</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span></code></pre>
</div>
<p>To recap, before we perform any analysis using the transcript
expression data, we need to:</p>
<ol style="list-style-type: decimal">
<li>normalize it by adjusting for library size and,<br>
</li>
<li>transform the expression of each gene to be Gaussian.</li>
</ol>
<div class="section level3">
<h3 id="normalizing-gene-expression">Normalizing Gene Expression<a class="anchor" aria-label="anchor" href="#normalizing-gene-expression"></a>
</h3>
<p>We will use the <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" class="external-link">DESeq2</a>
package to adjust the counts for library size. DESeq2 is a large package
which performs many types of analyses. Further details are in the <a href="https://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html" class="external-link">DESeq2
Tutorial</a>.</p>
<p>First, we must create a DESeq object. We need the raw counts, rounded
so that all values are integers, and the sample covariate data. We will
have to subset the sample covariates to include only the expression
samples, since we don’t have expression data for every mouse.</p>
<p>In order to create the DESeq2 object, we will need to transpose
(using <code>t()</code>) the expression data so that the mouse IDs
(samples) are moved to the columns. This is because DESeq2 requires that
the samples be in columns and the genes in rows. We will also tell
DESeq2 what the design variables are for our data, although they are not
used in this case. These would be used if we were searching for
differentially expressed genes. We specify no design with
<code>design = ~ 1</code>.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span>  <span class="op">=</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData <span class="op">=</span> <span class="fu">t</span><span class="op">(</span><span class="fu">round</span><span class="op">(</span><span class="va">raw</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                              colData   <span class="op">=</span> <span class="va">expr_covar</span>, </span>
<span>                              design    <span class="op">=</span> <span class="op">~</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>converting counts to integer mode</code></pre>
</div>
<p>The object <code>dds</code> contains the counts for all mice that
have expression data. Genes are in rows and samples are in columns.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>class: DESeqDataSet
dim: 21771 378
metadata(1): version
assays(1): counts
rownames(21771): ENSMUSG00000000001 ENSMUSG00000000028 ...
  ENSMUSG00000099322 ENSMUSG00000099329
rowData names(0):
colnames(378): DO021 DO022 ... DO417 DO420
colData names(4): mouse sex DOwave diet_days</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 21771   378</code></pre>
</div>
<p>This is a complex data object. Let’s look at the counts for the first
gene in the first 5 samples.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>DO021 DO022 DO023 DO024 DO025
10247 11838 12591 12424 10906 </code></pre>
</div>
<p>Now look at the counts for the first five genes in sample 1.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>ENSMUSG00000000001 ENSMUSG00000000028 ENSMUSG00000000037 ENSMUSG00000000049
             10247                108                 29                 15
ENSMUSG00000000056
               120 </code></pre>
</div>
<p>Next, we will run DESeq2 and let it adjust the expression data for
differing library sizes.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span>  <span class="op">=</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating size factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>mean-dispersion relationship</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>final dispersion estimates</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>fitting model and testing</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>-- replacing outliers and refitting for 155 genes
-- DESeq argument 'minReplicatesForReplace' = 7
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>fitting model and testing</code></pre>
</div>
<p>Once this is done, we will get the expression data after it has been
transformed using the <a href="https://en.wikipedia.org/wiki/Variance-stabilizing_transformation" class="external-link">Variance
Stabilizing Transformation</a> (VST). The VST adjusts the variance of
the genes such that it is not related to the mean gene expression
level.</p>
<div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor1" aria-labelledby="headingInstructor1">
<div class="accordion-body">
<p>The students don’t have to type the next block. You can show the plot
in the lesson or type it to show the plot live.</p>
</div>
</div>
</div>
</div>
<p>First, let’s look at the mean expression of each gene versus its
standard deviation.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tibble</span><span class="op">(</span>mean <span class="op">=</span> <span class="fu">colMeans</span><span class="op">(</span><span class="va">raw</span><span class="op">)</span>,</span>
<span>       sd   <span class="op">=</span> <span class="fu">apply</span><span class="op">(</span><span class="va">raw</span>, <span class="fl">2</span>, <span class="va">sd</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">mean</span>, <span class="va">sd</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mean vs SD of expression values"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"log(Mean)"</span>, y <span class="op">=</span> <span class="st">"log(Std. Dev.)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-show_expr_mean_var-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot above shows the mean expression value for each gene versus
the standard deviation of each gene. Both axes are log-transformed. As
you can see, there is a positive correlation between the mean and the
standard deviation. We would like each gene to have the same variance,
regardless of the mean, for each gene.</p>
<p>Next, we will apply the variance stabilizing transformation and will
transpose the expression values.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr</span> <span class="op">=</span> <span class="fu">assays</span><span class="op">(</span><span class="fu">vst</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">expr</span> <span class="op">=</span> <span class="fu">t</span><span class="op">(</span><span class="va">expr</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s look at the mean versus the standard deviation of each gene
after normalization.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tibble</span><span class="op">(</span>mean <span class="op">=</span> <span class="fu">colMeans</span><span class="op">(</span><span class="va">expr</span><span class="op">)</span>,</span>
<span>       sd   <span class="op">=</span> <span class="fu">apply</span><span class="op">(</span><span class="va">expr</span>, <span class="fl">2</span>, <span class="va">sd</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">mean</span>, <span class="va">sd</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mean vs. Std. Dev. of After VST"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"log(Mean)"</span>, y <span class="op">=</span> <span class="st">"log(Std. Dev.)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-mean_sd_after_vst-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The standard deviation is now largely unrelated to the mean. At lower
expression levels, the standard deviation is somewhat related to the
mean.</p>
<p>Let’s look at a the distribution of total counts per sample after
normalization.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="fu">rowSums</span><span class="op">(</span><span class="va">expr</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e-6</span>,</span>
<span>      breaks <span class="op">=</span> <span class="fl">50</span>, </span>
<span>      main   <span class="op">=</span> <span class="st">"Histogram of Total Expression per Sample (after Normalizaion)"</span>,</span>
<span>      xlab   <span class="op">=</span> <span class="st">"Total Counts (Millions)"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-unnamed-chunk-1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>At this point, while each gene has been normalized, each gene has a
different distribution. In QTL mapping, we often use permutations to
estimate significance thresholds. This approach works for one phenotype.
However, if other phenotypes have different distributions, then the
significance threshold for one phenotype cannot be used for another.
This means that we would have to perform 1,000 permutations for
<strong>each</strong> gene. While modern computing clusters can do this,
it is time consuming.</p>
<p>Another approach is to force the distribution of each gene to be
identical. Then, we can perform permutations on one gene and get a
significance threshold for all genes.</p>
<p>We can force the distribution of each gene to be Gaussian and
identical for all genes using an inverse-normal or rank-Z
transformation.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rankZ</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">x</span> <span class="op">=</span> <span class="fu">rank</span><span class="op">(</span><span class="va">x</span>,</span>
<span>           na.last     <span class="op">=</span> <span class="st">"keep"</span>,</span>
<span>           ties.method <span class="op">=</span> <span class="st">"average"</span><span class="op">)</span> <span class="op">/</span> <span class="op">(</span><span class="fu">sum</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span>
<span>  <span class="kw">return</span><span class="op">(</span><span class="fu">qnorm</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="co"># rankZ()</span></span>
<span></span>
<span><span class="va">expr_rz</span> <span class="op">=</span> <span class="fu">apply</span><span class="op">(</span><span class="va">expr</span>, <span class="fl">2</span>, <span class="va">rankZ</span><span class="op">)</span></span></code></pre>
</div>
<p>Boxplots of raw counts for six example genes are shown at left below.
Notice that the median count values (horizontal black bar in each
boxplot) are not comparable between the genes because the counts are not
on the same scale. At right, boxplots for the same genes show normalized
count data on the same scale.</p>
<div id="accordionInstructor2" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor2" aria-expanded="false" aria-controls="collapseInstructor2">
  <h3 class="accordion-header" id="headingInstructor2">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor2" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor2" aria-labelledby="headingInstructor2">
<div class="accordion-body">
<p>Show this in the lesson website. Don’t type all of this out or have
the students type it either.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">=</span> <span class="va">raw</span> <span class="op">|&gt;</span> </span>
<span>        <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>        <span class="fu">select</span><span class="op">(</span><span class="va">ENSMUSG00000000001</span><span class="op">:</span><span class="va">ENSMUSG00000000058</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>        <span class="fu">pivot_longer</span><span class="op">(</span>cols      <span class="op">=</span> <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>,</span>
<span>                     names_to  <span class="op">=</span> <span class="st">'gene'</span>,</span>
<span>                     values_to <span class="op">=</span> <span class="st">'value'</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>        <span class="fu">mutate</span><span class="op">(</span>type <span class="op">=</span> <span class="st">'raw'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">norm</span> <span class="op">=</span> <span class="va">expr</span> <span class="op">|&gt;</span> </span>
<span>         <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>         <span class="fu">select</span><span class="op">(</span><span class="va">ENSMUSG00000000001</span><span class="op">:</span><span class="va">ENSMUSG00000000058</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>         <span class="fu">pivot_longer</span><span class="op">(</span>cols      <span class="op">=</span> <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>,</span>
<span>                      names_to  <span class="op">=</span> <span class="st">'gene'</span>,</span>
<span>                      values_to <span class="op">=</span> <span class="st">'value'</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>         <span class="fu">mutate</span><span class="op">(</span>type <span class="op">=</span> <span class="st">'normalized'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">bind_rows</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">norm</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>type <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">type</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'raw'</span>, <span class="st">'normalized'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">gene</span>, <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">type</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'Count distributions for example genes'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,</span>
<span>          axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-view_example_boxplots-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">norm</span><span class="op">)</span></span></code></pre>
</div>
<p>In the rankZ-transformed data, every gene has the same
distribution.</p>
<div id="accordionInstructor3" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor3" aria-expanded="false" aria-controls="collapseInstructor3">
  <h3 class="accordion-header" id="headingInstructor3">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor3" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor3" aria-labelledby="headingInstructor3">
<div class="accordion-body">
<p>Show this in the lesson website. Don’t type all of this out or have
the students type it either.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_rz</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">ENSMUSG00000000001</span><span class="op">:</span><span class="va">ENSMUSG00000000058</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols      <span class="op">=</span> <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>,</span>
<span>               names_to  <span class="op">=</span> <span class="st">'gene'</span>,</span>
<span>               values_to <span class="op">=</span> <span class="st">'value'</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">gene</span>, <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'RankZ distributions for example genes'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,</span>
<span>          axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-rankz_expr-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Let’s save the rankZ-transformed expression data so that we will have
it when we need it again.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">expr_rz</span>, file <span class="op">=</span> <span class="st">"data/attie_do_expr_rz.rds"</span><span class="op">)</span></span></code></pre>
</div>
<p>Before moving on, let’s remove data objects that we won’t be using
again.</p>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">dds</span>, <span class="va">raw</span>, <span class="va">expr</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="the-marker-map">The Marker Map<a class="anchor" aria-label="anchor" href="#the-marker-map"></a>
</h3>
<p>The marker map contains a list of the genetic marker positions for
each marker in the genoprobs. Let’s read it in now.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_map.rds'</span><span class="op">)</span></span></code></pre>
</div>
<p>The marker map for each chromosome is stored in the <code>map</code>
object. This is used to plot the LOD scores calculated at each marker
during QTL mapping. Each list element is a numeric vector with each
marker position in megabases (Mb). Here we are using the 69K grid marker
file. Often when there are numerous genotype arrays used in a study, we
interpolate all to a 69k grid file so we are able to combine all samples
across different array types.</p>
<p>Look at the structure of <code>map</code> in the Environment tab by
clicking the triangle to the left or by running <code>str(map)</code> in
the Console.</p>
<p>Each element in map contains a list of marker positions and
names.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  1_3e+06 1_3041392 1_3346528 1_3651663 1_3657931 1_3664199
 3.000000  3.041392  3.346528  3.651663  3.657931  3.664199 </code></pre>
</div>
<p>The names of the markers consist of the chromosome and the bp
position, separated by an underscore (_).</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The marker positions in the map must be in Mb.</p>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="genotype-probabilities">Genotype probabilities<a class="anchor" aria-label="anchor" href="#genotype-probabilities"></a>
</h3>
<p>Previously, we loaded in the physiological phenotypes, the sample
covariates, and the transcript expression data and annotation. We also
normalized and rankZ transformed the expression data.</p>
<p>In order to perform QTL mapping, we also need the genotype
probabilities (i.e genoprobs). In this lesson, we have already processed
the genotypes and produced the genoprobs using <a href="https://github.com/kbroman/qtl2/blob/main/R/calc_genoprob.R" class="external-link">calc_genoprob</a>.</p>
<!-- We don't evaluate this block when building the website because the file is
too large (1.7 GB). We load the probs from Box above . -->
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">probs</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">"data/attie_DO500_genoprobs_v5.rds"</span><span class="op">)</span></span></code></pre>
</div>
<p>At this point, you should have several data objects in your
environment. Look at the Environment tab to see what data objects are in
your environment. It should look like the figure below.</p>
<p><img src="../fig/data_env_review_qtl.png" alt="Picture of Environment tab with data objects." class="figure"> Each element of
<code>probs</code> is a 3-dimensional array containing the founder
allele dosages for each sample at each marker on one chromosome. These
are the 8 state allele probabilities (not 32) using the 69k marker grid
for the same 500 DO mice that also have clinical phenotypes. We have
already calculated genotype probabilities for you, so you can skip the
step for <a href="https://smcclatchy.github.io/qtl-mapping/calc-genoprob/" class="external-link">calculating
genotype probabilities</a> and the optional step for calculating allele
probabilities.</p>
<p>Next, we look at the dimensions of <code>probs</code> for chromosome
1:</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1]  500    8 4711</code></pre>
</div>
<p>Each list element of the genoprobs has 500 samples, eight founders,
and a variable number of markers, depending on the chromosome.</p>
<p>As a reminder, this is what the genoprobs of one mouse look like
along one chromosome.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span><span class="va">probs</span>, <span class="va">map</span>, ind <span class="op">=</span> <span class="fl">1</span>, chr <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-geno_plot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, the founder contributions, which range between 0
and 1, are colored from white (= 0) to black (= 1.0). A value of ~0.5 is
grey. The markers are on the X-axis and the eight founders (denoted by
the letters A through H) on the Y-axis. Starting at the left, we see
that this sample has genotype GH because the rows for G and H are grey,
indicating values of 0.5 for both alleles. Moving along the genome to
the right, the genotype becomes HH where the row is black indicating a
value of 1.0. This is followed by CD, DD, DG, AD, AH, CE, etc. The
values at each marker sum to 1.0.</p>
</div>
<div class="section level3">
<h3 id="kinship-matrix">Kinship Matrix<a class="anchor" aria-label="anchor" href="#kinship-matrix"></a>
</h3>
<p>We also use a kinship matrix in the mapping model to adjust for the
relatedness between mice. We also use a different kinship matrix on each
chromosome by including all of the markers except the ones on the
current chromosome. This is called the “Leave-One-Chromosome-Out” (LOCO)
method. We use the genoprobs to create the kinship matrices in the <a href="https://github.com/kbroman/qtl2/blob/main/R/calc_kinship.R" class="external-link">calc-kinship</a>
function.</p>
<div id="accordionInstructor4" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor4" aria-expanded="false" aria-controls="collapseInstructor4">
  <h3 class="accordion-header" id="headingInstructor4">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor4" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor4" aria-labelledby="headingInstructor4">
<div class="accordion-body">
<p>We load in kinship at the top of the lesson to build the website.
This will take the students a while to build.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">K</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                  type <span class="op">=</span> <span class="st">'loco'</span><span class="op">)</span></span></code></pre>
</div>
<p>And let’s save the kinship matrices so that we don’t have to build
them again.</p>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">K</span>, file <span class="op">=</span> <span class="st">"data/attie_do_kinship.rds"</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s look at a part of one of the kinship matrices.</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="fu">heatmap</span><span class="op">(</span><span class="va">K</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">]</span>, main <span class="op">=</span> <span class="st">"Kinship Between Mice"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-kinship_probs-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The figure above shows kinship between all pairs of samples. Light
yellow indicates low kinship and dark red indicates higher kinship.
Orange values indicate varying levels of kinship between 0 and 1. The
dark red diagonal of the matrix indicates that each sample is identical
to itself. The orange blocks along the diagonal may indicate close
relatives (i.e. siblings or cousins).</p>
</div>
<div class="section level3">
<h3 id="covariates">Covariates<a class="anchor" aria-label="anchor" href="#covariates"></a>
</h3>
<p>Next, we need to create additive covariates that will be used in the
mapping model. First, we need to see which covariates are significant.
In the data set, we have <code>sex</code>, <code>DOwave</code>
(<em>i.e.</em>, batch) of DO mice) and <code>diet_days</code> (number of
days on diet) to test whether there are any sex, batch or diet
effects.</p>
<p>We will use <code>sex</code> and <code>DOwave</code> as additive
covariates. Sex and DO outbreeding generation are often sensible
covariates to add. We will convert <code>sex</code> and
<code>DOwave</code> to factors and then use <a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a> to
create the covariates matrix that <code>qtl2</code> will use.</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span><span class="op">$</span><span class="va">sex</span>    <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span></span>
<span><span class="va">pheno</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span>
<span></span>
<span><span class="va">addcovar</span>     <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span>, data <span class="op">=</span> <span class="va">pheno</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs must be in the rownames of <code>pheno</code>,
<code>addcovar</code>, <code>genoprobs</code> and <code>K</code>.
<code>qtl2</code> uses the sample IDs to align the samples between
objects.</p>
</div>
</div>
</div>
<!-- DMG: Not sure about this yet. -->
<p>Considering the paper included the covariate <code>diet_days</code>
we will include that as well.</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>      sexM DOwave2 DOwave3 DOwave4 DOwave5
DO021    0       0       0       0       0
DO022    0       0       0       0       0
DO023    0       0       0       0       0
DO024    0       0       0       0       0
DO025    0       0       0       0       0
DO026    0       0       0       0       0</code></pre>
</div>
</div>
</section><section><h2 class="section-heading" id="performing-a-genome-scan">Performing a Genome Scan<a class="anchor" aria-label="anchor" href="#performing-a-genome-scan"></a>
</h2>
<hr class="half-width">
<p>We will perform a genome scan for insulin AUC, comparing the results
of the untransformed and log-transformed results. Use the <a href="https://github.com/kbroman/qtl2/blob/main/R/scan1.R" class="external-link">scan1</a>
function to map insulin AUC.</p>
<div id="accordionInstructor5" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor5" aria-expanded="false" aria-controls="collapseInstructor5">
  <h3 class="accordion-header" id="headingInstructor5">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor5" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor5" aria-labelledby="headingInstructor5">
<div class="accordion-body">
<p>This takes about 15 to 30 seconds.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_ins</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                 pheno     <span class="op">=</span> <span class="va">ins_tauc</span>,</span>
<span>                 kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>After the genome scan, <code>lod_ins</code> contains the LOD scores
for both the untransformed and log-transformed insulin values.</p>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">lod_ins</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          Ins_tAUC Ins_tAUC_log
1_3000000 5.162655     4.333006
1_3041392 5.163071     4.333039
1_3346528 5.207254     4.396324
1_3651663 5.011606     4.261237
1_3657931 5.047916     4.286642
1_3664199 5.093272     4.314025</code></pre>
</div>
<p>Let’s plot both LOD curves.</p>
<div id="accordionInstructor6" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor6" aria-expanded="false" aria-controls="collapseInstructor6">
  <h3 class="accordion-header" id="headingInstructor6">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor6" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor6" aria-labelledby="headingInstructor6">
<div class="accordion-body">
<p>You don’t have to have the students type out the legend and title
code. Also, use “red3” instead of “rgb(0.8, 0, 0, 0.5)”.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x         <span class="op">=</span> <span class="va">lod_ins</span>, </span>
<span>           map       <span class="op">=</span> <span class="va">map</span>,</span>
<span>           lodcolumn <span class="op">=</span> <span class="st">"Ins_tAUC_log"</span>,</span>
<span>           main      <span class="op">=</span> <span class="st">"insulin AUC"</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x         <span class="op">=</span> <span class="va">lod_ins</span>, </span>
<span>           map       <span class="op">=</span> <span class="va">map</span>,</span>
<span>           lodcolumn <span class="op">=</span> <span class="st">"Ins_tAUC"</span>,</span>
<span>           col       <span class="op">=</span> <span class="fu">rgb</span><span class="op">(</span><span class="fl">0.8</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>           lty       <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>           add       <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">"topleft"</span>, </span>
<span>       legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"log-transformed"</span>, <span class="st">"untransformed"</span><span class="op">)</span>, </span>
<span>       col    <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span>,           <span class="st">"red3"</span><span class="op">)</span>, </span>
<span>       lwd    <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-plot_ins_tauc-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-1-which-phenotype-has-the-higher-lod-score-on-chromosomes-11-and-17" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-which-phenotype-has-the-higher-lod-score-on-chromosomes-11-and-17" class="callout-inner">
<h3 class="callout-title">Challenge 1: Which phenotype has the higher LOD score on chromosomes 11 and 17?</h3>
<div class="callout-content">

</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>The log-transformed data has a higher LOD score on chromosome 11.
However, the untransformed data has a higher LOD on chromosome 17.</p>
</div>
</div>
</div>
</div>
<p>The challenge above shows the value of transforming data to make it
more normally distributed. We do not have a peak for
<code>log(ins_tauc)</code> on chromosome 11 which we will work with for
the rest of the lesson.</p>
<p>Let’s save this LOD profile in case we need it again. We will only
save the results from the log-transformed data since that produces a
higher LOD at a peak that we will use later.</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">lod_ins</span><span class="op">[</span>, <span class="fl">2</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, file <span class="op">=</span> <span class="st">"data/ns_tauc_lod.rds"</span><span class="op">)</span></span></code></pre>
</div>
<p>Because we are working with the insulin AUC phenotype, which has a
QTL peak on chromosome 11, we will map a gene on chromosome 11 which may
influence insulin and glucose levels. This gene is called <a href="https://www.alliancegenome.org/gene/MGI:98505" class="external-link">Hnf1b</a>. Since
the expression data uses Ensembl IDs in its column names, we need to
find the Ensembl ID for this gene:</p>
<div class="codewrapper sourceCode" id="cb53">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ensid</span> <span class="op">&lt;-</span> <span class="va">annot</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">subset</span><span class="op">(</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Hnf1b"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">pull</span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span></span>
<span><span class="va">ensid</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "ENSMUSG00000020679"</code></pre>
</div>
<p>Next, we will create a variable which contains the rankZ-transformed
Hnf1b expression values to reduce our typing.</p>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hnf1b</span> <span class="op">=</span> <span class="va">expr_rz</span><span class="op">[</span>, <span class="va">ensid</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Remember to use the <code>drop = FALSE</code> argument so that R will
not convert the expression data from a matrix to a vector.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_hnf1b</span> <span class="op">=</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                  pheno     <span class="op">=</span> <span class="va">hnf1b</span>,</span>
<span>                  kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>                  addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-2-plot-hnf1b-genome-scan" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-plot-hnf1b-genome-scan" class="callout-inner">
<h3 class="callout-title">Challenge 2: Plot Hnf1b Genome Scan</h3>
<div class="callout-content">
<p>Use the <a href="https://github.com/kbroman/qtl2/blob/main/R/plot_scan1.R" class="external-link">plot_scan1</a>
function to plot the Hnf1b genome scan.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_hnf1b</span>,</span>
<span>           map  <span class="op">=</span> <span class="va">map</span>,</span>
<span>           main <span class="op">=</span> <span class="st">"Hnf1b"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-challenge2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="permutations">Permutations<a class="anchor" aria-label="anchor" href="#permutations"></a>
</h3>
<p>We now have a peaks on chromosome 11 for both insulin AUC and Hnf1b,
but we do not know if the LODs are significant. In order to assess
significance, we will use permutations.</p>
<div id="callout4" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Don’t run the permutation block below. It will take hours to
complete. We have pre-computed the permutations and have code for you to
load them in below.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">iperm</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno     <span class="op">=</span> <span class="va">ins_tauc</span><span class="op">[</span>,<span class="fl">2</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                   n_perm    <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></code></pre>
</div>
<p>Since calculating permutations takes a long time, we will read in
pre-computed permutations.</p>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">iperm</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/ins_tauc_log_perm_1000.rds'</span><span class="op">)</span></span></code></pre>
</div>
<p>We then obtain the permutation thresholds using the <a href="https://github.com/kbroman/qtl2/blob/main/R/summary_scan1perm.R" class="external-link">summary</a>
function. Let’s find the significance level for 0.1, 0.05 and 0.01.</p>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">thr_ins</span> <span class="op">&lt;-</span> <span class="fu">summary</span><span class="op">(</span>object <span class="op">=</span> <span class="va">iperm</span>,</span>
<span>                   alpha  <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.05</span>, <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">thr_ins</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     Ins_tAUC_log
0.1          7.06
0.05         7.42
0.01         8.34</code></pre>
</div>
<p>We also need to perform permutations of the Hnf1b values since they
have a different distribution than insulin AUC.</p>
<div id="callout5" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Don’t run the permutation block below. It will take hours to
complete. We have pre-computed the permutations and have code for you to
load them in below.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eperm</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno     <span class="op">=</span> <span class="va">expr_rz</span><span class="op">[</span>, <span class="va">ensid</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                   n_perm    <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></code></pre>
</div>
<p>We will read in the pre-computed permutations.</p>
<div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eperm</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">"data/"</span>, <span class="va">ensid</span>, <span class="st">"_perm_1000.rds"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will get the significance thresholds at the alpha = 0.1,
0.05, and 0.01 levels.</p>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">thr_hnf1b</span> <span class="op">&lt;-</span> <span class="fu">summary</span><span class="op">(</span><span class="va">eperm</span>, </span>
<span>                     alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.05</span>, <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">thr_hnf1b</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     ENSMUSG00000020679
0.1                7.08
0.05               7.47
0.01               8.35</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="finding-significant-peaks">Finding Significant Peaks<a class="anchor" aria-label="anchor" href="#finding-significant-peaks"></a>
</h3>
<p>Let’s use <code>find_peaks</code> to identify the significant peaks
in the insulin AUC genome scan. We will use the 0.05 significance
threshold.</p>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_ins</span> <span class="op">&lt;-</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_ins</span>, </span>
<span>                        map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>                        threshold    <span class="op">=</span> <span class="va">thr_ins</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, </span>
<span>                        prob         <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span><span class="va">peaks_ins</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">lodindex</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">chr</span>, <span class="va">pos</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">kable</span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"insulin AUC QTL Peaks"</span><span class="op">)</span></span></code></pre>
</div>
<table class="table">
<caption>insulin AUC QTL Peaks</caption>
<thead><tr class="header">
<th align="left">lodcolumn</th>
<th align="left">chr</th>
<th align="right">pos</th>
<th align="right">lod</th>
<th align="right">ci_lo</th>
<th align="right">ci_hi</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Ins_tAUC_log</td>
<td align="left">11</td>
<td align="right">83.59467</td>
<td align="right">11.258841</td>
<td align="right">83.58553</td>
<td align="right">84.95444</td>
</tr>
<tr class="even">
<td align="left">Ins_tAUC</td>
<td align="left">17</td>
<td align="right">31.69319</td>
<td align="right">7.445012</td>
<td align="right">25.57974</td>
<td align="right">73.89085</td>
</tr>
</tbody>
</table>
<p>We can see that we have a peak for insulin AUC on chromosome 17 at
31.693192 Mb.</p>
<div id="challenge-3-significant-peaks-for-hnf1b" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-significant-peaks-for-hnf1b" class="callout-inner">
<h3 class="callout-title">Challenge 3: Significant Peaks for Hnf1b</h3>
<div class="callout-content">
<p>Use the <a href="https://github.com/kbroman/qtl2/blob/main/R/find_peaks.R" class="external-link">find_peaks</a>
function to find the significant peaks for Hnf1b at the
<code>alpha = 0.05</code> threshold. Make a note of the QTL support
interval.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb67">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_hnf1b</span> <span class="op">&lt;-</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_hnf1b</span>,</span>
<span>                          map          <span class="op">=</span> <span class="va">map</span>,</span>
<span>                          threshold    <span class="op">=</span> <span class="va">thr_hnf1b</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,</span>
<span>                          prob         <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span></span>
<span><span class="va">peaks_hnf1b</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">lodindex</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">chr</span>, <span class="va">pos</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">kable</span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Hnf1b QTL Peaks"</span><span class="op">)</span></span></code></pre>
</div>
<table class="table">
<caption>Hnf1b QTL Peaks</caption>
<thead><tr class="header">
<th align="left">lodcolumn</th>
<th align="left">chr</th>
<th align="right">pos</th>
<th align="right">lod</th>
<th align="right">ci_lo</th>
<th align="right">ci_hi</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">ENSMUSG00000020679</td>
<td align="left">11</td>
<td align="right">84.40138</td>
<td align="right">36.92828</td>
<td align="right">83.64714</td>
<td align="right">84.40138</td>
</tr></tbody>
</table>
</div>
</div>
</div>
</div>
<div id="challenge-4-genomic-position-of-hnf1b" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-genomic-position-of-hnf1b" class="callout-inner">
<h3 class="callout-title">Challenge 4: Genomic Position of Hnf1b</h3>
<div class="callout-content">
<p>Find the position of the Hnf1b gene in the gene annotation. You may
want to use the <code>filter</code> or <code>subset</code> functions on
the <code>annot</code> object.</p>
<p>Where is Hnf1b in relation to the QTL interval in Challenge 3?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb68">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pos_hnf1b</span> <span class="op">&lt;-</span> <span class="fu">filter</span><span class="op">(</span><span class="va">annot</span>, <span class="va">symbol</span> <span class="op">==</span> <span class="st">"Hnf1b"</span><span class="op">)</span></span>
<span><span class="va">pos_hnf1b</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                              gene_id symbol chr    start      end strand
ENSMUSG00000020679 ENSMUSG00000020679  Hnf1b  11 83.85006 83.90592      1
                     middle nearest.marker.id        biotype       module
ENSMUSG00000020679 83.87799       11_84097611 protein_coding midnightblue
                   hotspot
ENSMUSG00000020679    &lt;NA&gt;</code></pre>
</div>
<p>The support interval ranges from 83.647144 to 84.401384 Mb. Hnf1b is
located on chromosome 11 at 83.850063 Mb, which is within the support
interval.</p>
</div>
</div>
</div>
</div>
<p>In the challenges above, we saw that Hnf1b has a QTL peak directly
over the gene’s genomic position. When this happens, we call is a
<em>local eQTL</em> because the QTL is co-located with the gene. We will
revisit this phenomenon more in later episodes.</p>
</div>
<div class="section level3">
<h3 id="estimating-founder-allele-effects">Estimating Founder Allele Effects<a class="anchor" aria-label="anchor" href="#estimating-founder-allele-effects"></a>
</h3>
<p>Let’s look at the QTL effects for insulin AUC on chromosome 11.</p>
<div id="accordionInstructor7" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor7" aria-expanded="false" aria-controls="collapseInstructor7">
  <h3 class="accordion-header" id="headingInstructor7">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor7" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor7" aria-labelledby="headingInstructor7">
<div class="accordion-body">
<p>This takes 40 minutes to run. Have the students read in the
pre-computed results.</p>
</div>
</div>
</div>
</div>
<p>Below is the code that you would use. It will take too long to run in
this class, so we will read the results in below.</p>
<div class="codewrapper sourceCode" id="cb70">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>      <span class="op">&lt;-</span> <span class="va">peaks_ins</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">blup_ins</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>,</span>
<span>                      pheno     <span class="op">=</span> <span class="va">ins_tauc</span><span class="op">[</span>,<span class="fl">2</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                      kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Read in the insulin AUC founder allele effects.</p>
<div class="codewrapper sourceCode" id="cb71">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup_ins</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/ins_tauc_blup_chr11.rds'</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will plot the founder allele effects.</p>
<div class="codewrapper sourceCode" id="cb72">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coefCC</span><span class="op">(</span>x      <span class="op">=</span> <span class="va">blup_ins</span>, </span>
<span>            map    <span class="op">=</span> <span class="va">map</span>, </span>
<span>            legend <span class="op">=</span> <span class="st">"bottomleft"</span>,</span>
<span>            scan1_output <span class="op">=</span> <span class="va">lod_ins</span><span class="op">[</span>, <span class="fl">2</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>            main   <span class="op">=</span> <span class="st">"insulin AUC"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-plot_ins_blup-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Next we will estimate the founder allele effects for Hnf1b.</p>
<div id="accordionInstructor8" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor8" aria-expanded="false" aria-controls="collapseInstructor8">
  <h3 class="accordion-header" id="headingInstructor8">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor8" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor8" aria-labelledby="headingInstructor8">
<div class="accordion-body">
<p>This takes 10 minutes to run. Have the students read in the
pre-computed results.</p>
</div>
</div>
</div>
</div>
<p>Below is the code that you would use. It will take too long to run in
this class, so we will read the results in below.</p>
<div class="codewrapper sourceCode" id="cb73">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>      <span class="op">&lt;-</span> <span class="va">peaks_hnf1b</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">blup_hnf1b</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>,</span>
<span>                      pheno     <span class="op">=</span> <span class="va">hnf1b</span>,</span>
<span>                      kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">blup_hnf1b</span>, file <span class="op">=</span> <span class="st">'data/hnf1b_blup_chr11.rds'</span><span class="op">)</span></span></code></pre>
</div>
<p>Read in the Hnf1b founder allele effects.</p>
<div class="codewrapper sourceCode" id="cb74">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup_hnf1b</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/hnf1b_blup_chr11.rds'</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will plot the founder allele effects.</p>
<div class="codewrapper sourceCode" id="cb75">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coefCC</span><span class="op">(</span>x      <span class="op">=</span> <span class="va">blup_hnf1b</span>, </span>
<span>            map    <span class="op">=</span> <span class="va">map</span>, </span>
<span>            legend <span class="op">=</span> <span class="st">"bottomleft"</span>,</span>
<span>            scan1_output <span class="op">=</span> <span class="va">lod_hnf1b</span>,</span>
<span>            main   <span class="op">=</span> <span class="st">"Hnf1b"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-one-eqtl-rendered-plot_hnf1b_blup-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-5-compare-founder-allele-effects" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5-compare-founder-allele-effects" class="callout-inner">
<h3 class="callout-title">Challenge 5: Compare Founder Allele Effects</h3>
<div class="callout-content">
<p>Compare at the pattern of founder allele effects at the QTL position
for insulin tAUC and Hnf1b.</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<p>In the insulin AUC allele effects, the A/J, C57BL/6J, 129S1/SvmJ, and
NOD/ShiLtJ alleles contribute to higher insulin AUC.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Gene expression values must be normalized to account for the library
size of each sample.</li>
<li>After normalization, gene expression values can be rankZ transformed
to make the distribution of every gene the same.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section><section id="aio-map-many-eqtls"><p>Content from <a href="map-many-eqtls.html">Mapping Many Gene Expression Traits</a></p>
<hr>
<p>Last updated on 2024-12-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/map-many-eqtls.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I map all of the genes in my data set?</li>
<li>What resources do I need to map all of the genes in my data
set?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To map several genes at the same time</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="mapping-all-genes">Mapping All Genes<a class="anchor" aria-label="anchor" href="#mapping-all-genes"></a>
</h3>
<p>Most people have laptops with enough memory and computing power to
map hundreds of genes. However, we have over 20,000 genes in our data
set. This would either take a very long time or would use up all of the
memory on most peoples’ laptops.</p>
<p>To map all of the genes, you will need to use a computing cluster,
such as <em>sumner2</em> at JAX. Since mapping all of the genes took 12
hours, you will not perform this operation in this workshop. Instead, we
will show you how to set up a computing job to run this analysis.</p>
<p><em>sumner2</em> uses <a href="https://slurm.schedmd.com/overview.html" class="external-link">slurm</a> to schedule
computing jobs. <code>slurm</code> is software that runs on the cluster
and manages how computing jobs are run. Users submit job requests and
<code>slurm</code> checks whether the requested memory, number of
processors, and time are available and then decides when to run the job.
It also balances the needs of different users so that everyone can get
their computing jobs done.</p>
<p><em>sumner2</em> also uses software containers to run software. JAX
uses <a href="https://docs.sylabs.io/guides/3.5/user-guide/introduction.html" class="external-link">Singularity</a>
containers. These contain software without the need to install it on the
cluster. The Computational Sciences group makes a large number of
containers available on <em>sumner2</em> and we will use their
<code>qtl2</code> container.</p>
<p>If you are unfamiliar with the JAX computing cluster, Research IT has
<a href="https://jacksonlaboratory.sharepoint.com/sites/ResearchIT/SitePages/Documentation.aspx" class="external-link">excellent
documentation</a> that explains how the cluster works, how to submit
jobs using <code>slurm</code>, and how to use <code>Singularity</code>
containers.</p>
<div class="section level4">
<h4 id="bash-script">bash Script<a class="anchor" aria-label="anchor" href="#bash-script"></a>
</h4>
<!-- DMG: Notes from sumner run. 11 hours with 20 cores & 100GB of memory.
11.4 G output file. 69005 rows x 21771 columns. -->
<p>To run a batch job on the cluster, we will create a bash script which
will request resources from <code>slurm</code> and will call an R
script, which will perform the mapping. The script that we used is shown
below.</p>
<pre><code>#!/bin/bash
#SBATCH --qos batch
#SBATCH --partition compute
#SBATCH --nodes 1
#SBATCH --ntasks 1
#SBATCH --cpus-per-task 22
#SBATCH --mem 200G
#SBATCH --time 1-0:00

CONTAINER=/projects/omics_share/meta/containers/quay.io-jaxcompsci-r-qtl2-deseq-biomart-tidy-v4.img

module load singularity

singularity exec ${CONTAINER} Rscript eqtl.R</code></pre>
<p>For each data set, you will need to determine the number of cores to
use. <code>qtl2</code> can perform the QTL mapping in parallel on one
node, so you will typically request only one node. In this case, we
requested 22 cores on one node. <em>sumner2</em> has nodes with up to 70
cores, but there is a point of diminishing returns when parallelizing
too much. We performed a short test run and found that we used about
100GB of memory, so we requested twice that amount in case there were
memory surges during computation. And we requested one day (24 hours) of
compute time.</p>
<p>Overall, the job took 12 hours and used 100GB of memory.</p>
<p>Note that there are many ways to set up a computing job, including
starting many separate mapping jobs and combining the results. We have
chosen to show you one simple way in this course. If you have
<code>slurm</code> expertise, you can try other methods of breaking up
the work.</p>
<p>Next, we call the R script using <code>singularity exec</code> to
execute the command which follows the container name. We call
<code>Rscript</code> to run the eQTL mapping script, which we placed in
the same directory as the bash script.</p>
<p>The inputs to the script are the genoprobs, expression data,
covariates, kinship matrix, and the marker map. You should have all of
these pre-computed and saved on the cluster in the same directory.</p>
</div>
<div class="section level4">
<h4 id="r-script">R Script<a class="anchor" aria-label="anchor" href="#r-script"></a>
</h4>
<pre><code><span><span class="co">################################################################################</span></span>
<span><span class="co"># Script to map all of the Attic DO 500 pancreatic islet RNASeq data.</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># Daniel Gatti</span></span>
<span><span class="co"># dan.gatti@jax.org</span></span>
<span><span class="co"># 2024-11-01</span></span>
<span><span class="co">################################################################################</span></span>
<span></span>
<span><span class="co">##### LIBRARIES #####</span></span>
<span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##### VARIABLES #####</span></span>
<span></span>
<span><span class="co"># Base directory for project.</span></span>
<span><span class="va">base_dir</span> <span class="op">=</span> <span class="st">'/projects/compsci/eqtl_course'</span></span>
<span></span>
<span><span class="co"># Number of cores. MUST MATCH SLURM REQUEST.</span></span>
<span><span class="va">n_cores</span> <span class="op">=</span> <span class="fl">20</span></span>
<span></span>
<span><span class="co"># Read in pre-built data files.</span></span>
<span></span>
<span><span class="va">probs</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_DO500_genoprobs_v5.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">K</span>     <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_kinship.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">covar</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_covar.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">expr</span>  <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_expr_rz.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">map</span>   <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_map.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##### MAIN #####</span></span>
<span></span>
<span><span class="co"># Convert covariates to factors.</span></span>
<span><span class="va">covar</span><span class="op">$</span><span class="va">sex</span>    <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">covar</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span></span>
<span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a matrix of additive covariates.</span></span>
<span><span class="va">addcovar</span>     <span class="op">=</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span>, data <span class="op">=</span> <span class="va">covar</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Map all genes.</span></span>
<span><span class="va">lod</span> <span class="op">=</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>            pheno     <span class="op">=</span> <span class="va">expr</span>,</span>
<span>            kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>            addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>            cores     <span class="op">=</span> <span class="va">n_cores</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save the LOD scores for all genes.</span></span>
<span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">lod</span>, file <span class="op">=</span> <span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_eqtl_lod.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get the peaks with LOD &gt;= 6.</span></span>
<span><span class="va">peaks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>,</span>
<span>                   map          <span class="op">=</span> <span class="va">map</span>,</span>
<span>                   threshold    <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                   prob         <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save the harvested LOD peaks.</span></span>
<span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">peaks</span>, file <span class="op">=</span> <span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_eqtl_peaks.rds'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
<p>The output of this script is two files.</p>
<ol style="list-style-type: decimal">
<li>
<code>attie_do_eqtl_lod.rds</code>: This is numeric matrix
containing LOD scores for all genes at all markers. We save it as a
compressed R data file (<code>*.rds</code>). In this case, the file is
11 GB.</li>
<li>
<code>attie_do_eqtl_peaks.rds</code>: This is a data.frame
containing the peaks with LOD over 6. We save it as a compressed R data
file (<code>*.rds</code>). In this case, the file is 674 KB. You should
have downloaded this file into your <code>data</code> directory.</li>
</ol>
</div>
</div>
<div class="section level3">
<h3 id="finding-qtl-peaks">Finding QTL Peaks<a class="anchor" aria-label="anchor" href="#finding-qtl-peaks"></a>
</h3>
<p>Let’s load in the QTL peaks that we pre-computed.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">"data/attie_do_eqtl_peaks.rds"</span><span class="op">)</span></span></code></pre>
</div>
<p><code>peaks</code> is a data frame which contains all of the peaks in
the same format as we saw in the previous lesson. Let’s remind ourselves
what the peaks table looks like.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">peaks</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr       pos       lod      ci_lo     ci_hi
1        2 ENSMUSG00000000028   6  71.73759  7.358089  64.958691  77.39716
2        2 ENSMUSG00000000028  11  40.54754  6.424368   5.107459 106.64066
3        2 ENSMUSG00000000028  16  18.54468  8.001125  15.918386  20.83108
4        3 ENSMUSG00000000037   5  99.00918  6.524121  92.704924 102.15720
5        3 ENSMUSG00000000037  12  77.55794  6.123602  76.998817 110.18071
6        3 ENSMUSG00000000037   X 161.30209 28.383984 160.417579 161.39338</code></pre>
</div>
<p>Each row contains information for one peak, including the gene ID,
chromosome, peak position, LOD, and support interval.</p>
<p>Notice that both genes <code>ENSMUSG00000000028</code> and
<code>ENSMUSG00000000037</code> have more than one peak.</p>
<div id="challenge-1-how-many-genes-have-qtl-peaks" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-how-many-genes-have-qtl-peaks" class="callout-inner">
<h3 class="callout-title">Challenge 1: How many genes have QTL peaks?</h3>
<div class="callout-content">
<p>Look at how many genes we have and how many rows there are in peaks.
Is this the same number? If not, can you explain why the numbers are
different.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Get the number of genes from <code>expr_rz</code>.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ncol</span><span class="op">(</span><span class="va">expr_rz</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 21771</code></pre>
</div>
<p>Get the number of QTL peaks in <code>peaks</code>.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">peaks</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 42981</code></pre>
</div>
<p>There are more peaks than there are genes.</p>
<p>There are several different possibilities when mapping a gene.</p>
<ol style="list-style-type: decimal">
<li>A gene may have no QTL peaks over a LOD of 6, and so it will not
appear in the list of peaks.</li>
<li>A gene may have more than one QTL peak with a LOD greater than 6,
and so will appear many times in the list of peaks.</li>
</ol>
</div>
</div>
</div>
</div>
<p>Let’s count the number of peaks that each gene has.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">lodcolumn</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">n</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  n   nn
1 1 9123
2 2 6814
3 3 3827
4 4 1441
5 5  445
6 6   95
7 7   20
8 8    4
9 9    2</code></pre>
</div>
<p>From the table above, we can see that most genes have one or two
peaks. However, some genes have 9 peaks with LOD &gt; 6!</p>
<p>When we built this table, we selected all of the peaks with LOD &gt;
6. This is well below the alpha = 0.05 significance threshold. Let’s get
that threshold and filter the list of peaks.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ethr</span> <span class="op">&lt;-</span> <span class="fu">summary</span><span class="op">(</span>object <span class="op">=</span> <span class="va">eperm</span>,</span>
<span>                alpha  <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">ethr</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     ENSMUSG00000020679
0.05               7.47</code></pre>
</div>
<p>Now filter the peaks to retain peaks with LOD greater than or equal
to 7.4745761.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_filt</span> <span class="op">&lt;-</span> <span class="fu">filter</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">lod</span> <span class="op">&gt;=</span> <span class="va">ethr</span><span class="op">[</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-2-how-many-peaks-are-there-after-filtering" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-how-many-peaks-are-there-after-filtering" class="callout-inner">
<h3 class="callout-title">Challenge 2: How many peaks are there after filtering?</h3>
<div class="callout-content">
<p>Get the number of QTL peaks that we have after filtering.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">peaks_filt</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 17192</code></pre>
</div>
<p>There are now fewer peaks.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3-how-many-peaks-does-each-gene-have" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-how-many-peaks-does-each-gene-have" class="callout-inner">
<h3 class="callout-title">Challenge 3: How many peaks does each gene have?</h3>
<div class="callout-content">
<p>Use the code above to count the number of QTL peaks that each gene
has.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">peaks_filt</span>, <span class="va">lodcolumn</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">n</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  n    nn
1 1 12680
2 2  1962
3 3   166
4 4    20
5 5     2</code></pre>
</div>
<p>Now most genes have only one peak and only two genes have five
peaks.</p>
</div>
</div>
</div>
</div>
<p>Let’s look at the number of genes with LOD scores above a certain
value. We do this by counting the number of peaks with LOD &gt; 6, then
&gt; 7, etc.</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Do not try to type the next block. We will look at it on the workshop
website.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_brks</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">results</span>  <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span><span class="va">lod_brks</span>, n <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">lod_brks</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">results</span><span class="op">$</span><span class="va">n</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">sum</span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">lod</span> <span class="op">&gt;=</span> <span class="va">i</span><span class="op">)</span></span>
<span>  </span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">lod_brks</span> <span class="op">&gt;=</span> <span class="fl">6</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">lod_brks</span>, <span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_line</span><span class="op">(</span>linewidth <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">10</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Number of Peaks above LOD Score"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"LOD"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"Number of Peaks with LOD &gt;= x"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-many-eqtls-rendered-lod_cum_dist-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">lod_brks</span>, <span class="va">results</span><span class="op">)</span></span></code></pre>
</div>
<p>As you can see from the plot above, most peaks have LODs less than
10. However, about 12,000 peaks have LODs over 10. As we increase the
LOD score, we see that fewer genes have high LODs. Let’s look at how
many peaks each gene has as we increase the LOD threshold.</p>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Do not try to type the next block. We will look at it on the workshop
website.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_brks</span> <span class="op">&lt;-</span> <span class="fl">0</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">result</span>   <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>lod_brks <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, n <span class="op">=</span> <span class="fl">0</span>, nn <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">lod_brks</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">smry</span> <span class="op">&lt;-</span> <span class="va">peaks</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">filter</span><span class="op">(</span><span class="va">lod</span> <span class="op">&gt;</span> <span class="va">i</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">lodcolumn</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">mutate</span><span class="op">(</span>lod_brks <span class="op">=</span> <span class="va">i</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">result</span>, <span class="va">smry</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span>
<span></span>
<span><span class="va">result</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">lod_brks</span> <span class="op">&gt;=</span> <span class="fl">6</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">as.character</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">lod_brks</span>, <span class="va">nn</span>, color <span class="op">=</span> <span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>group <span class="op">=</span> <span class="va">n</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">10</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Number of Peaks per Gene by LOD Threshold"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"LOD"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"Number of Peaks per Gene"</span>,</span>
<span>         color <span class="op">=</span> <span class="st">"Number of Peaks"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-many-eqtls-rendered-eqtl_num_peaks_per_gene-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">lod_brks</span>, <span class="va">result</span><span class="op">)</span></span></code></pre>
</div>
<p>From this plot, we can see that higher LOD scores occur only once per
gene. This makes sense since a high LOD score implies that the peak
explains a large proportion of the gene’s variance.</p>
</div>
<div class="section level3">
<h3 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h3>
<p>In this episode, you learned how to run an eQTL analysis on a
computing cluster and started to look at the resulting peaks. When we
filter the peaks by higher LOD scores, we tend to get fewer peaks per
gene.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Mapping all genes in a study requires a computing cluster.</li>
<li>Genes may have more than one QTL peak.</li>
<li>High LOD scores often occur only once per gene.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div></section><section id="aio-maximum-peaks-genes"><p>Content from <a href="maximum-peaks-genes.html">Maximum eQTL Peaks and Nearby Genes</a></p>
<hr>
<p>Last updated on 2024-12-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/maximum-peaks-genes.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 30 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I select QTL peaks for each gene?</li>
<li>How do I find genes near QTL peaks?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn how to filter LOD peaks to retain significant peaks.</li>
<li>Understand the two levels of multiple testing in eQTL mapping
studies.</li>
<li>Find genes which are located under QTL peaks.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="multiple-testing-in-qtl-mapping">Multiple Testing in QTL Mapping<a class="anchor" aria-label="anchor" href="#multiple-testing-in-qtl-mapping"></a>
</h3>
<p>When we perform multiple hypothesis tests, as we do in QTL mapping,
we must adjust for the fact that our overall false-positive rate is
higher than the nominal false-positive rate for a single test. There are
also two levels of multiple testing in expression QTL mapping, at the
marker level and at the gene level.</p>
<p>When we perform a single QTL scan, we are performing a hypothesis
test at each marker. We could perform a traditional multiple-testing
correction such as a Benjamini-Hochberg false discovery rate (FDR).
However, by performing permutations of the sample labels, scanning the
genome, and retaining the maximum LOD from each permutation, we are
effectively adjusting for multiple testing across the genome because we
are selecting only the maximum LOD across all markers in each
permutation.</p>
<p>Multiple testing at the gene level comes from mapping multiple genes.
If we use the 0.05 significance level for all genes, our overall
false-positive rate will be greater than 0.05. We must also perform a
multiple testing correction for each gene. However, we have more than
one peak per gene, so how should we proceed?</p>
<p>While genes that have multiple peaks are important and may have
interesting biology, we do not currently have a method of adjusting for
multiple QTL peaks for each gene. Instead, we will select the peak with
the highest LOD for each gene and will then perform a multiple-testing
correction.</p>
<p>First, we will filter the peaks to retain the peak with the highest
LOD score for each gene.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_filt</span> <span class="op">&lt;-</span> <span class="va">peaks</span> <span class="op">|&gt;</span></span>
<span>                <span class="fu">group_by</span><span class="op">(</span><span class="va">lodcolumn</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                <span class="fu">filter</span><span class="op">(</span><span class="va">lod</span> <span class="op">==</span> <span class="fu">max</span><span class="op">(</span><span class="va">lod</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s make sure that we have only one peak per gene.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_filt</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">lodcolumn</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">n</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 1 × 2
      n    nn
  &lt;int&gt; &lt;int&gt;
1     1 21771</code></pre>
</div>
<p>This looks good! All of the genes have only one peak.</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Note that only 21771 genes out of 21771 have QTL peaks.</p>
</div>
</div>
</div>
<p>Most multiple-testing correction methods rely upon p-values as input.
We currently have LOD scores. We will need to convert them into
genome-wide p-values using the permutations that we performed in Mapping
One eQTL. We have these in a variable called <code>eperm</code>. We will
estimate the p-value for each gene by getting the proportion of
permutations with a LOD greater than or equal to the LOD of each
gene.</p>
<div id="challenge-1-what-is-the-minimum-p-value-that-we-can-have" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-what-is-the-minimum-p-value-that-we-can-have" class="callout-inner">
<h3 class="callout-title">Challenge 1: What is the minimum p-value that we can have?</h3>
<div class="callout-content">
<p>We performed 1,000 permutations. What does this tell you about the
minimum possible p-value that we can get?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Technically, we could get a p-value of 0 if the gene’s LOD score is
above the highest LOD score in the permutations. In practice, we
recognize that the p-value isn’t zero, but some number less than one
over the number of permutations (1 / 1000) that we performed.</p>
</div>
</div>
</div>
</div>
<p>Let’s estimate the p-values by calculating the proportion of
permutations with LOD scores greater than or equal to each gene’s LOD.
We will also adjust p-values which are zero to be 0.001 to be
conservative.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_filt</span> <span class="op">&lt;-</span> <span class="va">peaks_filt</span> <span class="op">|&gt;</span></span>
<span>                <span class="fu">group_by</span><span class="op">(</span><span class="va">lodcolumn</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                <span class="fu">mutate</span><span class="op">(</span>pvalue <span class="op">=</span> <span class="fu">mean</span><span class="op">(</span><span class="va">lod</span> <span class="op">&lt;=</span> <span class="va">eperm</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                       pvalue <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">pvalue</span> <span class="op">&lt;</span> <span class="fl">0.001</span>, <span class="fl">0.001</span>, <span class="va">pvalue</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Now we can apply a procedure called <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC170937/" class="external-link">q-value</a> to
estimate the false discovery rate (FDR) for each gene.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_filt</span> <span class="op">&lt;-</span> <span class="va">peaks_filt</span> <span class="op">|&gt;</span></span>
<span>                <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                <span class="fu">mutate</span><span class="op">(</span>qvalue <span class="op">=</span> <span class="fu">qvalue</span><span class="op">(</span><span class="va">peaks_filt</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span><span class="op">$</span><span class="va">qvalue</span><span class="op">)</span></span></code></pre>
</div>
<p>Now we can plot the nominal p-values versus the q-values.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_filt</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">pvalue</span>, <span class="va">qvalue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_line</span><span class="op">(</span>linewidth <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_abline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>intercept <span class="op">=</span> <span class="fl">0</span>, slope <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">'red'</span>, linewidth <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"p-values versus q-values"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/maximum-peaks-genes-rendered-pv_vs_qv-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, we can see that the q-values are lower than the
p-values. The p-value versus q-value line is shown in black and the x =
y line in red. The q-values represent the FDR of each LOD peak and all
of the genes with lower q-values. A p-value of 0.25 has a q-value of
~0.06, meaning that the FDR of LOD peaks with p-values of 0.25 or less
have an FDR or 0.06.</p>
<p>We can filter the list of peaks to include ones with q-values less
than or equal to 0.05.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_filt</span> <span class="op">&lt;-</span> <span class="va">peaks_filt</span> <span class="op">|&gt;</span></span>
<span>                <span class="fu">filter</span><span class="op">(</span><span class="va">qvalue</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s see how many genes we have retained.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">peaks_filt</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 16781</code></pre>
</div>
<p>We still have almost 17,000 genes with an FDR of 5% or less.</p>
<p>Next, let’s look at what the range of LOD scores is.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">range</span><span class="op">(</span><span class="va">peaks_filt</span><span class="op">$</span><span class="va">lod</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1]   6.739219 194.063370</code></pre>
</div>
<p>The lowest LOD score is 6.7392192, which is lower than the
permutation threshold of 7.4745761.</p>
</div>
<div class="section level3">
<h3 id="finding-genes-under-qtl-peaks">Finding Genes under QTL Peaks<a class="anchor" aria-label="anchor" href="#finding-genes-under-qtl-peaks"></a>
</h3>
<p>We can use the gene annotation data to find genes under QTL peaks.
Let’s get the peak with the highest LOD score.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">high_peak</span> <span class="op">&lt;-</span> <span class="fu">slice_max</span><span class="op">(</span><span class="va">peaks_filt</span>, order_by <span class="op">=</span> <span class="va">lod</span>, n <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">high_peak</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 1 × 9
  lodindex lodcolumn          chr     pos   lod ci_lo ci_hi pvalue   qvalue
     &lt;int&gt; &lt;chr&gt;              &lt;fct&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;
1    12715 ENSMUSG00000048758 9      106.  194.  106.  106.  0.001 0.000369</code></pre>
</div>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>DMG: This is temporary. Remove later.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">high_peak</span><span class="op">$</span><span class="va">ci_lo</span> <span class="op">=</span> <span class="fl">106</span></span>
<span><span class="va">high_peak</span><span class="op">$</span><span class="va">ci_hi</span> <span class="op">=</span> <span class="fl">107</span></span></code></pre>
</div>
</div>
</div>
</div>
<p>This peak has a LOD of 194.0633696! Next, we will get the genes which
lie within the confidence interval from the gene annotation.</p>
<p>Let’s remind ourselves what the gene annotation contains.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">annot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                              gene_id symbol chr     start       end strand
ENSMUSG00000000001 ENSMUSG00000000001  Gnai3   3 108.10728 108.14615     -1
ENSMUSG00000000028 ENSMUSG00000000028  Cdc45  16  18.78045  18.81199     -1
ENSMUSG00000000037 ENSMUSG00000000037  Scml2   X 161.11719 161.25821      1
ENSMUSG00000000049 ENSMUSG00000000049   Apoh  11 108.34335 108.41440      1
ENSMUSG00000000056 ENSMUSG00000000056   Narf  11 121.23725 121.25586      1
ENSMUSG00000000058 ENSMUSG00000000058   Cav2   6  17.28119  17.28911      1
                      middle nearest.marker.id        biotype      module
ENSMUSG00000000001 108.12671       3_108090236 protein_coding   darkgreen
ENSMUSG00000000028  18.79622       16_18817262 protein_coding        grey
ENSMUSG00000000037 161.18770       X_161182677 protein_coding        grey
ENSMUSG00000000049 108.37887      11_108369225 protein_coding greenyellow
ENSMUSG00000000056 121.24655      11_121200487 protein_coding       brown
ENSMUSG00000000058  17.28515        6_17288298 protein_coding       brown
                   hotspot
ENSMUSG00000000001    &lt;NA&gt;
ENSMUSG00000000028    &lt;NA&gt;
ENSMUSG00000000037    &lt;NA&gt;
ENSMUSG00000000049    &lt;NA&gt;
ENSMUSG00000000056    &lt;NA&gt;
ENSMUSG00000000058    &lt;NA&gt;</code></pre>
</div>
<p>We can use the start, middle, or end columns to get the gene
positions. In this case, we will use the middle.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annot_filt</span> <span class="op">&lt;-</span> <span class="va">annot</span> <span class="op">|&gt;</span></span>
<span>               <span class="fu">filter</span><span class="op">(</span><span class="va">middle</span> <span class="op">&gt;=</span> <span class="va">high_peak</span><span class="op">$</span><span class="va">ci_lo</span> <span class="op">&amp;</span> <span class="va">middle</span> <span class="op">&lt;=</span> <span class="va">high_peak</span><span class="op">$</span><span class="va">ci_hi</span><span class="op">)</span></span></code></pre>
</div>
<p>There are 161 genes within the QTL support interval. This is a large
number and would require more research to find candidate genes.</p>
<p>Let’s see where the gene being mapped is located. Note that the
Ensembl ID of the gene is in <code>high_peak</code>.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">filter</span><span class="op">(</span><span class="va">annot</span>, <span class="va">gene_id</span> <span class="op">==</span> <span class="va">high_peak</span><span class="op">$</span><span class="va">lodcolumn</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                              gene_id symbol chr    start      end strand
ENSMUSG00000048758 ENSMUSG00000048758  Rpl29   9 106.4295 106.4316      1
                     middle nearest.marker.id        biotype module hotspot
ENSMUSG00000048758 106.4305       9_106387870 protein_coding   grey    &lt;NA&gt;</code></pre>
</div>
<div id="challenge-2-where-is-the-gene" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-where-is-the-gene" class="callout-inner">
<h3 class="callout-title">Challenge 2: Where is the gene?</h3>
<div class="callout-content">
<p>Look at the gene location above and compare it with its corresponding
QTL peak in <code>high_peak</code>. Is there any relationship between
the two genomic positions?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>Print out the QTL peak position.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">high_peak</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 1 × 9
  lodindex lodcolumn          chr     pos   lod ci_lo ci_hi pvalue   qvalue
     &lt;int&gt; &lt;chr&gt;              &lt;fct&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;
1    12715 ENSMUSG00000048758 9      106.  194.   106   107  0.001 0.000369</code></pre>
</div>
<p>The QTL is located on chromosome 9 at 106.38787 Mb. The gene is
located on chromosome 9 at Mb. These genomic positions are nearly
identical.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>There can be more than one significant QTL peak for each gene.</li>
<li>We sometimes focus on the largest peak for each gene.</li>
<li>A multiple-testing correction should be applied to all peaks.</li>
<li>Sometimes a gene s co-located with its QTL peak.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div></section><section id="aio-create-transcriptome-map"><p>Content from <a href="create-transcriptome-map.html">Creating A Transcriptome Map</a></p>
<hr>
<p>Last updated on 2024-12-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/create-transcriptome-map.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I create and interpret a transcriptome map?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe a transcriptome map.</li>
<li>Interpret a transcriptome map.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="local-and-distant-eqtl">Local and Distant eQTL<a class="anchor" aria-label="anchor" href="#local-and-distant-eqtl"></a>
</h3>
<p>In the previous lesson, we saw that the QTL peak for a gene can lie
directly over the gene that is being mapped. This is called a <em>local
eQTL</em> because the QTL peak is located near the gene. When the QTL
peak is located far from the gene being mapped, we call this a
<em>distal eQTL</em>. When the QTL peak is located on the same
chromosome as the gene, there are different heuristics regarding the
distance between the gene and its QTL peak that we use to call an eQTL
local or distal. This depends on the type of cross and the resolution of
the recombination block structure.</p>
<p>In this episode, we will create a table containing the QTL peaks for
all genes and the gene positions. We will then classify QTL into local
or distal and will make a plot of all off the eQTL in relation to the
gene positions.</p>
</div>
<div class="section level3">
<h3 id="transcriptome-map">Transcriptome Map<a class="anchor" aria-label="anchor" href="#transcriptome-map"></a>
</h3>
<p>We have encapsulated the code to create a transcriptome map in a file
in this lesson. You can copy this file from the Github repository to use
in your eQTL mapping project. We will read this file in now.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getwd</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dir</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">source</span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/smcclatchy/eqtl-mapping/refs/heads/main/episodes/code/gg_transcriptome_map.R"</span><span class="op">)</span></span></code></pre>
</div>
<p>This file loads in a function called <code>ggtmap</code>, which
requires the input data to be in a specific format.</p>
<p>In order to use the <code>ggtmap</code> function, we need to provide
specific column names. These are documented in the
<code>gg_transcriptome_map.R</code> file in the code directory of this
workshop. The required column names are:</p>
<ul>
<li>
<code>data</code>: data.frame (or tibble) with the following
columns:
<ul>
<li>
<code>gene_id</code>: (required) character string containing the
Ensembl gene ID.</li>
<li>
<code>qtl_chr</code>: (required) character string containing QTL
chromosome.</li>
<li>
<code>qtl_pos</code>: (required) floating point number containing
the QTL position in Mb.</li>
<li>
<code>qtl_lod</code>: (optional) floating point number containing
the LOD score.</li>
<li>
<code>gene_chr</code>: (optional) character string containing
transcript chromosome.</li>
<li>
<code>gene_start</code>: (optional) character string containing
transcript start position in Mb.</li>
<li>
<code>gene_end</code>: (optional) character string containing
transcript end position in Mb.</li>
</ul>
</li>
</ul>
<p>First, we will get the gene positions from the annotation and rename
the columns to match what <code>ggtmap</code> requires. We need to have
columns named <code>gene_id</code>, <code>gene_chr</code>, and
<code>gene_pos</code>. We must rename the columns because, when we are
finished, we will have <code>chr</code> and <code>pos</code> columns for
both the gene and its QTL. We will also add the <code>symbol</code>
column since it is nice to have gene symbols in the data.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_pos</span> <span class="op">&lt;-</span> <span class="va">annot</span> <span class="op">|&gt;</span></span>
<span>              <span class="fu">select</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">symbol</span>, </span>
<span>                     gene_chr   <span class="op">=</span> <span class="va">chr</span>, </span>
<span>                     gene_start <span class="op">=</span> <span class="va">start</span>,</span>
<span>                     gene_end   <span class="op">=</span> <span class="va">end</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will rename the columns in the filtered peaks and will join
them with the gene positions from above.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eqtl</span> <span class="op">&lt;-</span> <span class="va">peaks</span> <span class="op">|&gt;</span></span>
<span>          <span class="fu">select</span><span class="op">(</span>gene_id <span class="op">=</span> <span class="va">lodcolumn</span>,</span>
<span>                 qtl_chr <span class="op">=</span> <span class="va">chr</span>,</span>
<span>                 qtl_pos <span class="op">=</span> <span class="va">pos</span>,</span>
<span>                 qtl_lod <span class="op">=</span> <span class="va">lod</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>          <span class="fu">left_join</span><span class="op">(</span><span class="va">gene_pos</span>, by <span class="op">=</span> <span class="st">"gene_id"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>          <span class="fu">mutate</span><span class="op">(</span>marker.id <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">qtl_chr</span>,   <span class="va">qtl_pos</span> <span class="op">*</span> <span class="fl">1e6</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>,</span>
<span>                 gene_chr  <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">gene_chr</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                 qtl_chr   <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">qtl_chr</span>,  levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">"X"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">gene_pos</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-1-how-many-genes-have-qtl-on-the-same-chromosome" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-how-many-genes-have-qtl-on-the-same-chromosome" class="callout-inner">
<h3 class="callout-title">Challenge 1: How many genes have QTL on the same chromosome?</h3>
<div class="callout-content">
<p>Write a command to count the number of genes which are located on the
same chromosome as their corresponding QTL peak.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="va">eqtl</span><span class="op">$</span><span class="va">qtl_chr</span> <span class="op">==</span> <span class="va">eqtl</span><span class="op">$</span><span class="va">gene_chr</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 15061</code></pre>
</div>
<p>15061 genes have QTL on the same chromosome.</p>
</div>
</div>
</div>
</div>
<p>We can tabulate the number of local and distal eQTL that we have and
add this to our QTL summary table. A local eQTL occurs when the QTL
peaks is directly over the gene position. But what if it is 2 Mb away?
Or 10 Mb? It’s possible that a gene may have a distal eQTL on the same
chromosome if the QTL is “far enough” from the gene. In <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller et
al</a>, the authors selected a 4 Mb distance from the corresponding gene
and we will use this threshold.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eqtl</span> <span class="op">&lt;-</span> <span class="va">eqtl</span> <span class="op">|&gt;</span> </span>
<span>          <span class="fu">mutate</span><span class="op">(</span>local <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">qtl_chr</span> <span class="op">==</span> <span class="va">gene_chr</span> <span class="op">&amp;</span> </span>
<span>                                 <span class="fu">abs</span><span class="op">(</span><span class="va">gene_start</span> <span class="op">-</span> <span class="va">qtl_pos</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">4</span>, </span>
<span>                                     <span class="cn">TRUE</span>, </span>
<span>                                     <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">count</span><span class="op">(</span><span class="va">eqtl</span>, <span class="va">local</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  local     n
1 FALSE 28974
2  TRUE 14007</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="plot-transcriptome-map">Plot Transcriptome Map<a class="anchor" aria-label="anchor" href="#plot-transcriptome-map"></a>
</h3>
<p>In <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller
et al</a>, the authors used a LOD threshold of 7.2 to select genes to
use in their transcriptome map. We will use this threshold to reproduce
their results as closely as possible.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eqtl_thr</span> <span class="op">=</span> <span class="fl">7.2</span></span>
<span></span>
<span><span class="fu">ggtmap</span><span class="op">(</span>data <span class="op">=</span> <span class="va">eqtl</span> <span class="op">|&gt;</span> </span>
<span>                <span class="fu">filter</span><span class="op">(</span><span class="va">qtl_lod</span>      <span class="op">&gt;=</span> <span class="va">eqtl_thr</span><span class="op">)</span>, </span>
<span>                       local.points <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                       local.radius <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-transcriptome-map-rendered-tmap-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot above is called a <em>transcriptome map</em> because it
shows the positions of the genes (or transcripts) and their
corresponding QTL. The QTL position is shown on the X-axis and the gene
position is shown on the Y-axis. The chromosomes are listed along the
top and right of the plot.</p>
<div id="challenge-2-what-are-the-blue-points-on-the-diagonal" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-what-are-the-blue-points-on-the-diagonal" class="callout-inner">
<h3 class="callout-title">Challenge 2: What are the blue points on the diagonal?</h3>
<div class="callout-content">
<p>What type of QTL are the genes with QTL that are located along the
diagonal?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Points on the diagonal have QTL that are located in the same position
as the gene, and so are local eQTL.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3-are-there-any-genome-locations-with-many-qtl" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-are-there-any-genome-locations-with-many-qtl" class="callout-inner">
<h3 class="callout-title">Challenge 3: Are there any genome locations with many QTL?</h3>
<div class="callout-content">
<p>Look at the transcriptome map and see if you can find any vertical
stripes? What do these vertical stripes mean and what might cause
them?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>There are vertical stripes on chromosomes 2, 5, and 11. There may be
more, depending on how you look at the plot. Since the QTL position is
on the X-axis, these stripes represent genes which are located through
out the genome, but all have a QTL in the same location.</p>
<p>These stripes imply that there is some genomic feature at the QTL
position which regulates the expression levels of many genes. This might
be a transcription factor or some other molecule which can regulate
transcription, possibly through multiple steps.</p>
</div>
</div>
</div>
</div>
<p>If you look at the plot, there are vertical bands of points which
stack over a single QTL location. These are called <em>eQTL
hotspots</em>. Rather than look at the transcriptome map, it may be
easier to look at the density of the eQTL along the genome. We have
provided a function called <code>eqtl_density_plot</code> to do
this.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span><span class="va">eqtl</span>, lod_thr <span class="op">=</span> <span class="va">eqtl_thr</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-transcriptome-map-rendered-eqtl_density-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot above shows the mouse genome on the X-axis and the number of
transcripts in a 4 Mb window on the Y-axis. It is difficult to say how
many genes must be involved to call something and eQTL hotspot. There
are permutation-based methods which require a large amount of time and
memory. In this case, we have called hotspots involving more than 100
genes eQTL hotspots.</p>
<p>Note that there appears to be an eQTL hotspot on chromosome 7 in the
plot above, but this is not evident in the transcriptome map.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span><span class="fu">filter</span><span class="op">(</span><span class="va">eqtl</span>, <span class="va">local</span> <span class="op">==</span> <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>                  lod_thr <span class="op">=</span> <span class="va">eqtl_thr</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Distant eQTL Density"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-transcriptome-map-rendered-ldistant_eqtl_density-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">eqtl_density_plot</span><span class="op">(</span><span class="fu">filter</span><span class="op">(</span><span class="va">eqtl</span>, <span class="va">local</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span>, </span>
<span>                  lod_thr <span class="op">=</span> <span class="va">eqtl_thr</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Local eQTL Density"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-transcriptome-map-rendered-local_eqtl_density-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From these two plots, it appears that the eQTL hotspots on most
chromosomes contain distant eQTL.</p>
<div id="challenge-4-compare-results-to-keller-et-al." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-compare-results-to-keller-et-al." class="callout-inner">
<h3 class="callout-title">Challenge 4: Compare results to Keller et al.</h3>
<div class="callout-content">
<p>Look at figure 3 in <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller et
al</a>. How do the eQTL density plots compare to their results?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>The eQTL density plots are largely similar. The hotspot on chromosome
11 contains more genes in our analysis than in the paper.</p>
</div>
</div>
</div>
</div>
<p>How can we find the location of the eQTL hotspots and the genes which
lie within each eQTL? We have written a function called
<code>get_eqtl_hotspots</code> to help you and have provides it in
<code>gg_transcriptome_map.R</code>. It requires the eQTL data, a LOD
threshold to filter the peaks, the number of genes required to call a
locus a hotspot, and the radius in Mb around the hotspot to use when
selecting genes which are part of the hotspot. We will use a 2 MB radius
around the hotspot, but may refine that later.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hotspots</span> <span class="op">&lt;-</span> <span class="fu">get_eqtl_hotspots</span><span class="op">(</span>data           <span class="op">=</span> <span class="va">eqtl</span>, </span>
<span>                              lod_thr        <span class="op">=</span> <span class="fl">7</span>, </span>
<span>                              hotspot_thr    <span class="op">=</span> <span class="fl">200</span>,</span>
<span>                              hotspot_radius <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s see how many hotspots we have and how many genes are in each
hotspot.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sapply</span><span class="op">(</span><span class="va">hotspots</span>, <span class="va">nrow</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  2   5   7  11
274 259 265 279 </code></pre>
</div>
<p>In the table above, the first row of values are the chromosomes on
which hotspots occur. The second row contains the number of genes in
each hotspot.</p>
<p>Where does each hotspot occur? We can get this information by taking
the mean of the eQTL positions in each hotspot.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sapply</span><span class="op">(</span><span class="va">hotspots</span>, <span class="kw">function</span><span class="op">(</span><span class="va">z</span><span class="op">)</span> <span class="op">{</span> <span class="fu">mean</span><span class="op">(</span><span class="va">z</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span> <span class="op">}</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        2         5         7        11
164.01218 146.21618  45.45160  70.53259 </code></pre>
</div>
<p>Let’s look at some of the genes in the chromosome 2 hotspot.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             gene_id qtl_chr  qtl_pos   qtl_lod        symbol gene_chr
1 ENSMUSG00000059647       2 162.0824  7.073704       Gm10068        1
2 ENSMUSG00000021902       2 162.3910  7.207689          Phf7       14
3 ENSMUSG00000094856       2 162.5281 52.699793       Gm21962        3
4 ENSMUSG00000053141       2 162.6222 80.662602         Ptprt        2
5 ENSMUSG00000087267       2 162.6222 75.715207 4933427J07Rik        2
6 ENSMUSG00000025577       2 162.6285  8.038627          Cbx2       11
  gene_start  gene_end   marker.id local
1   58.64661  58.64832 2_162082435 FALSE
2   31.23770  31.25122 2_162391032 FALSE
3  137.67152 137.67254 2_162528109 FALSE
4  161.52199 162.66115 2_162622197  TRUE
5  128.95567 128.95786 2_162622197 FALSE
6  119.02296 119.03127 2_162628468 FALSE</code></pre>
</div>
<p>Let’s also look at the positions of the eQTL for these genes by
making a histogram of the positions.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">qtl_pos</span>, </span>
<span>     breaks <span class="op">=</span> <span class="fl">50</span>, </span>
<span>     las    <span class="op">=</span> <span class="fl">1</span>,</span>
<span>     main   <span class="op">=</span> <span class="st">"Positions of Chr 2 eQTL"</span>,</span>
<span>     xlab   <span class="op">=</span> <span class="st">"Position (Mb)"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-transcriptome-map-rendered-hotspot_chr2_pos_hist-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From the plot above, we can see that many genes have eQTL which stack
up around 164 Mb. Are the other genes part of the same regulatory
network? In this case, we will filter the chromosome 2 hotspot to only
include the genes with an eQTL near 164 Mb.</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hot_2</span> <span class="op">&lt;-</span> <span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">filter</span><span class="op">(</span><span class="fu">abs</span><span class="op">(</span><span class="va">qtl_pos</span> <span class="op">-</span> <span class="fl">164</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.1</span><span class="op">)</span></span></code></pre>
</div>
<p>How many genes are there in the chromosome 2 hotspot now?</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">hot_2</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 88</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="principal-component-of-eqtl-hotspot-genes">Principal Component of eQTL Hotspot Genes<a class="anchor" aria-label="anchor" href="#principal-component-of-eqtl-hotspot-genes"></a>
</h3>
<p>One way to summarize the expression of genes in an eQTL hotspot is to
take the first principal component (PC1) of all of the genes in the
hotspot. PC1 should capture the largest amount of variance in the eQTL
hotspot and we can use this as a phenotype to map the hotspot.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_2</span> <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">hot_2</span><span class="op">$</span><span class="va">gene_id</span><span class="op">]</span></span>
<span><span class="va">pca_2</span>  <span class="op">&lt;-</span> <span class="fu">princomp</span><span class="op">(</span><span class="va">expr_2</span><span class="op">)</span></span>
<span><span class="va">pc1_2</span>  <span class="op">&lt;-</span> <span class="va">pca_2</span><span class="op">$</span><span class="va">scores</span><span class="op">[</span>,<span class="fl">1</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>We can then map PC1 as a phenotype and we should have a large QTL
peak in the same location as the hotspot.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pc1_lod</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                 pheno     <span class="op">=</span> <span class="va">pc1_2</span>,</span>
<span>                 kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">pc1_lod</span>, <span class="va">map</span>, main <span class="op">=</span> <span class="st">"PC1 of Chr 2 Hotspot"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-transcriptome-map-rendered-map_pc1_2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We can also look at the founder allele effects at the peak. We will
use a function in <code>gg_transcriptome_map.R</code> called
<code>plot_fit1()</code>.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">find_peaks</span><span class="op">(</span><span class="va">pc1_lod</span>, <span class="va">map</span>, threshold <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="va">pr</span>    <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                          map       <span class="op">=</span> <span class="va">map</span>, </span>
<span>                          chr       <span class="op">=</span> <span class="va">peaks</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, </span>
<span>                          pos       <span class="op">=</span> <span class="va">peaks</span><span class="op">$</span><span class="va">pos</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">mod</span>   <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>,</span>
<span>              pheno     <span class="op">=</span> <span class="va">pc1_2</span>, </span>
<span>              kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">peaks</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>, </span>
<span>              addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_fit1</span><span class="op">(</span><span class="va">mod</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Chr 2 Hotspot Allele Effects"</span>,</span>
<span>       x     <span class="op">=</span> <span class="st">"Founder"</span>,</span>
<span>       y     <span class="op">=</span> <span class="st">"Founder Allele Effects"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-transcriptome-map-rendered-pc1_2_eff-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From the plot above, we can see that NZO and WSB have allele effects
which are different from the other strains. The direction of the allele
effects is arbitrary in the principal components, so we can’t determine
the direction of the allele effects from this plot.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Transcriptome maps aid in understanding gene expression
regulation.</li>
<li>Local eQTL occur more frequently than distant eQTL.</li>
<li>Local eQTL appear along the diagonal in a transcriptome map and
distant eQTL appear on the off-diagonal.</li>
<li>Stacks of eQTL which appear over a single locus are called <em>eQTL
hotspots</em> and represent sets of genes which are transcriptionally
regulated by a single locus.</li>
<li>The first principal component of genes in and eQTL hotspot can be
used to summarize the genes in the hotspot.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div></section><section id="aio-interpret-results"><p>Content from <a href="interpret-results.html">Interpreting qtl2 results</a></p>
<hr>
<p>Last updated on 2024-12-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/interpret-results.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 30 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I interpret qtl2 results?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Interpret the relationship between sequence, expression and
phenotype variation from qtl2 mapping results.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p><img src="../fig/SFig7_Keller_2018_chr11.png" alt="Keller et al, figure S7" class="figure"> :::::::::::::::::::::::::::::::::::::
challenge</p>
<section><h2 class="section-heading" id="challenge-interpreting-qtl2-results">Challenge: Interpreting qtl2 results<a class="anchor" aria-label="anchor" href="#challenge-interpreting-qtl2-results"></a>
</h2>
<hr class="half-width">
<p>Refer to the figure above.</p>
<p>1). What does panel A show? What conclusions could you draw from
panel A?<br>
2). What does panel B show? What conclusions could you draw from panel
B?<br>
3). What does panel C show? What conclusions could you draw from panel
C?<br>
4). How are panels A through C related to one another? What story do
they tell together?</p>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<p>::::::::::::::::::::::::::::::::::::::::::::::::</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-mediation-analysis"><p>Content from <a href="mediation-analysis.html">Mediation Analysis</a></p>
<hr>
<p>Last updated on 2024-12-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/mediation-analysis.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 30 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is mediation analysis?</li>
<li>How do I explore causal relations with mediation analysis?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Describe mediation analysis as applied in genetics and
genomics.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pheno</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_pheno.rds'</span><span class="op">)</span></span>
<span><span class="va">pheno_dict</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_pheno_dict.rds'</span><span class="op">)</span></span>
<span><span class="va">covar</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_covar.rds'</span><span class="op">)</span></span>
<span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span>
<span><span class="va">addcovar</span>   <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span>, data <span class="op">=</span> <span class="va">covar</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">annot</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_expr_annot.rds'</span><span class="op">)</span></span>
<span><span class="va">expr_rz</span>    <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_expr_rz.rds'</span><span class="op">)</span></span>
<span><span class="va">map</span>        <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_map.rds'</span><span class="op">)</span></span>
<span><span class="va">probs</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="fu">url</span><span class="op">(</span><span class="st">'https://thejacksonlaboratory.box.com/shared/static/4hy4hbjyrxjbrzh570i4g02r62bx3lgk.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">K</span>          <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_kinship.rds'</span><span class="op">)</span></span>
<span><span class="va">ins_tauc</span>   <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">[</span>, <span class="st">'Ins_tAUC'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC_log</span> <span class="op">&lt;-</span> <span class="fu">log</span><span class="op">(</span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC</span><span class="op">)</span></span>
<span><span class="va">ins_lod</span>    <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/ins_tauc_lod.rds'</span><span class="op">)</span></span>
<span><span class="va">eperm</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/ENSMUSG00000020679_perm_1000.rds'</span><span class="op">)</span></span>
<span><span class="va">hotspots</span>   <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/eqtl_hotspots.rds'</span><span class="op">)</span></span>
<span><span class="va">pc1_2</span>      <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/eqtl_hotspot_chr2_pc1.rds'</span><span class="op">)</span></span>
<span><span class="va">pc1_lod</span>    <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/eqtl_hotspot_chr2_pc1_lod.rds'</span><span class="op">)</span></span></code></pre>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>GWAS studies show that most disease-associated variants are found in
non-coding regions. This fact leads to the idea that regulation of gene
expression is an important mechanism enabling genetic variants to affect
complex traits. Mediation analysis can identify a causal chain from
genetic variants to molecular and clinical phenotypes. The graphic below
shows complete mediation, in which a predictor variable does not
directly impact the response variable. Rather, it directly influences
the mediator (path a). The mediator has a direct impact on the response
variable (path b). We would see (path c) the relationship between
predictor and response, not knowing that a mediator intervenes in this
relationship.</p>
<figure><img src="../fig/mediation-analysis.png" alt="Mediation Analysis" class="figure mx-auto d-block"><div class="figcaption">In complete mediation an predictor variable
influences a response variable indirectly through a mediator
variable.</div>
</figure><p>Mediation analysis is widely used in the social sciences including
psychology. In biomedicine, mediation analysis has been employed to
investigate how gene expression mediates the effects of genetic variants
on complex phenotypes and disease.</p>
<p>For example, a genetic variant (non-coding SNP) indirectly regulates
expression of gene 2 through a mediator, gene 1. The SNP regulates
expression of gene 1 in cis, and expression of gene 1 influences
expression of gene 2 in trans.</p>
<figure><img src="../fig/mediation-trans-noncoding-SNP.png" alt="mediation figure" class="figure mx-auto d-block"><div class="figcaption">A non-coding SNP affects expression of gene 1 in
cis. Gene 1 mediates expression of gene 2.</div>
</figure><p>Instead of the expression of one gene impacting another, expression
of gene 1 in the graphic above could impact a physiological phenotype
like blood glucose. Expression of gene 1 would mediate the relationship
between the non-coding SNP and the glucose phenotype.</p>
<p>Gene Akr1e1 is located on chromosome 13 in mouse.</p>
<figure><img src="../fig/LOD-plot-Akr1e1.png" alt="Akr1e1 QTL plot" class="figure mx-auto d-block"><div class="figcaption">Chromosome 13 gene Akr1e1 is affected by
expression in both local and distant by genes on chromosomes 13 and
4.</div>
</figure><div id="challenge-1-interpreting-akr1e1-eqtl-plot." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-interpreting-akr1e1-eqtl-plot." class="callout-inner">
<h3 class="callout-title">Challenge 1: Interpreting <em>Akr1e1</em> eQTL
plot.</h3>
<div class="callout-content">
<p>How would you interpret the <em>Akr1e1</em> LOD plot above? On which
chromosome(s) would you expect to find the driver gene(s)? The
SNP(s)?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p><em>Ark1e1</em> expression is regulated by some variants on
chromosome 13 and other variants on chromosome 4. The largest driver
gene occurs on chromosome 4 because the LOD peak is higher there,
indicating that the genotype on chromosome 4 explains more of the
variance than the one on chromosome 13.</p>
</div>
</div>
</div>
</div>
<p>Myo15b is located on chromosome 11.</p>
<figure><img src="../fig/LOD-plot-Myo15b.png" alt="Myo15b QTL plot" class="figure mx-auto d-block"><div class="figcaption">Chromosome 11 gene Myo15b is affected by
expression of a gene in the chromosome 2 hotspot.</div>
</figure><div id="challenge-2-interpreting-myo15b-eqtl-plot." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-interpreting-myo15b-eqtl-plot." class="callout-inner">
<h3 class="callout-title">Challenge 2: Interpreting <em>Myo15b</em> eQTL
plot.</h3>
<div class="callout-content">
<p>How would you interpret the <em>Myo15b</em> LOD plot above? On which
chromosome(s) would you expect to find the driver gene(s)? The
SNP(s)?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p><em>Myo15b</em> is regulated by variants on chromosome 2 because the
largest LOD peak is located there.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="using-gene-expression-as-a-covariate">Using Gene Expression as a Covariate<a class="anchor" aria-label="anchor" href="#using-gene-expression-as-a-covariate"></a>
</h2>
<hr class="half-width">
<p>When we are searching for a gene whose expression influences a
physiological phenotype, we may not be looking for a missense SNP in the
the gene. Instead, we may be looking for a SNP in a regulatory region
which influence the expression of the gene, which in turn influences the
physiological phenotype. However, the exact sequence of regulatory
regions and the genes that they influence is not as well annotated as
protein-coding regions of genes. So, rather than search for the
regulatory SNP, we will search for genes which influence the trait and
we will not seek to identify the causal SNP.</p>
<p>Next, we are going to create our own mediation function. We will walk
through it step-by-step so that you understand why we are performing
each step.</p>
<p>Mediation analysis bring together a lot of information, to the
function will have a LOT of variables.</p>
<p>First, we need to pass in the chromosome, position, and LOD socre of
the QTL at which we are mapping. Remember, in a mediation scan, we map a
phenotype at <strong>ONE</strong> locus and use different genes as
covariates.</p>
<p>Next, since we are mapping, we need to pass in the genoprobs, the
original phenotype which produced the QTL peak, the kinship matrices,
the covariates matrix, and the markers map. In this case, our
“phenotype” will be PC1 from the genes in the chromosome 2 eQTL
hotspot.</p>
<p>The last two arguments will be the expression matrix and the gene
annotation.</p>
<p>With all of those pieces in place, we will build the function
below.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform mediation analysis on one locus using all of the available genes on</span></span>
<span><span class="co"># the corresponding chromosome.</span></span>
<span><span class="co"># qtl_chr: character string containing one of the chromosome names in genoprobs.</span></span>
<span><span class="co">#      Must be 1:19 or X.</span></span>
<span><span class="co"># qtl_pos: floating point number that is the Mb position of the QTL to map at.</span></span>
<span><span class="co"># qtl_lod: floating point number that is the LOD score at the QTL.</span></span>
<span><span class="co"># genoprobs: qtl2-style list containing genoprobs for all chromosomes.</span></span>
<span><span class="co"># pheno: numeric matrix containing one column with the phenotype values. Sample</span></span>
<span><span class="co">#        IDs must be in rownames.</span></span>
<span><span class="co"># K: qtl2-style list of LOCO kinship matrices.</span></span>
<span><span class="co"># addcovar: numeric matrix of additive covariate created with something like</span></span>
<span><span class="co">#           model.matrix(). Must be ready to pass into scan1() directly.</span></span>
<span><span class="co"># map: qtl2-style list containing marker map. Markers must match the markers in</span></span>
<span><span class="co">#      genoprobs.</span></span>
<span><span class="co"># expr: numeric matrix of normalized expression values. Sample in rows and genes</span></span>
<span><span class="co">#       in columns.</span></span>
<span><span class="co"># annot: data.frame containing the gene annotation. Genes in rows must be in the</span></span>
<span><span class="co">#        same order as the genes in columns in "expr".</span></span>
<span><span class="va">mediation</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">qtl_chr</span>, <span class="va">qtl_pos</span>, <span class="va">qtl_lod</span>, <span class="va">genoprobs</span>, <span class="va">pheno</span>, <span class="va">K</span>, <span class="va">addcovar</span>, </span>
<span>                      <span class="va">map</span>, <span class="va">expr</span>, <span class="va">annot</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>  <span class="co"># Get the genoprobs at the QTL peak.</span></span>
<span>  <span class="va">pr</span> <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                         map       <span class="op">=</span> <span class="va">map</span>, </span>
<span>                         chr       <span class="op">=</span> <span class="va">qtl_chr</span>, </span>
<span>                         pos       <span class="op">=</span> <span class="va">qtl_pos</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Get the genes on chromosome 11.</span></span>
<span>  <span class="va">genes_chr</span> <span class="op">&lt;-</span> <span class="va">annot</span> <span class="op">|&gt;</span></span>
<span>                 <span class="fu">filter</span><span class="op">(</span><span class="va">chr</span> <span class="op">==</span> <span class="va">qtl_chr</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                 <span class="fu">pull</span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span></span>
<span>  <span class="va">expr_chr</span>  <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>, <span class="va">genes_chr</span><span class="op">]</span></span>
<span></span>
<span>  <span class="co"># Subset the covariates to include only the samples in the expression data.</span></span>
<span>  <span class="va">addcovar_expr</span> <span class="op">&lt;-</span> <span class="va">addcovar</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">expr_chr</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span>  <span class="co"># Create the results.</span></span>
<span>  <span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>gene_id  <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">expr_chr</span><span class="op">)</span>,</span>
<span>                        base_lod <span class="op">=</span> <span class="va">qtl_lod</span>,</span>
<span>                        med_lod  <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">left_join</span><span class="op">(</span><span class="fu">select</span><span class="op">(</span><span class="va">annot</span>, <span class="va">gene_id</span>, <span class="va">symbol</span>, <span class="va">middle</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Loop through each gene, add it to the covariates, and map it at the QTL </span></span>
<span>  <span class="co"># marker.</span></span>
<span>  <span class="va">t_init</span> <span class="op">&lt;-</span> <span class="fu">proc.time</span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu">ncol</span><span class="op">(</span><span class="va">expr_chr</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    </span>
<span>    <span class="co"># Show progress.</span></span>
<span>    <span class="kw">if</span><span class="op">(</span><span class="va">i</span> <span class="op">%%</span> <span class="fl">50</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>      </span>
<span>      <span class="va">t2</span> <span class="op">&lt;-</span> <span class="fu">proc.time</span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span></span>
<span>      <span class="fu">print</span><span class="op">(</span><span class="fu">paste</span><span class="op">(</span><span class="va">i</span>, <span class="st">'of'</span>, <span class="fu">ncol</span><span class="op">(</span><span class="va">expr_chr</span><span class="op">)</span>, <span class="st">":"</span>, <span class="op">(</span><span class="va">t2</span> <span class="op">-</span> <span class="va">t_init</span><span class="op">)</span>, <span class="st">"seconds elapsed."</span><span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>    <span class="op">}</span> <span class="co"># for(i)</span></span>
<span>  </span>
<span>    <span class="co"># Create covariates with the current gene's expression.</span></span>
<span>    <span class="va">addcovar_tmp</span> <span class="op">=</span> <span class="fu">cbind</span><span class="op">(</span><span class="va">addcovar_expr</span>, <span class="va">expr_chr</span><span class="op">[</span>,<span class="va">i</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Fit the QTL model at the QTL marker with the new covariates.</span></span>
<span>    <span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>,</span>
<span>                pheno     <span class="op">=</span> <span class="va">pheno</span>,</span>
<span>                kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">qtl_chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                addcovar  <span class="op">=</span> <span class="va">addcovar_tmp</span><span class="op">)</span></span>
<span>  </span>
<span>    <span class="co"># Add the LOD to the results.</span></span>
<span>    <span class="va">results</span><span class="op">$</span><span class="va">med_lod</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">lod</span></span>
<span>    </span>
<span>  <span class="op">}</span> <span class="co"># for(i)</span></span>
<span></span>
<span>  <span class="co"># Subtract the meditation LODs from the baseline LOD.</span></span>
<span>  <span class="va">results</span> <span class="op">&lt;-</span> <span class="va">results</span> <span class="op">|&gt;</span></span>
<span>               <span class="fu">mutate</span><span class="op">(</span>lod_drop <span class="op">=</span> <span class="va">med_lod</span> <span class="op">-</span> <span class="va">base_lod</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Return the results.</span></span>
<span>  <span class="kw">return</span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span></span>
<span><span class="op">}</span> <span class="co"># mediation()</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl_pos</span> <span class="op">&lt;-</span> <span class="fu">mean</span><span class="op">(</span><span class="va">hotspots</span><span class="op">[[</span><span class="st">"2"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">med_2</span> <span class="op">&lt;-</span> <span class="fu">mediation</span><span class="op">(</span>qtl_chr  <span class="op">=</span> <span class="st">"2"</span>, </span>
<span>                   qtl_pos  <span class="op">=</span> <span class="va">qtl_pos</span>,</span>
<span>                   qtl_lod  <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">pc1_lod</span><span class="op">)</span>, </span>
<span>                   genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno    <span class="op">=</span> <span class="va">pc1_2</span>, </span>
<span>                   K        <span class="op">=</span> <span class="va">K</span>, </span>
<span>                   addcovar <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                   map      <span class="op">=</span> <span class="va">map</span>, </span>
<span>                   expr     <span class="op">=</span> <span class="va">expr_rz</span>, </span>
<span>                   annot    <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "50 of 1892 : 3.422 seconds elapsed."
[1] "100 of 1892 : 6.81 seconds elapsed."
[1] "150 of 1892 : 10.219 seconds elapsed."
[1] "200 of 1892 : 13.701 seconds elapsed."
[1] "250 of 1892 : 17.114 seconds elapsed."
[1] "300 of 1892 : 20.67 seconds elapsed."
[1] "350 of 1892 : 24.139 seconds elapsed."
[1] "400 of 1892 : 27.582 seconds elapsed."
[1] "450 of 1892 : 31.073 seconds elapsed."
[1] "500 of 1892 : 34.457 seconds elapsed."
[1] "550 of 1892 : 37.813 seconds elapsed."
[1] "600 of 1892 : 41.17 seconds elapsed."
[1] "650 of 1892 : 44.514 seconds elapsed."
[1] "700 of 1892 : 47.86 seconds elapsed."
[1] "750 of 1892 : 51.213 seconds elapsed."
[1] "800 of 1892 : 54.57 seconds elapsed."
[1] "850 of 1892 : 57.926 seconds elapsed."
[1] "900 of 1892 : 61.274 seconds elapsed."
[1] "950 of 1892 : 64.624 seconds elapsed."
[1] "1000 of 1892 : 68.006 seconds elapsed."
[1] "1050 of 1892 : 71.36 seconds elapsed."
[1] "1100 of 1892 : 74.721 seconds elapsed."
[1] "1150 of 1892 : 78.076 seconds elapsed."
[1] "1200 of 1892 : 81.435 seconds elapsed."
[1] "1250 of 1892 : 84.798 seconds elapsed."
[1] "1300 of 1892 : 88.157 seconds elapsed."
[1] "1350 of 1892 : 91.539 seconds elapsed."
[1] "1400 of 1892 : 94.916 seconds elapsed."
[1] "1450 of 1892 : 98.272 seconds elapsed."
[1] "1500 of 1892 : 101.625 seconds elapsed."
[1] "1550 of 1892 : 104.992 seconds elapsed."
[1] "1600 of 1892 : 108.358 seconds elapsed."
[1] "1650 of 1892 : 111.725 seconds elapsed."
[1] "1700 of 1892 : 115.086 seconds elapsed."
[1] "1750 of 1892 : 118.44 seconds elapsed."
[1] "1800 of 1892 : 121.931 seconds elapsed."
[1] "1850 of 1892 : 125.305 seconds elapsed."</code></pre>
</div>
<p>Now that we have the change in LOD scores associated with using each
gene as a covariate, we can plot the LOD change along chromosome 11.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">end</span>  <span class="op">&lt;-</span> <span class="fu">ceiling</span><span class="op">(</span><span class="fu">max</span><span class="op">(</span><span class="va">med_2</span><span class="op">$</span><span class="va">middle</span> <span class="op">/</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">med_2</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">middle</span>, <span class="va">lod_drop</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="va">end</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mediation Analysis for Chr 2 PC1"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"Position (Mb)"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"LOD Drop"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>size <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in plot_theme(plot): The `size` theme element is not defined in the
element hierarchy.</code></pre>
</div>
<figure><img src="../fig/mediation-analysis-rendered-plot_med-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, the position of each gene along chromosome 2 is
shown on the X-axis and the LOD drop is shown on the Y-axes. Most genes
don’t change the LOD score by very much. It seems that, by chance, LOD
scores can vary by plus or minus 3. In this case, we are searching for
genes which decrease the LOD score by the largest amount. In this case,
there is one gene which reduces the LOD .</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">med_2</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">lod_drop</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">20</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             gene_id base_lod med_lod symbol    middle  lod_drop
1 ENSMUSG00000017950 97.61195 60.5270  Hnf4a 163.53986 -37.08495
2 ENSMUSG00000035000 97.61195 73.3269   Dpp4  62.37115 -24.28505</code></pre>
</div>
<div id="challenge-3-keller-et-al-chromosome-2-hotspot-candidate-gene" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-keller-et-al-chromosome-2-hotspot-candidate-gene" class="callout-inner">
<h3 class="callout-title">Challenge 3: Keller et al chromosome 2 hotspot candidate gene?</h3>
<div class="callout-content">
<p>Look at Figure 4C in <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller et
al</a> and see which gene the selected in their mediaiton analysis.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>The authors found that <em>Hnf4a</em> reduces the LOD more than any
other gene.</p>
</div>
</div>
</div>
</div>
<div id="challenge-4-look-up-dpp4-in-pubmed" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-look-up-dpp4-in-pubmed" class="callout-inner">
<h3 class="callout-title">Challenge 4: Look up <em>Dpp4</em> in <a href="https://pubmed.ncbi.nlm.nih.gov/" class="external-link">Pubmed</a>
</h3>
<div class="callout-content">
<p>or <a href="https://www.alliancegenome.org/" class="external-link">Alliance Genome</a> and
see if it has any known association with type 2 diabetes or insulin
metabolism.</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p><a href="https://www.alliancegenome.org/gene/MGI:94919" class="external-link">Alliance
Genome</a> says that “Human ortholog(s) of this gene implicated in …
type 2 diabetes mellitus.”</p>
</div>
</div>
</div>
</div>
<p>Let’s look more closely at <em>Hnf4a</em>. First, let’s map the
expresson of <em>Hnf4a</em>.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hnf4a</span>     <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Hnf4a"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="va">hnf4a_lod</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                  pheno     <span class="op">=</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">hnf4a</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                  kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>                  addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">hnf4a_lod</span>, <span class="va">map</span>, main <span class="op">=</span> <span class="st">"Hnf4a Genome Scan"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/mediation-analysis-rendered-map_hnf4a-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, we can see that <em>Hnf4a</em> has a QTL peak on
chromosome 2.</p>
<div id="challenge-5-find-the-hnf4a-qtl-peak-location." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5-find-the-hnf4a-qtl-peak-location." class="callout-inner">
<h3 class="callout-title">Challenge 5: Find the <em>Hnf4a</em> QTL peak
location.</h3>
<div class="callout-content">
<p>Use find_peaks() to find the position of the <em>Hnf4a</em> QTL peak.
What kind of eQTL would you call this peak? Local or distant? How does
this peak location compare with the location of the chromosome 2 eQTL
hotspot?</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span><span class="va">hnf4a_lod</span>, map <span class="op">=</span> <span class="va">map</span>, threshold <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr      pos      lod
1        1 ENSMUSG00000017950   2 164.0224 49.60808</code></pre>
</div>
<p><em>Hnf4a</em> has a QTL peak on chromosome 2 at 164 Mb.</p>
<p>This is the same position as the chromosome 2 eQTL hotspot.</p>
<p>Since <em>Hnf4a</em> is located at 163.4 Mb and its largest eQTL is
located at 164 Mb, we could call this a local eQTL.</p>
</div>
</div>
</div>
</div>
<p>These are the primary steps in mediation analysis. As you can see,
sometimes, meditation analysis can point to a candidate gene that is
plausible. It is always important to remember that there may be more
than one causal gene and that causal genes may contain missense or
splice SNPs which affect gene expression levels.</p>
<div class="section level3">
<h3 id="interactive-qtl-viewer">Interactive QTL Viewer<a class="anchor" aria-label="anchor" href="#interactive-qtl-viewer"></a>
</h3>
<p>The <a href="https://churchilllab.jax.org/qtlviewer/attie/islets#" class="external-link">QTL
Viewer</a> for the Attie islet data integrates mediation into
exploration of the data. Below, mediation analysis identifies gene Hnf4a
as the chromosome 2 gene that impacts Myo15b expression.</p>
<figure><img src="../fig/mediation-Hnf4a-Myo15b.png" alt="Hnf4a mediation" class="figure mx-auto d-block"><div class="figcaption">Mediating expression of Myo15b identifies Hnf4a
as the gene that drops the LOD score from greater than 70 to less than
50.</div>
</figure><div id="challenge-6-perform-mediation-at-another-eqtl-hotspot." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-perform-mediation-at-another-eqtl-hotspot." class="callout-inner">
<h3 class="callout-title">Challenge 6: Perform mediation at another eQTL hotspot.</h3>
<div class="callout-content">
<p>Select another eQTL hotspot and perform mediation analysis.</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" data-bs-parent="#accordionSolution6" aria-labelledby="headingSolution6">
<div class="accordion-body">
<p>Set “hot_chr” to one of the other eQTL hotspot locations: 5, 7, or
11</p>
<p>First, we look at the positions of the eQTL in this region.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hot_chr</span> <span class="op">&lt;-</span> <span class="st">"11"</span></span>
<span><span class="va">hot</span>     <span class="op">&lt;-</span> <span class="va">hotspots</span><span class="op">[[</span><span class="va">hot_chr</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="fu">table</span><span class="op">(</span><span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span>, las <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/mediation-analysis-rendered-challenge6a-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Next, we subset the genes to select ones which have eQTL at the same
location. Some of the eQTL hotspots have large numbers of of genes at
more than one marker. Feel free to explore different markers. Below, we
sort the markers by the number of genes with eQTL at each marker so that
we can select ones to pursue.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sort</span><span class="op">(</span><span class="fu">table</span><span class="op">(</span><span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
68.122257  68.42035 68.640199 68.660126 68.825289 68.888033 68.889671  68.89049
        1         1         1         1         1         1         1         1
 69.21825 69.224642 69.231034 69.244859 69.505599  69.66328 69.768364 69.869428
        1         1         1         1         1         1         1         1
70.196497  70.60917 70.616235 70.782062 71.198629 71.422464 71.425934 71.432875
        1         1         1         1         1         1         1         1
71.510771 71.793079 71.952385 68.656091 69.198036 69.733033 69.844926 69.876397
        1         1         1         2         2         2         2         2
69.924127 70.772323 71.449172 71.675712  71.83438 69.131341  69.23423 69.592546
        2         2         2         2         2         3         3         3
69.876917 69.978753 71.276599 71.418821 71.429405 71.937774 68.892129 69.416362
        3         3         3         3         3         3         4         4
71.658293 71.803404 71.859513 68.987403 69.542556 69.796313 69.064645 69.227838
        4         4         4         5         5         5         6         6
70.661194 70.750402 70.785072  71.12066 69.484146 69.837051 71.782692 69.810287
        6         6         6         6         8         9        10        14
71.813729 69.829176 71.693132 71.354568
       14        22        26        31 </code></pre>
</div>
<p>Set the thresholds below to match the markers which you would like to
include.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hot</span> <span class="op">&lt;-</span> <span class="va">hot</span><span class="op">[</span><span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span> <span class="op">&gt;=</span> <span class="fl">71.3</span> <span class="op">&amp;</span> <span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span> <span class="op">&lt;=</span> <span class="fl">71.7</span>,<span class="op">]</span></span>
<span><span class="fu">nrow</span><span class="op">(</span><span class="va">hot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 75</code></pre>
</div>
<p>Next, we get the expression of the genes at the marker(s) we are
pursuing and calculate the first pricipal component.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_hot</span> <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">hot</span><span class="op">$</span><span class="va">gene_id</span><span class="op">]</span></span>
<span><span class="va">hot_pca</span>  <span class="op">&lt;-</span> <span class="fu">princomp</span><span class="op">(</span><span class="va">expr_hot</span><span class="op">)</span></span>
<span><span class="va">pc1_hot</span>  <span class="op">&lt;-</span> <span class="va">hot_pca</span><span class="op">$</span><span class="va">scores</span><span class="op">[</span>,<span class="fl">1</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>Plot the genome scan of PC1.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pc1_lod</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                 pheno     <span class="op">=</span> <span class="va">pc1_hot</span>,</span>
<span>                 kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">pc1_lod</span>, <span class="va">map</span>, main <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chr"</span>, <span class="va">hot_chr</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/mediation-analysis-rendered-challenge6d-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>At this point, we have everything that we need to perform mediation
analysis.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl_pos</span> <span class="op">&lt;-</span> <span class="fu">median</span><span class="op">(</span><span class="va">hot</span><span class="op">$</span><span class="va">qtl_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">med_hot</span> <span class="op">&lt;-</span> <span class="fu">mediation</span><span class="op">(</span>qtl_chr  <span class="op">=</span> <span class="va">hot_chr</span>, </span>
<span>                     qtl_pos  <span class="op">=</span> <span class="va">qtl_pos</span>,</span>
<span>                     qtl_lod  <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">pc1_lod</span><span class="op">)</span>, </span>
<span>                     genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                     pheno    <span class="op">=</span> <span class="va">pc1_hot</span>, </span>
<span>                     K        <span class="op">=</span> <span class="va">K</span>, </span>
<span>                     addcovar <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                     map      <span class="op">=</span> <span class="va">map</span>, </span>
<span>                     expr     <span class="op">=</span> <span class="va">expr_rz</span>, </span>
<span>                     annot    <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in data.frame(gene_id = colnames(expr_chr), base_lod = qtl_lod, : row
names were found from a short variable and have been discarded</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Joining with `by = join_by(gene_id)`</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "50 of 1810 : 3.31800000000001 seconds elapsed."
[1] "100 of 1810 : 6.67500000000001 seconds elapsed."
[1] "150 of 1810 : 10.106 seconds elapsed."
[1] "200 of 1810 : 13.567 seconds elapsed."
[1] "250 of 1810 : 16.93 seconds elapsed."
[1] "300 of 1810 : 20.323 seconds elapsed."
[1] "350 of 1810 : 23.697 seconds elapsed."
[1] "400 of 1810 : 27.095 seconds elapsed."
[1] "450 of 1810 : 30.462 seconds elapsed."
[1] "500 of 1810 : 33.807 seconds elapsed."
[1] "550 of 1810 : 37.18 seconds elapsed."
[1] "600 of 1810 : 40.541 seconds elapsed."
[1] "650 of 1810 : 43.909 seconds elapsed."
[1] "700 of 1810 : 47.282 seconds elapsed."
[1] "750 of 1810 : 50.679 seconds elapsed."
[1] "800 of 1810 : 54.056 seconds elapsed."
[1] "850 of 1810 : 57.407 seconds elapsed."
[1] "900 of 1810 : 60.909 seconds elapsed."
[1] "950 of 1810 : 64.392 seconds elapsed."
[1] "1000 of 1810 : 67.74 seconds elapsed."
[1] "1050 of 1810 : 71.105 seconds elapsed."
[1] "1100 of 1810 : 74.46 seconds elapsed."
[1] "1150 of 1810 : 77.883 seconds elapsed."
[1] "1200 of 1810 : 81.238 seconds elapsed."
[1] "1250 of 1810 : 84.612 seconds elapsed."
[1] "1300 of 1810 : 87.966 seconds elapsed."
[1] "1350 of 1810 : 91.355 seconds elapsed."
[1] "1400 of 1810 : 94.703 seconds elapsed."
[1] "1450 of 1810 : 98.084 seconds elapsed."
[1] "1500 of 1810 : 101.488 seconds elapsed."
[1] "1550 of 1810 : 104.884 seconds elapsed."
[1] "1600 of 1810 : 108.255 seconds elapsed."
[1] "1650 of 1810 : 111.609 seconds elapsed."
[1] "1700 of 1810 : 115.024 seconds elapsed."
[1] "1750 of 1810 : 118.414 seconds elapsed."
[1] "1800 of 1810 : 121.766 seconds elapsed."</code></pre>
</div>
<p>Now that we have completed the mediation analysis, we plot the
results.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">end</span>  <span class="op">&lt;-</span> <span class="fu">ceiling</span><span class="op">(</span><span class="fu">max</span><span class="op">(</span><span class="va">med_hot</span><span class="op">$</span><span class="va">middle</span> <span class="op">/</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">med_hot</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">middle</span>, <span class="va">lod_drop</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="va">end</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">"Mediation Analysis for Chr "</span>, <span class="va">hot_chr</span>, <span class="st">" PC1"</span><span class="op">)</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"Position (Mb)"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"LOD Drop"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>size <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in plot_theme(plot): The `size` theme element is not defined in the
element hierarchy.</code></pre>
</div>
<figure><img src="../fig/mediation-analysis-rendered-challeng6f-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Let’s list the genes with LOD drops less than -10.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">filter</span><span class="op">(</span><span class="va">med_hot</span>, <span class="va">lod_drop</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             gene_id base_lod  med_lod  symbol   middle  lod_drop
1 ENSMUSG00000020846 18.59729 6.641443 Fam101b 76.02349 -11.95584
2 ENSMUSG00000069835 18.59729 5.301991    Sat2 69.62295 -13.29530
3 ENSMUSG00000072640 18.59729 6.594574   Lyrm9 78.83673 -12.00271</code></pre>
</div>
<p>here are three genes with large LOD drops. In figure S7 of <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5937189/" class="external-link">Keller et
al</a>, they select <em>Sat2</em> as a candidate gene.</p>
<p>But notice that there are also gene with large LOD increases. Let’s
list out the genes with LOD increases over 12.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">filter</span><span class="op">(</span><span class="va">med_hot</span>, <span class="va">lod_drop</span> <span class="op">&gt;</span> <span class="fl">12</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             gene_id base_lod  med_lod  symbol   middle lod_drop
1 ENSMUSG00000020829 18.59729 34.72247 Slc46a1 78.46888 16.12518
2 ENSMUSG00000040746 18.59729 35.13738  Rnf167 70.64933 16.54009
3 ENSMUSG00000045287 18.59729 31.31408 Rtn4rl1 75.23078 12.71679</code></pre>
</div>
<p>There are three genes with large LOD increase. If a LOD drop means
that the gene is absorbing variance and may be a causal mediator, what
does a LOD increase mean?</p>
<p>Let’s start by looking at the allele effects of the hotspot PC1 at
the QTL.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pr</span>      <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, map <span class="op">=</span> <span class="va">map</span>, chr <span class="op">=</span> <span class="va">hot_chr</span>, pos <span class="op">=</span> <span class="va">qtl_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pc1_eff</span> <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>,</span>
<span>                pheno     <span class="op">=</span> <span class="va">pc1_hot</span>,</span>
<span>                kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">hot_chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, <span class="va">pc1_eff</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>, xaxt <span class="op">=</span> <span class="st">"n"</span>, main <span class="op">=</span> <span class="st">"PC1"</span>, xlab <span class="op">=</span> <span class="st">"Founder"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"Allele Effects"</span><span class="op">)</span></span>
<span><span class="fu">axis</span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, at <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, labels <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">pc1_eff</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/mediation-analysis-rendered-unnamed-chunk-2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From the plot above, we can see that CAST (F) and PWK (G) have lower
effects than the other founders. Remember that the sign of the PC is
arbitrary, so the effects may be positive.</p>
<p>Next, let’s look at the allele effects of the gene with the highest
LOD increase, <em>Rnf167</em>.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_id</span>  <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Rnf167"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span></span>
<span></span>
<span><span class="va">gene_lod</span> <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">pr</span>,</span>
<span>                  pheno     <span class="op">=</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">gene_id</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                  kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">hot_chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                  addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, <span class="va">gene_lod</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,  xaxt <span class="op">=</span> <span class="st">"n"</span>, main <span class="op">=</span> <span class="st">"Rnf167"</span>, xlab <span class="op">=</span> <span class="st">"Founder"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"Allele Effects"</span><span class="op">)</span></span>
<span><span class="fu">axis</span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, at <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, labels <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">pc1_eff</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/mediation-analysis-rendered-unnamed-chunk-3-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p><em>Rnf167</em> has allele effects which seem quite correlated with
PC1. Both CAST (F) and PWK (G) have different allele effects from the
other strains.</p>
<p>Next, let’s look at <em>Sat2</em>, which is the gene with the largest
LOD <strong>drop</strong>.</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_id</span>  <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Sat2"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="va">pr</span>       <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, map <span class="op">=</span> <span class="va">map</span>, chr <span class="op">=</span> <span class="va">hot_chr</span>, pos <span class="op">=</span> <span class="va">qtl_pos</span><span class="op">)</span></span>
<span><span class="va">gene_lod</span> <span class="op">&lt;-</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">pr</span>,</span>
<span>                  pheno     <span class="op">=</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="va">gene_id</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                  kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">hot_chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                  addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">gene_lod</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,xaxt <span class="op">=</span> <span class="st">"n"</span>, main <span class="op">=</span> <span class="st">"Sat2"</span>, xlab <span class="op">=</span> <span class="st">"Founder"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"Allele Effects"</span><span class="op">)</span></span>
<span><span class="fu">axis</span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, at <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, labels <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">pc1_eff</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/mediation-analysis-rendered-unnamed-chunk-4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The allele effects for <em>Sat2</em> appear to be less correlated,
with PWK (G) having lower effects than the other strains.</p>
<p>Let’s look at the correlation of <em>Sat2</em>, <em>Rnf167</em>, and
the hotspot PC1.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_tmp</span> <span class="op">&lt;-</span> <span class="va">expr_rz</span><span class="op">[</span>,<span class="fu">c</span><span class="op">(</span><span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Sat2"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span>, <span class="va">annot</span><span class="op">[</span><span class="va">annot</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Rnf167"</span>,<span class="op">]</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">expr_tmp</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sat2"</span>, <span class="st">"Rnf167"</span><span class="op">)</span></span>
<span><span class="va">expr_tmp</span> <span class="op">&lt;-</span> <span class="fu">cbind</span><span class="op">(</span><span class="va">expr_tmp</span>, <span class="va">pc1_hot</span><span class="op">)</span></span>
<span><span class="fu">cor</span><span class="op">(</span><span class="va">expr_tmp</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             Sat2     Rnf167    Comp.1
Sat2    1.0000000 -0.1007433 0.4116323
Rnf167 -0.1007433  1.0000000 0.4086829
Comp.1  0.4116323  0.4086829 1.0000000</code></pre>
</div>
<p>Both genes have almost identical correlation (~0.41) with PC1.</p>
</div>
</div>
</div>
</div>
<p>Mediation analysis is one way of identifying candidate genes under
QTL peaks. In this episode, we have focused on eQTL hotspots, but you
could do this with physiological phenotypes as well.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Mediation analysis investigates an intermediary between an
independent variable and its effect on a dependent variable.</li>
<li>Mediation analysis is used in high-throughput genomics studies to
identify molecular phenotypes, such as gene expression or methylation
traits, that mediate the effect of genetic variation on disease
phenotypes or other outcomes of interest.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.10" class="external-link">sandpaper (0.16.10)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.7" class="external-link">pegboard (0.7.7)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene expression, quantitative trait, genetic mapping, Diversity Outbred population",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/instructor/aio.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/instructor/aio.html",
  "dateCreated": "2024-11-07",
  "dateModified": "2024-12-09",
  "datePublished": "2024-12-09"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

