<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Expression Quantitative Trait Locus (eQTL) Mapping: Load and Explore Data</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png"><link rel="manifest" href="../favicons/incubator/site.webmanifest"><link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../load-explore-data.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Expression Quantitative Trait Locus (eQTL) Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Expression Quantitative Trait Locus (eQTL) Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr></ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Expression Quantitative Trait Locus (eQTL) Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 9%" class="percentage">
    9%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 9%" aria-valuenow="9" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../load-explore-data.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="review-eqtl-study.html">2. Genetic Drivers of Pancreatic Islet Function</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        3. Load and Explore Data
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#physiological-phenotypes">Physiological Phenotypes</a></li>
<li><a href="#physiological-phenotypes-1">Physiological Phenotypes</a></li>
<li><a href="#gene-expression-data">Gene Expression Data</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="map-one-eqtl.html">4. Mapping A Single Gene Expression Trait</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="map-many-eqtls.html">5. Mapping Many Gene Expression Traits</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="maximum-peaks-genes.html">6. Maximum eQTL Peaks and Nearby Genes</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="create-transcriptome-map.html">7. Creating A Transcriptome Map</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="mediation-analysis.html">8. Mediation Analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr></ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/review-eqtl-study.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/map-one-eqtl.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/review-eqtl-study.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Genetic Drivers of
        </a>
        <a class="chapter-link float-end" href="../instructor/map-one-eqtl.html" rel="next">
          Next: Mapping A Single...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Load and Explore Data</h1>
        <p>Last updated on 2024-12-11 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/load-explore-data.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 45 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What data are required for eqtl mapping?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>To provide an example and exploration of data used for eqtl
mapping.</li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">ggbeeswarm</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">DESeq2</span><span class="op">)</span></span></code></pre>
</div>
<section><h2 class="section-heading" id="physiological-phenotypes">Physiological Phenotypes<a class="anchor" aria-label="anchor" href="#physiological-phenotypes"></a></h2>
<hr class="half-width"><p>You should have downloaded data files already when following the <a href="https://smcclatchy.github.io/eqtl-mapping/index.html" class="external-link">setup
instructions</a>. The complete data used in these analyses are available
from <a href="https://doi.org/10.5061/dryad.pj105" class="external-link">Data Dryad</a>. The
files we will use are in a simpler format than those on Data Dryad.</p>
<p>Load in the physiological phenotypes.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the data</span></span>
<span><span class="va">pheno</span>       <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_pheno.rds'</span><span class="op">)</span></span>
<span><span class="va">pheno_dict</span>  <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_pheno_dict.rds'</span><span class="op">)</span></span>
<span><span class="va">covar</span>       <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_covar.rds'</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="physiological-phenotypes-1">Physiological Phenotypes<a class="anchor" aria-label="anchor" href="#physiological-phenotypes-1"></a></h2>
<hr class="half-width"><p>In this data set, we have 20 phenotypes for 500 Diversity Outbred
mice. <code>pheno</code> is a data frame containing the phenotype data
as well as covariates. Click on the triangle to the left of
<code>pheno</code> in the Environment pane to view its contents. Run
<code>names(pheno)</code> to list the variables.</p>
<p><code>pheno_dict</code> is the phenotype dictionary. This data frame
contains information on each variable in <code>pheno</code>, including
<code>name</code>, <code>short name</code>,
<code>pheno_type</code>,<br><code>formula</code> (if used) and <code>description</code>. You can
view a table of the data dictionary.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_dict</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">description</span>, <span class="va">formula</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">kable</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<table class="table"><colgroup><col width="6%"><col width="82%"><col width="11%"></colgroup><thead><tr class="header"><th align="left"></th>
<th align="left">description</th>
<th align="left">formula</th>
</tr></thead><tbody><tr class="odd"><td align="left">mouse</td>
<td align="left">Animal identifier.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">sex</td>
<td align="left">Male (M) or female (F).</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">sac_date</td>
<td align="left">Date when mouse was sacrificed; used to compute days on
diet, using birth dates.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">partial_inflation</td>
<td align="left">Some mice showed a partial pancreatic inflation which
would negatively effect the total number of islets collected from these
mice.</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">coat_color</td>
<td align="left">Visual inspection by Kathy Schuler on coat color.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">oGTT_date</td>
<td align="left">Date the oGTT was performed.</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">FAD_NAD_paired</td>
<td align="left">A change in the method that was used to make this
measurement by Matt Merrins’ lab. Paired was the same islet for the
value at 3.3mM vs. 8.3mM glucose; unpaired was where averages were used
for each glucose concentration and used to compute ratio.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">FAD_NAD_filter_set</td>
<td align="left">A different filter set was used on the microscope to
make the fluorescent measurement; may have influenced the values.</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">crumblers</td>
<td align="left">Some mice store food in their bedding (hoarders) which
would be incorrectly interpreted as consumed.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">birthdate</td>
<td align="left">Birth date</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">diet_days</td>
<td align="left">Number of days.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">num_islets</td>
<td align="left">Total number of islets harvested per mouse; negatively
impacted by those with partial inflation.</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">Ins_per_islet</td>
<td align="left">Amount of insulin per islet in units of
ng/ml/islet.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">WPIC</td>
<td align="left">Derived number; equal to total number of islets times
insulin content per islet.</td>
<td align="left">Ins_per_islet * num_islets</td>
</tr><tr class="odd"><td align="left">HOMA_IR_0min</td>
<td align="left">glucose*insulin/405 at time t=0 for the oGTT</td>
<td align="left">Glu_0min * Ins_0min / 405</td>
</tr><tr class="even"><td align="left">HOMA_B_0min</td>
<td align="left">360 * Insulin / (Glucose - 63) at time t=0 for the
oGTT</td>
<td align="left">360 * Ins_0min / (Glu_0min - 63)</td>
</tr><tr class="odd"><td align="left">Glu_tAUC</td>
<td align="left">Area under the curve (AUC) calculation without any
correction for baseline differences.</td>
<td align="left">complicated</td>
</tr><tr class="even"><td align="left">Ins_tAUC</td>
<td align="left">Area under the curve (AUC) calculation without any
correction for baseline differences.</td>
<td align="left">complicated</td>
</tr><tr class="odd"><td align="left">Glu_6wk</td>
<td align="left">Plasma glucose with units of mg/dl; fasting.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">Ins_6wk</td>
<td align="left">Plasma insulin with units of ng/ml; fasting.</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">TG_6wk</td>
<td align="left">Plasma triglyceride (TG) with units of mg/dl;
fasting.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">Glu_10wk</td>
<td align="left">Plasma glucose with units of mg/dl; fasting.</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">Ins_10wk</td>
<td align="left">Plasma insulin with units of ng/ml; fasting.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">TG_10wk</td>
<td align="left">Plasma triglyceride (TG) with units of mg/dl;
fasting.</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">Glu_14wk</td>
<td align="left">Plasma glucose with units of mg/dl; fasting.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">Ins_14wk</td>
<td align="left">Plasma insulin with units of ng/ml; fasting.</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">TG_14wk</td>
<td align="left">Plasma triglyceride (TG) with units of mg/dl;
fasting.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">food_ave</td>
<td align="left">Average food consumption over the measurements made for
each mouse.</td>
<td align="left">complicated</td>
</tr><tr class="odd"><td align="left">weight_2wk</td>
<td align="left">Body weight at indicated date; units are gm.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">weight_6wk</td>
<td align="left">Body weight at indicated date; units are gm.</td>
<td align="left">NA</td>
</tr><tr class="odd"><td align="left">weight_10wk</td>
<td align="left">Body weight at indicated date; units are gm.</td>
<td align="left">NA</td>
</tr><tr class="even"><td align="left">DOwave</td>
<td align="left">Wave (i.e., batch) of DO mice</td>
<td align="left">NA</td>
</tr></tbody></table><p>Since the paper is interested in type 2 diabetes and insulin
secretion, we will choose insulin AUC (area under the curve which was
calculated without any correction for baseline differences) for this
review.</p>
<div class="section level3">
<h3 id="phenotype-distributions">Phenotype Distributions<a class="anchor" aria-label="anchor" href="#phenotype-distributions"></a></h3>
<p>Many statistical models, including the QTL mapping model in
<code>qtl2</code>, expect that the incoming data will be normally
distributed. You may use transformations such as log or square root to
make your data more normally distributed. Here, we will log transform
the data.</p>
<p>Let’s make a variable for insulin AUC so that we don’t have to type
as much.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ins_tauc</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">[</span>, <span class="st">'Ins_tAUC'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>Next, let’s look at the distribution of insulin AUC using a
histogram.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">ins_tauc</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, </span>
<span>     breaks <span class="op">=</span> <span class="fl">20</span>,</span>
<span>     main   <span class="op">=</span> <span class="st">"Insulin Area Under the Curve"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-hist_untransformed-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>This is clearly <strong>not</strong> normally distributed. In fact,
this type of distribution is often log normal.</p>
<p>Now, let’s apply the <code>log()</code> function to this data in an
effort to make the distribution more normal. By default the
<code>log()</code> function calculates the natural log (base
<em>e</em>).</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC_log</span> <span class="op">&lt;-</span> <span class="fu">log</span><span class="op">(</span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s make a histogram of the log-transformed data.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">ins_tauc</span><span class="op">$</span><span class="va">Ins_tAUC_log</span>, </span>
<span>     breaks <span class="op">=</span> <span class="fl">20</span>,</span>
<span>     main   <span class="op">=</span> <span class="st">"insulin AUC (log-transformed)"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-hist_log_transform-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>This looks much better! The data has a somewhat Gaussian shape.
Technically, the assumptions of a linear model require that the
<strong>residuals</strong> be normally distributed. In practice,
transforming the input data to be normally distributed helps to make the
residuals normally distributed. As a reminder, a residual is the
vertical distance from a data point to the line described by a linear
model.</p>
<figure><img src=".././fig/residual.png" alt="a residual for one data point located above the line described by a linear model" class="figure mx-auto d-block"><div class="figcaption">A linear model showing one residual</div>
</figure><p>Boxplots are another great way to view the distribution of the data
and to identify any outliers. We will log-transform insulin AUC using
the <a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link"><code>scale_y_log10()</code></a>
function. This transforms the data using base 10 and creates a base 10
y-axis for plotting. We have overlaid the data points using
<code>ggbeeswarm</code>’s <a href="https://www.rdocumentation.org/packages/ggbeeswarm/versions/0.7.2/topics/geom_beeswarm" class="external-link"><code>geom_beeswarm</code></a>.
We have told <code>geom_beeswarm()</code> to plot the points with some
transparency using the argument <code>alpha = 0.2</code>. The
<code>alpha</code> argument ranges between 0 (completely transparent) to
1 (completely opaque). A value of 0.1 means mostly transparent.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot Insulin on a log 10 scale</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">pheno</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">sex</span>, <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_beeswarm</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Insulin area under the curve"</span>, y <span class="op">=</span> <span class="st">"insulin AUC"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-Ins_tAUC_boxplot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>How many orders of magnitude (powers of 10) does insulin AUC
span?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>insulin AUC spans three orders of magnitude, from near 10 to over
1000.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p>Which sex has higher median insulin AUC values?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Males have higher insulin AUC than females.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3</h3>
<div class="callout-content">
<p>What does the heavy line bisecting the boxes indicate?<br>
What do the lines at the top and bottom of the boxes indicate?<br>
What does the <em>whisker</em> extending from the top and bottom of the
boxes indicate?<br>
What do the black dots extending from the whiskers indicate?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>The heavy line bisecting the boxes shows the median value (not the
mean!). Half of the data points are above and half are below this
line.<br>
The lines at the top and bottom of the boxes indicate the first and 3rd
quartiles of the data (the <em>hinges</em>). One-fourth of the data
points are beneath the box and another one-fourth are above the box. The
box itself contains 50% of the data points.<br>
The whiskers represent some multiple of the interquartile range (IQR),
which is the height of the box between the first and third quartiles.
<code>geom_boxplot()</code> produces <em>Tukey-style boxplots</em> in
which the whiskers are 1.5 <span class="math inline">\(\times\)</span>
the IQR. Any data points that lie beyond the whiskers are considered
outliers and are shown as heavy black dots.</p>
</div>
</div>
</div>
</div>
<p>The boxplot is a useful plot to visualize the distribution of your
data.</p>
</div>
<div class="section level3">
<h3 id="quality-control-of-data">Quality Control of Data<a class="anchor" aria-label="anchor" href="#quality-control-of-data"></a></h3>
<p>Many statistical tests rely upon the data having a normal (or
Gaussian) distribution. Many biological phenotypes do not follow this
distribution and must be transformed before analysis. This is why we
log-transformed the data in the plots above.</p>
<p>While we can “eyeball” the distributions in the boxplot, it would be
better to use a <em>quantile-quantile</em> plot.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Quantile-Quantile Plot of Ins_tAUC"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"Normal Quantiles"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"Ins_tAUC"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-qqplot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In these plots, the <em>quantiles</em> of the normal distribution are
plotted on the X-axis and the data are plotted on the Y-axis. A quantile
evenly divides data observations into a specific number of groups. The
boxplot above evenly divides the observations into quartiles, a quantile
containing four groups that each contain one-fourth of the data. A
percentile similarly divides data into 100 quantiles, with each
percentile containing 1% of the observations.</p>
<p>In a quantile-quantile (Q-Q) plot the straight line indicates the
quantiles a normal distribution would follow. The untransformed insulin
AUC data values do <strong>not</strong> follow a normal distribution
because the points are far from the line. If they were reasonably
normally distributed, most of the data points would fall directly on the
straight line.</p>
<p>Next, we will log-transform the data and then create a
quantile-quantile plot.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Ins_tAUC <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Quantile-Quantile Plot of log(Ins_tAUC)"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"Normal Quantiles"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"log(Ins_tAUC)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-qqplot_log-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-4" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4" class="callout-inner">
<h3 class="callout-title">Challenge 4</h3>
<div class="callout-content">
<p>Does the log transformation make the data more normally distributed?
Explain your answer.</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>Yes. The log transformation makes the data more normally distributed
because the data points follow the normality line more closely.</p>
</div>
</div>
</div>
</div>
<div id="challenge-5" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5" class="callout-inner">
<h3 class="callout-title">Challenge 5</h3>
<div class="callout-content">
<p>Do any data points look suspicious to you? Explain your answer.</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<p>The data points that deviate from the normality line would be worth
investigating. All data deviates somewhat from normality, but the three
lowest points in the male data plot would be worth investigating. They
may be real, but there may also have been mishap in the assay.</p>
</div>
</div>
</div>
</div>
<p>Another way to identify outliers is to standardize the data and look
for data points that are more than four standard deviations from the
mean.</p>
<p>To do this, we will log transform and standardize insulin AUC.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ins_tauc</span> <span class="op">=</span> <span class="va">pheno</span> <span class="op">|&gt;</span> </span>
<span>             <span class="fu">select</span><span class="op">(</span><span class="va">mouse</span>, <span class="va">sex</span>, <span class="va">Ins_tAUC</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">group_by</span><span class="op">(</span><span class="va">sex</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>             <span class="fu">mutate</span><span class="op">(</span>Ins_tAUC <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">Ins_tAUC</span><span class="op">)</span>,</span>
<span>                    Ins_tAUC <span class="op">=</span> <span class="fu">scale</span><span class="op">(</span><span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ins_tauc</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">sex</span>, y <span class="op">=</span> <span class="va">Ins_tAUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_beeswarm</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="op">-</span><span class="fl">4</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span>  <span class="fl">4</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Distribution of Standardized Ins_tAUC"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-pheno_std-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>There are no data points outside of the four standard deviation
limits.</p>
</div>
</section><section><h2 class="section-heading" id="gene-expression-data">Gene Expression Data<a class="anchor" aria-label="anchor" href="#gene-expression-data"></a></h2>
<hr class="half-width"><p>Let’s read in the gene expression data.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_expr_annot.rds'</span><span class="op">)</span></span>
<span><span class="va">raw</span>   <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_do_expr_raw.rds'</span><span class="op">)</span></span></code></pre>
</div>
<p>We have loaded in two data objects:</p>
<ol style="list-style-type: decimal"><li>
<code>annot</code>: a data frame containing gene annotation,
and</li>
<li>
<code>raw</code>: a numeric matrix containing the un-normalized
expression counts.</li>
</ol><div id="challenge-6-how-many-samples-and-genes-are-there" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-how-many-samples-and-genes-are-there" class="callout-inner">
<h3 class="callout-title">Challenge 6: How many samples and genes are there?</h3>
<div class="callout-content">

</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<p>Use the <code>dim</code> command or the Environment tab to determine
the number of samples and genes in <code>raw</code>.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">raw</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1]   378 21771</code></pre>
</div>
<p>There are 378 samples and 21,771 genes.</p>
</div>
</div>
</div>
</div>
<p>The expression objects that we have loaded in are organized such that
the transcripts and samples are aligned between the objects. The figure
below may help you to visualize the relationship between the expression,
annotation, and covariates.</p>
<figure><img src="../fig/expr_sample_annot.png" alt="Figure showing relationship between samples, expression, and transcripts." class="figure mx-auto d-block"><div class="figcaption">Data Diagram</div>
</figure><p>Let’s look at the rows in the gene annotation object.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">annot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                              gene_id symbol chr     start       end strand
ENSMUSG00000000001 ENSMUSG00000000001  Gnai3   3 108.10728 108.14615     -1
ENSMUSG00000000028 ENSMUSG00000000028  Cdc45  16  18.78045  18.81199     -1
ENSMUSG00000000037 ENSMUSG00000000037  Scml2   X 161.11719 161.25821      1
ENSMUSG00000000049 ENSMUSG00000000049   Apoh  11 108.34335 108.41440      1
ENSMUSG00000000056 ENSMUSG00000000056   Narf  11 121.23725 121.25586      1
ENSMUSG00000000058 ENSMUSG00000000058   Cav2   6  17.28119  17.28911      1
                      middle nearest.marker.id        biotype      module
ENSMUSG00000000001 108.12671       3_108090236 protein_coding   darkgreen
ENSMUSG00000000028  18.79622       16_18817262 protein_coding        grey
ENSMUSG00000000037 161.18770       X_161182677 protein_coding        grey
ENSMUSG00000000049 108.37887      11_108369225 protein_coding greenyellow
ENSMUSG00000000056 121.24655      11_121200487 protein_coding       brown
ENSMUSG00000000058  17.28515        6_17288298 protein_coding       brown
                   hotspot
ENSMUSG00000000001    &lt;NA&gt;
ENSMUSG00000000028    &lt;NA&gt;
ENSMUSG00000000037    &lt;NA&gt;
ENSMUSG00000000049    &lt;NA&gt;
ENSMUSG00000000056    &lt;NA&gt;
ENSMUSG00000000058    &lt;NA&gt;</code></pre>
</div>
<p>There are many columns in the gene annotation file, including the
Ensembl ID, gene symbol, chromosome, start and end of the gene.</p>
<p>Next, let’s look at the sample covariates.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">covar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>      mouse sex DOwave diet_days
DO021 DO021   F      1       112
DO022 DO022   F      1       112
DO023 DO023   F      1       112
DO024 DO024   F      1       112
DO025 DO025   F      1       114
DO026 DO026   F      1       114</code></pre>
</div>
<p>The sample covariates have information about the sex and DO
generation, indicated as <code>DOwave</code>, of each mouse. These are
<em>metadata</em>, or data about the data.</p>
<p>In order to make reasonable gene comparisons between samples, the
count data needs to be normalized. In the quantile-quantile (Q-Q) plot
below, count data for the first gene are plotted over a diagonal line
tracing a normal distribution for those counts. Notice that most of the
count data values lie off of this line, indicating that these gene
counts are not normally distributed.</p>
<figure><img src="../fig/load-explore-data-rendered-view_manual_qqplot_raw-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Q-Q plots for the first six genes show that count data for these
genes are not normally distributed. They are also not on the same scale.
The y-axis values for each subplot range to 20,000 counts in the first
subplot, 250 in the second, 90 in the third, and so on.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">ENSMUSG00000000001</span><span class="op">:</span><span class="va">ENSMUSG00000000058</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">everything</span><span class="op">(</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">'gene'</span>, values_to <span class="op">=</span> <span class="st">'value'</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">stat_qq</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_qq_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">gene</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">'Count distribution for six genes'</span>,</span>
<span>         xlab <span class="op">=</span> <span class="st">'Normal percentiles'</span>, y <span class="op">=</span> <span class="st">'Count percentiles'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/load-explore-data-rendered-view_qqplots_raw-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Since each gene has a different distribution, we will need to
normalize the gene expression data. We will do this in a future
lesson.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>It is important to inspect the phenotype distributions and to
transform them to be nearly normal.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/review-eqtl-study.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/map-one-eqtl.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/review-eqtl-study.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Genetic Drivers of
        </a>
        <a class="chapter-link float-end" href="../instructor/map-one-eqtl.html" rel="next">
          Next: Mapping A Single...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/load-explore-data.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.10" class="external-link">sandpaper (0.16.10)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.7" class="external-link">pegboard (0.7.7)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/instructor/load-explore-data.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene expression, quantitative trait, genetic mapping, Diversity Outbred population",
  "name": "Load and Explore Data",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/instructor/load-explore-data.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/instructor/load-explore-data.html",
  "dateCreated": "2024-11-07",
  "dateModified": "2024-12-11",
  "datePublished": "2024-12-11"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

