<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Expression Quantitative Trait Locus (eQTL) Mapping: Mapping A Single Gene Expression Trait</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png"><link rel="manifest" href="../favicons/incubator/site.webmanifest"><link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../map-one-eqtl.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Expression Quantitative Trait Locus (eQTL) Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Expression Quantitative Trait Locus (eQTL) Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr></ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Expression Quantitative Trait Locus (eQTL) Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 33%" class="percentage">
    33%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 33%" aria-valuenow="33" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../map-one-eqtl.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="review-eqtl-study.html">2. Genetic Drivers of Pancreatic Islet Function</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="load-explore-data.html">3. Load and Explore Data</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="review-mapping-steps.html">4. Review Mapping Steps</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        5. Mapping A Single Gene Expression Trait
        </span>
      
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="map-many-eqtls.html">6. Mapping Many Gene Expression Traits</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="create-transcriptome-map.html">7. Creating A Transcriptome Map</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="full-transcriptome-map.html">8. Transcriptome Map of cis and trans eQTL</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="maximum-peaks-genes.html">9. Maximum eQTL Peaks and Nearby Genes</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="interpret-results.html">10. Interpreting qtl2 results</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="mediation-analysis.html">11. Mediation Analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr></ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/review-mapping-steps.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/map-many-eqtls.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/review-mapping-steps.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Review Mapping Steps
        </a>
        <a class="chapter-link float-end" href="../instructor/map-many-eqtls.html" rel="next">
          Next: Mapping Many Gene...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Mapping A Single Gene Expression Trait</h1>
        <p>Last updated on 2024-11-20 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/map-one-eqtl.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How do I map one gene expression trait?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>To run a QTL analysis for expression data</li>
</ul></div>
</div>
</div>
</div>
</div>
<p>In this tutorial, we are going to use the gene expression data as our
phenotype to map QTLs. That is, we are looking to find any QTLs (cis or
trans) that are responsible for variation in gene expression across
samples.</p>
<p>We are using the same directory as our previous tutorial, which you
can check by typing <code>getwd()</code> in the Console. If you are not
in the correct directory, type in <code>setwd("code")</code> in the
console or Session -&gt; Set Working Directory -&gt; Choose Directory in
the RStudio menu to set your working directory to the main
directory.</p>
<p>If you have started a new R session, you will need to load the
libraries again. If you haven’t, the libraries listed below are already
loaded.</p>
<div class="section level3">
<h3 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a></h3>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">knitr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">broom</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a></h3>
<p>In this lesson, we are loading in gene expression data for
<code>21,771</code> genes: <code>attie_DO500_expr.datasets.RData</code>
that includes normalised and raw gene expression data.
<code>dataset.islet.rnaseq</code> is the dataset you can download
directly from Dryad. Again, if you are using the same R session, you
will not need to the load the mapping, phenotypes and genotype
probabilities data, again.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># expression data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_expr.datasets.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_expr.datasets.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># data from paper</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/dataset.islet.rnaseq.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/dataset.islet.rnaseq.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># phenotypes</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_clinical.phenotypes.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_clinical.phenotypes.RData', probable reason 'No such file
or directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># mapping data</span></span>
<span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/attie_DO500_mapping.data.RData"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'../data/attie_DO500_mapping.data.RData', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># genotype probabilities</span></span>
<span><span class="va">probs</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">"../data/attie_DO500_genoprobs_v5.rds"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in gzfile(file, "rb"): cannot open compressed file
'../data/attie_DO500_genoprobs_v5.rds', probable reason 'No such file or
directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in gzfile(file, "rb"): cannot open the connection</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="expression-data">Expression Data<a class="anchor" aria-label="anchor" href="#expression-data"></a></h3>
<p>Raw gene expression counts are in the <code>counts</code> data
object. These counts have been normalised and saved in the
<code>norm</code> data object. More information is about normalisation
is <a href="https://whoatemycookie.github.io/gene-expression-qtl/03-load-explore-data/index.html#gene-expression-phenotypes" class="external-link">here</a>.</p>
<p>To view the contents of either one of these data , click the table on
the right hand side either <code>norm</code> or <code>counts</code> in
the Environment tab. If you type in <code>names(counts)</code>, you will
see the names all start with <code>ENSMUSG</code>. These are Ensembl
IDs. If we want to see which gene these IDs correspond to, type in
<code>dataset.islet.rnaseq$annots</code>, which gives information about
each gene, including ensemble id, gene symbol as well as start &amp;
stop location of the gene and chromsome on which the gene lies.</p>
<p>Because we are working with the <code>insulin tAUC</code> phenotype,
let’s map the expression counts for <code>Hnf1b</code> which is known to
influence this phenotype is these data. First, we need to find the
Ensembl ID for this gene:</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">annots</span><span class="op">[</span><span class="va">dataset.islet.rnaseq</span><span class="op">$</span><span class="va">annots</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"Hnf1b"</span>,<span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'dataset.islet.rnaseq' not found</code></pre>
</div>
<p>We can see that the ensembl ID of <code>Hnf1b</code> is
<code>ENSMUSG00000020679</code>. If we check the distribution for
<code>Hnf1b</code> expression data between the raw and normalised data,
we can see there distribution has been corrected. Here is the
distribution of the raw counts:</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">counts</span><span class="op">$</span><span class="va">ENSMUSG00000020679</span>, main <span class="op">=</span> <span class="st">"Hnf1b"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'counts' not found</code></pre>
</div>
<p>and here is the distribution of the normalised counts:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">norm</span><span class="op">$</span><span class="va">ENSMUSG00000020679</span>, main <span class="op">=</span> <span class="st">"Hnf1b"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in norm$ENSMUSG00000020679: object of type 'closure' is not subsettable</code></pre>
</div>
<p>The histogram indicates that distribution of these counts are
normalised.</p>
</div>
<div class="section level3">
<h3 id="the-marker-map">The Marker Map<a class="anchor" aria-label="anchor" href="#the-marker-map"></a></h3>
<p>We are using the same marker map as in the previous <a href="https://smcclatchy.github.io/eqtl-mapping/review-mapping-steps/index.html#the-marker-map" class="external-link">lesson</a></p>
</div>
<div class="section level3">
<h3 id="genotype-probabilities">Genotype probabilities<a class="anchor" aria-label="anchor" href="#genotype-probabilities"></a></h3>
<p>We have explored this earlier in th previous <a href="https://smcclatchy.github.io/eqtl-mapping/review-mapping-steps/index.html#genotype-probabilities" class="external-link">lesson</a>.
But, as a reminder, we have already calculated genotype probabilities
which we loaded above called <code>probs</code>. This contains the 8
state genotype probabilities using the 69k grid map of the same 500 DO
mice that also have clinical phenotypes.</p>
</div>
<div class="section level3">
<h3 id="kinship-matrix"><a href="https://smcclatchy.github.io/qtl-mapping/calc-kinship/" class="external-link">Kinship
Matrix</a><a class="anchor" aria-label="anchor" href="#kinship-matrix"></a></h3>
<p>We have explored the kinship matrix in the previous <a href="https://smcclatchy.github.io/eqtl-mapping/review-mapping-steps/index.html#kinship-matrix" class="external-link">lesson</a>.
It has already been calculated and loaded in above.</p>
</div>
<div class="section level3">
<h3 id="covariates">Covariates<a class="anchor" aria-label="anchor" href="#covariates"></a></h3>
<p>Now let’s add the necessary covariates. For <code>Hnf1b</code>
expression data, let’s see which covariates are significant.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">###merging covariate data and expression data to test for sex, wave and diet_days.</span></span>
<span></span>
<span><span class="va">cov.counts</span> <span class="op">&lt;-</span> <span class="fu">merge</span><span class="op">(</span><span class="va">covar</span>, <span class="va">norm</span>, by<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="st">"row.names"</span><span class="op">)</span>, sort<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'covar' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#testing covairates on expression data</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="va">cov.counts</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">mouse</span>, <span class="va">sex</span>, <span class="va">DOwave</span>, <span class="va">diet_days</span>, <span class="va">ENSMUSG00000020679</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">gather</span><span class="op">(</span><span class="va">expression</span>, <span class="va">value</span>, <span class="op">-</span><span class="va">mouse</span>, <span class="op">-</span><span class="va">sex</span>, <span class="op">-</span><span class="va">DOwave</span>, <span class="op">-</span><span class="va">diet_days</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">group_by</span><span class="op">(</span><span class="va">expression</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">nest</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'cov.counts' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod_fxn</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">lm</span><span class="op">(</span><span class="va">value</span> <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span> <span class="op">+</span> <span class="va">diet_days</span>, data <span class="op">=</span> <span class="va">df</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">tmp</span> <span class="op">=</span> <span class="va">tmp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="va">mod_fxn</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>summ <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">model</span>, <span class="va">tidy</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">unnest</span><span class="op">(</span><span class="va">summ</span><span class="op">)</span> </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'tmp' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#  kable(tmp, caption = "Effects of Sex, Wave &amp; Diet Days on Expression")</span></span>
<span></span>
<span><span class="va">tmp</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'tmp' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">term</span> <span class="op">!=</span> <span class="st">"(Intercept)"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>neg.log.p <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">term</span>, <span class="va">neg.log.p</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">expression</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Significance of Sex, Wave &amp; Diet Days on Expression"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'tmp' not found</code></pre>
</div>
<p>We can see that sex and DOwave are significant. Here DOwave is the
group or batch number as not all mice were submitted for genotyping at
the same time. Because of this, we now have to correct for it.
Considering the paper included the covariate, <code>diet_days</code>, we
will include that as well.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># convert sex and DO wave (batch) to factors</span></span>
<span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">sex</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">diet_days</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">pheno_clin</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covar</span> <span class="op">=</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span> <span class="op">+</span> <span class="va">diet_days</span>, data <span class="op">=</span> <span class="va">pheno_clin</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'pheno_clin' not found</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="performing-a-genome-scan"><a href="https://smcclatchy.github.io/qtl-mapping/perform-genome-scan/" class="external-link">Performing
a genome scan</a><a class="anchor" aria-label="anchor" href="#performing-a-genome-scan"></a></h3>
<div class="section level4">
<h4 id="permutations">[Permutations]<a class="anchor" aria-label="anchor" href="#permutations"></a></h4>
<p>First, we need to work out the signifcance level. Let’s find the
signifance level for 0.1, 0.05 and 0.01.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">operm</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno <span class="op">=</span> <span class="va">norm</span><span class="op">[</span>,<span class="st">"ENSMUSG00000020679"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   addcovar<span class="op">=</span><span class="va">covar</span>,</span>
<span>                   n_perm<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span></code></pre>
</div>
<p><em>Note</em> DO NOT RUN THIS (it will take too long). Instead, I
have run it earlier and will load it in here. We will also perform a
summary to find the summary level for 0.1, 0.05 and 0.01</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"../data/operm_ENSMUSG00000020679_1000.Rdata"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>,alpha<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.05</span>, <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="genome-scan">Genome Scan<a class="anchor" aria-label="anchor" href="#genome-scan"></a></h4>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl</span> <span class="op">=</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>            pheno <span class="op">=</span> <span class="va">norm</span><span class="op">[</span>,<span class="st">"ENSMUSG00000020679"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>            kinship <span class="op">=</span> <span class="va">K</span>, </span>
<span>            addcovar <span class="op">=</span> <span class="va">covar</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we plot the genome scan.</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">qtl</span>, </span>
<span>           map <span class="op">=</span> <span class="va">map</span>, </span>
<span>           lodcolumn <span class="op">=</span> <span class="st">"ENSMUSG00000020679"</span>,</span>
<span>           main <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">qtl</span><span class="op">)</span><span class="op">)</span></span>
<span>           <span class="fu">add_threshold</span><span class="op">(</span><span class="va">map</span>,  <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'purple'</span><span class="op">)</span></span>
<span>           <span class="fu">add_threshold</span><span class="op">(</span><span class="va">map</span>,  <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span>
<span>           <span class="fu">add_threshold</span><span class="op">(</span><span class="va">map</span>,  <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'blue'</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
<div class="section level3">
<h3 id="finding-lod-peaks"><a href="https://smcclatchy.github.io/qtl-mapping/find-lod-peaks/" class="external-link">Finding
LOD peaks</a><a class="anchor" aria-label="anchor" href="#finding-lod-peaks"></a></h3>
<p>Let’s find LOD peaks</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_threshold</span> <span class="op">=</span> <span class="fu">summary</span><span class="op">(</span><span class="va">operm</span>, alpha<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">peaks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">qtl</span>, </span>
<span>                   map <span class="op">=</span> <span class="va">map</span>, </span>
<span>                   threshold <span class="op">=</span> <span class="va">lod_threshold</span>, </span>
<span>                   peakdrop <span class="op">=</span> <span class="fl">4</span>, prob <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">peaks</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">lodindex</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">arrange</span><span class="op">(</span><span class="va">chr</span>, <span class="va">pos</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Phenotype QTL Peaks with LOD &gt;= 6"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="qtl-effects">QTL effects<a class="anchor" aria-label="anchor" href="#qtl-effects"></a></h3>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs<span class="op">=</span><span class="va">probs</span><span class="op">[</span>,<span class="va">peaks</span><span class="op">$</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                  <span class="va">norm</span><span class="op">[</span>,<span class="va">peaks</span><span class="op">$</span><span class="va">lodcolumn</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, drop<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_coefCC</span><span class="op">(</span><span class="va">blup</span>, </span>
<span>       map<span class="op">=</span><span class="va">map</span>, </span>
<span>       columns<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,</span>
<span>       bgcolor<span class="op">=</span><span class="st">"gray95"</span>, </span>
<span>       legend<span class="op">=</span><span class="st">"bottomleft"</span>,</span>
<span>       scan1_output <span class="op">=</span> <span class="va">qtl</span> <span class="op">)</span></span></code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1:</h3>
<div class="callout-content">
<p>Now choose another gene expression trait in <code>norm</code> data
object and perform the same steps.<br>
1). Check the distribution of the raw counts and normalised counts 2).
Are there any sex, batch, diet effects?<br>
3). Run a genome scan with the genotype probabilities and kinship
provided.<br>
4). Plot the genome scan for this gene.<br>
5). Find the peaks above LOD score of 6.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Replace <code>&lt;ensembl id&gt;</code> with your choice of gene
expression trait</p>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a><span class="co">#1).</span></span>
<span id="cb47-2"><a href="#cb47-2" tabindex="-1"></a></span>
<span id="cb47-3"><a href="#cb47-3" tabindex="-1"></a><span class="fu">hist</span>(pheno_clin<span class="sc">$</span><span class="er">&lt;</span>ensembl id<span class="sc">&gt;</span>)</span>
<span id="cb47-4"><a href="#cb47-4" tabindex="-1"></a></span>
<span id="cb47-5"><a href="#cb47-5" tabindex="-1"></a>pheno_clin<span class="sc">$</span><span class="er">&lt;</span>ensembl id<span class="sc">&gt;</span>_log <span class="ot">&lt;-</span> <span class="fu">log</span>(pheno_clin<span class="sc">$</span><span class="er">&lt;</span>ensembl id<span class="sc">&gt;</span>)</span>
<span id="cb47-6"><a href="#cb47-6" tabindex="-1"></a></span>
<span id="cb47-7"><a href="#cb47-7" tabindex="-1"></a><span class="fu">hist</span>(pheno_clin<span class="sc">$</span><span class="er">&lt;</span>ensembl id<span class="sc">&gt;</span>_log)</span>
<span id="cb47-8"><a href="#cb47-8" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" tabindex="-1"></a></span>
<span id="cb47-10"><a href="#cb47-10" tabindex="-1"></a><span class="co">#2).</span></span>
<span id="cb47-11"><a href="#cb47-11" tabindex="-1"></a></span>
<span id="cb47-12"><a href="#cb47-12" tabindex="-1"></a>tmp <span class="ot">=</span> pheno_clin <span class="sc">%&gt;%</span></span>
<span id="cb47-13"><a href="#cb47-13" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(mouse, sex, DOwave, diet_days, <span class="sc">&lt;</span>ensembl id<span class="sc">&gt;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-14"><a href="#cb47-14" tabindex="-1"></a>        <span class="fu">gather</span>(expression, value, <span class="sc">-</span>mouse, <span class="sc">-</span>sex, <span class="sc">-</span>DOwave, <span class="sc">-</span>diet_days) <span class="sc">%&gt;%</span></span>
<span id="cb47-15"><a href="#cb47-15" tabindex="-1"></a>        <span class="fu">group_by</span>(expression) <span class="sc">%&gt;%</span></span>
<span id="cb47-16"><a href="#cb47-16" tabindex="-1"></a>        <span class="fu">nest</span>()</span>
<span id="cb47-17"><a href="#cb47-17" tabindex="-1"></a>mod_fxn <span class="ot">=</span> <span class="cf">function</span>(df) {</span>
<span id="cb47-18"><a href="#cb47-18" tabindex="-1"></a>  <span class="fu">lm</span>(value <span class="sc">~</span> sex <span class="sc">+</span> DOwave <span class="sc">+</span> diet_days, <span class="at">data =</span> df)</span>
<span id="cb47-19"><a href="#cb47-19" tabindex="-1"></a>}</span>
<span id="cb47-20"><a href="#cb47-20" tabindex="-1"></a>tmp <span class="ot">=</span> tmp <span class="sc">%&gt;%</span></span>
<span id="cb47-21"><a href="#cb47-21" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model =</span> <span class="fu">map</span>(data, mod_fxn)) <span class="sc">%&gt;%</span></span>
<span id="cb47-22"><a href="#cb47-22" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">summ =</span> <span class="fu">map</span>(model, tidy)) <span class="sc">%&gt;%</span></span>
<span id="cb47-23"><a href="#cb47-23" tabindex="-1"></a>  <span class="fu">unnest</span>(summ) </span>
<span id="cb47-24"><a href="#cb47-24" tabindex="-1"></a></span>
<span id="cb47-25"><a href="#cb47-25" tabindex="-1"></a>tmp</span>
<span id="cb47-26"><a href="#cb47-26" tabindex="-1"></a></span>
<span id="cb47-27"><a href="#cb47-27" tabindex="-1"></a>tmp <span class="sc">%&gt;%</span></span>
<span id="cb47-28"><a href="#cb47-28" tabindex="-1"></a>  <span class="fu">filter</span>(term <span class="sc">!=</span> <span class="st">"(Intercept)"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-29"><a href="#cb47-29" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neg.log.p =</span> <span class="sc">-</span><span class="fu">log10</span>(p.value)) <span class="sc">%&gt;%</span></span>
<span id="cb47-30"><a href="#cb47-30" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(term, neg.log.p)) <span class="sc">+</span></span>
<span id="cb47-31"><a href="#cb47-31" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb47-32"><a href="#cb47-32" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>expression) <span class="sc">+</span></span>
<span id="cb47-33"><a href="#cb47-33" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Significance of Sex, Wave &amp; Diet Days on Phenotypes"</span>) <span class="sc">+</span></span>
<span id="cb47-34"><a href="#cb47-34" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb47-35"><a href="#cb47-35" tabindex="-1"></a><span class="fu">rm</span>(tmp)</span>
<span id="cb47-36"><a href="#cb47-36" tabindex="-1"></a></span>
<span id="cb47-37"><a href="#cb47-37" tabindex="-1"></a><span class="co">#3).</span></span>
<span id="cb47-38"><a href="#cb47-38" tabindex="-1"></a></span>
<span id="cb47-39"><a href="#cb47-39" tabindex="-1"></a>qtl <span class="ot">=</span> <span class="fu">scan1</span>(<span class="at">genoprobs =</span> probs, </span>
<span id="cb47-40"><a href="#cb47-40" tabindex="-1"></a>           <span class="at">pheno =</span> pheno_clin[,<span class="st">"&lt;ensembl id&gt;"</span>, <span class="at">drop =</span> <span class="cn">FALSE</span>], </span>
<span id="cb47-41"><a href="#cb47-41" tabindex="-1"></a>        <span class="at">kinship =</span> K, </span>
<span id="cb47-42"><a href="#cb47-42" tabindex="-1"></a>        <span class="at">addcovar =</span> covar)</span>
<span id="cb47-43"><a href="#cb47-43" tabindex="-1"></a></span>
<span id="cb47-44"><a href="#cb47-44" tabindex="-1"></a><span class="co">#4).</span></span>
<span id="cb47-45"><a href="#cb47-45" tabindex="-1"></a></span>
<span id="cb47-46"><a href="#cb47-46" tabindex="-1"></a><span class="fu">plot_scan1</span>(<span class="at">x =</span> qtl, <span class="at">map =</span> map, <span class="at">lodcolumn =</span> <span class="st">"&lt;ensembl id&gt;"</span>)</span>
<span id="cb47-47"><a href="#cb47-47" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">6</span>, <span class="at">col =</span> <span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb47-48"><a href="#cb47-48" tabindex="-1"></a></span>
<span id="cb47-49"><a href="#cb47-49" tabindex="-1"></a><span class="co">#5). </span></span>
<span id="cb47-50"><a href="#cb47-50" tabindex="-1"></a></span>
<span id="cb47-51"><a href="#cb47-51" tabindex="-1"></a>peaks <span class="ot">=</span> <span class="fu">find_peaks</span>(<span class="at">scan1_output =</span> qtl, <span class="at">map =</span> map, </span>
<span id="cb47-52"><a href="#cb47-52" tabindex="-1"></a>               <span class="at">threshold =</span> lod_threshold, </span>
<span id="cb47-53"><a href="#cb47-53" tabindex="-1"></a>               <span class="at">peakdrop =</span> <span class="dv">4</span>, </span>
<span id="cb47-54"><a href="#cb47-54" tabindex="-1"></a>               <span class="at">prob =</span> <span class="fl">0.95</span>)</span></code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>Use <code>.md</code> files for episodes when you want static
content</li>
<li>Use <code>.Rmd</code> files for episodes when you need to generate
output</li>
<li>Run <code>sandpaper::check_lesson()</code> to identify any issues
with your lesson</li>
<li>Run <code>sandpaper::build_lesson()</code> to preview your lesson
locally</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>



      </div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/review-mapping-steps.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/map-many-eqtls.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/review-mapping-steps.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Review Mapping Steps
        </a>
        <a class="chapter-link float-end" href="../instructor/map-many-eqtls.html" rel="next">
          Next: Mapping Many Gene...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/map-one-eqtl.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.10" class="external-link">sandpaper (0.16.10)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.7" class="external-link">pegboard (0.7.7)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/instructor/map-one-eqtl.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene expression, quantitative trait, genetic mapping, Diversity Outbred population",
  "name": "Mapping A Single Gene Expression Trait",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/instructor/map-one-eqtl.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/instructor/map-one-eqtl.html",
  "dateCreated": "2024-11-07",
  "dateModified": "2024-11-20",
  "datePublished": "2024-11-20"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

