<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Expression Quantitative Trait Locus (eQTL) Mapping: Mapping Many Gene Expression Traits</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png"><link rel="manifest" href="../favicons/incubator/site.webmanifest"><link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-info">
          <abbr title="This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#polishing-beta-stage" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-circle" style="border-radius: 5px"></i>
              Beta
            </a>
            <span class="visually-hidden">This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../map-many-eqtls.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Expression Quantitative Trait Locus (eQTL) Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Expression Quantitative Trait Locus (eQTL) Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr></ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Expression Quantitative Trait Locus (eQTL) Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 40%" class="percentage">
    40%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 40%" aria-valuenow="40" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../map-many-eqtls.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="review-eqtl-study.html">2. Genetic Drivers of Pancreatic Islet Function</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="load-explore-data.html">3. Load and Explore Data</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="map-one-eqtl.html">4. Mapping A Single Gene Expression Trait</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        5. Mapping Many Gene Expression Traits
        </span>
      
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="maximum-peaks-genes.html">6. Maximum eQTL Peaks and Nearby Genes</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="create-transcriptome-map.html">7. Creating A Transcriptome Map</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="mediation-analysis.html">8. Mediation Analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr></ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/map-one-eqtl.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/maximum-peaks-genes.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/map-one-eqtl.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Mapping A Single
        </a>
        <a class="chapter-link float-end" href="../instructor/maximum-peaks-genes.html" rel="next">
          Next: Maximum eQTL Peaks...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Mapping Many Gene Expression Traits</h1>
        <p>Last updated on 2025-03-11 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/map-many-eqtls.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 50 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How do I map all of the genes in my data set?</li>
<li>What resources do I need to map all of the genes in my data
set?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>To map several genes at the same time</li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="mapping-all-genes">Mapping All Genes<a class="anchor" aria-label="anchor" href="#mapping-all-genes"></a></h3>
<p>Most people have laptops with enough memory and computing power to
map hundreds of genes. However, we have over 20,000 genes in our data
set. This would either take a very long time or would use up all of the
memory on most peoples’ laptops.</p>
<p>To map all of the genes, you will need to use a computing cluster,
such as <em>sumner2</em> at JAX. Since mapping all of the genes took 12
hours, you will not perform this operation in this workshop. Instead, we
will show you how to set up a computing job to run this analysis.</p>
<p><em>sumner2</em> uses <a href="https://slurm.schedmd.com/overview.html" class="external-link">slurm</a> to schedule
computing jobs. <code>slurm</code> is software that runs on the cluster
and manages how computing jobs are run. Users submit job requests and
<code>slurm</code> checks whether the requested memory, number of
processors, and time are available and then decides when to run the job.
It also balances the needs of different users so that everyone can get
their computing jobs done.</p>
<p><em>sumner2</em> also uses software containers to run software. JAX
uses <a href="https://docs.sylabs.io/guides/3.5/user-guide/introduction.html" class="external-link">Singularity</a>
containers. These contain software without the need to install it on the
cluster. The Computational Sciences group makes a large number of
containers available on <em>sumner2</em> and we will use their
<code>qtl2</code> container.</p>
<p>If you are unfamiliar with the JAX computing cluster, Research IT has
<a href="https://jacksonlaboratory.sharepoint.com/sites/ResearchIT/SitePages/Documentation.aspx" class="external-link">excellent
documentation</a> that explains how the cluster works, how to submit
jobs using <code>slurm</code>, and how to use <code>Singularity</code>
containers.</p>
<div class="section level4">
<h4 id="bash-script">bash Script<a class="anchor" aria-label="anchor" href="#bash-script"></a></h4>
<p>To run a batch job on the cluster, we will create a bash script which
will request resources from <code>slurm</code> and will call an R
script, which will perform the mapping. The script that we used is shown
below.</p>
<pre><code>#!/bin/bash
#SBATCH --qos batch
#SBATCH --partition compute
#SBATCH --nodes 1
#SBATCH --ntasks 1
#SBATCH --cpus-per-task 22
#SBATCH --mem 200G
#SBATCH --time 1-0:00

CONTAINER=/projects/omics_share/meta/containers/quay.io-jaxcompsci-r-qtl2-deseq-biomart-tidy-v4.img

module load singularity

singularity exec ${CONTAINER} Rscript eqtl.R</code></pre>
<p>For each data set, you will need to determine the number of cores to
use. <code>qtl2</code> can perform the QTL mapping in parallel on one
node, so you will typically request only one node. In this case, we
requested 22 cores on one node. <em>sumner2</em> has nodes with up to 70
cores, but there is a point of diminishing returns when parallelizing
too much. We performed a short test run and found that we used about
100GB of memory, so we requested twice that amount in case there were
memory surges during computation. And we requested one day (24 hours) of
compute time.</p>
<p>Overall, the job took 12 hours and used 100GB of memory.</p>
<p>Note that there are many ways to set up a computing job, including
starting many separate mapping jobs and combining the results. We have
chosen to show you one simple way in this course. If you have
<code>slurm</code> expertise, you can try other methods of breaking up
the work.</p>
<p>Next, we call the R script using <code>singularity exec</code> to
execute the command which follows the container name. We call
<code>Rscript</code> to run the eQTL mapping script, which we placed in
the same directory as the bash script.</p>
<p>The inputs to the script are the genoprobs, expression data,
covariates, kinship matrix, and the marker map. You should have all of
these pre-computed and saved on the cluster in the same directory.</p>
</div>
<div class="section level4">
<h4 id="r-script">R Script<a class="anchor" aria-label="anchor" href="#r-script"></a></h4>
<pre><code><span><span class="co">################################################################################</span></span>
<span><span class="co"># Script to map all of the Attic DO 500 pancreatic islet RNASeq data.</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># Daniel Gatti</span></span>
<span><span class="co"># dan.gatti@jax.org</span></span>
<span><span class="co"># 2024-11-01</span></span>
<span><span class="co">################################################################################</span></span>
<span></span>
<span><span class="co">##### LIBRARIES #####</span></span>
<span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##### VARIABLES #####</span></span>
<span></span>
<span><span class="co"># Base directory for project.</span></span>
<span><span class="va">base_dir</span> <span class="op">=</span> <span class="st">'/projects/compsci/eqtl_course'</span></span>
<span></span>
<span><span class="co"># Number of cores. MUST MATCH SLURM REQUEST.</span></span>
<span><span class="va">n_cores</span> <span class="op">=</span> <span class="fl">20</span></span>
<span></span>
<span><span class="co"># Read in pre-built data files.</span></span>
<span></span>
<span><span class="va">probs</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_DO500_genoprobs_v5.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">K</span>     <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_kinship.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">covar</span> <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_covar.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">expr</span>  <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_expr_rz.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">map</span>   <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_map.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##### MAIN #####</span></span>
<span></span>
<span><span class="co"># Convert covariates to factors.</span></span>
<span><span class="va">covar</span><span class="op">$</span><span class="va">sex</span>    <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">covar</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span></span>
<span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span> <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">covar</span><span class="op">$</span><span class="va">DOwave</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a matrix of additive covariates.</span></span>
<span><span class="va">addcovar</span>     <span class="op">=</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span> <span class="op">+</span> <span class="va">DOwave</span>, data <span class="op">=</span> <span class="va">covar</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Map all genes.</span></span>
<span><span class="va">lod</span> <span class="op">=</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>            pheno     <span class="op">=</span> <span class="va">expr</span>,</span>
<span>            kinship   <span class="op">=</span> <span class="va">K</span>,</span>
<span>            addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>            cores     <span class="op">=</span> <span class="va">n_cores</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save the LOD scores for all genes.</span></span>
<span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">lod</span>, file <span class="op">=</span> <span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_eqtl_lod.rds'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get the peaks with LOD &gt;= 6.</span></span>
<span><span class="va">peaks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>,</span>
<span>                   map          <span class="op">=</span> <span class="va">map</span>,</span>
<span>                   threshold    <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                   prob         <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save the harvested LOD peaks.</span></span>
<span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">peaks</span>, file <span class="op">=</span> <span class="fu">file.path</span><span class="op">(</span><span class="va">base_dir</span>, <span class="st">'attie_do_eqtl_peaks.rds'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
<p>The output of this script is two files.</p>
<ol style="list-style-type: decimal"><li>
<code>attie_do_eqtl_lod.rds</code>: This is numeric matrix
containing LOD scores for all genes at all markers. We save it as a
compressed R data file (<code>*.rds</code>). In this case, the file is
11 GB.</li>
<li>
<code>attie_do_eqtl_peaks.rds</code>: This is a data.frame
containing the peaks with LOD over 6. We save it as a compressed R data
file (<code>*.rds</code>). In this case, the file is 674 KB. You should
have downloaded this file into your <code>data</code> directory.</li>
</ol></div>
</div>
<div class="section level3">
<h3 id="finding-qtl-peaks">Finding QTL Peaks<a class="anchor" aria-label="anchor" href="#finding-qtl-peaks"></a></h3>
<p>Let’s load in the QTL peaks that we pre-computed.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span>file <span class="op">=</span> <span class="st">"data/attie_do_eqtl_peaks.rds"</span><span class="op">)</span></span></code></pre>
</div>
<p><code>peaks</code> is a data frame which contains all of the peaks in
the same format as we saw in the previous lesson. Let’s remind ourselves
what the peaks table looks like.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">peaks</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr       pos       lod      ci_lo     ci_hi
1        2 ENSMUSG00000000028   6  71.73759  7.402989  65.005336  77.42623
2        2 ENSMUSG00000000028  11  40.55159  6.028608   3.031502 119.01535
3        2 ENSMUSG00000000028  16  18.54468  7.875773  15.918386  20.81798
4        3 ENSMUSG00000000037   5  99.00918  6.837536  93.289032  99.67072
5        3 ENSMUSG00000000037  12  77.55794  6.153038  77.002567 110.22383
6        3 ENSMUSG00000000037   X 161.30209 26.698858 160.417579 161.80999</code></pre>
</div>
<p>Each row contains information for one peak, including the gene ID,
chromosome, peak position, LOD, and support interval.</p>
<p>Notice that both genes <code>ENSMUSG00000000028</code> and
<code>ENSMUSG00000000037</code> have more than one peak.</p>
<div id="challenge-1-how-many-genes-have-qtl-peaks" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-how-many-genes-have-qtl-peaks" class="callout-inner">
<h3 class="callout-title">Challenge 1: How many genes have QTL peaks?</h3>
<div class="callout-content">
<p>Look at how many genes we have and how many rows there are in peaks.
Is this the same number? If not, can you explain why the numbers are
different.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>Get the number of genes from <code>expr_rz</code>.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ncol</span><span class="op">(</span><span class="va">expr_rz</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 21771</code></pre>
</div>
<p>Get the number of QTL peaks in <code>peaks</code>.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">peaks</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 42941</code></pre>
</div>
<p>There are more peaks than there are genes.</p>
<p>There are several different possibilities when mapping a gene.</p>
<ol style="list-style-type: decimal"><li>A gene may have no QTL peaks over a LOD of 6, and so it will not
appear in the list of peaks.</li>
<li>A gene may have more than one QTL peak with a LOD greater than 6,
and so will appear many times in the list of peaks.</li>
</ol></div>
</div>
</div>
</div>
<p>Let’s count the number of peaks that each gene has.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">lodcolumn</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">n</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>    n   nn
1   1 9201
2   2 6755
3   3 3789
4   4 1425
5   5  483
6   6   87
7   7   25
8   8    4
9   9    1
10 10    1</code></pre>
</div>
<p>From the table above, we can see that most genes have one or two
peaks. However, some genes have 9 peaks with LOD &gt; 6!</p>
<p>When we built this table, we selected all of the peaks with LOD &gt;
6. This is well below the alpha = 0.05 significance threshold. Let’s get
that threshold and filter the list of peaks.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ethr</span> <span class="op">&lt;-</span> <span class="fu">summary</span><span class="op">(</span>object <span class="op">=</span> <span class="va">eperm</span>,</span>
<span>                alpha  <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">ethr</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     ENSMUSG00000020679
0.05               7.47</code></pre>
</div>
<p>Now filter the peaks to retain peaks with LOD greater than or equal
to 7.4745761.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks_filt</span> <span class="op">&lt;-</span> <span class="fu">filter</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">lod</span> <span class="op">&gt;=</span> <span class="va">ethr</span><span class="op">[</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-2-how-many-peaks-are-there-after-filtering" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-how-many-peaks-are-there-after-filtering" class="callout-inner">
<h3 class="callout-title">Challenge 2: How many peaks are there after filtering?</h3>
<div class="callout-content">
<p>Get the number of QTL peaks that we have after filtering.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">peaks_filt</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 17164</code></pre>
</div>
<p>There are now fewer peaks.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3-how-many-peaks-does-each-gene-have" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-how-many-peaks-does-each-gene-have" class="callout-inner">
<h3 class="callout-title">Challenge 3: How many peaks does each gene have?</h3>
<div class="callout-content">
<p>Use the code above to count the number of QTL peaks that each gene
has.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">peaks_filt</span>, <span class="va">lodcolumn</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">n</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  n    nn
1 1 12707
2 2  1959
3 3   151
4 4    19
5 5     2</code></pre>
</div>
<p>Now most genes have only one peak and only two genes have five
peaks.</p>
</div>
</div>
</div>
</div>
<p>Let’s look at the number of genes with LOD scores above a certain
value. We do this by counting the number of peaks with LOD &gt; 6, then
&gt; 7, etc.</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Do not try to type the next block. We will look at it on the workshop
website.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_brks</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">results</span>  <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span><span class="va">lod_brks</span>, n <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">lod_brks</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">results</span><span class="op">$</span><span class="va">n</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">sum</span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">lod</span> <span class="op">&gt;=</span> <span class="va">i</span><span class="op">)</span></span>
<span>  </span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">lod_brks</span> <span class="op">&gt;=</span> <span class="fl">6</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">lod_brks</span>, <span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_line</span><span class="op">(</span>linewidth <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">10</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Number of Peaks above LOD Score"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"LOD"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"Number of Peaks with LOD &gt;= x"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-many-eqtls-rendered-lod_cum_dist-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">lod_brks</span>, <span class="va">results</span><span class="op">)</span></span></code></pre>
</div>
<p>As you can see from the plot above, most peaks have LODs less than
10. However, about 12,000 peaks have LODs over 10. As we increase the
LOD score, we see that fewer genes have high LODs. Let’s look at how
many peaks each gene has as we increase the LOD threshold.</p>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Do not try to type the next block. We will look at it on the workshop
website.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_brks</span> <span class="op">&lt;-</span> <span class="fl">0</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">result</span>   <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>lod_brks <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, n <span class="op">=</span> <span class="fl">0</span>, nn <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">lod_brks</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">smry</span> <span class="op">&lt;-</span> <span class="va">peaks</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">filter</span><span class="op">(</span><span class="va">lod</span> <span class="op">&gt;</span> <span class="va">i</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">lodcolumn</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">mutate</span><span class="op">(</span>lod_brks <span class="op">=</span> <span class="va">i</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">result</span>, <span class="va">smry</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span>
<span></span>
<span><span class="va">result</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">lod_brks</span> <span class="op">&gt;=</span> <span class="fl">6</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">as.character</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">lod_brks</span>, <span class="va">nn</span>, color <span class="op">=</span> <span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>group <span class="op">=</span> <span class="va">n</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">10</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Number of Peaks per Gene by LOD Threshold"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"LOD"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"Number of Peaks per Gene"</span>,</span>
<span>         color <span class="op">=</span> <span class="st">"Number of Peaks"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/map-many-eqtls-rendered-eqtl_num_peaks_per_gene-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">lod_brks</span>, <span class="va">result</span><span class="op">)</span></span></code></pre>
</div>
<p>From this plot, we can see that higher LOD scores occur only once per
gene. This makes sense since a high LOD score implies that the peak
explains a large proportion of the gene’s variance.</p>
</div>
<div class="section level3">
<h3 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a></h3>
<p>In this episode, you learned how to run an eQTL analysis on a
computing cluster and started to look at the resulting peaks. When we
filter the peaks by higher LOD scores, we tend to get fewer peaks per
gene.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>Mapping all genes in a study requires a computing cluster.</li>
<li>Genes may have more than one QTL peak.</li>
<li>High LOD scores often occur only once per gene.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>



      </div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/map-one-eqtl.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/maximum-peaks-genes.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/map-one-eqtl.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Mapping A Single
        </a>
        <a class="chapter-link float-end" href="../instructor/maximum-peaks-genes.html" rel="next">
          Next: Maximum eQTL Peaks...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/map-many-eqtls.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.11" class="external-link">sandpaper (0.16.11)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/instructor/map-many-eqtls.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene expression, quantitative trait, genetic mapping, Diversity Outbred population",
  "name": "Mapping Many Gene Expression Traits",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/instructor/map-many-eqtls.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/instructor/map-many-eqtls.html",
  "dateCreated": "2024-11-07",
  "dateModified": "2025-03-11",
  "datePublished": "2025-03-11"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

