---
title: "Creating A Transcriptome Map"
teaching: 30
exercises: 30
---

:::::::::::::::::::::::::::::::::::::: questions 

- How do I create and interpret a transcriptome map?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Describe a transcriptome map.
- Interpret a transcriptome map.

::::::::::::::::::::::::::::::::::::::::::::::::

```{r setup, echo=FALSE, warning=FALSE}
# Load libraries & data for website build.
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(qtl2))

annot      <- readRDS(file = 'data/attie_do_expr_annot.rds')
peaks      <- readRDS(file = 'data/attie_do_eqtl_peaks.rds')
peaks_filt <- readRDS(file = 'data/peaks_filtered.rds')
eperm      <- readRDS(file = 'data/ENSMUSG00000020679_perm_1000.rds')
ethr       <- summary(object = eperm, alpha  = 0.05)
```

### Local and Distant eQTL

In the previous lesson, we saw that the QTL peak for a gene can lie directly
over the gene that is being mapped. This is called a "local eQTL" because the
QTL peak is located near the gene. When the QTL peak is located far from the
gene being mapped, we call this a "distant eQTL". When the QTL peak is located
on the same chromosome as the gene, there are different heuristics regarding the
distance between the gene and its QTL peak that we use to call an eQTL local
or distant. This depends on the type of cross and the resolution of the 
recombination block structure.

In this episode, we will create a table containing the QTL peaks for all genes
and the gene positions. We will then classify QTL into local or distant and 
will make a plot of all off the eQTL in relation to the gene positions.

### Transcriptome Map

We have encapsulated the code to create a transcriptome map in a file in this
lesson. You can copy this file from the Github repository to use in your eQTL 
mapping project. We will read this file in now.

```{r}
getwd()
```

```{r}
dir()
```


```{r source_ggtmap,message=FALSE,warning=FALSE}
source("https://raw.githubusercontent.com/smcclatchy/eqtl-mapping/refs/heads/main/episodes/code/gg_transcriptome_map.R")
```

This file loads in a function called `ggtmap`, which requires the input data
to be in a specific format. 

In order to use the `ggtmap` function, we need to provide specific column names. 
These are documented in the `gg_transcriptome_map.R` file in the code directory 
of this workshop. The required column names are:

* `data`: data.frame (or tibble) with the following columns:
    * `gene_id`: (required) character string containing the Ensembl gene ID.
    * `qtl_chr`: (required) character string containing QTL chromosome.
    * `qtl_pos`: (required) floating point number containing the QTL position in 
    Mb.
    * `qtl_lod`: (optional) floating point number containing the LOD score.
    * `gene_chr`:  (optional) character string containing transcript chromosome.
    * `gene_start`: (optional) character string containing transcript start 
    position in Mb.
    * `gene_end`:  (optional) character string containing transcript end 
    position in Mb.

First, we will get the gene positions from the annotation and rename the 
columns to match what `ggtmap` requires. We need to have columns named "gene_id",
"gene_chr", and "gene_pos". We must rename the columns because, when we are 
finished, we will have "chr" and "pos" columns for both the gene and its QTL. We
will also add the "symbol" column since it is nice to have gene symbols in the
data. 

```{r get_gene_pos}
gene_pos <- annot |>
              select(gene_id, symbol, 
                     gene_chr   = chr, 
                     gene_start = start,
                     gene_end   = end)
```

Next, we will rename the columns in the filtered peaks and will join them with
the gene positions from above. 

```{r lod_summary, warning=FALSE, message=FALSE}
eqtl <- peaks |>
          select(gene_id = lodcolumn,
                 qtl_chr = chr,
                 qtl_pos = pos,
                 qtl_lod = lod) |> 
          left_join(gene_pos, by = "gene_id") |> 
          mutate(marker.id = str_c(qtl_chr,   qtl_pos * 1e6, sep = "_"),
                 gene_chr  = factor(gene_chr, levels = c(1:19, "X")),
                 qtl_chr   = factor(qtl_chr,  levels = c(1:19, "X")))

rm(gene_pos)
```


::::::::::::::::::::::::::::::::::::: challenge 

#### Challenge 1: How many genes have QTL on the same chromosome?

Write a command to count the number of genes which are located on the same
chromosome as their corresponding QTL peak.

:::::::::::::::::::::::: solution 

```{r challenge1}
sum(eqtl$qtl_chr == eqtl$gene_chr, na.rm = TRUE)
```

`r sum(eqtl$qtl_chr == eqtl$gene_chr, na.rm = TRUE)` genes have QTL on the same
chromosome.

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

We can tabulate the number of local and distant eQTL that we have and add this to 
our QTL summary table. A local eQTL occurs when the QTL peaks is directly over the 
gene position. But what if it is 2 Mb away? Or 10 Mb? It's possible that a gene 
may have a trans eQTL on the same chromosome if the QTL is "far enough" from the 
gene. We have selected 2 Mb as a good rule of thumb in the DO.

```{r local_summary, warning=FALSE, message=FALSE}
eqtl <- eqtl |> 
          mutate(local = if_else(qtl_chr == gene_chr & 
                                 abs(gene_start - qtl_pos) < 2, 
                                     TRUE, 
                                     FALSE))
count(eqtl, local)
```

### Plot Transcriptome Map

```{r tmap,fig.width=8,fig.height=8,warning=FALSE,message=FALSE}
ggtmap(data = eqtl |> 
                filter(qtl_lod      >= 7), 
                       local.points = TRUE, 
                       local.radius = 2)
```

The plot above is called a "Transcriptome Map" because it shows the positions of 
the genes (or transcripts) and their corresponding QTL. The QTL position is 
shown on the X-axis and the gene position is shown on the Y-axis. The 
chromosomes are listed along the top and right of the plot. 


:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: challenge

#### Challenge 2: What are the blue points on the diagonal?

What type of QTL are the genes with QTL that are located along the diagonal?

::::::::::::::::::::::::::::::::::::: solution

Points on the diagonal have QTL that are located in the same position as the
gene, and so are local eQTL.

::::::::::::::::::::::::::::::::::::::::::::::

#### Challenge 3: Are there any genome locations with many QTL?

Look at the transcriptome map and see if you can find any vertical stripes?
What do these vertical stripes mean and what might cause them?

::::::::::::::::::::::::::::::::::::: solution

There are vertical stripes on chromosomes 2, 5, and 11. There may be more, 
depending on how you look at the plot. Since the QTL position is on the X-axis,
these stripes represent genes which are located through out the genome, but all
have a QTL in the same location.

These stripes imply that there is some genomic feature at the QTL position
which regulates the expression levels of many genes. This might be a 
transcription factor or some other molecule which can regulate transcription,
possibly through multiple steps.

::::::::::::::::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

If you look at the plot, there are vertical bands of points which stack over a
single QTL location. These are called "eQTL hotspots". Rather than look at the
transcriptome map, it may be easier to look at the density of the eQTL along
the genome. We have provided a function called "eqtl_density_plot" to do this.


```{r eqtl_density,fig.height=6,fig.width=8,warning=FALSE,message=FALSE}
eqtl_density_plot(eqtl, lod_thr = 6)
```

The plot above shows the mouse genome on the X-axis and the number of transcripts
in a 4 Mb window on the Y-axis. It is difficult to say
how many genes must be involved to call something and eQTL hotspot. There are
permutation-based methods which require a large amount of time and memory. In
this case, we have called hotspots involving more than 100 genes eQTL hotspots.

Note that there appears to be an eQTL hotspot on chromosome 7 in the plot above,
but this is not evident in the transcriptome map.

```{r ldistant_eqtl_density,fig.height=6,fig.width=8,warning=FALSE,message=FALSE}
eqtl_density_plot(filter(eqtl, local == FALSE),
                  lod_thr = 6) +
  labs(title = "Distant eQTL Density")
```

```{r local_eqtl_density,fig.height=6,fig.width=8,warning=FALSE,message=FALSE}
eqtl_density_plot(filter(eqtl, local == TRUE), 
                  lod_thr = 6) +
  labs(title = "Local eQTL Density")
```

From these two plots, it appears that the eQTL hotspots on most chromosomes 
contain distant eQTL. 

How can we find the location of the eQTL hotspots and the genes which lie
within each eQTL? We have written a function called "get_eqtl_hotspots" to help 
you and have provides it in "gg_transcriptome_map.R". It requires the eQTL data,
a LOD threshold to filter the peaks, the number of genes required to call a
locus a hotspot, and the radius in Mb around the hotspot to use when selecting
genes which are part of the hotspot

```{r get_hotspots,message=FALSE,warning=FALSE}
hotspots <- get_eqtl_hotspots(data           = eqtl, 
                              lod_thr        = 7, 
                              hotspot_thr    = 200,
                              hotspot_radius = 2)
```

Let's see how many hotspots we have and how many genes are in each hotspot.

```{r num_hotspot}
sapply(hotspots, nrow)
```

In the table above, the first row of values are the chromosomes on which 
hotspots occur. The second row contains the number of genes in each hotspot.

Where does each hotspot occur? We can get this information by taking the mean
of the eQTL positions in each hotspot.

```{r get_hotspot_pos}
sapply(hotspots, function(z) { mean(z$qtl_pos) })
```




::::::::::::::::::::::::::::::::::::: keypoints 

- Transcriptome maps aid in understanding gene expression regulation.
- Local eQTL occur more frequently than distant eQTL.
- Local eQTL appear along the diagonal in a transcriptome map and distant eQTL
appear on the off-diagonal.
- Stacks of eQTL which appear over a single locus are called "eQTL hotspots" and
represent sets of genes which are transcriptionally regulated by a single locus.

::::::::::::::::::::::::::::::::::::::::::::::::
