---
title: "Load and Explore Data"
teaching: 15
exercises: 30
---

:::::::::::::::::::::::::::::::::::::: questions 

- What data are required for eqtl mapping?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- To provide an example and exploration of data used for eqtl mapping.

::::::::::::::::::::::::::::::::::::::::::::::::

```{r load_libraries, warning=FALSE, message=FALSE}
library(knitr)
library(ggbeeswarm)
library(tidyverse)
library(qtl2)
library(DESeq2)
```

## Physiological Phenotypes

You should have downloaded data files already when following the 
[setup instructions](https://smcclatchy.github.io/eqtl-mapping/index.html). 
The complete data used in these analyses are available from 
[Data Dryad](https://doi.org/10.5061/dryad.pj105). The files we will use are in
a simpler format than those on Data Dryad.

Load in the physiological phenotypes.

```{r load_pheno}
# load the data
pheno       <- readRDS(file = 'data/attie_do_pheno.rds')
pheno_dict  <- readRDS(file = 'data/attie_do_pheno_dict.rds')
covar       <- readRDS(file = 'data/attie_do_covar.rds')
```

## Physiological Phenotypes

In this data set, we have 20 phenotypes for 500 Diversity Outbred mice. `pheno` 
is a data frame containing the phenotype data as well as covariates. Click on 
the triangle to the left of `pheno` in the Environment pane to view its 
contents. Run `names(pheno)` to list the variables. 

`pheno_dict` is the phenotype dictionary.  This data frame contains information 
on each variable in `pheno`, including `name`, `short name`, `pheno_type`,  
`formula` (if used) and `description`.  You can view a table of the data 
dictionary.

```{r dict_phenos}
pheno_dict |> 
  select(description, formula) |> 
  kable()
```

Since the paper is interested in type 2 diabetes and insulin secretion, we will 
choose insulin AUC (area  under the curve which was calculated without any
correction for baseline differences) for this review.

### Phenotype Distributions

Many statistical models, including the QTL mapping model in `qtl2`, expect that 
the incoming data will be normally distributed. You may use transformations such 
as log or square root to make your data more normally distributed. Here, we will 
log transform the data. 

Let's make a variable for insulin AUC so that we don't have to type as much.

```{r make_itauc}
ins_tauc <- pheno[, 'Ins_tAUC', drop = FALSE]
```

Next, let's look at the distribution of insulin AUC using a histogram.

```{r hist_untransformed, fig.width=8, fig.height=6}
hist(ins_tauc[,1], 
     breaks = 20,
     main   = "Insulin Area Under the Curve")
```

This is clearly **not** normally distributed. In fact, this type of distribution
is often log normal. 

Now, let's apply the `log()` function to this data in an effort to make the
distribution more normal. By default the `log()` function calculates the natural
log (base *e*).

```{r log_transform}
ins_tauc$Ins_tAUC_log <- log(ins_tauc$Ins_tAUC)
```

Let's make a histogram of the log-transformed data.

```{r hist_log_transform, fig.width=8, fig.height=6}
hist(ins_tauc$Ins_tAUC_log, 
     breaks = 20,
     main   = "insulin AUC (log-transformed)")
```

This looks much better! The data has a somewhat Gaussian shape. Technically, the
assumptions of a linear model require that the **residuals** be normally
distributed. In practice, transforming the input data to be normally distributed
helps to make the residuals normally distributed. As a reminder, a residual is 
the vertical distance from a data point to the line described by a linear model.

![A linear model showing one residual](./fig/residual.png){alt="a residual for one data point located above the line described by a linear model"}

Boxplots are another great way to view the distribution of the data and to 
identify any outliers. We will log-transform insulin AUC using the 
[`scale_y_log10()`](https://ggplot2.tidyverse.org/reference/scale_continuous.html)
function. This transforms the data using base 10 and creates a base 10 y-axis 
for plotting. We have overlaid the data points using `ggbeeswarm`'s
[`geom_beeswarm`](https://www.rdocumentation.org/packages/ggbeeswarm/versions/0.7.2/topics/geom_beeswarm).
We have told `geom_beeswarm()` to plot the points with some transparency using 
the argument `alpha = 0.2`. The `alpha` argument ranges between 0 (completely 
transparent) to 1 (completely opaque). A value of 0.1 means mostly transparent.

```{r Ins_tAUC_boxplot, fig.width=8, fig.height=6, warning=FALSE}
# plot Insulin on a log 10 scale
ggplot(pheno, aes(sex, Ins_tAUC)) +
  geom_boxplot() +
  geom_beeswarm(alpha = 0.2) +
  scale_y_log10() +
  labs(title = "Insulin area under the curve", y = "insulin AUC") +
  theme(text = element_text(size = 20))
```

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 1

How many orders of magnitude (powers of 10) does insulin AUC span?  

:::::::::::::::::::::::: solution 

insulin AUC spans three orders of magnitude, from near 10 to over 1000.  

:::::::::::::::::::::::::::::::::


## Challenge 2

Which sex has higher median insulin AUC values?  

:::::::::::::::::::::::: solution 

Males have higher insulin AUC than females.  

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 3

What does the heavy line bisecting the boxes indicate?  
What do the lines at the top and bottom of the boxes indicate?  
What does the *whisker* extending from the top and bottom of the boxes indicate?    
What do the black dots extending from the whiskers indicate? 

:::::::::::::::::::::::: solution 

The heavy line bisecting the boxes shows the median value (not the mean!). Half
of the data points are above and half are below this line.   
The lines at the top and bottom of the boxes indicate the first and 3rd 
quartiles of the data (the *hinges*). One-fourth of the data points are 
beneath the box and another one-fourth are above the box. The box itself 
contains 50% of the data points.  
The whiskers represent some multiple of the interquartile range (IQR), which is 
the height of the box between the first and third quartiles. `geom_boxplot()`
produces *Tukey-style boxplots* in which the whiskers are 1.5 $\times$ the IQR.
Any data points that lie beyond the whiskers are considered outliers and are
shown as heavy black dots.


:::::::::::::::::::::::::::::::::

:::::::::::::::::::::::::::::::::::::

The boxplot is a useful plot to visualize the distribution of your data.

### Quality Control of Data

Many statistical tests rely upon the data having a normal (or Gaussian) 
distribution. Many biological phenotypes do not follow this distribution and
must be transformed before analysis. This is why we log-transformed the data
in the plots above. 

While we can "eyeball" the distributions in the boxplot, it would be  better to 
use a *quantile-quantile* plot. 

```{r qqplot, fig.width=8, fig.height=6, warning=FALSE}
pheno |> 
  ggplot(aes(sample = Ins_tAUC)) +
    stat_qq() +
    geom_qq_line() +
    facet_wrap(~sex) +
    labs(title = "Quantile-Quantile Plot of Ins_tAUC",
         x     = "Normal Quantiles",
         y     = "Ins_tAUC") +
    theme(text = element_text(size = 20))
```

In these plots, the *quantiles* of the normal distribution are plotted on the 
X-axis and the data are plotted on the Y-axis. A quantile evenly divides data
observations into a specific number of groups. The boxplot above evenly divides
the observations into quartiles, a quantile containing four groups that each
contain one-fourth of the data. A percentile similarly divides data into 100 
quantiles, with each percentile containing 1% of the observations. 

In a quantile-quantile (Q-Q) plot the straight line indicates the quantiles a 
normal distribution would follow. The untransformed insulin AUC data values do 
**not** follow a normal distribution because the points are far from the line.
If they were reasonably normally distributed, most of the data points would fall
directly on the straight line.

Next, we will log-transform the data and then create a quantile-quantile plot.

```{r qqplot_log, fig.height=6, fig.width=8, warning=FALSE}
pheno |> 
  mutate(Ins_tAUC = log(Ins_tAUC)) |> 
  ggplot(aes(sample = Ins_tAUC)) +
    stat_qq() +
    geom_qq_line() +
    facet_wrap(~sex) +
    labs(title = "Quantile-Quantile Plot of log(Ins_tAUC)",
         x     = "Normal Quantiles",
         y     = "log(Ins_tAUC)") +
    theme(text = element_text(size = 20))
```

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 4

Does the log transformation make the data more normally distributed? Explain 
your answer.

:::::::::::::::::::::::: solution 

Yes. The log transformation makes the data more normally distributed because
the data points follow the normality line more closely. 

:::::::::::::::::::::::::::::::::

## Challenge 5

Do any data points look suspicious to you? Explain your answer.

:::::::::::::::::::::::: solution 

The data points that deviate from the normality line would be worth
investigating. All data deviates somewhat from normality, but the three lowest 
points in the male data plot would be worth investigating. They may be real, but
there may also have been mishap in the assay.

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

Another way to identify outliers is to standardize the data and look for data 
points that are more than four standard deviations from the mean.

To do this, we will log transform and standardize insulin AUC. 

```{r pheno_std, fig.width=8, fig.height=6, warning=FALSE}
ins_tauc = pheno |> 
             select(mouse, sex, Ins_tAUC) |>
             group_by(sex) |> 
             mutate(Ins_tAUC = log(Ins_tAUC),
                    Ins_tAUC = scale(Ins_tAUC))

ins_tauc |> 
  ggplot(aes(x = sex, y = Ins_tAUC)) +
    geom_boxplot() +
    geom_beeswarm(alpha = 0.2) +
    geom_hline(aes(yintercept = -4), color = 'red') +
    geom_hline(aes(yintercept =  4), color = 'red') +
    labs(title = "Distribution of Standardized Ins_tAUC") +
    theme(text = element_text(size = 20))
```

There are no data points outside of the four standard deviation limits.

## Gene Expression Data

Let's read in the gene expression data.

```{r load_expression_data}
annot <- readRDS(file = 'data/attie_do_expr_annot.rds')
raw   <- readRDS(file = 'data/attie_do_expr_raw.rds')
```

We have loaded in two data objects: 

1. `annot`: a data frame containing gene annotation, and
2. `raw`: a numeric matrix containing the un-normalized expression counts.

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 6: How many samples and genes are there?

:::::::::::::::::::::::: solution 

Use the `dim` command or the Environment tab to determine the number of samples 
and genes in `raw`.

```{r challenge5}
dim(raw)
```

There are 378 samples and 21,771 genes.

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

The expression objects that we have loaded in are organized such that the 
transcripts and samples are aligned between the objects. The figure below may
help you to visualize the relationship between the expression, annotation,
and covariates.

![Data Diagram](fig/expr_sample_annot.png){alt="Figure showing relationship between samples, expression, and transcripts."}

Let's look at the rows in the gene annotation object.

```{r view_annot_rawCounts_metadata}
head(annot)
```

There are many columns in the gene annotation file, including the Ensembl ID,
gene symbol, chromosome, start and end of the gene.

Next, let's look at the sample covariates.

```{r view_covar}
head(covar)
```

The sample covariates have information about the sex and DO generation, 
indicated as `DOwave`, of each mouse. These are *metadata*, or data about the 
data.

In order to make reasonable gene comparisons between samples, the count data 
needs to be normalized. In the quantile-quantile (Q-Q) plot below, count data 
for the first gene are plotted over a diagonal line tracing a normal 
distribution for those counts. Notice that most of the count data values lie off
of this line, indicating that these gene counts are not normally distributed. 

```{r view_manual_qqplot_raw, fig.height=6, fig.width=8, echo=FALSE, warning=FALSE}
raw |> 
  as.data.frame() |> 
  ggplot(aes(sample = ENSMUSG00000000001)) +
    stat_qq() +
    geom_qq_line() +
    labs(title = 'Count distribution for ENSMUSG00000000001',
         xlab = 'Normal percentiles', y = 'Count percentiles') +
    theme(text = element_text(size = 20))
```

Q-Q plots for the first six genes show that count data for these genes are not
normally distributed. They are also not on the same scale. The y-axis values for
each subplot range to 20,000 counts in the first subplot, 250 in the second, 90
in the third, and so on. 

```{r view_qqplots_raw, fig.width=10, fig.height=8}
raw |> 
  as.data.frame() |>
  select(ENSMUSG00000000001:ENSMUSG00000000058) |> 
  pivot_longer(cols = everything(), names_to = 'gene', values_to = 'value') |> 
  ggplot(aes(sample = value)) +
    stat_qq() +
    geom_qq_line() +
    facet_wrap(~gene, scales = 'free') +
    labs(title = 'Count distribution for six genes',
         xlab = 'Normal percentiles', y = 'Count percentiles') +
    theme(text = element_text(size = 20))
```

Since each gene has a different distribution, we will need to normalize the gene
expression data. We will do this in a future lesson.

::::::::::::::::::::::::::::::::::::: keypoints 

- It is important to inspect the phenotype distributions and to transform them
to be nearly normal.

::::::::::::::::::::::::::::::::::::::::::::::::
