---
title: "Review Mapping Steps"
teaching: 30
exercises: 30
---

:::::::::::::::::::::::::::::::::::::: questions 

- What are the steps involved in running QTL mapping in Diversity Outbred mice?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- To understand the key steps running a QTL mapping analysis.

::::::::::::::::::::::::::::::::::::::::::::::::

Before we begin to run QTL mapping on gene expression data to find eQTLs, let's 
review the main QTL mapping steps that we learnt in the 
[QTL mapping course](https://smcclatchy.github.io/qtl-mapping/). As a reminder, 
we are using data from 
[Keller et al., 2018](https://academic.oup.com/genetics/article/209/1/335/5931013?login=false) 
which are freely available to download from 
[Data Dryad](https://datadryad.org/stash/dataset/doi:10.5061/dryad.pj105).

Make sure that you have opened the project which you created so that you will
be in the correct directory and will be able to load the data.

```{r setup,include=FALSE}
pheno      <- readRDS(file = 'data/attie_do_pheno.rds')
pheno_dict <- readRDS(file = 'data/attie_do_pheno_dict.rds')
covar      <- readRDS(file = 'data/attie_do_covar.rds')
annot      <- readRDS(file = 'data/attie_do_expr_annot.rds')
expr_rz    <- readRDS(file = 'data/attie_do_expr_rz.rds')
probs      <- readRDS(file = url('https://thejacksonlaboratory.box.com/shared/static/4hy4hbjyrxjbrzh570i4g02r62bx3lgk.rds'))
ins_tauc   <- pheno[,'Ins_tAUC', drop = FALSE]
expr_covar <- covar[rownames(expr_rz),]
```


## Load Libraries  

Below are the necessary libraries that we require for this review.  They are 
already installed on your machines so go ahead an load them using the following 
code:

```{r load_libraries, message=FALSE, warning=FALSE}
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(qtl2))
```



## The Marker Map  

The marker map contains a list of the genetic marker positions for each marker
in the genoprobs. Let's read it in now.

```{r read_map}
map <- readRDS(file = 'data/attie_do_map.rds')
```

The marker map for each chromosome is stored in the `map` object. This is used 
to plot the LOD scores calculated at each marker during QTL mapping. Each list 
element is a numeric vector with each marker position in megabases (Mb). Here we 
are using the 69K grid marker file. Often when there are numerous genotype 
arrays used in a study, we interpolate all to a 69k grid file so we are able to 
combine all samples across different array types. 

Look at the structure of `map` in the Environment tab by clicking the triangle 
to the left or by running `str(map)` in the Console.  

Each element in map cointains a list of marker positions and names.

```{r head_map}
head(map[[1]])
```

The names of the markers consist of the chromosome and the bp position, 
separated by an underscore (_).

::::::::::::::::::::::::::::::::::::::::::: callout

The marker positions in the map must be in Mb.

:::::::::::::::::::::::::::::::::::::::::::::::::::

## Genoprobs Data

Previously, we loaded in the physiological phenotypes, the sample covariates,
and the transcript expression data and annotation. We also normalized and rankZ 
transformed the expression data.

In order to perform QTL mapping, we also need the genotype probabilities 
(i.e genoprobs). In this lesson, we have already processed the genotypes and 
produced the genoprobs using 
[calc_genoprob](https://github.com/kbroman/qtl2/blob/main/R/calc_genoprob.R).

<!-- We don't evaluate this block when building the website because the file is 
too large (1.7 GB). We load the probs from Box above . -->

```{r load_genoprobs,eval=FALSE,message=FALSE}
probs <- readRDS("data/attie_DO500_genoprobs_v5.rds")
```

At this point, you should have several data objects in your environment. Look at
the Environment tab to see what data objects are in your environment. It should
look like the figure below.

![Data Environment](fig/data_env_review_qtl.png){alt="Picture of Environment tab with data objects."}

Each element of `probs` is a 3 dimensional array containing the founder allele 
dosages for each sample at each marker on one chromosome. These are the 8 state 
allelle probabilities (not 32) using the 69k marker grid for same 500 DO mice 
that also have clinical phenotypes. We have already calculated genotype 
probabilities for you, so you can skip the step for
[calculating genotype probabilities](https://smcclatchy.github.io/qtl-mapping/calc-genoprob/) 
and the optional step for calculating allele probabilities.

Next, we look at the dimensions of `probs` for chromosome 1: 

```{r dim_genoprobs}
dim(probs[[1]])
```

Each list element of the genoprobs has 500 samples, eight founders, and a 
variable number of markers, depending on the chromosome.

As a reminder, this is what the genoprobs of one mouse look like along one
chromosome.

```{r geno_plot, fig.width=8, fig.height=6}
plot_genoprob(probs, map, ind = 1, chr = 1)
```

In the plot above, the founder contributions, which range between 0 and 1, are 
colored from white (= 0) to black (= 1.0). A value of ~0.5 is grey. The markers 
are on the X-axis and the eight founders (denoted by the letters A through H) on 
the Y-axis. Starting at the left, we see that this sample has genotype GH 
because the rows for G & H are grey, indicating values of 0.5 for both alleles. 
Moving along the genome to the right, the genotype becomes HH where where the 
row is black indicating a value of 1.0.  This is followed by CD, DD, DG, AD, AH,
CE, etc. The values at each marker sum to 1.0.  


## [Kinship Matrix](https://smcclatchy.github.io/qtl-mapping/calc-kinship/)

We also use a kinship matrix in the mapping model to adjust the relatedness
between mice. We also use a different kinship matrix on each chromosome by 
including all of the markers except the ones on the current chromosome. This 
is called the "Leave-One-Chromosome-Out" (LOCO) method. We use the genoprobs 
to create the kinship matrices in the 
[calc-kinship](https://github.com/kbroman/qtl2/blob/main/R/calc_kinship.R) 
function.

```{r calc_kinsihp}
K <- calc_kinship(probs = probs,
                  type = 'loco')
```

Let's look at a part of one of the kinship matrices.

```{r kinship_probs, fig.width=8, fig.height=8}
n_samples <- 50
heatmap(K[[1]][1:n_samples, 1:n_samples], main = "Kinship Between Mice")
```

The figure above shows kinship between all pairs of samples. Light yellow 
indicates low kinship and dark red indicates higher kinship. Orange values 
indicate varying levels of kinship between 0 and 1. The dark red diagonal of the 
matrix indicates that each sample is identical to itself. The orange blocks 
along the diagonal may indicate close relatives (i.e. siblings or cousins).

### Covariates    

Next, we need to create additive covariates that will be used in the mapping 
model.  First, we need to see which covariates are significant. In the data set, 
we have `sex`, `DOwave` (Wave (i.e., batch) of DO mice) 
and `diet_days` (number of days on diet) to test whether there are any sex,
batch or diet effects.

We will use `sex` and `DOwave` as additive covariates. Sex and DO outbreeding
generation are often sensible covariates to add. We will convert `sex` and 
`DOwave` to factors and then use 
[model.matrix](https://rdrr.io/r/stats/model.matrix.html)
to create the covariates matrix that `qtl2` will use.


```{r covariates}
pheno$sex    <- factor(pheno$sex)
pheno$DOwave <- factor(pheno$DOwave)

addcovar     <- model.matrix(~sex + DOwave, data = pheno)[,-1]
```

::::::::::::::::::::::::::::::::::::::::::::::: callout

The sample IDs must be in the rownames of `pheno`, `addcovar`, 
`genoprobs` and `K`. `qtl2` uses the sample IDs to align the samples between 
objects.

:::::::::::::::::::::::::::::::::::::::::::::::::::::::

## [Performing a genome scan](https://smcclatchy.github.io/qtl-mapping/perform-genome-scan/) 

At each marker on the genotyping array, we will fit a model that regresses the 
phenotype (insulin secretion AUC) on covariates and the founder allele 
proportions.  Note that this model will give us an estimate of the effect of 
each founder allele at each marker. There are eight founder strains that 
contributed to the DO, so we will get eight founder allele effects.

### Using Permutations to Asses Significance Thresholds

First, we need to work out the signifcance level.  Let's find the significance 
level for 0.1, 0.05 and 0.01. We would run this code to get the thresholds.
Since this takes several minutes, we have pre-computed the permutations and
have made these available for you to read in.

```{r QTL_perm, warning=FALSE, eval=FALSE}
perm <- scan1perm(genoprobs = probs, 
                  pheno     = ins_tauc,
                  addcovar  = addcovar,
                  n_perm    = 1000)
```

```{r QTL_perm_saving, warning=FALSE, eval=FALSE, eval=F, echo=F}
saveRDS(perm, file = "data/ins_tauc_perm_1000.rds")
```

Read in the pre-computed permutations.

```{r QTL_perm_file, warning=FALSE, eval=FALSE}
perm <- readRDS("data/ins_tauc_perm_1000.rds")

thr = summary(perm,alpha = c(0.1, 0.05, 0.01))
```

### Genome Scan

Now that we have the phenotypes, genoprobs, covariates, and kinship matrices,
we are ready to perform a genome scan on insulin tAUC. We will map both the
untransformed and log-transformed data.

```{r first_scan1, warning=FALSE}
lod = scan1(genoprobs = probs, 
            pheno     = ins_tauc, 
            kinship   = K, 
            addcovar  = addcovar)
```

Next, we plot the genome scan of the untransformed data.

```{r qtl_plot1, fig.width=8, fig.height=6, warning=FALSE}
plot_scan1(x         = qtl,
           map       = map, 
           lodcolumn = "Ins_tAUC", 
           ylim      = c(0, 8.5),
           main      = "Insulin tAUC (untransformed)")
add_threshold(map        = map, 
              thresholdA = thr[2], 
              col        = 'red')
```

We then find the significant peaks using 
[find_peaks](https://github.com/kbroman/qtl2/blob/main/R/find_peaks.R).

```{r find_peaks1}
find_peaks(scan1_output = lod[,1, drop = FALSE], 
           map          = map, 
           threshold    = thr[2],
           prob         = 0.95)
```

There are two peaks on chromosomes 11 and 17. The peak on chromosome 17 has
the highest LOD score.

Next, let's plot the log-transformed data and find the peaks.

```{r qtl_plot2, fig.width=8, fig.height=6, warning=FALSE}
plot_scan1(x         = lod,
           map       = map, 
           lodcolumn = "Ins_tAUC_log", 
           main      = "Insulin tAUC (log-transformed)")
add_threshold(map        = map, 
              thresholdA = thr[2], 
              col        = 'red')
```

We can see that the peaks on chromosome 11 has a higher LOD score when using
the log-transformed data. Let's find the significant peaks.

```{r find_peaks2}
find_peaks(scan1_output = lod[,2,drop = FALSE], 
           map          = map, 
           threshold    = thr[2],
           prob         = 0.95)
```

::::::::::::::::::::::::::::::::::::::::::::::: challenge

#### Challenge 1: Width of QTL Interval

Compare the results of `find_peaks` function for the untransformed and 
log-transformed mapping results. Which one has the narrower support interval?

:::::::::::::::::::::::::::: solution

```{r find_peaks3,echo=FALSE}
p1 <- find_peaks(scan1_output = lod, 
                 map          = map, 
                 threshold    = thr[2],
                 prob         = 0.95)
```


The untransformed data has a `r p1$ci_hi[1] - p1$ci_lo[1]` Mb support interval 
from `r p1$ci_lo[1]` to `r p1$ci_hi[1]` Mb. 

The log-transformed data has a `r p1$ci_hi[4] - p1$ci_lo[4]` Mb support interval 
from `r p1$ci_lo[4]` to `r p1$ci_hi[4]` Mb. 

The log-transformed data produces a higher LOD and a narrower support interval.

:::::::::::::::::::::::::::::::::::::

:::::::::::::::::::::::::::::::::::::::::::::::::::::::::


We can see a very strong peak on chromosome 11 with no other distinguishable 
peaks. Since the log-transformed data produced the highest LOD score, we will
proceed with that phenotype and discard the untransformed tAUC.

```{r discard_untransformed}
ins_tauc = ins_tauc[, 2, drop = FALSE]
```

### QTL effects

We then estimate the founder allele effects on the chromosome containing the
peak that we are interested in. In this case, we will look at chromosome 11.

```{r blups}
chr  <- '11'
blup <- scan1blup(genoprobs = probs[,chr], 
                  pheno     = ins_tauc,
                  kinship   = K[[chr]],
                  addcovar  = addcovar)
```

We then plot the founder allele effects.

```{r plot_blub,fig.width=8,fig.height=6}
plot_coefCC(x       = blup, 
            map     = map, 
            bgcolor = "gray95", 
            legend  = "bottomleft",
            scan1_output = lod,
            main    = "log(Insulin tAUC)")
```

::::::::::::::::::::::::::::::::::::: challenge 

#### Challenge 2: Founder Allele Effects

Which founder alleles contribute to higher insulin tAUC? Use the colors on the
plot and the legend to find the founder strain names.

:::::::::::::::::::::::: solution

The strains whose alleles contribute to higher tAUC are:

A/J  
C57BL/6J  
129S1/SvImJ  
NOD/ShiLtJ  

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: keypoints 

- QTL mapping requires phenotypes and genotypes on the same set of mice.
- We calculate genotype probabilities at each marker and map using founder
allele dosages.
- We use a list of kinship matrices to adjust for the relatedness between mice.
- We add covariates to the model to adjust for the effects of sex and 
generation.
- Log-transforming data to make the distribution more Normal satisfies the 
mapping model assumptions and can lead to higher LOD scores.

::::::::::::::::::::::::::::::::::::::::::::::::
